---
title: "HTG_vignette"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{HTG_vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# DETAILED TUTURAL

This is a detailed tutorial to guide you through the usage of the
'HTGAnalyzer' package. The first step is to load the library:

```{r setup, echo=FALSE}
# Lista de paquetes de Imports
imports <- c(
  "ggplot2", "stats", "apeglm", "data.table", "SummarizedExperiment", "ggrepel", 
  "ggridges", "scales", "ggupset", "pheatmap", "DESeq2", "PoiClaClu", 
  "EnhancedVolcano", "clusterProfiler", "RColorBrewer", "survival", "dplyr", 
  "ggpubr", "tidyr", "maxstat", "readxl", "utils", "graphics", "grDevices", 
  "purrr", "reshape2", "rlang", "tibble"
)

# Lista de paquetes de Suggests
suggests <- c(
  "clusterProfiler", "enrichplot", "ComplexHeatmap", "IOBR", "immunedeconv", 
  "EPIC", "xCell", "org.Hs.eg.db", "knitr", "rmarkdown", "testthat"
)

# Lista de paquetes de Remotes
remotes <- c(
  "omnideconv/immunedeconv", "dviraran/xCell", "GfellerLab/EPIC", 
  "IOBR/IOBR", "kevinblighe/EnhancedVolcano"
)

# Función para cargar un paquete de manera condicional
load_package <- function(pkg) {
  if (!requireNamespace(pkg, quietly = TRUE)) {
    message(paste("El paquete", pkg, "no está instalado."))
  } else {
    library(pkg, character.only = TRUE)
  }
}

# Cargar los paquetes de Imports
for (pkg in imports) {
  load_package(pkg)
}

# Cargar los paquetes de Suggests
for (pkg in suggests) {
  load_package(pkg)
}
library(HTGAnalyzer)

```

## HTG technology:

### HTG_auto() for HTG:

We will start explaining the function `HTG_auto`. This function was
designed to simplify the process of performing multiple analyses on HTG
data. It has a lot of variables that can be turned on and off to skip an
analysis if you wanted or also the function has a lot of variables in
case you want to change the stablished threshold that we propose.

You can download the example Excel files from the following link:
<https://github.com/ISGLOBAL-Rakislova-Lab/HTGAnalyzer/tree/main/SUPLEMENTARY_OUTPUT>

#### HTG_auto(): All the analysis for HTG:

Imagine you have HTG data and want to perform a comprehensive set of
analyses, including Differential Expression Analysis (DEA), Tumor
Microenvironment (TME) analysis, Survival Analysis, and Gene Set
Enrichment Analysis (GSEA).

The code below demonstrates how to conduct each of these analyses. We've
highlighted all the variables that can be modified in the functions,
just to make you aware of them, but feel free to leave them unchanged if
they suit your needs.

```{r, eval=FALSE}
result <- HTG_auto(
  counts_file_path = "~/HTGAnalyzer/counts.xlsx",
  file_type = "HTG",
  AnnotData_file_path = "~/HTGAnalyzer/annotdata.xlsx",
  pattern = "^NC-|^POS-|^GDNA-|^ERCC-",
  threshold_superior_pos = 5, 
  threshold_line_pos = 4, 
  threshold_inferior_lib = 5e+06,
  threshold_lib = 7e+06,
  threshold_superior_nc = 0.05,
  threshold_line_nc = 0.045,
  threshold_superior_gdna = 0.025,
  threshold_line_gdna = 0.02,
  threshold_superior_ercc = 0.03,
  threshold_line_ercc = 0.025,
  threshold_inferior_median = 3,
  threshold_line_median = 5,
  save_csv = TRUE,
  csv_file = "QC_results.csv",
  design_formula = "HPV_status",
  percentage_gene = 0.2,
  threshold_gene = 200,
  threshold_subject = 10,
  genes_to_use = c("CCND1", "MMP10", "CTTN"),
  heatmap_columns = c("Recurrence", "FIGO_2021_STAGE"),
  contrast = c("HPV_status", "Associated", "Independent"),
  pCutoff = 5e-2,
  variable_01 = "Recurrence_01",
  time = "time_to_recurrence",
  DEA = TRUE,
  remove_outliers = TRUE,
  GSEA = TRUE,
  generate_heatmap = TRUE,
  TME = TRUE,
  survival_analysis = TRUE
)

```

The code above is an example. Some of the parameters have default values
in the function, but they are included here for clarity. Refer to the
function documentation for more details on each parameter.

#### HTG_auto(): skipping an analysis for HTG:

If you want to perform some analyses but exclude others (e.g., skipping
survival analysis and GSEA), you can set the relevant parameter to
FALSE. Below is an example where survival and GSEA analyses are
excluded:

If an analysis requires a specific column that you don’t have (e.g.,
survival analysis needs the variable_01 and time columns), you can set
those variables to NULL to skip that particular analysis.

```{r, eval=FALSE}
## LET'S SKIP survival analysis and GSEA

result <- HTG_auto(
  counts_file_path = "~/HTGAnalyzer/counts.xlsx",  # Path to counts data
  file_type = "HTG",  # Type of file
  AnnotData_file_path = "~/HTGAnalyzer/annotdata.xlsx",  # Path to annotation data
  pattern = "^NC-|^POS-|^GDNA-|^ERCC-",  # HTG probes that start with one of these prefixes
  # These thresholds are specific for the HTG transcriptomic panel
  threshold_superior_pos = 5,
  threshold_line_pos = 4,
  threshold_inferior_lib = 5e+06,
  threshold_lib = 7e+06,
  threshold_superior_nc = 0.05,
  threshold_line_nc = 0.045,
  threshold_superior_gdna = 0.025,
  threshold_line_gdna = 0.02,
  threshold_superior_ercc = 0.03,
  threshold_line_ercc = 0.025,
  threshold_inferior_median = 3,
  threshold_line_median = 5,
  save_csv = TRUE,  
  csv_file = "QC_results.csv", 
  design_formula = "HPV_status", 
  percentage_gene = 0.2,  
  threshold_gene = 200,  
  threshold_subject = 10,  
  genes_to_use = c("CCND1", "MMP10", "CTTN"),  # Genes used for survival analysis (optional)
  heatmap_columns = c("HPV_status", "Cyclin_D1"),  
  contrast = c("HPV_status", "Associated", "Independent"),  # Contrast for differential analysis
  pCutoff = 5e-2,  
  variable_01 = "Recurrence_01",  # Variable for survival analysis
  time = "time_to_recurrence",  # Time for survival analysis
  DEA = TRUE,  
  remove_outliers = TRUE,  
  GSEA = FALSE,  # Turn FALSE to skip GSEA
  generate_heatmap = TRUE,  
  TME = TRUE,  
  survival_analysis = FALSE  # Skip survival analysis by setting this parameter to FALSE
)
```

In this example, survival analysis is skipped by setting
`survival_analysis = FALSE`and GSA by setting `GSEA = FALSE`. The output will indicate that the survival
analysis step has been skipped.

### STEP BY STEP for HTG:

What if you already had some analysis or you want to get more control
over it? Don't worry we have the solution.Let's go `step-by-step`,
starting with data import:

#### Load data: HTG_import_counts() for HTG:

You can upload data directly from HTG Edge Reveal using this format. R
needs a quadratic information structure, so maybe you will have to edit
your data to look like this on excel before importing:

| **id**           | VSCC_1     | VSCC_2     | VSCC_3     | VSCC_4     | VSCC_5     | VSCC_6     | VSCC_7     | VSCC_8     |
|--------|--------|--------|--------|--------|--------|--------|--------|--------|
| **Sample ID**    | 1          | 2          | 3          | 4          | 5          | 6          | 7          | 8          |
| **Well**         | A1         | B1         | C1         | D1         | E1         | F1         | G1         | H1         |
| **Date Parsed**  | 14/08/2024 | 14/08/2024 | 14/08/2024 | 14/08/2024 | 14/08/2024 | 14/08/2024 | 14/08/2024 | 14/08/2024 |
| **Total Counts** | 9650628    | 33950226   | 24680689   | 10605963   | 7395173    | 4057094    | 20049502   | 13607678   |
| **A3GALT2**      | 0          | 0          | 164        | 19         | 46         | 1          | 0          | 0          |
| **A4GALT**       | 2          | 662        | 665        | 830        | 748        | 89         | 256        | 333        |
| **A4GNT**        | 0          | 1041       | 103        | 0          | 0          | 2          | 0          | 0          |
| **AAAS**         | 0          | 368        | 70         | 634        | 707        | 425        | 587        | 872        |
| **AACS**         | 1          | 0          | 562        | 728        | 250        | 209        | 365        | 445        |
| **AADAC**        | 0          | 382        | 0          | 5          | 144        | 0          | 0          | 1          |

Or an alternative format:

| **id**      | VSCC_10    | VSCC_11    | VSCC_5    | VSCC_6    | VSCC_4    | VSCC_1    | VSCC_2    | VSCC_3    | VSCC_7    | VSCC_9    | VSCC_8    |
|------|------|------|------|------|------|------|------|------|------|------|------|
| **A3GALT2** | 0          | 0          | 164       | 19        | 46        | 1         | 0         | 0         | 0         | 0         | 0         |
| **A4GALT**  | 2          | 662        | 665       | 830       | 748       | 89        | 256       | 333       | 210       | 2         | 504       |
| **A4GNT**   | 0          | 1041       | 103       | 0         | 0         | 2         | 0         | 0         | 84        | 0         | 0         |
| **AAAS**    | 0          | 368        | 70        | 634       | 707       | 425       | 587       | 872       | 542       | 0         | 370       |
| **AACS**    | 1          | 0          | 562       | 728       | 250       | 209       | 365       | 445       | 62        | 46        | 785       |
| **AADAC**   | 0          | 382        | 0         | 5         | 144       | 0         | 0         | 1         | 260       | 683       | 77        |

```{r, eval=FALSE}
## LOAD COUNTS DATA 

htg_data <- HTG_import_counts("~/HTGAnalyzer/counts.xlsx", "HTG")
head(htg_data)

## LOAD ANNOTDATA
annotdata<- readxl::read_excel("~/HTGAnalyzer/annotdata.xlsx")
head(annotdata)

```

**NOTE:** For the following functions, we'll use tutorial data from the
package. If you want to try with your data, replace
`counts_data_tutorial` with `htg_data`.

#### HTG_quant_to_qual() for HTG:

If you need to transform a quantitative column to a qualitative one, you
can use this function.

```{r}
head(AnnotData_tutorial)

AnnotData_tutorial <- HTG_quant_to_qual(AnnotData_tutorial,  # Dataset to modify
                                        "Cyclin_D1",  # Column to transform
                                        60,  # Threshold to categorize the data
                                        "high",  # Value for entries greater than 60
                                        "low")  # Value for entries less than 60
#Last column will be the generated one 
head(AnnotData_tutorial)
head(AnnotData_tutorial$Cyclin_D12)

```

#### HTG_subset() for HTG:

Now that you have the counts data loaded, you can search for certain
genes of interest, look for specific probes, or filter results from DEA.
This function can also normalize the data by TPM (Transcripts Per
Million).

```{r}
# Subset rows with the prefix "ERCC"
cts_ERCC <- HTG_subset(counts_data_tutorial, "ERCC")

# Subset rows with the prefix "POS"
cts_POS <- HTG_subset(counts_data_tutorial, "POS")

 # Subset rows with the prefix "NC"
cts_NC <- HTG_subset(counts_data_tutorial, "NC")
 
# Subset rows with the prefix "GDNA"
cts_GDNA <- HTG_subset(counts_data_tutorial, "GDNA")

# Subset rows with the prefix "ZZZ3"
cts_ZZZ3 <- HTG_subset(counts_data_tutorial, "ZZZ3")

# Subset rows with the prefix "ZZZ3" and normalize the data
cts_ZZZ3 <- HTG_subset(counts_data_tutorial, "ZZZ3", normalize = TRUE)

resAAAS<- HTG_subset(res_tutorial, "AAAS")
```

#### Perform a QC: HTG_QC() for HTG:

Now, let's perform quality control (QC) on the tutorial data. As you may
know, HTG uses specific probes to perform QC to detect outliers and
assess data quality. Default thresholds are provided, but feel free to
adjust them. This function will:

1.  Detect outliers and save them in an object called `outliers`.

2.  Generate visualization plots.

3.  Perform summary statistics and save them in a .csv file.

4.  Generate a table with the QC results for further review.

```{r}
outliers <- HTG_QC(counts_data_tutorial, pattern = "^NC-|^POS-|^GDNA-|^ERCC-", 
                             threshold_superior_pos = 5,
                             threshold_line_pos = 4,
                             threshold_inferior_lib = 5e+06,
                             threshold_lib = 7e+06,
                             threshold_superior_nc = 0.05,
                             threshold_line_nc = 0.045,
                             threshold_superior_gdna = 0.025,
                             threshold_line_gdna = 0.02,
                             threshold_superior_ercc = 0.03,
                             threshold_line_ercc = 0.025,
                             threshold_inferior_median = 3,
                             threshold_line_median = 5,
                             save_csv = TRUE,
                             csv_file = "QC_results.csv")

```

#### HTG_calculate_summary_stats() for HTG:

If you only want to perform the summary statistics without full QC, this
function will generate summary stats and save them in a .csv file as
well as an R object.

```{r}
summary <- HTG_calculate_summary_stats(counts_data_tutorial, pattern = "^NC-|^POS-|^GDNA-|^ERCC-")
```

#### Perform all the analysis: HTG_analysis() for HTG:

Now that you know which samples are outliers, you can perform any of
the analyses. This function will remove the outliers, perform DEA, TME,
survival analysis, and GSEA. If you want to perform each step
separately, you can use the functions `HTG_GSEA()`, `HTG_DEA()`,
`HTG_TME()`, and `HTG_survival()`.
Running the analyses individually offers greater control, allowing you to adjust additional parameters. Refer to each function's section for detailed instructions on how they work.

```{r}
ALL_done <- HTG_analysis(outliers = outliers_tutorial, 
                         pattern = "^NC-|^POS-|^GDNA-|^ERCC-", 
                         counts_data = counts_data_tutorial, 
                         col_data = AnnotData_tutorial, 
                         design_formula = "HPV_status", 
                         percentage_gene = 0.2, 
                         threshold_gene = 200, 
                         threshold_subject = 10,
                         genes_to_use = c("CCND1", "MMP10", "CTTN"),
                         heatmap_columns = c("HPV_status","Cyclin_D1"),
                         contrast = c("HPV_status", "Associated", "Independent"), 
                         pCutoff = 5e-2, 
                         variable_01 = "Recurrence_01",
                         time = "time_to_recurrence", 
                         DEA = TRUE, 
                         remove_outliers = TRUE,
                         GSEA = FALSE, 
                         generate_heatmap = TRUE, 
                         TME = TRUE, 
                         survival_analysis = FALSE)
```

#### Differentia expression analysis: HTG_DEA() for HTG:

If you want more control, such as applying extra filtering or performing
LFC shrinkage, `HTG_DEA()` will provide more options.

```{r}
### DEA:
results <- HTG_DEA(outliers= outliers_tutorial, 
                   counts_data= counts_data_tutorial, 
                   AnnotData_tutorial, 
                   design_formula = "HPV_status", 
                   heatmap_columns = c("HPV_status", "Recurrence"),
                   contrast = c("HPV_status", "Associated", "Independent"), 
                   pattern = "^NC-|^POS-|^GDNA-|^ERCC-", 
                   remove_outliers = TRUE,
                   percentage_gene = 0.2, 
                   threshold_gene = 200,
                   threshold_subject = 10,
                   pCutoff = 5e-2, 
                   apply_filtering = TRUE, 
                   apply_lfc_shrinkage = TRUE, 
                   extract_shrinkage = FALSE)

head(results)
```

#### GSEA: GSEA() for HTG:

A GSEA analysis can be performed using the results from the DEA
contrast. HTG_GSEA will generate plots and Excel files for easy
interpretation.

```{r}
HTG_GSEA(res_tutorial)
```

#### TME: HTG_TME () for HTG:

This function performs Tumor Microenvironment (TME) analysis, which will
be stored in `TME_results`. You can access each method using `$`. This
object contains results from three methodologies.

```{r}

TME_results<- HTG_TME(outliers= outliers_tutorial, 
                      pattern= "^NC-|^POS-|^GDNA-|^ERCC-",
                      counts_data= counts_data_tutorial, 
                      AnnotData= AnnotData_tutorial, 
                      design_formula= "HPV_status", 
                      remove_outliers = TRUE, 
                      DEA = NULL)

head(TME_results$EPIC)
head(TME_results$QTI)
head(TME_results$XCELL)
```

#### Survival analysis: HTG_survival() for HTG:

Survival analysis can be performed using the results from DEA (res),
specific genes, or TME results. The function prioritizes inputs: first
genes_to_use, then DEA results, and finally TME results if none are
provided.

```{r}
############# survival using res
 HTG_survival(variable_01 = "Recurrence_01",
              time = "time_to_recurrence",
              col_data = AnnotData_tutorial,
              counts_data = counts_data_tutorial,
              res = res_tutorial,
              method = "median",
              genes_to_use = NULL,
              TME = NULL,
              outliers = outliers_tutorial,
              pattern = "^NC-|^POS-|^GDNA-|^ERCC-",
              remove_outliers = TRUE)

 # USING SPECIFIC GENES

 HTG_survival(variable_01 = "Recurrence_01",
              time = "time_to_recurrence",
              col_data = AnnotData_tutorial,
              counts_data = counts_data_tutorial,
              res = NULL,
              method = "quartiles",
              genes_to_use = c("LCP1", "OMA1"),
              TME = NULL,
              outliers = outliers_tutorial,
              pattern = "^NC-|^POS-|^GDNA-|^ERCC-",
              remove_outliers = TRUE)
 

 # USING TME DATA
 HTG_survival(variable_01 = "Recurrence_01",
              time = "time_to_recurrence",
              col_data = AnnotData_tutorial,
              counts_data = NULL,
              res = NULL,
              method = "maxstat",
              genes_to_use = NULL,
              TME = TME_data_tutorial$EPIC,
              outliers = outliers_tutorial,
              pattern = "^NC-|^POS-|^GDNA-|^ERCC-",
              remove_outliers = TRUE)
```

## RNAseq:

Now let's show how it works with RNAseq You can download the example
Excel files from the following link:
<https://github.com/ISGLOBAL-Rakislova-Lab/HTGAnalyzer/tree/main/SUPLEMENTARY_OUTPUT>

### HTG_auto() for RNAseq:

#### HTG_auto(): All the analysis for RNAseq:

Imagine you have RNAseq data and aim to perform a full suite of analyses, including Differential Expression Analysis (DEA), Tumor Microenvironment (TME) analysis, Survival Analysis, and Gene Set Enrichment Analysis (GSEA).

The following code shows how to run each of these analyses. We’ve highlighted the variables that can be adjusted in the functions, so you're aware of them. However, you can leave them as they are if the default settings meet your requirements.

```{r, eval=FALSE}
result <- HTG_auto(
  counts_file_path = "~/HTGAnalyzer/counts_RNAseq.xlsx",  # path to counts data
  file_type = "RNAseq",  # Tipe of file
  AnnotData_file_path = "~/HTGAnalyzer/annotdata.xlsx",  # path to annotdata.
  pattern = NULL,
  save_csv = TRUE,
  csv_file = "QC_results.csv",
  design_formula = "HPV_status",
  percentage_gene = 0.2,
  threshold_gene = 200,
  threshold_subject = 10,
  genes_to_use = c("CCND1", "MMP10", "CTTN"), # Genes used for survival analysis (optional)
  heatmap_columns = c("HPV_status", "Cyclin_D1"),  # Columns for the heatmap
  contrast = c("HPV_status", "Associated", "Independent"),  # Contrast for differential analysis
  pCutoff = 5e-2,
  variable_01 = "Recurrence_01",   # Variable for analysis
  time = "time_to_recurrence",  # Time for survival analysis
  DEA = TRUE,
  remove_outliers = TRUE,
  GSEA = TRUE,
  generate_heatmap = TRUE,
  TME = TRUE,
  survival_analysis = TRUE
)

```

**NOTE:** remember that RNAseq data doesn't have probes and doesn't pass
for `HTG_QC()`as we did with HTG data. 

The code above is an example. Some of the parameters have default values
in the function, but they are included here for clarity. Refer to the
function documentation for more details on each parameter.

#### HTG_auto(): skipping an analysis for RNAseq:

If you'd like to run some analyses but skip others (for example, leaving out the survival analysis and GSEA), simply set the corresponding parameter to FALSE. Here's an example where the survival and GSEA analyses are not included:

In cases where a specific analysis requires certain data columns that you don't have (such as the survival analysis needing variable_01 and time columns), you can set those variables to NULL to bypass that analysis.

```{r,eval=FALSE}
result <- HTG_auto(
  counts_file_path = "~/HTGAnalyzer/counts_RNAseq.xlsx",  # path to counts data
  file_type = "RNAseq",  # Tipe of file
  AnnotData_file_path = "~/HTGAnalyzer/annotdata_RNAseq.xlsx",  # path to annotdata.
  pattern = NULL,
  save_csv = TRUE,  
  csv_file = "QC_results.csv", 
  design_formula = "HPV_status", 
  percentage_gene = 0.2,  
  threshold_gene = 200,  
  threshold_subject = 10,  
  genes_to_use = c("CCND1", "MMP10", "CTTN"),  # Genes used for survival analysis (optional)
  heatmap_columns = c("HPV_status", "Cyclin_D1"),  
  contrast = c("HPV_status", "Associated", "Independent"),  # Contrast for differential analysis
  pCutoff = 5e-2,  
  variable_01 = "Recurrence_01",  # Variable for survival analysis
  time = "time_to_recurrence",  # Time for survival analysis
  DEA = TRUE,  
  remove_outliers = TRUE,  
  GSEA = FALSE,  
  generate_heatmap = TRUE,  
  TME = TRUE,  
  survival_analysis = FALSE  # Skip survival analysis by setting this parameter to FALSE
)
```

In this example, survival analysis is skipped by setting
`survival_analysis = FALSE` and GSA by setting `GSEA = FALSE`. The output will indicate that the survival
analysis step has been skipped.

### STEP BY STEP for RNAseq:

What if you already had some analysis or you want to get more control
over it? Don't worry we have the solution.Let's go `step-by-step`,
starting with data import:

#### Load data: HTG_import_counts() for RNAseq:

You can upload data directly from HTG Edge Reveal using this function.

```{r, eval=FALSE}
## LOAD COUNTS DATA 

htg_data <- HTG_import_counts("~/HTGAnalyzer/counts.xlsx", "RNAseq")
head(htg_data)

## LOAD ANNOTDATA
annotdata<- readxl::read_excel("~/HTGAnalyzer/annotdata.xlsx")
head(annotdata)

```

**NOTE:** For the following functions, we'll use tutorial data from the
package. If you want to try with your data, replace
`counts_data_tutorial` with `htg_data`.

#### HTG_quant_to_qual() for RNAseq:

If you need to transform a quantitative column to a qualitative one, you
can use this function.

```{r}
head(AnnotData_tutorial)

AnnotData_tutorial <- HTG_quant_to_qual(AnnotData_tutorial,  # Dataset to modify
                                        "Cyclin_D1",  # Column to transform
                                        60,  # Threshold to categorize the data
                                        "high",  # Value for entries greater than 60
                                        "low")  # Value for entries less than 60
head(AnnotData_tutorial)

```

#### HTG_subset() for RNAseq:

Now that you have the counts data loaded, you can search for certain
genes of interest, look for specific probes, or filter results from DEA.
This function can also normalize the data by TPM (Transcripts Per
Million).

```{r}
 # Subset rows with the prefix "ERCC"
 cts_ERCC <- HTG_subset(counts_data_RNAseq, "ERCC")

 # Subset rows with the prefix "POS"
 cts_POS <- HTG_subset(counts_data_RNAseq, "POS")

 # Subset rows with the prefix "NC"
 cts_NC <- HTG_subset(counts_data_RNAseq, "NC")
 
# Subset rows with the prefix "GDNA"
 cts_GDNA <- HTG_subset(counts_data_RNAseq, "GDNA")

 # Subset rows with the prefix "ZZZ3"
 cts_ZZZ3 <- HTG_subset(counts_data_RNAseq, "ZZZ3")

  # Subset rows with the prefix "ZZZ3" and normalize the data
 cts_ZZZ3 <- HTG_subset(counts_data_RNAseq, "ZZZ3", normalize = TRUE)

 resAAAS<- HTG_subset(res_tutorial, "AAAS")
```

#### HTG_calculate_summary_stats() for RNAseq:

As RNaseq doen't have a QC() if is needed you can perform a summary
stats.

```{r}
summary <- HTG_calculate_summary_stats(counts_data_RNAseq, pattern = NULL)
```

#### Perform all the analysis: HTG_analysis() for RNAseq:

Now that you know which variables are outliers, you can perform any of
the analyses.This function will remove the outliers, perform DEA, TME,
survival analysis, and GSEA. If you want to perform each step
separately, you can use the functions `HTG_GSEA()`, `HTG_DEA()`,
`HTG_TME()`, and `HTG_survival()`.

```{r}
ALL_done <- HTG_analysis(outliers = NULL, # RNAseq doen't have outliers. 
                         pattern = NULL, 
                         counts_data = counts_data_RNAseq, 
                         col_data = AnnotData_tutorial, 
                         design_formula = "HPV_status", 
                         percentage_gene = 0.2, 
                         threshold_gene = 200, 
                         threshold_subject = 10,
                         genes_to_use = c("CCND1", "MMP10", "CTTN"),
                         heatmap_columns = c("HPV_status","Cyclin_D1"),
                         contrast = c("HPV_status", "Associated", "Independent"), 
                         pCutoff = 5e-2, 
                         variable_01 = "Recurrence_01",
                         time = "time_to_recurrence", 
                         DEA = TRUE, 
                         remove_outliers = FALSE, #set false because don't have outliers. 
                         GSEA = FALSE, 
                         generate_heatmap = TRUE, 
                         TME = TRUE, 
                         survival_analysis = FALSE)
```

#### Differentia expression analysis: HTG_DEA() for RNAseq:

If you want more control, such as applying extra filtering or performing
LFC shrinkage, `HTG_DEA()` will provide more options.

```{r}
### DEA:
results <- HTG_DEA(outliers= NULL, 
                   counts_data= counts_data_RNAseq, 
                   AnnotData_tutorial, 
                   design_formula = "HPV_status", 
                   heatmap_columns = c("HPV_status", "Recurrence"),
                   contrast = c("HPV_status", "Associated", "Independent"), 
                   pattern = NULL, 
                   remove_outliers = TRUE,
                   percentage_gene = 0.2, 
                   threshold_gene = 200,
                   threshold_subject = 10,
                   pCutoff = 5e-2, 
                   apply_filtering = TRUE, 
                   apply_lfc_shrinkage = TRUE, 
                   extract_shrinkage = FALSE)

head(results)
```

#### GSEA: GSEA() for RNAseq:

A GSEA analysis can be performed using the results from the DEA
contrast. HTG_GSEA will generate plots and Excel files for easy
interpretation.

```{r}
HTG_GSEA(res_tutorial)
```

#### TME: HTG_TME () for RNAseq:

This function performs Tumor Microenvironment (TME) analysis, which will
be stored in `TME_results`. You can access each method using `$`. This
object contains results from three methodologies.

```{r}

TME_results<- HTG_TME(outliers= NULL, 
                      pattern= NULL,
                      counts_data= counts_data_RNAseq, 
                      AnnotData= AnnotData_tutorial, 
                      design_formula= "HPV_status", 
                      remove_outliers = FALSE, 
                      DEA = NULL)


head(TME_results$EPIC)
head(TME_results$QTI)
head(TME_results$XCELL)
```

#### Survival analysis: HTG_survival() for RNAseq:

Survival analysis can be performed using the results from DEA (res),
specific genes, or TME results. The function prioritizes inputs: first
genes_to_use, then DEA results, and finally TME results if none are
provided.

```{r}
# survival using res
HTG_survival(variable_01 = "Recurrence_01",
             time = "time_to_recurrence",
             col_data = AnnotData_tutorial,
             counts_data = counts_data_tutorial,
             res = res_tutorial,
             method = "median",
             genes_to_use = NULL,
             TME = NULL,
             outliers = NULL,
             pattern = NULL,
             remove_outliers = FALSE)

# USING SPECIFIC GENES
HTG_survival(variable_01 = "Recurrence_01",
             time = "time_to_recurrence",
             col_data = AnnotData_tutorial,
             counts_data = counts_data_tutorial,
             res = NULL,
             method = "quartiles",
             genes_to_use = c("LCP1", "OMA1"),
             TME = NULL,
             outliers = NULL,
             pattern = NULL,
             remove_outliers = FALSE)

# USING TME DATA
HTG_survival(variable_01 = "Recurrence_01",
             time = "time_to_recurrence",
             col_data = AnnotData_tutorial,
             counts_data = NULL,
             res = NULL,
             method = "maxstat",
             genes_to_use = NULL,
             TME = TME_data_tutorial$EPIC,
             outliers = NULL,
             pattern = NULL,
             remove_outliers = FALSE)

```
