<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>HTG_vignette</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">HTG_vignette</h1>



<div id="detailed-tutural" class="section level1">
<h1>DETAILED TUTURAL</h1>
<p>This is a detailed tutorial to guide you through the usage of the
‘HTGAnalyzer’ package. The first step is to load the library:</p>
<pre><code>#&gt; Warning: package &#39;ggplot2&#39; was built under R version 4.3.3
#&gt; Warning: package &#39;data.table&#39; was built under R version 4.3.3
#&gt; Loading required package: MatrixGenerics
#&gt; Loading required package: matrixStats
#&gt; Warning: package &#39;matrixStats&#39; was built under R version 4.3.3
#&gt; 
#&gt; Attaching package: &#39;MatrixGenerics&#39;
#&gt; The following objects are masked from &#39;package:matrixStats&#39;:
#&gt; 
#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,
#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
#&gt;     rowWeightedSds, rowWeightedVars
#&gt; Loading required package: GenomicRanges
#&gt; Loading required package: stats4
#&gt; Loading required package: BiocGenerics
#&gt; 
#&gt; Attaching package: &#39;BiocGenerics&#39;
#&gt; The following objects are masked from &#39;package:stats&#39;:
#&gt; 
#&gt;     IQR, mad, sd, var, xtabs
#&gt; The following objects are masked from &#39;package:base&#39;:
#&gt; 
#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,
#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
#&gt;     table, tapply, union, unique, unsplit, which.max, which.min
#&gt; Loading required package: S4Vectors
#&gt; 
#&gt; Attaching package: &#39;S4Vectors&#39;
#&gt; The following objects are masked from &#39;package:data.table&#39;:
#&gt; 
#&gt;     first, second
#&gt; The following object is masked from &#39;package:utils&#39;:
#&gt; 
#&gt;     findMatches
#&gt; The following objects are masked from &#39;package:base&#39;:
#&gt; 
#&gt;     expand.grid, I, unname
#&gt; Loading required package: IRanges
#&gt; 
#&gt; Attaching package: &#39;IRanges&#39;
#&gt; The following object is masked from &#39;package:data.table&#39;:
#&gt; 
#&gt;     shift
#&gt; The following object is masked from &#39;package:grDevices&#39;:
#&gt; 
#&gt;     windows
#&gt; Loading required package: GenomeInfoDb
#&gt; Warning: package &#39;GenomeInfoDb&#39; was built under R version 4.3.3
#&gt; Loading required package: Biobase
#&gt; Welcome to Bioconductor
#&gt; 
#&gt;     Vignettes contain introductory material; view with
#&gt;     &#39;browseVignettes()&#39;. To cite Bioconductor, see
#&gt;     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.
#&gt; 
#&gt; Attaching package: &#39;Biobase&#39;
#&gt; The following object is masked from &#39;package:MatrixGenerics&#39;:
#&gt; 
#&gt;     rowMedians
#&gt; The following objects are masked from &#39;package:matrixStats&#39;:
#&gt; 
#&gt;     anyMissing, rowMedians
#&gt; Warning: package &#39;ggrepel&#39; was built under R version 4.3.3
#&gt; Warning: package &#39;ggupset&#39; was built under R version 4.3.3
#&gt; Warning: package &#39;DESeq2&#39; was built under R version 4.3.3
#&gt; 
#&gt; Warning: package &#39;clusterProfiler&#39; was built under R version 4.3.3
#&gt; clusterProfiler v4.10.1  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/
#&gt; 
#&gt; If you use clusterProfiler in published research, please cite:
#&gt; T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141
#&gt; 
#&gt; Attaching package: &#39;clusterProfiler&#39;
#&gt; The following object is masked from &#39;package:IRanges&#39;:
#&gt; 
#&gt;     slice
#&gt; The following object is masked from &#39;package:S4Vectors&#39;:
#&gt; 
#&gt;     rename
#&gt; The following object is masked from &#39;package:stats&#39;:
#&gt; 
#&gt;     filter
#&gt; Warning: package &#39;survival&#39; was built under R version 4.3.3
#&gt; 
#&gt; Attaching package: &#39;dplyr&#39;
#&gt; The following object is masked from &#39;package:Biobase&#39;:
#&gt; 
#&gt;     combine
#&gt; The following objects are masked from &#39;package:GenomicRanges&#39;:
#&gt; 
#&gt;     intersect, setdiff, union
#&gt; The following object is masked from &#39;package:GenomeInfoDb&#39;:
#&gt; 
#&gt;     intersect
#&gt; The following objects are masked from &#39;package:IRanges&#39;:
#&gt; 
#&gt;     collapse, desc, intersect, setdiff, slice, union
#&gt; The following objects are masked from &#39;package:S4Vectors&#39;:
#&gt; 
#&gt;     first, intersect, rename, setdiff, setequal, union
#&gt; The following objects are masked from &#39;package:BiocGenerics&#39;:
#&gt; 
#&gt;     combine, intersect, setdiff, union
#&gt; The following object is masked from &#39;package:matrixStats&#39;:
#&gt; 
#&gt;     count
#&gt; The following objects are masked from &#39;package:data.table&#39;:
#&gt; 
#&gt;     between, first, last
#&gt; The following objects are masked from &#39;package:stats&#39;:
#&gt; 
#&gt;     filter, lag
#&gt; The following objects are masked from &#39;package:base&#39;:
#&gt; 
#&gt;     intersect, setdiff, setequal, union
#&gt; 
#&gt; Attaching package: &#39;tidyr&#39;
#&gt; The following object is masked from &#39;package:S4Vectors&#39;:
#&gt; 
#&gt;     expand
#&gt; 
#&gt; Attaching package: &#39;purrr&#39;
#&gt; The following object is masked from &#39;package:clusterProfiler&#39;:
#&gt; 
#&gt;     simplify
#&gt; The following object is masked from &#39;package:scales&#39;:
#&gt; 
#&gt;     discard
#&gt; The following object is masked from &#39;package:GenomicRanges&#39;:
#&gt; 
#&gt;     reduce
#&gt; The following object is masked from &#39;package:IRanges&#39;:
#&gt; 
#&gt;     reduce
#&gt; The following object is masked from &#39;package:data.table&#39;:
#&gt; 
#&gt;     transpose
#&gt; 
#&gt; Attaching package: &#39;reshape2&#39;
#&gt; The following object is masked from &#39;package:tidyr&#39;:
#&gt; 
#&gt;     smiths
#&gt; The following objects are masked from &#39;package:data.table&#39;:
#&gt; 
#&gt;     dcast, melt
#&gt; Warning: package &#39;rlang&#39; was built under R version 4.3.3
#&gt; 
#&gt; Attaching package: &#39;rlang&#39;
#&gt; The following objects are masked from &#39;package:purrr&#39;:
#&gt; 
#&gt;     %@%, flatten, flatten_chr, flatten_dbl, flatten_int, flatten_lgl,
#&gt;     flatten_raw, invoke, splice
#&gt; The following object is masked from &#39;package:Biobase&#39;:
#&gt; 
#&gt;     exprs
#&gt; The following object is masked from &#39;package:data.table&#39;:
#&gt; 
#&gt;     :=
#&gt; 
#&gt; Attaching package: &#39;enrichplot&#39;
#&gt; The following object is masked from &#39;package:ggpubr&#39;:
#&gt; 
#&gt;     color_palette
#&gt; Loading required package: grid
#&gt; ========================================
#&gt; ComplexHeatmap version 2.18.0
#&gt; Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
#&gt; Github page: https://github.com/jokergoo/ComplexHeatmap
#&gt; Documentation: http://jokergoo.github.io/ComplexHeatmap-reference
#&gt; 
#&gt; If you use it in published research, please cite either one:
#&gt; - Gu, Z. Complex Heatmap Visualization. iMeta 2022.
#&gt; - Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
#&gt;     genomic data. Bioinformatics 2016.
#&gt; 
#&gt; 
#&gt; The new InteractiveComplexHeatmap package can directly export static 
#&gt; complex heatmaps into an interactive Shiny app with zero effort. Have a try!
#&gt; 
#&gt; This message can be suppressed by:
#&gt;   suppressPackageStartupMessages(library(ComplexHeatmap))
#&gt; ========================================
#&gt; ! pheatmap() has been masked by ComplexHeatmap::pheatmap(). Most of the arguments
#&gt;    in the original pheatmap() are identically supported in the new function. You 
#&gt;    can still use the original function by explicitly calling pheatmap::pheatmap().
#&gt; 
#&gt; Attaching package: &#39;ComplexHeatmap&#39;
#&gt; The following object is masked from &#39;package:pheatmap&#39;:
#&gt; 
#&gt;     pheatmap
#&gt; Loading required package: patchwork
#&gt; Warning: package &#39;patchwork&#39; was built under R version 4.3.3
#&gt; Loading required package: survminer
#&gt; 
#&gt; Attaching package: &#39;survminer&#39;
#&gt; The following object is masked from &#39;package:survival&#39;:
#&gt; 
#&gt;     myeloma
#&gt; ==========================================================================
#&gt;   IOBR v0.99.8  Immuno-Oncology Biological Research 
#&gt;   For Tutorial: https://iobr.github.io/book/
#&gt;   For Help: https://github.com/IOBR/IOBR/issues
#&gt; 
#&gt;  If you use IOBR in published research, please cite:
#&gt;  DQ Zeng, YR Fang, WJ Qiu, ..., GC Yu*, WJ Liao*, (2024) 
#&gt;  IOBR2: Multidimensional Decoding Tumor Microenvironment for Immuno-Oncology Research. 
#&gt;  bioRxiv, 2024.01. 13.575484; 
#&gt;  https://www.biorxiv.org/content/10.1101/2024.01.13.575484v2.full.pdf 
#&gt; ==========================================================================
#&gt;  DQ Zeng, ZL Ye, RF Sheng, GC Yu, Y Xiong, ..., WJ Liao*.
#&gt;  IOBR: Multi-omics Immuno-Oncology Biological Research to decode 
#&gt;   tumor microenvironment and signatures. Frontiers in Immunology. 12:687975,(2021). 
#&gt;  DOI: 10.3389/fimmu.2021.687975
#&gt;  Higly Cited Paper and Hot Paper of WOS
#&gt; ==========================================================================
#&gt; Loading required package: EPIC
#&gt; 
#&gt; Attaching package: &#39;EPIC&#39;
#&gt; The following object is masked from &#39;package:IOBR&#39;:
#&gt; 
#&gt;     EPIC
#&gt; 
#&gt; Loading required package: AnnotationDbi
#&gt; 
#&gt; Attaching package: &#39;AnnotationDbi&#39;
#&gt; The following object is masked from &#39;package:dplyr&#39;:
#&gt; 
#&gt;     select
#&gt; The following object is masked from &#39;package:clusterProfiler&#39;:
#&gt; 
#&gt;     select
#&gt; Warning: package &#39;knitr&#39; was built under R version 4.3.3
#&gt; Warning: package &#39;rmarkdown&#39; was built under R version 4.3.3
#&gt; 
#&gt; Attaching package: &#39;testthat&#39;
#&gt; The following objects are masked from &#39;package:rlang&#39;:
#&gt; 
#&gt;     is_false, is_null, is_true
#&gt; The following object is masked from &#39;package:purrr&#39;:
#&gt; 
#&gt;     is_null
#&gt; The following object is masked from &#39;package:tidyr&#39;:
#&gt; 
#&gt;     matches
#&gt; The following object is masked from &#39;package:dplyr&#39;:
#&gt; 
#&gt;     matches</code></pre>
<div id="htg-technology" class="section level2">
<h2>HTG technology:</h2>
<div id="htg_auto-for-htg" class="section level3">
<h3>HTG_auto() for HTG:</h3>
<p>We will start explaining the function <code>HTG_auto</code>. This
function was designed to simplify the process of performing multiple
analyses on HTG data. It has a lot of variables that can be turned on
and off to skip an analysis if you wanted or also the function has a lot
of variables in case you want to change the stablished threshold that we
propose.</p>
<p>You can download the example Excel files from the following link: <a href="https://github.com/ISGLOBAL-Rakislova-Lab/HTGAnalyzer/tree/main/SUPLEMENTARY_OUTPUT" class="uri">https://github.com/ISGLOBAL-Rakislova-Lab/HTGAnalyzer/tree/main/SUPLEMENTARY_OUTPUT</a></p>
<div id="htg_auto-all-the-analysis-for-htg" class="section level4">
<h4>HTG_auto(): All the analysis for HTG:</h4>
<p>Imagine you have HTG data and want to perform a comprehensive set of
analyses, including Differential Expression Analysis (DEA), Tumor
Microenvironment (TME) analysis, Survival Analysis, and Gene Set
Enrichment Analysis (GSEA).</p>
<p>The code below demonstrates how to conduct each of these analyses.
We’ve highlighted all the variables that can be modified in the
functions, just to make you aware of them, but feel free to leave them
unchanged if they suit your needs.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">HTG_auto</span>(</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>  <span class="at">counts_file_path =</span> <span class="st">&quot;~/HTGAnalyzer/counts.xlsx&quot;</span>,</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  <span class="at">file_type =</span> <span class="st">&quot;HTG&quot;</span>,</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>  <span class="at">AnnotData_file_path =</span> <span class="st">&quot;~/HTGAnalyzer/annotdata.xlsx&quot;</span>,</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>  <span class="at">pattern =</span> <span class="st">&quot;^NC-|^POS-|^GDNA-|^ERCC-&quot;</span>,</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>  <span class="at">threshold_superior_pos =</span> <span class="dv">5</span>, </span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>  <span class="at">threshold_line_pos =</span> <span class="dv">4</span>, </span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>  <span class="at">threshold_inferior_lib =</span> <span class="fl">5e+06</span>,</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>  <span class="at">threshold_lib =</span> <span class="fl">7e+06</span>,</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>  <span class="at">threshold_superior_nc =</span> <span class="fl">0.05</span>,</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>  <span class="at">threshold_line_nc =</span> <span class="fl">0.045</span>,</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>  <span class="at">threshold_superior_gdna =</span> <span class="fl">0.025</span>,</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>  <span class="at">threshold_line_gdna =</span> <span class="fl">0.02</span>,</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>  <span class="at">threshold_superior_ercc =</span> <span class="fl">0.03</span>,</span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>  <span class="at">threshold_line_ercc =</span> <span class="fl">0.025</span>,</span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a>  <span class="at">threshold_inferior_median =</span> <span class="dv">3</span>,</span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a>  <span class="at">threshold_line_median =</span> <span class="dv">5</span>,</span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a>  <span class="at">save_csv =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a>  <span class="at">csv_file =</span> <span class="st">&quot;QC_results.csv&quot;</span>,</span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a>  <span class="at">design_formula =</span> <span class="st">&quot;HPV_status&quot;</span>,</span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a>  <span class="at">percentage_gene =</span> <span class="fl">0.2</span>,</span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a>  <span class="at">threshold_gene =</span> <span class="dv">200</span>,</span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a>  <span class="at">threshold_subject =</span> <span class="dv">10</span>,</span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a>  <span class="at">genes_to_use =</span> <span class="fu">c</span>(<span class="st">&quot;CCND1&quot;</span>, <span class="st">&quot;MMP10&quot;</span>, <span class="st">&quot;CTTN&quot;</span>),</span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a>  <span class="at">heatmap_columns =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>, <span class="st">&quot;Ciclina_D1&quot;</span>),</span>
<span id="cb2-26"><a href="#cb2-26" tabindex="-1"></a>  <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>, <span class="st">&quot;Associated&quot;</span>, <span class="st">&quot;Independent&quot;</span>),</span>
<span id="cb2-27"><a href="#cb2-27" tabindex="-1"></a>  <span class="at">pCutoff =</span> <span class="fl">5e-2</span>,</span>
<span id="cb2-28"><a href="#cb2-28" tabindex="-1"></a>  <span class="at">variable_01 =</span> <span class="st">&quot;Recurrence_01&quot;</span>,</span>
<span id="cb2-29"><a href="#cb2-29" tabindex="-1"></a>  <span class="at">time =</span> <span class="st">&quot;Time_to_death_surv&quot;</span>,</span>
<span id="cb2-30"><a href="#cb2-30" tabindex="-1"></a>  <span class="at">DEA =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-31"><a href="#cb2-31" tabindex="-1"></a>  <span class="at">remove_outliers =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-32"><a href="#cb2-32" tabindex="-1"></a>  <span class="at">GSEA =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-33"><a href="#cb2-33" tabindex="-1"></a>  <span class="at">generate_heatmap =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-34"><a href="#cb2-34" tabindex="-1"></a>  <span class="at">TME =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-35"><a href="#cb2-35" tabindex="-1"></a>  <span class="at">survival_analysis =</span> <span class="cn">TRUE</span></span>
<span id="cb2-36"><a href="#cb2-36" tabindex="-1"></a>)</span></code></pre></div>
<p>The code above is an example. Some of the parameters have default
values in the function, but they are included here for clarity. Refer to
the function documentation for more details on each parameter.</p>
</div>
<div id="htg_auto-skipping-an-analysis-for-htg" class="section level4">
<h4>HTG_auto(): skipping an analysis for HTG:</h4>
<p>If you want to perform some analyses but exclude others (e.g.,
skipping survival analysis and GSEA), you can set the relevant parameter
to FALSE. Below is an example where survival and GSEA analyses are
excluded:</p>
<p>If an analysis requires a specific column that you don’t have (e.g.,
survival analysis needs the variable_01 and time columns), you can set
those variables to NULL to skip that particular analysis.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="do">## LET&#39;S SKIP survival analysis and GSEA</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">HTG_auto</span>(</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  <span class="at">counts_file_path =</span> <span class="st">&quot;~/HTGAnalyzer/counts.xlsx&quot;</span>,  <span class="co"># Path to counts data</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="at">file_type =</span> <span class="st">&quot;HTG&quot;</span>,  <span class="co"># Type of file</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="at">AnnotData_file_path =</span> <span class="st">&quot;~/HTGAnalyzer/annotdata.xlsx&quot;</span>,  <span class="co"># Path to annotation data</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  <span class="at">pattern =</span> <span class="st">&quot;^NC-|^POS-|^GDNA-|^ERCC-&quot;</span>,  <span class="co"># HTG probes that start with one of these prefixes</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>  <span class="co"># These thresholds are specific for the HTG transcriptomic panel</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>  <span class="at">threshold_superior_pos =</span> <span class="dv">5</span>,</span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>  <span class="at">threshold_line_pos =</span> <span class="dv">4</span>,</span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>  <span class="at">threshold_inferior_lib =</span> <span class="fl">5e+06</span>,</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>  <span class="at">threshold_lib =</span> <span class="fl">7e+06</span>,</span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>  <span class="at">threshold_superior_nc =</span> <span class="fl">0.05</span>,</span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>  <span class="at">threshold_line_nc =</span> <span class="fl">0.045</span>,</span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>  <span class="at">threshold_superior_gdna =</span> <span class="fl">0.025</span>,</span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a>  <span class="at">threshold_line_gdna =</span> <span class="fl">0.02</span>,</span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a>  <span class="at">threshold_superior_ercc =</span> <span class="fl">0.03</span>,</span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a>  <span class="at">threshold_line_ercc =</span> <span class="fl">0.025</span>,</span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a>  <span class="at">threshold_inferior_median =</span> <span class="dv">3</span>,</span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a>  <span class="at">threshold_line_median =</span> <span class="dv">5</span>,</span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a>  <span class="at">save_csv =</span> <span class="cn">TRUE</span>,  </span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a>  <span class="at">csv_file =</span> <span class="st">&quot;QC_results.csv&quot;</span>, </span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a>  <span class="at">design_formula =</span> <span class="st">&quot;HPV_status&quot;</span>, </span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a>  <span class="at">percentage_gene =</span> <span class="fl">0.2</span>,  </span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a>  <span class="at">threshold_gene =</span> <span class="dv">200</span>,  </span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a>  <span class="at">threshold_subject =</span> <span class="dv">10</span>,  </span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a>  <span class="at">genes_to_use =</span> <span class="fu">c</span>(<span class="st">&quot;CCND1&quot;</span>, <span class="st">&quot;MMP10&quot;</span>, <span class="st">&quot;CTTN&quot;</span>),  <span class="co"># Genes used for survival analysis (optional)</span></span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a>  <span class="at">heatmap_columns =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>, <span class="st">&quot;Ciclina_D1&quot;</span>),  </span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a>  <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>, <span class="st">&quot;Associated&quot;</span>, <span class="st">&quot;Independent&quot;</span>),  <span class="co"># Contrast for differential analysis</span></span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a>  <span class="at">pCutoff =</span> <span class="fl">5e-2</span>,  </span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a>  <span class="at">variable_01 =</span> <span class="st">&quot;Recurrence_01&quot;</span>,  <span class="co"># Variable for survival analysis</span></span>
<span id="cb3-32"><a href="#cb3-32" tabindex="-1"></a>  <span class="at">time =</span> <span class="st">&quot;Time_to_death_surv&quot;</span>,  <span class="co"># Time for survival analysis</span></span>
<span id="cb3-33"><a href="#cb3-33" tabindex="-1"></a>  <span class="at">DEA =</span> <span class="cn">TRUE</span>,  </span>
<span id="cb3-34"><a href="#cb3-34" tabindex="-1"></a>  <span class="at">remove_outliers =</span> <span class="cn">TRUE</span>,  </span>
<span id="cb3-35"><a href="#cb3-35" tabindex="-1"></a>  <span class="at">GSEA =</span> <span class="cn">FALSE</span>,  </span>
<span id="cb3-36"><a href="#cb3-36" tabindex="-1"></a>  <span class="at">generate_heatmap =</span> <span class="cn">TRUE</span>,  </span>
<span id="cb3-37"><a href="#cb3-37" tabindex="-1"></a>  <span class="at">TME =</span> <span class="cn">TRUE</span>,  </span>
<span id="cb3-38"><a href="#cb3-38" tabindex="-1"></a>  <span class="at">survival_analysis =</span> <span class="cn">FALSE</span>  <span class="co"># Skip survival analysis by setting this parameter to FALSE</span></span>
<span id="cb3-39"><a href="#cb3-39" tabindex="-1"></a>)</span></code></pre></div>
<p>In this example, survival analysis is skipped by setting
<code>survival_analysis = FALSE</code>and GSA by setting
<code>GSEA = FALSE</code>. The output will indicate that the survival
analysis step has been skipped.</p>
</div>
</div>
<div id="step-by-step-for-htg" class="section level3">
<h3>STEP BY STEP for HTG:</h3>
<p>What if you already had some analysis or you want to get more control
over it? Don’t worry we have the solution.Let’s go
<code>step-by-step</code>, starting with data import:</p>
<div id="load-data-htg_import_counts-for-htg" class="section level4">
<h4>Load data: HTG_import_counts() for HTG:</h4>
<p>You can upload data directly from HTG Edge Reveal using this format.
R needs a quadratic information structure, so maybe you will have to
edit your data to look like this on excel before importing:</p>
<table style="width:100%;">
<colgroup>
<col width="11%" />
<col width="11%" />
<col width="11%" />
<col width="11%" />
<col width="11%" />
<col width="11%" />
<col width="11%" />
<col width="11%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>id</strong></th>
<th>patient_1</th>
<th>patient_2</th>
<th>patient_3</th>
<th>patient_4</th>
<th>patient_5</th>
<th>patient_6</th>
<th>patient_7</th>
<th>patient_8</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Sample ID</strong></td>
<td>1</td>
<td>2</td>
<td>3</td>
<td>4</td>
<td>5</td>
<td>6</td>
<td>7</td>
<td>8</td>
</tr>
<tr class="even">
<td><strong>Well</strong></td>
<td>A1</td>
<td>B1</td>
<td>C1</td>
<td>D1</td>
<td>E1</td>
<td>F1</td>
<td>G1</td>
<td>H1</td>
</tr>
<tr class="odd">
<td><strong>Date Parsed</strong></td>
<td>14/08/2024</td>
<td>14/08/2024</td>
<td>14/08/2024</td>
<td>14/08/2024</td>
<td>14/08/2024</td>
<td>14/08/2024</td>
<td>14/08/2024</td>
<td>14/08/2024</td>
</tr>
<tr class="even">
<td><strong>Total Counts</strong></td>
<td>9650628</td>
<td>33950226</td>
<td>24680689</td>
<td>10605963</td>
<td>7395173</td>
<td>4057094</td>
<td>20049502</td>
<td>13607678</td>
</tr>
<tr class="odd">
<td><strong>A3GALT2</strong></td>
<td>0</td>
<td>0</td>
<td>164</td>
<td>19</td>
<td>46</td>
<td>1</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td><strong>A4GALT</strong></td>
<td>2</td>
<td>662</td>
<td>665</td>
<td>830</td>
<td>748</td>
<td>89</td>
<td>256</td>
<td>333</td>
</tr>
<tr class="odd">
<td><strong>A4GNT</strong></td>
<td>0</td>
<td>1041</td>
<td>103</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td><strong>AAAS</strong></td>
<td>0</td>
<td>368</td>
<td>70</td>
<td>634</td>
<td>707</td>
<td>425</td>
<td>587</td>
<td>872</td>
</tr>
<tr class="odd">
<td><strong>AACS</strong></td>
<td>1</td>
<td>0</td>
<td>562</td>
<td>728</td>
<td>250</td>
<td>209</td>
<td>365</td>
<td>445</td>
</tr>
<tr class="even">
<td><strong>AADAC</strong></td>
<td>0</td>
<td>382</td>
<td>0</td>
<td>5</td>
<td>144</td>
<td>0</td>
<td>0</td>
<td>1</td>
</tr>
</tbody>
</table>
<p>Or an alternative format:</p>
<table>
<colgroup>
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>id</strong></th>
<th>patient_10</th>
<th>patient_11</th>
<th>patient_5</th>
<th>patient_6</th>
<th>patient_4</th>
<th>patient_1</th>
<th>patient_2</th>
<th>patient_3</th>
<th>patient_7</th>
<th>patient_9</th>
<th>patient_8</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>A3GALT2</strong></td>
<td>0</td>
<td>0</td>
<td>164</td>
<td>19</td>
<td>46</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td><strong>A4GALT</strong></td>
<td>2</td>
<td>662</td>
<td>665</td>
<td>830</td>
<td>748</td>
<td>89</td>
<td>256</td>
<td>333</td>
<td>210</td>
<td>2</td>
<td>504</td>
</tr>
<tr class="odd">
<td><strong>A4GNT</strong></td>
<td>0</td>
<td>1041</td>
<td>103</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>0</td>
<td>0</td>
<td>84</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td><strong>AAAS</strong></td>
<td>0</td>
<td>368</td>
<td>70</td>
<td>634</td>
<td>707</td>
<td>425</td>
<td>587</td>
<td>872</td>
<td>542</td>
<td>0</td>
<td>370</td>
</tr>
<tr class="odd">
<td><strong>AACS</strong></td>
<td>1</td>
<td>0</td>
<td>562</td>
<td>728</td>
<td>250</td>
<td>209</td>
<td>365</td>
<td>445</td>
<td>62</td>
<td>46</td>
<td>785</td>
</tr>
<tr class="even">
<td><strong>AADAC</strong></td>
<td>0</td>
<td>382</td>
<td>0</td>
<td>5</td>
<td>144</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>260</td>
<td>683</td>
<td>77</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="do">## LOAD COUNTS DATA </span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>htg_data <span class="ot">&lt;-</span> <span class="fu">HTG_import_counts</span>(<span class="st">&quot;~/HTGAnalyzer/counts.xlsx&quot;</span>, <span class="st">&quot;HTG&quot;</span>)</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="fu">head</span>(htg_data)</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="do">## LOAD ANNOTDATA</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>annotdata<span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_excel</span>(<span class="st">&quot;~/HTGAnalyzer/annotdata.xlsx&quot;</span>)</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="fu">head</span>(annotdata)</span></code></pre></div>
<p><strong>NOTE:</strong> For the following functions, we’ll use
tutorial data from the package. If you want to try with your data,
replace <code>counts_data_tutorial</code> with
<code>htg_data</code>.</p>
</div>
<div id="htg_quant_to_qual-for-htg" class="section level4">
<h4>HTG_quant_to_qual() for HTG:</h4>
<p>If you need to transform a quantitative column to a qualitative one,
you can use this function.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">head</span>(AnnotData_tutorial)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="co">#&gt;                  id  HPV_status Ciclina_D1 FIGO_2021_STAGE Recurrence</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="co">#&gt; patient_1 patient_1  Associated         10            IIIB        yes</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co">#&gt; patient_2 patient_2  Associated         50              II         no</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#&gt; patient_3 patient_3  Associated         NA            IIIC         no</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#&gt; patient_4 patient_4  Associated          0              IB        yes</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#&gt; patient_5 patient_5 Independent         50              IB        yes</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co">#&gt; patient_6 patient_6 Independent         80              IB         no</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co">#&gt;           Recurrence_01 Time_to_death_surv Ciclina_D12</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="co">#&gt; patient_1             1               1287         low</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="co">#&gt; patient_2             0                510         low</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="co">#&gt; patient_3             0                762        &lt;NA&gt;</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="co">#&gt; patient_4             1               1164         low</span></span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a><span class="co">#&gt; patient_5             1               5844         low</span></span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a><span class="co">#&gt; patient_6             0               1436        high</span></span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a>AnnotData_tutorial <span class="ot">&lt;-</span> <span class="fu">HTG_quant_to_qual</span>(AnnotData_tutorial,  <span class="co"># Dataset to modify</span></span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a>                                        <span class="st">&quot;Ciclina_D1&quot;</span>,  <span class="co"># Column to transform</span></span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a>                                        <span class="dv">60</span>,  <span class="co"># Threshold to categorize the data</span></span>
<span id="cb5-20"><a href="#cb5-20" tabindex="-1"></a>                                        <span class="st">&quot;high&quot;</span>,  <span class="co"># Value for entries greater than 60</span></span>
<span id="cb5-21"><a href="#cb5-21" tabindex="-1"></a>                                        <span class="st">&quot;low&quot;</span>)  <span class="co"># Value for entries less than 60</span></span>
<span id="cb5-22"><a href="#cb5-22" tabindex="-1"></a><span class="co">#Last column will be the generated one </span></span>
<span id="cb5-23"><a href="#cb5-23" tabindex="-1"></a><span class="fu">head</span>(AnnotData_tutorial)</span>
<span id="cb5-24"><a href="#cb5-24" tabindex="-1"></a><span class="co">#&gt;                  id  HPV_status Ciclina_D1 FIGO_2021_STAGE Recurrence</span></span>
<span id="cb5-25"><a href="#cb5-25" tabindex="-1"></a><span class="co">#&gt; patient_1 patient_1  Associated         10            IIIB        yes</span></span>
<span id="cb5-26"><a href="#cb5-26" tabindex="-1"></a><span class="co">#&gt; patient_2 patient_2  Associated         50              II         no</span></span>
<span id="cb5-27"><a href="#cb5-27" tabindex="-1"></a><span class="co">#&gt; patient_3 patient_3  Associated         NA            IIIC         no</span></span>
<span id="cb5-28"><a href="#cb5-28" tabindex="-1"></a><span class="co">#&gt; patient_4 patient_4  Associated          0              IB        yes</span></span>
<span id="cb5-29"><a href="#cb5-29" tabindex="-1"></a><span class="co">#&gt; patient_5 patient_5 Independent         50              IB        yes</span></span>
<span id="cb5-30"><a href="#cb5-30" tabindex="-1"></a><span class="co">#&gt; patient_6 patient_6 Independent         80              IB         no</span></span>
<span id="cb5-31"><a href="#cb5-31" tabindex="-1"></a><span class="co">#&gt;           Recurrence_01 Time_to_death_surv Ciclina_D12</span></span>
<span id="cb5-32"><a href="#cb5-32" tabindex="-1"></a><span class="co">#&gt; patient_1             1               1287         low</span></span>
<span id="cb5-33"><a href="#cb5-33" tabindex="-1"></a><span class="co">#&gt; patient_2             0                510         low</span></span>
<span id="cb5-34"><a href="#cb5-34" tabindex="-1"></a><span class="co">#&gt; patient_3             0                762        &lt;NA&gt;</span></span>
<span id="cb5-35"><a href="#cb5-35" tabindex="-1"></a><span class="co">#&gt; patient_4             1               1164         low</span></span>
<span id="cb5-36"><a href="#cb5-36" tabindex="-1"></a><span class="co">#&gt; patient_5             1               5844         low</span></span>
<span id="cb5-37"><a href="#cb5-37" tabindex="-1"></a><span class="co">#&gt; patient_6             0               1436        high</span></span>
<span id="cb5-38"><a href="#cb5-38" tabindex="-1"></a><span class="fu">head</span>(AnnotData_tutorial<span class="sc">$</span>Ciclina_D12)</span>
<span id="cb5-39"><a href="#cb5-39" tabindex="-1"></a><span class="co">#&gt; [1] &quot;low&quot;  &quot;low&quot;  NA     &quot;low&quot;  &quot;low&quot;  &quot;high&quot;</span></span></code></pre></div>
</div>
<div id="htg_subset-for-htg" class="section level4">
<h4>HTG_subset() for HTG:</h4>
<p>Now that you have the counts data loaded, you can search for certain
genes of interest, look for specific probes, or filter results from DEA.
This function can also normalize the data by TPM (Transcripts Per
Million).</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co"># Subset rows with the prefix &quot;ERCC&quot;</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>cts_ERCC <span class="ot">&lt;-</span> <span class="fu">HTG_subset</span>(counts_data_tutorial, <span class="st">&quot;ERCC&quot;</span>)</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co">#&gt; [32mSubsetting without normalization...[0m</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co">#&gt; Dimensions: 101 11</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co"># Subset rows with the prefix &quot;POS&quot;</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>cts_POS <span class="ot">&lt;-</span> <span class="fu">HTG_subset</span>(counts_data_tutorial, <span class="st">&quot;POS&quot;</span>)</span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co">#&gt; [32mSubsetting without normalization...[0m</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt; Dimensions: 5 11</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a> <span class="co"># Subset rows with the prefix &quot;NC&quot;</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>cts_NC <span class="ot">&lt;-</span> <span class="fu">HTG_subset</span>(counts_data_tutorial, <span class="st">&quot;NC&quot;</span>)</span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="co">#&gt; [32mSubsetting without normalization...[0m</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="co">#&gt; Dimensions: 146 11</span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a> </span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="co"># Subset rows with the prefix &quot;GDNA&quot;</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a>cts_GDNA <span class="ot">&lt;-</span> <span class="fu">HTG_subset</span>(counts_data_tutorial, <span class="st">&quot;GDNA&quot;</span>)</span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a><span class="co">#&gt; [32mSubsetting without normalization...[0m</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a><span class="co">#&gt; Dimensions: 22 11</span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a></span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a><span class="co"># Subset rows with the prefix &quot;ZZZ3&quot;</span></span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a>cts_ZZZ3 <span class="ot">&lt;-</span> <span class="fu">HTG_subset</span>(counts_data_tutorial, <span class="st">&quot;ZZZ3&quot;</span>)</span>
<span id="cb6-23"><a href="#cb6-23" tabindex="-1"></a><span class="co">#&gt; [32mSubsetting without normalization...[0m</span></span>
<span id="cb6-24"><a href="#cb6-24" tabindex="-1"></a><span class="co">#&gt; Dimensions: 1 11</span></span>
<span id="cb6-25"><a href="#cb6-25" tabindex="-1"></a></span>
<span id="cb6-26"><a href="#cb6-26" tabindex="-1"></a><span class="co"># Subset rows with the prefix &quot;ZZZ3&quot; and normalize the data</span></span>
<span id="cb6-27"><a href="#cb6-27" tabindex="-1"></a>cts_ZZZ3 <span class="ot">&lt;-</span> <span class="fu">HTG_subset</span>(counts_data_tutorial, <span class="st">&quot;ZZZ3&quot;</span>, <span class="at">normalize =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-28"><a href="#cb6-28" tabindex="-1"></a><span class="co">#&gt; [32mNormalizing data using TPM (Transcripts Per Million)...[0m</span></span>
<span id="cb6-29"><a href="#cb6-29" tabindex="-1"></a><span class="co">#&gt; Dimensions: 1 11</span></span>
<span id="cb6-30"><a href="#cb6-30" tabindex="-1"></a></span>
<span id="cb6-31"><a href="#cb6-31" tabindex="-1"></a>resAAAS<span class="ot">&lt;-</span> <span class="fu">HTG_subset</span>(res_tutorial, <span class="st">&quot;AAAS&quot;</span>)</span>
<span id="cb6-32"><a href="#cb6-32" tabindex="-1"></a><span class="co">#&gt; [32mSubsetting without normalization...[0m</span></span>
<span id="cb6-33"><a href="#cb6-33" tabindex="-1"></a><span class="co">#&gt; Dimensions: 1 6</span></span></code></pre></div>
</div>
<div id="perform-a-qc-htg_qc-for-htg" class="section level4">
<h4>Perform a QC: HTG_QC() for HTG:</h4>
<p>Now, let’s perform quality control (QC) on the tutorial data. As you
may know, HTG uses specific probes to perform QC to detect outliers and
assess data quality. Default thresholds are provided, but feel free to
adjust them. This function will:</p>
<ol style="list-style-type: decimal">
<li><p>Detect outliers and save them in an object called
<code>outliers</code>.</p></li>
<li><p>Generate visualization plots.</p></li>
<li><p>Perform summary statistics and save them in a .csv file.</p></li>
<li><p>Generate a table with the QC results for further review.</p></li>
</ol>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>outliers <span class="ot">&lt;-</span> <span class="fu">HTG_QC</span>(counts_data_tutorial, <span class="at">pattern =</span> <span class="st">&quot;^NC-|^POS-|^GDNA-|^ERCC-&quot;</span>, </span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>                             <span class="at">threshold_superior_pos =</span> <span class="dv">5</span>,</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>                             <span class="at">threshold_line_pos =</span> <span class="dv">4</span>,</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>                             <span class="at">threshold_inferior_lib =</span> <span class="fl">5e+06</span>,</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>                             <span class="at">threshold_lib =</span> <span class="fl">7e+06</span>,</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>                             <span class="at">threshold_superior_nc =</span> <span class="fl">0.05</span>,</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>                             <span class="at">threshold_line_nc =</span> <span class="fl">0.045</span>,</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>                             <span class="at">threshold_superior_gdna =</span> <span class="fl">0.025</span>,</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>                             <span class="at">threshold_line_gdna =</span> <span class="fl">0.02</span>,</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>                             <span class="at">threshold_superior_ercc =</span> <span class="fl">0.03</span>,</span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>                             <span class="at">threshold_line_ercc =</span> <span class="fl">0.025</span>,</span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>                             <span class="at">threshold_inferior_median =</span> <span class="dv">3</span>,</span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>                             <span class="at">threshold_line_median =</span> <span class="dv">5</span>,</span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a>                             <span class="at">save_csv =</span> <span class="cn">TRUE</span>,</span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>                             <span class="at">csv_file =</span> <span class="st">&quot;QC_results.csv&quot;</span>)</span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a><span class="co">#&gt; [33mINITIATING DATA FILTERING...[0m</span></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a><span class="co">#&gt; [32mSummary statistics saved as &#39;summary_stats.csv&#39;[0m</span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a><span class="co">#&gt; [33mINITIATING QC PLOTS...[0m</span></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a><span class="co">#&gt;            ID total_POS total_NC total_GDNA total_ERCC    pos/gens   Q0</span></span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a><span class="co">#&gt; 1   patient_1     80307      696        172        256  0.64028783 PASS</span></span>
<span id="cb7-21"><a href="#cb7-21" tabindex="-1"></a><span class="co">#&gt; 2   patient_2      6381       29         37         15  0.04489742 PASS</span></span>
<span id="cb7-22"><a href="#cb7-22" tabindex="-1"></a><span class="co">#&gt; 3   patient_3      4418        9         24         11  0.02725260 PASS</span></span>
<span id="cb7-23"><a href="#cb7-23" tabindex="-1"></a><span class="co">#&gt; 4   patient_4     68866     3034        884       1355  0.51098424 PASS</span></span>
<span id="cb7-24"><a href="#cb7-24" tabindex="-1"></a><span class="co">#&gt; 5   patient_5    214604     2738       1598       1523  1.75876807 PASS</span></span>
<span id="cb7-25"><a href="#cb7-25" tabindex="-1"></a><span class="co">#&gt; 6   patient_6     31934      393        146        236  0.24849258 PASS</span></span>
<span id="cb7-26"><a href="#cb7-26" tabindex="-1"></a><span class="co">#&gt; 7   patient_7     88459     1760        844       1247  0.65164132 PASS</span></span>
<span id="cb7-27"><a href="#cb7-27" tabindex="-1"></a><span class="co">#&gt; 8   patient_8    168625      695        413       1001  1.25664410 PASS</span></span>
<span id="cb7-28"><a href="#cb7-28" tabindex="-1"></a><span class="co">#&gt; 9   patient_9    910985     1714       7202        103  7.73949173 FAIL</span></span>
<span id="cb7-29"><a href="#cb7-29" tabindex="-1"></a><span class="co">#&gt; 10 patient_10   3347486    13712       2372       3580 31.62646098 FAIL</span></span>
<span id="cb7-30"><a href="#cb7-30" tabindex="-1"></a><span class="co">#&gt; 11 patient_11     32239      274      17785        186  0.22045428 PASS</span></span>
<span id="cb7-31"><a href="#cb7-31" tabindex="-1"></a><span class="co">#&gt;    total_gens   Q1      nc/gens   Q2    gdna/gens   Q3    ERCC/gens   Q4 median</span></span>
<span id="cb7-32"><a href="#cb7-32" tabindex="-1"></a><span class="co">#&gt; 1    12542328 PASS 5.549209e-03 PASS 0.0013713563 PASS 2.041088e-03 PASS     69</span></span>
<span id="cb7-33"><a href="#cb7-33" tabindex="-1"></a><span class="co">#&gt; 2    14212399 PASS 2.040472e-04 PASS 0.0002603361 PASS 1.055416e-04 PASS    139</span></span>
<span id="cb7-34"><a href="#cb7-34" tabindex="-1"></a><span class="co">#&gt; 3    16211297 PASS 5.551684e-05 PASS 0.0001480449 PASS 6.785392e-05 PASS    189</span></span>
<span id="cb7-35"><a href="#cb7-35" tabindex="-1"></a><span class="co">#&gt; 4    13477128 PASS 2.251221e-02 PASS 0.0065592610 PASS 1.005407e-02 PASS    167</span></span>
<span id="cb7-36"><a href="#cb7-36" tabindex="-1"></a><span class="co">#&gt; 5    12201950 PASS 2.243904e-02 PASS 0.0130962674 PASS 1.248161e-02 PASS    131</span></span>
<span id="cb7-37"><a href="#cb7-37" tabindex="-1"></a><span class="co">#&gt; 6    12851088 PASS 3.058107e-03 PASS 0.0011360906 PASS 1.836420e-03 PASS    126</span></span>
<span id="cb7-38"><a href="#cb7-38" tabindex="-1"></a><span class="co">#&gt; 7    13574799 PASS 1.296520e-02 PASS 0.0062174033 PASS 9.186140e-03 PASS     73</span></span>
<span id="cb7-39"><a href="#cb7-39" tabindex="-1"></a><span class="co">#&gt; 8    13418676 PASS 5.179349e-03 PASS 0.0030777999 PASS 7.459752e-03 PASS     96</span></span>
<span id="cb7-40"><a href="#cb7-40" tabindex="-1"></a><span class="co">#&gt; 9    11770605 PASS 1.456170e-02 PASS 0.0611863196 FAIL 8.750612e-04 PASS      0</span></span>
<span id="cb7-41"><a href="#cb7-41" tabindex="-1"></a><span class="co">#&gt; 10   10584447 PASS 1.295486e-01 PASS 0.0224102402 FAIL 3.382321e-02 FAIL      0</span></span>
<span id="cb7-42"><a href="#cb7-42" tabindex="-1"></a><span class="co">#&gt; 11   14623894 PASS 1.873646e-03 PASS 0.1216160347 FAIL 1.271891e-03 PASS    503</span></span>
<span id="cb7-43"><a href="#cb7-43" tabindex="-1"></a><span class="co">#&gt;      Q5 min     max     mean QC_status</span></span>
<span id="cb7-44"><a href="#cb7-44" tabindex="-1"></a><span class="co">#&gt; 1  PASS   0  461925 646.7118      PASS</span></span>
<span id="cb7-45"><a href="#cb7-45" tabindex="-1"></a><span class="co">#&gt; 2  PASS   0  162327 732.8245      PASS</span></span>
<span id="cb7-46"><a href="#cb7-46" tabindex="-1"></a><span class="co">#&gt; 3  PASS   0  189344 835.8924      PASS</span></span>
<span id="cb7-47"><a href="#cb7-47" tabindex="-1"></a><span class="co">#&gt; 4  PASS   0  225483 694.9122      PASS</span></span>
<span id="cb7-48"><a href="#cb7-48" tabindex="-1"></a><span class="co">#&gt; 5  PASS   0  285334 629.1611      PASS</span></span>
<span id="cb7-49"><a href="#cb7-49" tabindex="-1"></a><span class="co">#&gt; 6  PASS   0  264344 662.6322      PASS</span></span>
<span id="cb7-50"><a href="#cb7-50" tabindex="-1"></a><span class="co">#&gt; 7  PASS   0  658315 699.9484      PASS</span></span>
<span id="cb7-51"><a href="#cb7-51" tabindex="-1"></a><span class="co">#&gt; 8  PASS   0  278813 691.8983      PASS</span></span>
<span id="cb7-52"><a href="#cb7-52" tabindex="-1"></a><span class="co">#&gt; 9  FAIL   0 2325356 606.9199      FAIL</span></span>
<span id="cb7-53"><a href="#cb7-53" tabindex="-1"></a><span class="co">#&gt; 10 FAIL   0 1638654 545.7588      FAIL</span></span>
<span id="cb7-54"><a href="#cb7-54" tabindex="-1"></a><span class="co">#&gt; 11 PASS   0 1527692 754.0422      FAIL</span></span>
<span id="cb7-55"><a href="#cb7-55" tabindex="-1"></a><span class="co">#&gt; [32mQC DATA SAVED AS &#39; QC_results.csv &#39;[0m</span></span>
<span id="cb7-56"><a href="#cb7-56" tabindex="-1"></a><span class="co">#&gt; &#39;data.frame&#39;:    11 obs. of  12 variables:</span></span>
<span id="cb7-57"><a href="#cb7-57" tabindex="-1"></a><span class="co">#&gt;  $ total_POS : num  3347486 32239 214604 31934 68866 ...</span></span>
<span id="cb7-58"><a href="#cb7-58" tabindex="-1"></a><span class="co">#&gt;  $ total_GDNA: num  2372 17785 1598 146 884 ...</span></span>
<span id="cb7-59"><a href="#cb7-59" tabindex="-1"></a><span class="co">#&gt;  $ total_gens: num  10584447 14623894 12201950 12851088 13477128 ...</span></span>
<span id="cb7-60"><a href="#cb7-60" tabindex="-1"></a><span class="co">#&gt;  $ total_NC  : num  13712 274 2738 393 3034 ...</span></span>
<span id="cb7-61"><a href="#cb7-61" tabindex="-1"></a><span class="co">#&gt;  $ total_ERCC: num  3580 186 1523 236 1355 ...</span></span>
<span id="cb7-62"><a href="#cb7-62" tabindex="-1"></a><span class="co">#&gt;  $ pos/gens  : num  31.626 0.22 1.759 0.248 0.511 ...</span></span>
<span id="cb7-63"><a href="#cb7-63" tabindex="-1"></a><span class="co">#&gt;  $ gdna/gens : num  0.02241 0.12162 0.0131 0.00114 0.00656 ...</span></span>
<span id="cb7-64"><a href="#cb7-64" tabindex="-1"></a><span class="co">#&gt;  $ nc/gens   : num  0.12955 0.00187 0.02244 0.00306 0.02251 ...</span></span>
<span id="cb7-65"><a href="#cb7-65" tabindex="-1"></a><span class="co">#&gt;  $ ERCC/gens : num  0.03382 0.00127 0.01248 0.00184 0.01005 ...</span></span>
<span id="cb7-66"><a href="#cb7-66" tabindex="-1"></a><span class="co">#&gt;  $ median    : num  0 503 131 126 167 69 139 189 73 0 ...</span></span>
<span id="cb7-67"><a href="#cb7-67" tabindex="-1"></a><span class="co">#&gt;  $ samples   : chr  &quot;patient_10&quot; &quot;patient_11&quot; &quot;patient_5&quot; &quot;patient_6&quot; ...</span></span>
<span id="cb7-68"><a href="#cb7-68" tabindex="-1"></a><span class="co">#&gt;  $ Size      : num  10584447 14623894 12201950 12851088 13477128 ...</span></span>
<span id="cb7-69"><a href="#cb7-69" tabindex="-1"></a><span class="co">#&gt; Warning: Ensembl will soon enforce the use of https.</span></span>
<span id="cb7-70"><a href="#cb7-70" tabindex="-1"></a><span class="co">#&gt; Ensure the &#39;host&#39; argument includes &quot;https://&quot;</span></span>
<span id="cb7-71"><a href="#cb7-71" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt;--- This function is being optimised and we strongly recommend that you should set `source` as `local`....</span></span>
<span id="cb7-72"><a href="#cb7-72" tabindex="-1"></a><span class="co">#&gt; Warning in IOBR::count2tpm(counts_data, idType = &quot;Symbol&quot;, org = &quot;hsa&quot;, :</span></span>
<span id="cb7-73"><a href="#cb7-73" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt;--- Omit 1304 genes of which length is not available !</span></span>
<span id="cb7-74"><a href="#cb7-74" tabindex="-1"></a><span class="co">#&gt;         Sample          Gene                TPM              LogTPM      </span></span>
<span id="cb7-75"><a href="#cb7-75" tabindex="-1"></a><span class="co">#&gt;  patient_10:17392   Length:191312      Min.   :  0.000   Min.   :0.0000  </span></span>
<span id="cb7-76"><a href="#cb7-76" tabindex="-1"></a><span class="co">#&gt;  patient_11:17392   Class :character   1st Qu.:  0.000   1st Qu.:0.0000  </span></span>
<span id="cb7-77"><a href="#cb7-77" tabindex="-1"></a><span class="co">#&gt;  patient_5 :17392   Mode  :character   Median :  1.384   Median :0.8689  </span></span>
<span id="cb7-78"><a href="#cb7-78" tabindex="-1"></a><span class="co">#&gt;  patient_6 :17392                      Mean   : 10.179   Mean   :1.3078  </span></span>
<span id="cb7-79"><a href="#cb7-79" tabindex="-1"></a><span class="co">#&gt;  patient_4 :17392                      3rd Qu.:  9.660   3rd Qu.:2.3665  </span></span>
<span id="cb7-80"><a href="#cb7-80" tabindex="-1"></a><span class="co">#&gt;  patient_1 :17392                      Max.   :227.599   Max.   :5.4320  </span></span>
<span id="cb7-81"><a href="#cb7-81" tabindex="-1"></a><span class="co">#&gt;  (Other)   :86960                                                        </span></span>
<span id="cb7-82"><a href="#cb7-82" tabindex="-1"></a><span class="co">#&gt;    Threshold    </span></span>
<span id="cb7-83"><a href="#cb7-83" tabindex="-1"></a><span class="co">#&gt;  Min.   :3.938  </span></span>
<span id="cb7-84"><a href="#cb7-84" tabindex="-1"></a><span class="co">#&gt;  1st Qu.:4.555  </span></span>
<span id="cb7-85"><a href="#cb7-85" tabindex="-1"></a><span class="co">#&gt;  Median :4.697  </span></span>
<span id="cb7-86"><a href="#cb7-86" tabindex="-1"></a><span class="co">#&gt;  Mean   :4.694  </span></span>
<span id="cb7-87"><a href="#cb7-87" tabindex="-1"></a><span class="co">#&gt;  3rd Qu.:4.878  </span></span>
<span id="cb7-88"><a href="#cb7-88" tabindex="-1"></a><span class="co">#&gt;  Max.   :5.432  </span></span>
<span id="cb7-89"><a href="#cb7-89" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-90"><a href="#cb7-90" tabindex="-1"></a><span class="co">#&gt; [32mViolin plot saved as &#39;QC_plots_violin_plot.pdf&#39;[0m</span></span>
<span id="cb7-91"><a href="#cb7-91" tabindex="-1"></a><span class="co">#&gt; [32mQC plots saved as &#39;plot_QC.pdf&#39;[0m</span></span>
<span id="cb7-92"><a href="#cb7-92" tabindex="-1"></a><span class="co">#&gt; [32mHeatmap PLOTS SAVED AS &#39;HTG_QC_heatmap.pdf&#39;[0m</span></span>
<span id="cb7-93"><a href="#cb7-93" tabindex="-1"></a><span class="co">#&gt; [32m                              ***[0m</span></span>
<span id="cb7-94"><a href="#cb7-94" tabindex="-1"></a><span class="co">#&gt; [32mThese are the samples plotted at least once in the heatmap:  [0m</span></span>
<span id="cb7-95"><a href="#cb7-95" tabindex="-1"></a><span class="co">#&gt; The number of samples that are outliers are: 3patient_10 patient_11 patient_9</span></span></code></pre></div>
</div>
<div id="htg_calculate_summary_stats-for-htg" class="section level4">
<h4>HTG_calculate_summary_stats() for HTG:</h4>
<p>If you only want to perform the summary statistics without full QC,
this function will generate summary stats and save them in a .csv file
as well as an R object.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>summary <span class="ot">&lt;-</span> <span class="fu">HTG_calculate_summary_stats</span>(counts_data_tutorial, <span class="at">pattern =</span> <span class="st">&quot;^NC-|^POS-|^GDNA-|^ERCC-&quot;</span>)</span></code></pre></div>
</div>
<div id="perform-all-the-analysis-htg_analysis-for-htg" class="section level4">
<h4>Perform all the analysis: HTG_analysis() for HTG:</h4>
<p>Now that you know which samples are outliers, you can perform any of
the analyses. This function will remove the outliers, perform DEA, TME,
survival analysis, and GSEA. If you want to perform each step
separately, you can use the functions <code>HTG_GSEA()</code>,
<code>HTG_DEA()</code>, <code>HTG_TME()</code>, and
<code>HTG_survival()</code>. Running the analyses individually offers
greater control, allowing you to adjust additional parameters. Refer to
each function’s section for detailed instructions on how they work.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>ALL_done <span class="ot">&lt;-</span> <span class="fu">HTG_analysis</span>(<span class="at">outliers =</span> outliers_tutorial, </span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>                         <span class="at">pattern =</span> <span class="st">&quot;^NC-|^POS-|^GDNA-|^ERCC-&quot;</span>, </span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>                         <span class="at">counts_data =</span> counts_data_tutorial, </span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>                         <span class="at">col_data =</span> AnnotData_tutorial, </span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>                         <span class="at">design_formula =</span> <span class="st">&quot;HPV_status&quot;</span>, </span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>                         <span class="at">percentage_gene =</span> <span class="fl">0.2</span>, </span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>                         <span class="at">threshold_gene =</span> <span class="dv">200</span>, </span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>                         <span class="at">threshold_subject =</span> <span class="dv">10</span>,</span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>                         <span class="at">genes_to_use =</span> <span class="fu">c</span>(<span class="st">&quot;CCND1&quot;</span>, <span class="st">&quot;MMP10&quot;</span>, <span class="st">&quot;CTTN&quot;</span>),</span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>                         <span class="at">heatmap_columns =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>,<span class="st">&quot;Ciclina_D1&quot;</span>),</span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>                         <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>, <span class="st">&quot;Associated&quot;</span>, <span class="st">&quot;Independent&quot;</span>), </span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>                         <span class="at">pCutoff =</span> <span class="fl">5e-2</span>, </span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a>                         <span class="at">variable_01 =</span> <span class="st">&quot;Recurrence_01&quot;</span>,</span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a>                         <span class="at">time =</span> <span class="st">&quot;Time_to_death_surv&quot;</span>, </span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a>                         <span class="at">DEA =</span> <span class="cn">TRUE</span>, </span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a>                         <span class="at">remove_outliers =</span> <span class="cn">TRUE</span>,</span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a>                         <span class="at">GSEA =</span> <span class="cn">FALSE</span>, </span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a>                         <span class="at">generate_heatmap =</span> <span class="cn">TRUE</span>, </span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a>                         <span class="at">TME =</span> <span class="cn">TRUE</span>, </span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a>                         <span class="at">survival_analysis =</span> <span class="cn">FALSE</span>)</span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a><span class="co">#&gt; [33mFILTERING THE COUNT DATA. DELETING THE PROVES...[0m</span></span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a><span class="co">#&gt; [33mREMOVING OUTLIERS...[0m</span></span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a><span class="co">#&gt; [33mSTARTING THE DIFERENTIAN EXPRESSION ANALYSIS.[0m</span></span>
<span id="cb9-24"><a href="#cb9-24" tabindex="-1"></a><span class="co">#&gt; [1] &quot;HPV_status&quot;</span></span>
<span id="cb9-25"><a href="#cb9-25" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb9-26"><a href="#cb9-26" tabindex="-1"></a><span class="co">#&gt; [32m[0m</span></span>
<span id="cb9-27"><a href="#cb9-27" tabindex="-1"></a><span class="co">#&gt; [32mBEFORE FILTERING[0m</span></span>
<span id="cb9-28"><a href="#cb9-28" tabindex="-1"></a><span class="co">#&gt; class: DESeqDataSet </span></span>
<span id="cb9-29"><a href="#cb9-29" tabindex="-1"></a><span class="co">#&gt; dim: 19394 8 </span></span>
<span id="cb9-30"><a href="#cb9-30" tabindex="-1"></a><span class="co">#&gt; metadata(1): version</span></span>
<span id="cb9-31"><a href="#cb9-31" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb9-32"><a href="#cb9-32" tabindex="-1"></a><span class="co">#&gt; rownames(19394): A3GALT2 A4GALT ... ZZEF1 ZZZ3</span></span>
<span id="cb9-33"><a href="#cb9-33" tabindex="-1"></a><span class="co">#&gt; rowData names(0):</span></span>
<span id="cb9-34"><a href="#cb9-34" tabindex="-1"></a><span class="co">#&gt; colnames(8): patient_1 patient_2 ... patient_7 patient_8</span></span>
<span id="cb9-35"><a href="#cb9-35" tabindex="-1"></a><span class="co">#&gt; colData names(8): id HPV_status ... Time_to_death_surv Ciclina_D12</span></span>
<span id="cb9-36"><a href="#cb9-36" tabindex="-1"></a><span class="co">#&gt; [32m[0m</span></span>
<span id="cb9-37"><a href="#cb9-37" tabindex="-1"></a><span class="co">#&gt; [32m [0m</span></span>
<span id="cb9-38"><a href="#cb9-38" tabindex="-1"></a><span class="co">#&gt; [32m[32mAFTER FILTERING[0m</span></span>
<span id="cb9-39"><a href="#cb9-39" tabindex="-1"></a><span class="co">#&gt; class: DESeqDataSet </span></span>
<span id="cb9-40"><a href="#cb9-40" tabindex="-1"></a><span class="co">#&gt; dim: 10293 8 </span></span>
<span id="cb9-41"><a href="#cb9-41" tabindex="-1"></a><span class="co">#&gt; metadata(1): version</span></span>
<span id="cb9-42"><a href="#cb9-42" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb9-43"><a href="#cb9-43" tabindex="-1"></a><span class="co">#&gt; rownames(10293): A4GALT AAAS ... ZZEF1 ZZZ3</span></span>
<span id="cb9-44"><a href="#cb9-44" tabindex="-1"></a><span class="co">#&gt; rowData names(0):</span></span>
<span id="cb9-45"><a href="#cb9-45" tabindex="-1"></a><span class="co">#&gt; colnames(8): patient_1 patient_2 ... patient_7 patient_8</span></span>
<span id="cb9-46"><a href="#cb9-46" tabindex="-1"></a><span class="co">#&gt; colData names(8): id HPV_status ... Time_to_death_surv Ciclina_D12</span></span>
<span id="cb9-47"><a href="#cb9-47" tabindex="-1"></a><span class="co">#&gt; [32m[32m [0m</span></span>
<span id="cb9-48"><a href="#cb9-48" tabindex="-1"></a><span class="co">#&gt; [32m[32mRESULTS OF VST NORMALIZATION[0m</span></span>
<span id="cb9-49"><a href="#cb9-49" tabindex="-1"></a><span class="co">#&gt; class: DESeqTransform </span></span>
<span id="cb9-50"><a href="#cb9-50" tabindex="-1"></a><span class="co">#&gt; dim: 10293 8 </span></span>
<span id="cb9-51"><a href="#cb9-51" tabindex="-1"></a><span class="co">#&gt; metadata(1): version</span></span>
<span id="cb9-52"><a href="#cb9-52" tabindex="-1"></a><span class="co">#&gt; assays(1): &#39;&#39;</span></span>
<span id="cb9-53"><a href="#cb9-53" tabindex="-1"></a><span class="co">#&gt; rownames(10293): A4GALT AAAS ... ZZEF1 ZZZ3</span></span>
<span id="cb9-54"><a href="#cb9-54" tabindex="-1"></a><span class="co">#&gt; rowData names(4): baseMean baseVar allZero dispFit</span></span>
<span id="cb9-55"><a href="#cb9-55" tabindex="-1"></a><span class="co">#&gt; colnames(8): patient_1 patient_2 ... patient_7 patient_8</span></span>
<span id="cb9-56"><a href="#cb9-56" tabindex="-1"></a><span class="co">#&gt; colData names(9): id HPV_status ... Ciclina_D12 sizeFactor</span></span>
<span id="cb9-57"><a href="#cb9-57" tabindex="-1"></a><span class="co">#&gt; [32m[32m [0m</span></span>
<span id="cb9-58"><a href="#cb9-58" tabindex="-1"></a><span class="co">#&gt; estimating size factors</span></span>
<span id="cb9-59"><a href="#cb9-59" tabindex="-1"></a><span class="co">#&gt; estimating dispersions</span></span>
<span id="cb9-60"><a href="#cb9-60" tabindex="-1"></a><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span id="cb9-61"><a href="#cb9-61" tabindex="-1"></a><span class="co">#&gt; mean-dispersion relationship</span></span>
<span id="cb9-62"><a href="#cb9-62" tabindex="-1"></a><span class="co">#&gt; final dispersion estimates</span></span>
<span id="cb9-63"><a href="#cb9-63" tabindex="-1"></a><span class="co">#&gt; fitting model and testing</span></span>
<span id="cb9-64"><a href="#cb9-64" tabindex="-1"></a><span class="co">#&gt; [32m[32mHEATMAP[0m</span></span>
<span id="cb9-65"><a href="#cb9-65" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING HEATMAP[0m</span></span>
<span id="cb9-66"><a href="#cb9-66" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING CONTRAST RESULTS[0m</span></span>
<span id="cb9-67"><a href="#cb9-67" tabindex="-1"></a><span class="co">#&gt; [32mSUMMARY OF RESULT OF THE CONTRAST[0m</span></span>
<span id="cb9-68"><a href="#cb9-68" tabindex="-1"></a><span class="co">#&gt; [1] &quot;DESeqResults object of length 6 with 2 metadata columns&quot;</span></span>
<span id="cb9-69"><a href="#cb9-69" tabindex="-1"></a><span class="co">#&gt; [32m [0m</span></span>
<span id="cb9-70"><a href="#cb9-70" tabindex="-1"></a><span class="co">#&gt; [32mRESULTS OF THE CONTRAST (TOP 10 p-adj)[0m</span></span>
<span id="cb9-71"><a href="#cb9-71" tabindex="-1"></a><span class="co">#&gt; log2 fold change (MLE): HPV_status Associated vs Independent </span></span>
<span id="cb9-72"><a href="#cb9-72" tabindex="-1"></a><span class="co">#&gt; Wald test p-value: HPV_status Associated vs Independent </span></span>
<span id="cb9-73"><a href="#cb9-73" tabindex="-1"></a><span class="co">#&gt; DataFrame with 10 rows and 6 columns</span></span>
<span id="cb9-74"><a href="#cb9-74" tabindex="-1"></a><span class="co">#&gt;          baseMean log2FoldChange     lfcSE      stat      pvalue        padj</span></span>
<span id="cb9-75"><a href="#cb9-75" tabindex="-1"></a><span class="co">#&gt;         &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span id="cb9-76"><a href="#cb9-76" tabindex="-1"></a><span class="co">#&gt; LCP1     4433.349        3.22176  0.331278   9.72524 2.35348e-22 2.07483e-18</span></span>
<span id="cb9-77"><a href="#cb9-77" tabindex="-1"></a><span class="co">#&gt; HLA-DMB  1781.568        2.64005  0.282567   9.34312 9.35407e-21 4.12328e-17</span></span>
<span id="cb9-78"><a href="#cb9-78" tabindex="-1"></a><span class="co">#&gt; LYPD3    2294.363       -7.58069  0.944092  -8.02962 9.77771e-16 2.87334e-12</span></span>
<span id="cb9-79"><a href="#cb9-79" tabindex="-1"></a><span class="co">#&gt; SLAMF7   1706.337        2.63203  0.371327   7.08817 1.35898e-12 2.99520e-09</span></span>
<span id="cb9-80"><a href="#cb9-80" tabindex="-1"></a><span class="co">#&gt; OMA1      606.110        2.20549  0.316706   6.96385 3.31100e-12 5.83796e-09</span></span>
<span id="cb9-81"><a href="#cb9-81" tabindex="-1"></a><span class="co">#&gt; VOPP1    1296.988        2.83390  0.408990   6.92901 4.23781e-12 6.22676e-09</span></span>
<span id="cb9-82"><a href="#cb9-82" tabindex="-1"></a><span class="co">#&gt; COL5A3    550.436       -3.18286  0.467776  -6.80423 1.01589e-11 1.27944e-08</span></span>
<span id="cb9-83"><a href="#cb9-83" tabindex="-1"></a><span class="co">#&gt; MMP10    1804.866      -10.42896  1.593830  -6.54333 6.01644e-11 6.63012e-08</span></span>
<span id="cb9-84"><a href="#cb9-84" tabindex="-1"></a><span class="co">#&gt; CDCA4     676.770       -2.44454  0.379031  -6.44945 1.12255e-10 1.09960e-07</span></span>
<span id="cb9-85"><a href="#cb9-85" tabindex="-1"></a><span class="co">#&gt; TACSTD2  1909.633       -8.54117  1.332003  -6.41228 1.43360e-10 1.26386e-07</span></span>
<span id="cb9-86"><a href="#cb9-86" tabindex="-1"></a><span class="co">#&gt; [32m [0m</span></span>
<span id="cb9-87"><a href="#cb9-87" tabindex="-1"></a><span class="co">#&gt; [32mRESULTS of DEA WILL BE SAVED IN .CSV IN YOUR CURRENT DIRECTORY[0m</span></span>
<span id="cb9-88"><a href="#cb9-88" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING POISSON DISTANCES PLOT[0m</span></span>
<span id="cb9-89"><a href="#cb9-89" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING VSD DISTANCE HEATMAP[0m</span></span>
<span id="cb9-90"><a href="#cb9-90" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING VSD BOXPLOT[0m</span></span>
<span id="cb9-91"><a href="#cb9-91" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING PCA PLOT[0m</span></span>
<span id="cb9-92"><a href="#cb9-92" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING VOLCANO PLOT[0m</span></span>
<span id="cb9-93"><a href="#cb9-93" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING MA PLOT[0m</span></span>
<span id="cb9-94"><a href="#cb9-94" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING DISPERSION PLOT[0m</span></span>
<span id="cb9-95"><a href="#cb9-95" tabindex="-1"></a><span class="co">#&gt; [33mSTARTING TME[0m</span></span>
<span id="cb9-96"><a href="#cb9-96" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Inicial gene number: 19394&quot;</span></span>
<span id="cb9-97"><a href="#cb9-97" tabindex="-1"></a><span class="co">#&gt; [32mTPM normalization performed and stored on tpm_counts.csv[0m</span></span>
<span id="cb9-98"><a href="#cb9-98" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt;--- This function is being optimised and we strongly recommend that you should set `source` as `local`....</span></span>
<span id="cb9-99"><a href="#cb9-99" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Number of genes omitted during TPM normalization due to their length not being available in Biomart:   1086&quot;</span></span>
<span id="cb9-100"><a href="#cb9-100" tabindex="-1"></a><span class="co">#&gt; [32mWe are going to use information from DEA[0m</span></span>
<span id="cb9-101"><a href="#cb9-101" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb9-102"><a href="#cb9-102" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt; Running epic</span></span>
<span id="cb9-103"><a href="#cb9-103" tabindex="-1"></a><span class="co">#&gt; Warning in (function (bulk, reference = NULL, mRNA_cell = NULL, mRNA_cell_sub = NULL, : The optimization didn&#39;t fully converge for some samples:</span></span>
<span id="cb9-104"><a href="#cb9-104" tabindex="-1"></a><span class="co">#&gt; patient_3</span></span>
<span id="cb9-105"><a href="#cb9-105" tabindex="-1"></a><span class="co">#&gt;  - check fit.gof for the convergeCode and convergeMessage</span></span>
<span id="cb9-106"><a href="#cb9-106" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb9-107"><a href="#cb9-107" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt; Running quantiseq</span></span>
<span id="cb9-108"><a href="#cb9-108" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb9-109"><a href="#cb9-109" tabindex="-1"></a><span class="co">#&gt; Running quanTIseq deconvolution module</span></span>
<span id="cb9-110"><a href="#cb9-110" tabindex="-1"></a><span class="co">#&gt; Gene expression normalization and re-annotation (arrays: FALSE)</span></span>
<span id="cb9-111"><a href="#cb9-111" tabindex="-1"></a><span class="co">#&gt; Removing 17 noisy genes</span></span>
<span id="cb9-112"><a href="#cb9-112" tabindex="-1"></a><span class="co">#&gt; Removing 15 genes with high expression in tumors</span></span>
<span id="cb9-113"><a href="#cb9-113" tabindex="-1"></a><span class="co">#&gt; Signature genes found in data set: 132/138 (95.65%)</span></span>
<span id="cb9-114"><a href="#cb9-114" tabindex="-1"></a><span class="co">#&gt; Mixture deconvolution (method: lsei)</span></span>
<span id="cb9-115"><a href="#cb9-115" tabindex="-1"></a><span class="co">#&gt; Deconvolution successful!</span></span>
<span id="cb9-116"><a href="#cb9-116" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb9-117"><a href="#cb9-117" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt; Running xcell</span></span>
<span id="cb9-118"><a href="#cb9-118" tabindex="-1"></a><span class="co">#&gt; [32mresults of the TME will be stored in imm_epic.csv, imm_qti.csv and imm_xcell.csv [0m</span></span>
<span id="cb9-119"><a href="#cb9-119" tabindex="-1"></a><span class="co">#&gt; [32mAre they in the same order?[0m</span></span>
<span id="cb9-120"><a href="#cb9-120" tabindex="-1"></a><span class="co">#&gt; [32mHave to be true.[0m</span></span>
<span id="cb9-121"><a href="#cb9-121" tabindex="-1"></a><span class="co">#&gt; [32mEPIC[0m</span></span>
<span id="cb9-122"><a href="#cb9-122" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb9-123"><a href="#cb9-123" tabindex="-1"></a><span class="co">#&gt; [32mqti[0m</span></span>
<span id="cb9-124"><a href="#cb9-124" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb9-125"><a href="#cb9-125" tabindex="-1"></a><span class="co">#&gt; [32mxcell[0m</span></span>
<span id="cb9-126"><a href="#cb9-126" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb9-127"><a href="#cb9-127" tabindex="-1"></a><span class="co">#&gt; [33mApply the filtering and normality test for each dataset[0m</span></span>
<span id="cb9-128"><a href="#cb9-128" tabindex="-1"></a><span class="co">#&gt; [32mChecking if All Values in Each Column Are Equal in dataset: quanTIseq [0m</span></span>
<span id="cb9-129"><a href="#cb9-129" tabindex="-1"></a><span class="co">#&gt; [33mIn dataset quanTIseq the following columns have the same value in all rows and will be excluded from the analysis:[0m</span></span>
<span id="cb9-130"><a href="#cb9-130" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Neutrophil&quot;</span></span>
<span id="cb9-131"><a href="#cb9-131" tabindex="-1"></a><span class="co">#&gt; [32mChecking if All Values in Each Column Are Equal in dataset: EPIC [0m</span></span>
<span id="cb9-132"><a href="#cb9-132" tabindex="-1"></a><span class="co">#&gt; [33mIn dataset EPIC the following columns have the same value in all rows and will be excluded from the analysis:[0m</span></span>
<span id="cb9-133"><a href="#cb9-133" tabindex="-1"></a><span class="co">#&gt; character(0)</span></span>
<span id="cb9-134"><a href="#cb9-134" tabindex="-1"></a><span class="co">#&gt; [32mChecking if All Values in Each Column Are Equal in dataset: xcell [0m</span></span>
<span id="cb9-135"><a href="#cb9-135" tabindex="-1"></a><span class="co">#&gt; [33mIn dataset xcell the following columns have the same value in all rows and will be excluded from the analysis:[0m</span></span>
<span id="cb9-136"><a href="#cb9-136" tabindex="-1"></a><span class="co">#&gt; character(0)</span></span>
<span id="cb9-137"><a href="#cb9-137" tabindex="-1"></a><span class="co">#&gt; [32mPerforming Shapiro-Wilk test for quanTIseq[0m</span></span>
<span id="cb9-138"><a href="#cb9-138" tabindex="-1"></a><span class="co">#&gt; # A tibble: 10 × 3</span></span>
<span id="cb9-139"><a href="#cb9-139" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                    shapiro_test    p.value</span></span>
<span id="cb9-140"><a href="#cb9-140" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                        &lt;list&gt;            &lt;dbl&gt;</span></span>
<span id="cb9-141"><a href="#cb9-141" tabindex="-1"></a><span class="co">#&gt;  1 B_cell                       &lt;htest&gt;      0.247     </span></span>
<span id="cb9-142"><a href="#cb9-142" tabindex="-1"></a><span class="co">#&gt;  2 Macrophage_M1                &lt;htest&gt;      0.00361   </span></span>
<span id="cb9-143"><a href="#cb9-143" tabindex="-1"></a><span class="co">#&gt;  3 Macrophage_M2                &lt;htest&gt;      0.266     </span></span>
<span id="cb9-144"><a href="#cb9-144" tabindex="-1"></a><span class="co">#&gt;  4 Monocyte                     &lt;htest&gt;      0.00178   </span></span>
<span id="cb9-145"><a href="#cb9-145" tabindex="-1"></a><span class="co">#&gt;  5 Myeloid_dendritic_cell       &lt;htest&gt;      0.00000105</span></span>
<span id="cb9-146"><a href="#cb9-146" tabindex="-1"></a><span class="co">#&gt;  6 NK_cell                      &lt;htest&gt;      0.498     </span></span>
<span id="cb9-147"><a href="#cb9-147" tabindex="-1"></a><span class="co">#&gt;  7 T_cell_CD4+_(non-regulatory) &lt;htest&gt;      0.000202  </span></span>
<span id="cb9-148"><a href="#cb9-148" tabindex="-1"></a><span class="co">#&gt;  8 T_cell_CD8+                  &lt;htest&gt;      0.0212    </span></span>
<span id="cb9-149"><a href="#cb9-149" tabindex="-1"></a><span class="co">#&gt;  9 T_cell_regulatory_(Tregs)    &lt;htest&gt;      0.0284    </span></span>
<span id="cb9-150"><a href="#cb9-150" tabindex="-1"></a><span class="co">#&gt; 10 uncharacterized_cell         &lt;htest&gt;      0.162     </span></span>
<span id="cb9-151"><a href="#cb9-151" tabindex="-1"></a><span class="co">#&gt; [32mPerforming parametric tests for quanTIseq[0m</span></span>
<span id="cb9-152"><a href="#cb9-152" tabindex="-1"></a><span class="co">#&gt; # A tibble: 10 × 3</span></span>
<span id="cb9-153"><a href="#cb9-153" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                    Test    p.value</span></span>
<span id="cb9-154"><a href="#cb9-154" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                        &lt;chr&gt;     &lt;dbl&gt;</span></span>
<span id="cb9-155"><a href="#cb9-155" tabindex="-1"></a><span class="co">#&gt;  1 B_cell                       t-test 0.0117  </span></span>
<span id="cb9-156"><a href="#cb9-156" tabindex="-1"></a><span class="co">#&gt;  2 Macrophage_M1                t-test 0.538   </span></span>
<span id="cb9-157"><a href="#cb9-157" tabindex="-1"></a><span class="co">#&gt;  3 Macrophage_M2                t-test 0.138   </span></span>
<span id="cb9-158"><a href="#cb9-158" tabindex="-1"></a><span class="co">#&gt;  4 Monocyte                     t-test 0.139   </span></span>
<span id="cb9-159"><a href="#cb9-159" tabindex="-1"></a><span class="co">#&gt;  5 NK_cell                      t-test 0.887   </span></span>
<span id="cb9-160"><a href="#cb9-160" tabindex="-1"></a><span class="co">#&gt;  6 T_cell_CD4+_(non-regulatory) t-test 0.393   </span></span>
<span id="cb9-161"><a href="#cb9-161" tabindex="-1"></a><span class="co">#&gt;  7 T_cell_CD8+                  t-test 0.0735  </span></span>
<span id="cb9-162"><a href="#cb9-162" tabindex="-1"></a><span class="co">#&gt;  8 T_cell_regulatory_(Tregs)    t-test 0.345   </span></span>
<span id="cb9-163"><a href="#cb9-163" tabindex="-1"></a><span class="co">#&gt;  9 Myeloid_dendritic_cell       t-test 0.391   </span></span>
<span id="cb9-164"><a href="#cb9-164" tabindex="-1"></a><span class="co">#&gt; 10 uncharacterized_cell         t-test 0.000286</span></span>
<span id="cb9-165"><a href="#cb9-165" tabindex="-1"></a><span class="co">#&gt; [32mPerforming Shapiro-Wilk test for EPIC[0m</span></span>
<span id="cb9-166"><a href="#cb9-166" tabindex="-1"></a><span class="co">#&gt; # A tibble: 8 × 3</span></span>
<span id="cb9-167"><a href="#cb9-167" tabindex="-1"></a><span class="co">#&gt;   Cell_Type                    shapiro_test    p.value</span></span>
<span id="cb9-168"><a href="#cb9-168" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                        &lt;list&gt;            &lt;dbl&gt;</span></span>
<span id="cb9-169"><a href="#cb9-169" tabindex="-1"></a><span class="co">#&gt; 1 B_cell                       &lt;htest&gt;      0.0225    </span></span>
<span id="cb9-170"><a href="#cb9-170" tabindex="-1"></a><span class="co">#&gt; 2 Cancer_associated_fibroblast &lt;htest&gt;      0.382     </span></span>
<span id="cb9-171"><a href="#cb9-171" tabindex="-1"></a><span class="co">#&gt; 3 Endothelial_cell             &lt;htest&gt;      0.292     </span></span>
<span id="cb9-172"><a href="#cb9-172" tabindex="-1"></a><span class="co">#&gt; 4 Macrophage                   &lt;htest&gt;      0.00703   </span></span>
<span id="cb9-173"><a href="#cb9-173" tabindex="-1"></a><span class="co">#&gt; 5 NK_cell                      &lt;htest&gt;      0.000335  </span></span>
<span id="cb9-174"><a href="#cb9-174" tabindex="-1"></a><span class="co">#&gt; 6 T_cell_CD4+                  &lt;htest&gt;      0.0902    </span></span>
<span id="cb9-175"><a href="#cb9-175" tabindex="-1"></a><span class="co">#&gt; 7 T_cell_CD8+                  &lt;htest&gt;      0.00000114</span></span>
<span id="cb9-176"><a href="#cb9-176" tabindex="-1"></a><span class="co">#&gt; 8 uncharacterized_cell         &lt;htest&gt;      0.116     </span></span>
<span id="cb9-177"><a href="#cb9-177" tabindex="-1"></a><span class="co">#&gt; [32mPerforming parametric tests for EPIC[0m</span></span>
<span id="cb9-178"><a href="#cb9-178" tabindex="-1"></a><span class="co">#&gt; # A tibble: 8 × 3</span></span>
<span id="cb9-179"><a href="#cb9-179" tabindex="-1"></a><span class="co">#&gt;   Cell_Type                    Test   p.value</span></span>
<span id="cb9-180"><a href="#cb9-180" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                        &lt;chr&gt;    &lt;dbl&gt;</span></span>
<span id="cb9-181"><a href="#cb9-181" tabindex="-1"></a><span class="co">#&gt; 1 B_cell                       t-test  0.0806</span></span>
<span id="cb9-182"><a href="#cb9-182" tabindex="-1"></a><span class="co">#&gt; 2 Cancer_associated_fibroblast t-test  0.453 </span></span>
<span id="cb9-183"><a href="#cb9-183" tabindex="-1"></a><span class="co">#&gt; 3 T_cell_CD4+                  t-test  0.761 </span></span>
<span id="cb9-184"><a href="#cb9-184" tabindex="-1"></a><span class="co">#&gt; 4 T_cell_CD8+                  t-test  0.389 </span></span>
<span id="cb9-185"><a href="#cb9-185" tabindex="-1"></a><span class="co">#&gt; 5 Endothelial_cell             t-test  0.590 </span></span>
<span id="cb9-186"><a href="#cb9-186" tabindex="-1"></a><span class="co">#&gt; 6 Macrophage                   t-test  0.837 </span></span>
<span id="cb9-187"><a href="#cb9-187" tabindex="-1"></a><span class="co">#&gt; 7 NK_cell                      t-test  0.745 </span></span>
<span id="cb9-188"><a href="#cb9-188" tabindex="-1"></a><span class="co">#&gt; 8 uncharacterized_cell         t-test  0.528 </span></span>
<span id="cb9-189"><a href="#cb9-189" tabindex="-1"></a><span class="co">#&gt; [32mPerforming Shapiro-Wilk test for xcell[0m</span></span>
<span id="cb9-190"><a href="#cb9-190" tabindex="-1"></a><span class="co">#&gt; # A tibble: 39 × 3</span></span>
<span id="cb9-191"><a href="#cb9-191" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                    shapiro_test    p.value</span></span>
<span id="cb9-192"><a href="#cb9-192" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                        &lt;list&gt;            &lt;dbl&gt;</span></span>
<span id="cb9-193"><a href="#cb9-193" tabindex="-1"></a><span class="co">#&gt;  1 B_cell                       &lt;htest&gt;      0.494     </span></span>
<span id="cb9-194"><a href="#cb9-194" tabindex="-1"></a><span class="co">#&gt;  2 B_cell_memory                &lt;htest&gt;      0.180     </span></span>
<span id="cb9-195"><a href="#cb9-195" tabindex="-1"></a><span class="co">#&gt;  3 B_cell_naive                 &lt;htest&gt;      0.108     </span></span>
<span id="cb9-196"><a href="#cb9-196" tabindex="-1"></a><span class="co">#&gt;  4 B_cell_plasma                &lt;htest&gt;      0.0104    </span></span>
<span id="cb9-197"><a href="#cb9-197" tabindex="-1"></a><span class="co">#&gt;  5 Cancer_associated_fibroblast &lt;htest&gt;      0.128     </span></span>
<span id="cb9-198"><a href="#cb9-198" tabindex="-1"></a><span class="co">#&gt;  6 Class-switched_memory_B_cell &lt;htest&gt;      0.200     </span></span>
<span id="cb9-199"><a href="#cb9-199" tabindex="-1"></a><span class="co">#&gt;  7 Common_lymphoid_progenitor   &lt;htest&gt;      0.0145    </span></span>
<span id="cb9-200"><a href="#cb9-200" tabindex="-1"></a><span class="co">#&gt;  8 Common_myeloid_progenitor    &lt;htest&gt;      0.00000811</span></span>
<span id="cb9-201"><a href="#cb9-201" tabindex="-1"></a><span class="co">#&gt;  9 Endothelial_cell             &lt;htest&gt;      0.00539   </span></span>
<span id="cb9-202"><a href="#cb9-202" tabindex="-1"></a><span class="co">#&gt; 10 Eosinophil                   &lt;htest&gt;      0.0443    </span></span>
<span id="cb9-203"><a href="#cb9-203" tabindex="-1"></a><span class="co">#&gt; # ℹ 29 more rows</span></span>
<span id="cb9-204"><a href="#cb9-204" tabindex="-1"></a><span class="co">#&gt; [32mPerforming parametric tests for xcell[0m</span></span>
<span id="cb9-205"><a href="#cb9-205" tabindex="-1"></a><span class="co">#&gt; # A tibble: 39 × 3</span></span>
<span id="cb9-206"><a href="#cb9-206" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                        Test   p.value</span></span>
<span id="cb9-207"><a href="#cb9-207" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                            &lt;chr&gt;    &lt;dbl&gt;</span></span>
<span id="cb9-208"><a href="#cb9-208" tabindex="-1"></a><span class="co">#&gt;  1 Myeloid_dendritic_cell_activated t-test  0.198 </span></span>
<span id="cb9-209"><a href="#cb9-209" tabindex="-1"></a><span class="co">#&gt;  2 B_cell                           t-test  0.0484</span></span>
<span id="cb9-210"><a href="#cb9-210" tabindex="-1"></a><span class="co">#&gt;  3 T_cell_CD4+_memory               t-test  0.220 </span></span>
<span id="cb9-211"><a href="#cb9-211" tabindex="-1"></a><span class="co">#&gt;  4 T_cell_CD4+_naive                t-test  0.185 </span></span>
<span id="cb9-212"><a href="#cb9-212" tabindex="-1"></a><span class="co">#&gt;  5 T_cell_CD4+_(non-regulatory)     t-test  0.391 </span></span>
<span id="cb9-213"><a href="#cb9-213" tabindex="-1"></a><span class="co">#&gt;  6 T_cell_CD4+_central_memory       t-test  0.776 </span></span>
<span id="cb9-214"><a href="#cb9-214" tabindex="-1"></a><span class="co">#&gt;  7 T_cell_CD4+_effector_memory      t-test  0.182 </span></span>
<span id="cb9-215"><a href="#cb9-215" tabindex="-1"></a><span class="co">#&gt;  8 T_cell_CD8+_naive                t-test  0.171 </span></span>
<span id="cb9-216"><a href="#cb9-216" tabindex="-1"></a><span class="co">#&gt;  9 T_cell_CD8+                      t-test  0.0164</span></span>
<span id="cb9-217"><a href="#cb9-217" tabindex="-1"></a><span class="co">#&gt; 10 T_cell_CD8+_central_memory       t-test  0.0146</span></span>
<span id="cb9-218"><a href="#cb9-218" tabindex="-1"></a><span class="co">#&gt; # ℹ 29 more rows</span></span>
<span id="cb9-219"><a href="#cb9-219" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING HEATMAPS[0m</span></span>
<span id="cb9-220"><a href="#cb9-220" tabindex="-1"></a><span class="co">#&gt; [32mHeatmap of qti, EPIC and xcell will be stored on plots_TME_heatmap.pdf[0m</span></span>
<span id="cb9-221"><a href="#cb9-221" tabindex="-1"></a><span class="co">#&gt; [32mSkipping survival analysis.[0m</span></span></code></pre></div>
</div>
<div id="differentia-expression-analysis-htg_dea-for-htg" class="section level4">
<h4>Differentia expression analysis: HTG_DEA() for HTG:</h4>
<p>If you want more control, such as applying extra filtering or
performing LFC shrinkage, <code>HTG_DEA()</code> will provide more
options.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="do">### DEA:</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">HTG_DEA</span>(<span class="at">outliers=</span> outliers_tutorial, </span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>                   <span class="at">counts_data=</span> counts_data_tutorial, </span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>                   AnnotData_tutorial, </span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>                   <span class="at">design_formula =</span> <span class="st">&quot;HPV_status&quot;</span>, </span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>                   <span class="at">heatmap_columns =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>, <span class="st">&quot;Recurrence&quot;</span>),</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>                   <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>, <span class="st">&quot;Associated&quot;</span>, <span class="st">&quot;Independent&quot;</span>), </span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>                   <span class="at">pattern =</span> <span class="st">&quot;^NC-|^POS-|^GDNA-|^ERCC-&quot;</span>, </span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>                   <span class="at">remove_outliers =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a>                   <span class="at">percentage_gene =</span> <span class="fl">0.2</span>, </span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a>                   <span class="at">threshold_gene =</span> <span class="dv">200</span>,</span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a>                   <span class="at">threshold_subject =</span> <span class="dv">10</span>,</span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a>                   <span class="at">pCutoff =</span> <span class="fl">5e-2</span>, </span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a>                   <span class="at">apply_filtering =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a>                   <span class="at">apply_lfc_shrinkage =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a>                   <span class="at">extract_shrinkage =</span> <span class="cn">FALSE</span>)</span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a><span class="co">#&gt; [33mINITIATING DATA FILTERING...[0m</span></span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a><span class="co">#&gt; [33mREMOVING SPECIFIED OUTLIERS...[0m</span></span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a><span class="co">#&gt; [33mPREPARING DATA FOR DESeq2 ANALYSIS...[0m</span></span>
<span id="cb10-20"><a href="#cb10-20" tabindex="-1"></a><span class="co">#&gt; [33mCREATING DESeqDataSet OBJECT...[0m</span></span>
<span id="cb10-21"><a href="#cb10-21" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb10-22"><a href="#cb10-22" tabindex="-1"></a><span class="co">#&gt; [33mAPPLYING GENE FILTERING BASED ON THRESHOLDS...[0m</span></span>
<span id="cb10-23"><a href="#cb10-23" tabindex="-1"></a><span class="co">#&gt; [33mPERFORMING DESeq2 ANALYSIS...[0m</span></span>
<span id="cb10-24"><a href="#cb10-24" tabindex="-1"></a><span class="co">#&gt; estimating size factors</span></span>
<span id="cb10-25"><a href="#cb10-25" tabindex="-1"></a><span class="co">#&gt; estimating dispersions</span></span>
<span id="cb10-26"><a href="#cb10-26" tabindex="-1"></a><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span id="cb10-27"><a href="#cb10-27" tabindex="-1"></a><span class="co">#&gt; mean-dispersion relationship</span></span>
<span id="cb10-28"><a href="#cb10-28" tabindex="-1"></a><span class="co">#&gt; final dispersion estimates</span></span>
<span id="cb10-29"><a href="#cb10-29" tabindex="-1"></a><span class="co">#&gt; fitting model and testing</span></span>
<span id="cb10-30"><a href="#cb10-30" tabindex="-1"></a><span class="co">#&gt; [32mResults saved as &#39;results_HTG_DEA.csv&#39;[0m</span></span>
<span id="cb10-31"><a href="#cb10-31" tabindex="-1"></a><span class="co">#&gt; [32mDisplaying top 10 differential expression results...[0m</span></span>
<span id="cb10-32"><a href="#cb10-32" tabindex="-1"></a><span class="co">#&gt; log2 fold change (MLE): HPV_status Associated vs Independent </span></span>
<span id="cb10-33"><a href="#cb10-33" tabindex="-1"></a><span class="co">#&gt; Wald test p-value: HPV_status Associated vs Independent </span></span>
<span id="cb10-34"><a href="#cb10-34" tabindex="-1"></a><span class="co">#&gt; DataFrame with 10 rows and 6 columns</span></span>
<span id="cb10-35"><a href="#cb10-35" tabindex="-1"></a><span class="co">#&gt;          baseMean log2FoldChange     lfcSE      stat      pvalue        padj</span></span>
<span id="cb10-36"><a href="#cb10-36" tabindex="-1"></a><span class="co">#&gt;         &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span id="cb10-37"><a href="#cb10-37" tabindex="-1"></a><span class="co">#&gt; LCP1     4433.349        3.22176  0.331278   9.72524 2.35348e-22 2.07483e-18</span></span>
<span id="cb10-38"><a href="#cb10-38" tabindex="-1"></a><span class="co">#&gt; HLA-DMB  1781.568        2.64005  0.282567   9.34312 9.35407e-21 4.12328e-17</span></span>
<span id="cb10-39"><a href="#cb10-39" tabindex="-1"></a><span class="co">#&gt; LYPD3    2294.363       -7.58069  0.944092  -8.02962 9.77771e-16 2.87334e-12</span></span>
<span id="cb10-40"><a href="#cb10-40" tabindex="-1"></a><span class="co">#&gt; SLAMF7   1706.337        2.63203  0.371327   7.08817 1.35898e-12 2.99520e-09</span></span>
<span id="cb10-41"><a href="#cb10-41" tabindex="-1"></a><span class="co">#&gt; OMA1      606.110        2.20549  0.316706   6.96385 3.31100e-12 5.83796e-09</span></span>
<span id="cb10-42"><a href="#cb10-42" tabindex="-1"></a><span class="co">#&gt; VOPP1    1296.988        2.83390  0.408990   6.92901 4.23781e-12 6.22676e-09</span></span>
<span id="cb10-43"><a href="#cb10-43" tabindex="-1"></a><span class="co">#&gt; COL5A3    550.436       -3.18286  0.467776  -6.80423 1.01589e-11 1.27944e-08</span></span>
<span id="cb10-44"><a href="#cb10-44" tabindex="-1"></a><span class="co">#&gt; MMP10    1804.866      -10.42896  1.593830  -6.54333 6.01644e-11 6.63012e-08</span></span>
<span id="cb10-45"><a href="#cb10-45" tabindex="-1"></a><span class="co">#&gt; CDCA4     676.770       -2.44454  0.379031  -6.44945 1.12255e-10 1.09960e-07</span></span>
<span id="cb10-46"><a href="#cb10-46" tabindex="-1"></a><span class="co">#&gt; TACSTD2  1909.633       -8.54117  1.332003  -6.41228 1.43360e-10 1.26386e-07</span></span>
<span id="cb10-47"><a href="#cb10-47" tabindex="-1"></a><span class="co">#&gt; [33mAPPLYING LOG-FOLD CHANGE SHRINKAGE...[0m</span></span>
<span id="cb10-48"><a href="#cb10-48" tabindex="-1"></a><span class="co">#&gt; using &#39;apeglm&#39; for LFC shrinkage. If used in published research, please cite:</span></span>
<span id="cb10-49"><a href="#cb10-49" tabindex="-1"></a><span class="co">#&gt;     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for</span></span>
<span id="cb10-50"><a href="#cb10-50" tabindex="-1"></a><span class="co">#&gt;     sequence count data: removing the noise and preserving large differences.</span></span>
<span id="cb10-51"><a href="#cb10-51" tabindex="-1"></a><span class="co">#&gt;     Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</span></span>
<span id="cb10-52"><a href="#cb10-52" tabindex="-1"></a><span class="co">#&gt; log2 fold change (MAP): HPV status Independent vs Associated </span></span>
<span id="cb10-53"><a href="#cb10-53" tabindex="-1"></a><span class="co">#&gt; Wald test p-value: HPV status Independent vs Associated </span></span>
<span id="cb10-54"><a href="#cb10-54" tabindex="-1"></a><span class="co">#&gt; DataFrame with 10 rows and 5 columns</span></span>
<span id="cb10-55"><a href="#cb10-55" tabindex="-1"></a><span class="co">#&gt;          baseMean log2FoldChange     lfcSE      pvalue        padj</span></span>
<span id="cb10-56"><a href="#cb10-56" tabindex="-1"></a><span class="co">#&gt;         &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span id="cb10-57"><a href="#cb10-57" tabindex="-1"></a><span class="co">#&gt; LCP1     4433.349       -3.14866  0.334146 2.35348e-22 2.07483e-18</span></span>
<span id="cb10-58"><a href="#cb10-58" tabindex="-1"></a><span class="co">#&gt; HLA-DMB  1781.568       -2.58470  0.285422 9.35407e-21 4.12328e-17</span></span>
<span id="cb10-59"><a href="#cb10-59" tabindex="-1"></a><span class="co">#&gt; LYPD3    2294.363        7.34425  0.961843 9.77771e-16 2.87334e-12</span></span>
<span id="cb10-60"><a href="#cb10-60" tabindex="-1"></a><span class="co">#&gt; SLAMF7   1706.337       -2.52504  0.377241 1.35898e-12 2.99520e-09</span></span>
<span id="cb10-61"><a href="#cb10-61" tabindex="-1"></a><span class="co">#&gt; OMA1      606.110       -2.12104  0.321592 3.31100e-12 5.83796e-09</span></span>
<span id="cb10-62"><a href="#cb10-62" tabindex="-1"></a><span class="co">#&gt; VOPP1    1296.988       -2.71818  0.416609 4.23781e-12 6.22676e-09</span></span>
<span id="cb10-63"><a href="#cb10-63" tabindex="-1"></a><span class="co">#&gt; COL5A3    550.436        3.04905  0.477775 1.01589e-11 1.27944e-08</span></span>
<span id="cb10-64"><a href="#cb10-64" tabindex="-1"></a><span class="co">#&gt; MMP10    1804.866        9.93535  1.640901 6.01644e-11 6.63012e-08</span></span>
<span id="cb10-65"><a href="#cb10-65" tabindex="-1"></a><span class="co">#&gt; CDCA4     676.770        2.33824  0.386402 1.12255e-10 1.09960e-07</span></span>
<span id="cb10-66"><a href="#cb10-66" tabindex="-1"></a><span class="co">#&gt; TACSTD2  1909.633        8.11419  1.378398 1.43360e-10 1.26386e-07</span></span>
<span id="cb10-67"><a href="#cb10-67" tabindex="-1"></a><span class="co">#&gt; [32mSaving plots as &#39;DEA_plots.pdf&#39;[0m</span></span></code></pre></div>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAB71BMVEUAAAAAAA0AACgADQ0ADSgADVEAKIENAAANAA0NACgNDQANDSgNDVENKCgNKFENKIENUYENUbwoAAAoAA0oACgoDQAoDQ0oDSgoDVEoKA0oUYEoUbwogYEogbwogf9FdbRJebZOfbhRDQBRDQ1RDShRKA1RgYFRgbxRvP9SgrtXhr1ci79gj8JllMRpmMZunclzoct3ps58qtCAr9KBKACBKA2BUQ2BUSiBgVGBvFGBvLyBvP+B/7yB//+Fs9WKuNeOvNmTwNuYw92ZmZmcxt+hyuGmzeKr0OSw0+a01ui52em8UQ28USi8gSi8gVG8gYG8vIG8vLy8/4G8//++3OvD4O3I4+/M5vDR6fLW7PTXMCfZNSrbOy3b7/bdQDDfRjPg8/fh8/TiTDbj9PHkUTnl9e3mVzzn9eroXT/p9ubrYkLr9+PtaEXt+N/vbUjv+Nzw+djxc0vy+tX0eU70+9L2flH2+874hFT4/Mv6ilf6/cf8j1r8lF38mWH8nmT8o2f8qGv8rW78snH8/cT9t3X9vHj9wXv9xn/9y4L90IX91Yn92oz+34/+4ZL+45X+5Zj+55v+6Z7+66H+7aP+76b+8Kn+8qz+9K/+9rL++LX++rf+/Lr+/r3+/sD/gSj/vFH/vIH/vLz//4H//7z///9vPxuUAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAKCUlEQVR4nO3diZvkRBkG8JqFmVbAhRlcL9xWB9FR8e5dlXW9oqA2oNjqKt73jfctIt4KonjgSbd25w+1ksrxVvL1VKqTmtl03vfZZ2enNlWV/LpSyUzSaRUzx0ad9gpc7SGQIwRyhECOeADdPVLq+ksNFpyrM5drhfc+s17WvtFnKLVzdoPqydqILdbSHGiqkjRpU+p5zdq0a/RCWlvtjH2rp991DLQ8UDfO/nug9n1XZn1Z20a17vk4vmekdmee1ZvZpPEB0psxSVZmlbxyZ2fxYrTzonPq2qOF/utSssBzb1V6g033q9eN1M4tee1J8krv6aF9TqlrX9xNo6tDAztN97G8ZV1606E6c4f5cimvYapnfZi1MUp5Pd3T+duUuqG6vzYG0h2r6+8ot1Zv1GKU/uOaUfqd7kEZhrRns1A+NCb5/6QZd9KopjwqVq9oOWlS77RvNF8u5zWg+u4MgIp6eU/V0dh8DkrXfOd8tmIXdOu6ZG+mx3ny15nLuofdS6uJ2jmap/+nF1qMipGcro1eZG+mFynHd5tG56oEKlvWQLr4vuxLUaOofqGYfuam/axe3lN13/M4iq0ujMx+Ev/v9c9WZlvGZjWzvsbpmo6T78zsWw6WbJWSTTILtm8UgcqWtUzSWvalqGH2p6yPEqisZ3qqT05+50HvPacbXN2WHXqyl6fYlqyvfWtbFmZfgeNGskhWummjaT3cxcqWs5kp+2IBFX0gUF6vNZCpmqyUVn/OfVNhW+wXO9t8Gyh/vbB0g0bTLLOj31zPYtYIsoCyGtlwMX0EGUF6k26crS7qBqfJweEwnU8r26LLLxbThbolnirHHNSuUXOYv3dkzSU2UFHD+GZ9rJmDWu5i0/J4smZvOOYollbay9sYd9QonCgWLdtAMR7Fij7M2qQtFvU6mIPuvk4ffZOTEP3jwdn3pftJZVv0iQScc+B5UHoKsjernge1bDRtLvtRA86DECivkVbP+zBrc0/1PKiDSfq4WAenq7pRrxDIEQI5wt8HOUIgR04CSIVM8JUP3UHgPgh0im2fTAft+zjd/a8PQKfVcWdtBO6DQMGqEyhsx521EbgPAgWrTqCwHXfWRuA+CBSsOoHCdtxZG4H7IFCw6gQK23FnbQTug0DBqhMobMedtRG4DwIFq06gsB131kbgPggUrDqBwnbcWRuB+yBQsOoECttxZ20E7oNAwaoTKGzHnbURuI/tA+r4postBOq2T99V7PZOmG0E6qKR9hVbtEmgbvskULeLE2ij5QnkWJ5AjuUJ5FieQI7lCeRYnkCO5QnkWP7qB9r8R8ihAG3cO4E2aa6T7kP0vnGbBHK0SSBHmwRytEkgR5sehS1/57/9QJs037bRjXvrrE0COdokkKNNAjnaJJCjzSCHtpbdN0lvgBoXEshRSCBHIYEchQRyFBLIUUggR2EboMVTaw8qJxBkeVB/kjuBykzV00KOoKgvMat7P8YUvWcWdBeL7hcS/VtI9Ech0V1CojcLid4tJHpSSHRFyHqgwHNQz4A+giEQgbyBPoohEIG8gT6OIVBDICGDBfoEhkAE8gb6NIZAgYGcV8uT9Azos5jWQE0WItB2AX0eQyACeQN9EUMgAnkDfRlDIAJ5A30VQyACeQM9gCHQVQHUl5jV/SbmZIDEwSIOq98Kib4nJPqakOg/QqIPCYluF0KgZkDfxhCIQN5A38UQqBHQ6rD4GNrFKP+Q6GTb3b8t3EKg72NM0WQv+ZNkebAfT/dKINe2DwNoefNRfuUw+ZJ8O2SgH2LiigqBHEDLg3E8NZ8RPFigH2OqQMkkfdPhsIF+gqkBxXGLXWzddbG+A8EknXttCCQv1DOgBzGmCA7z2mqy8WF+a4GWB+mJ4mQ/jufKOlEcItBPMY5tJxCB6kAPYQhEIG+gn2EIRCBvoIcxBAoN1OTXZ6d9NadxzOr+HNMWqF5FAhIvB4qDRRxWHxQSvVNI9DEh8gh6nhACNQP6JYZABPIG+hWGQATyBvo1hkAE8gb6DYZABPIGstaIQCcLtOayT7+AfofpFmhNCLRdQL/HmCL7/qCdIwJVgPD+oHE8PdP8yuo2Aj2CMSqb3x80DCBQWR2O47m5tDpYoEcxVaBkOvK59Dw4oPQexYHPQX/AVIE4B0lAnKQtoMcwpqg8zK8OuYsJQHB/UJATxb7ErK71Oh6zWR0Cie+IE6+XioNFHFYPC4keFBLdKUS+MEugRkB/whCoDvQ4hkAE8gb6M4ZABPIG+guGQATyBvorhkAE8gb6GyYIUO26GIHqNTE9A3oCQyACeQP9HUMgAnkD/QNDIAJ5A/0TQyACeQP9C0OgRkDl/UHz9GeD/UEDWW9rM0XlhcMUqe2FwxpQX7IOyH40RXIPVcdA4kMhxddQvF4qDhZxWD0gJHqtkOgLQjIg6/2KaYn9cJNp69tfthtoZR6OM2Ag65mmdaBiXyOQDDTdg81Ua28O93k0Xs+AfoFJS6xJerJfARrcCKoD4WG+mIKGC/QjjCmC+4PKoxmBSqC1mzlIoB9gCHRqQHjZp19A1k+HAYHK7wm0XUDfwRCIQN5A38IQiEDeQN/AEIhA3kBfxxCoDmStEYEI5A30FUwYoMqvFU/7ak7jnBRQUT0HEj/2S3wopPiOOPFyoDhYxGElXrYUx69Z3S9hCEQgbyBrXiIQgbyBPochEIG8gT6DMUXw/CD9T3N7EIEACC4cTnZni6fgkxeGB/QpTFoCl57hI34IVABVHm4ydKBPYqpA892LSll3mG0MBNfFtglIz9DznS7mIEjPgKzOa0C7s/wGDwIVQJXnByHQ2k862GKgD2NMER7m9/E24HiII0gAgvuDkhPFcpIeJJD1JKZjNotABJKBrmAIRCBvIOudNAQikDfQBzAnA9SXnBrQk0Lk5z+KheI74sTrpeJgEYfVE0IyoPdjCEQgbyBrqiQQgbyB3oUhEIG8gaybkwhEIG8gqzaBQgM1eY9vz4CsftoDHf/faQi0XUDvwBCoDvR2DIEaAdmfL6bKDz8aJNDbMKYILhzOi+cIEagAwkdTFE+mGC7QWzFpCX6+2K3WDVQEqgItb74OP7pmkEBvwVSBkvsTl0/3uHlhaEBJ4OOzBglkvWurEIGbEytAfg9TEoH6knVAcJhP7g1a3IDnQY40AroiJLpdiPyxX+JFPvEdceKS4mARh5VZ3TdhsjFT3h80Vfk5I4FKoI03fSuBrGmNQATyBnoNhkAE8gZ6NYZABPIGehWGQATyBnolpj1Qk+tiQwZqkp4BvQJDIAJ5A70cQyACeQNZa0QgAnkDvQxDIAJ5A70UQyACeQO9BHMyQH3JWiC4P6h8vkl3QD3LCzCmyPp8sYnXdbEtTB3IuvQ8v2boI+j5mLQEb15YvfANBDoOaLo/+DnIOoSmJXgD1bNmgweqB4Am42I2+j/SKVUwyGIkFgAAAABJRU5ErkJggg==" /><!-- --><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAB71BMVEUAAAAAAA0AACgADQ0ADSgADVEAKIENAAANAA0NACgNDQANDSgNDVENKCgNKFENKIENUYENUbwoAAAoAA0oACgoDQAoDQ0oDSgoDVEoKA0oUYEoUbwogYEogbwogf9FdbRJebZOfbhRDQBRDQ1RDShRKA1RgYFRgbxRvP9SgrtXhr1ci79gj8JllMRpmMZunclzoct3ps58qtCAr9KBKACBKA2BUQ2BUSiBgVGBvFGBvLyBvP+B/7yB//+Fs9WKuNeOvNmTwNuYw92ZmZmcxt+hyuGmzeKr0OSw0+a01ui52em8UQ28USi8gSi8gVG8gYG8vIG8vLy8/4G8//++3OvD4O3I4+/M5vDR6fLW7PTXMCfZNSrbOy3b7/bdQDDfRjPg8/fh8/TiTDbj9PHkUTnl9e3mVzzn9eroXT/p9ubrYkLr9+PtaEXt+N/vbUjv+Nzw+djxc0vy+tX0eU70+9L2flH2+874hFT4/Mv6ilf6/cf8j1r8lF38mWH8nmT8o2f8qGv8rW78snH8/cT9t3X9vHj9wXv9xn/9y4L90IX91Yn92oz+34/+4ZL+45X+5Zj+55v+6Z7+66H+7aP+76b+8Kn+8qz+9K/+9rL++LX++rf+/Lr+/r3+/sD/gSj/vFH/vIH/vLz//4H//7z///9vPxuUAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAKCUlEQVR4nO3diZvkRBkG8JqFmVbAhRlcL9xWB9FR8e5dlXW9oqA2oNjqKt73jfctIt4KonjgSbd25w+1ksrxVvL1VKqTmtl03vfZZ2enNlWV/LpSyUzSaRUzx0ad9gpc7SGQIwRyhECOeADdPVLq+ksNFpyrM5drhfc+s17WvtFnKLVzdoPqydqILdbSHGiqkjRpU+p5zdq0a/RCWlvtjH2rp991DLQ8UDfO/nug9n1XZn1Z20a17vk4vmekdmee1ZvZpPEB0psxSVZmlbxyZ2fxYrTzonPq2qOF/utSssBzb1V6g033q9eN1M4tee1J8krv6aF9TqlrX9xNo6tDAztN97G8ZV1606E6c4f5cimvYapnfZi1MUp5Pd3T+duUuqG6vzYG0h2r6+8ot1Zv1GKU/uOaUfqd7kEZhrRns1A+NCb5/6QZd9KopjwqVq9oOWlS77RvNF8u5zWg+u4MgIp6eU/V0dh8DkrXfOd8tmIXdOu6ZG+mx3ny15nLuofdS6uJ2jmap/+nF1qMipGcro1eZG+mFynHd5tG56oEKlvWQLr4vuxLUaOofqGYfuam/axe3lN13/M4iq0ujMx+Ev/v9c9WZlvGZjWzvsbpmo6T78zsWw6WbJWSTTILtm8UgcqWtUzSWvalqGH2p6yPEqisZ3qqT05+50HvPacbXN2WHXqyl6fYlqyvfWtbFmZfgeNGskhWummjaT3cxcqWs5kp+2IBFX0gUF6vNZCpmqyUVn/OfVNhW+wXO9t8Gyh/vbB0g0bTLLOj31zPYtYIsoCyGtlwMX0EGUF6k26crS7qBqfJweEwnU8r26LLLxbThbolnirHHNSuUXOYv3dkzSU2UFHD+GZ9rJmDWu5i0/J4smZvOOYollbay9sYd9QonCgWLdtAMR7Fij7M2qQtFvU6mIPuvk4ffZOTEP3jwdn3pftJZVv0iQScc+B5UHoKsjernge1bDRtLvtRA86DECivkVbP+zBrc0/1PKiDSfq4WAenq7pRrxDIEQI5wt8HOUIgR04CSIVM8JUP3UHgPgh0im2fTAft+zjd/a8PQKfVcWdtBO6DQMGqEyhsx521EbgPAgWrTqCwHXfWRuA+CBSsOoHCdtxZG4H7IFCw6gQK23FnbQTug0DBqhMobMedtRG4DwIFq06gsB131kbgPggUrDqBwnbcWRuB+yBQsOoECttxZ20E7oNAwaoTKGzHnbURuI/tA+r4postBOq2T99V7PZOmG0E6qKR9hVbtEmgbvskULeLE2ij5QnkWJ5AjuUJ5FieQI7lCeRYnkCO5QnkWP7qB9r8R8ihAG3cO4E2aa6T7kP0vnGbBHK0SSBHmwRytEkgR5sehS1/57/9QJs037bRjXvrrE0COdokkKNNAjnaJJCjzSCHtpbdN0lvgBoXEshRSCBHIYEchQRyFBLIUUggR2EboMVTaw8qJxBkeVB/kjuBykzV00KOoKgvMat7P8YUvWcWdBeL7hcS/VtI9Ech0V1CojcLid4tJHpSSHRFyHqgwHNQz4A+giEQgbyBPoohEIG8gT6OIVBDICGDBfoEhkAE8gb6NIZAgYGcV8uT9Azos5jWQE0WItB2AX0eQyACeQN9EUMgAnkDfRlDIAJ5A30VQyACeQM9gCHQVQHUl5jV/SbmZIDEwSIOq98Kib4nJPqakOg/QqIPCYluF0KgZkDfxhCIQN5A38UQqBHQ6rD4GNrFKP+Q6GTb3b8t3EKg72NM0WQv+ZNkebAfT/dKINe2DwNoefNRfuUw+ZJ8O2SgH2LiigqBHEDLg3E8NZ8RPFigH2OqQMkkfdPhsIF+gqkBxXGLXWzddbG+A8EknXttCCQv1DOgBzGmCA7z2mqy8WF+a4GWB+mJ4mQ/jufKOlEcItBPMY5tJxCB6kAPYQhEIG+gn2EIRCBvoIcxBAoN1OTXZ6d9NadxzOr+HNMWqF5FAhIvB4qDRRxWHxQSvVNI9DEh8gh6nhACNQP6JYZABPIG+hWGQATyBvo1hkAE8gb6DYZABPIGstaIQCcLtOayT7+AfofpFmhNCLRdQL/HmCL7/qCdIwJVgPD+oHE8PdP8yuo2Aj2CMSqb3x80DCBQWR2O47m5tDpYoEcxVaBkOvK59Dw4oPQexYHPQX/AVIE4B0lAnKQtoMcwpqg8zK8OuYsJQHB/UJATxb7ErK71Oh6zWR0Cie+IE6+XioNFHFYPC4keFBLdKUS+MEugRkB/whCoDvQ4hkAE8gb6M4ZABPIG+guGQATyBvorhkAE8gb6GyYIUO26GIHqNTE9A3oCQyACeQP9HUMgAnkD/QNDIAJ5A/0TQyACeQP9C0OgRkDl/UHz9GeD/UEDWW9rM0XlhcMUqe2FwxpQX7IOyH40RXIPVcdA4kMhxddQvF4qDhZxWD0gJHqtkOgLQjIg6/2KaYn9cJNp69tfthtoZR6OM2Ag65mmdaBiXyOQDDTdg81Ua28O93k0Xs+AfoFJS6xJerJfARrcCKoD4WG+mIKGC/QjjCmC+4PKoxmBSqC1mzlIoB9gCHRqQHjZp19A1k+HAYHK7wm0XUDfwRCIQN5A38IQiEDeQN/AEIhA3kBfxxCoDmStEYEI5A30FUwYoMqvFU/7ak7jnBRQUT0HEj/2S3wopPiOOPFyoDhYxGElXrYUx69Z3S9hCEQgbyBrXiIQgbyBPochEIG8gT6DMUXw/CD9T3N7EIEACC4cTnZni6fgkxeGB/QpTFoCl57hI34IVABVHm4ydKBPYqpA892LSll3mG0MBNfFtglIz9DznS7mIEjPgKzOa0C7s/wGDwIVQJXnByHQ2k862GKgD2NMER7m9/E24HiII0gAgvuDkhPFcpIeJJD1JKZjNotABJKBrmAIRCBvIOudNAQikDfQBzAnA9SXnBrQk0Lk5z+KheI74sTrpeJgEYfVE0IyoPdjCEQgbyBrqiQQgbyB3oUhEIG8gaybkwhEIG8gqzaBQgM1eY9vz4CsftoDHf/faQi0XUDvwBCoDvR2DIEaAdmfL6bKDz8aJNDbMKYILhzOi+cIEagAwkdTFE+mGC7QWzFpCX6+2K3WDVQEqgItb74OP7pmkEBvwVSBkvsTl0/3uHlhaEBJ4OOzBglkvWurEIGbEytAfg9TEoH6knVAcJhP7g1a3IDnQY40AroiJLpdiPyxX+JFPvEdceKS4mARh5VZ3TdhsjFT3h80Vfk5I4FKoI03fSuBrGmNQATyBnoNhkAE8gZ6NYZABPIGehWGQATyBnolpj1Qk+tiQwZqkp4BvQJDIAJ5A70cQyACeQNZa0QgAnkDvQxDIAJ5A70UQyACeQO9BHMyQH3JWiC4P6h8vkl3QD3LCzCmyPp8sYnXdbEtTB3IuvQ8v2boI+j5mLQEb15YvfANBDoOaLo/+DnIOoSmJXgD1bNmgweqB4Am42I2+j/SKVUwyGIkFgAAAABJRU5ErkJggg==" /><!-- --><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAAxlBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZpAAZrY6AAA6ADo6AGY6OgA6OmY6OpA6ZmY6ZpA6ZrY6kLY6kNtHk69mAABmADpmOgBmOjpmZmZmkLZmkNtmtttmtv+QOgCQOjqQZgCQZjqQkGaQkLaQtpCQttuQtv+Q29uQ2/+2ZgC2Zjq2ZpC2kDq2kGa227a229u22/+2/7a2/9u2///bkDrbkGbbtmbbtpDb27bb29vb2//b/7bb////tmb/25D/27b//7b//9v///+MIMhKAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAL2ElEQVR4nO2de3/bthWG6cSetLrdOtXN1sZu0y2rmW2tu7JN1oqOxe//pYYrCZIAzyEhUJT5vn/4Jx8dXPgQN4IAlFXQoLJTZ2DpAiBCAEQIgAgBECEAIgRAhACIEAARAiBCAEQIgAgBECEAIgRAhACIEAARAiBCAEQIgAgBECEAIgRAhACIEAARAiBCAEQIgAgBECEAIgRAhACIUAygw112ca8+PW6zq311+M91ll38+UFa8qzWpvb/18+jon+3zbLPJ+Ytz16YxAqdC5OvVgKc/ESVIJH2rvkgeOms3FZeQB9fvRgFqJSBdxOz1gNk8uWIl58oQE83suCooiTSqnMi0/UAavLMU2HL5xR5AHWj4+Unrg3KdaKihm3UP6/FfRHlyNwqwW3j+ip24rL/8Sp78VC9fyX+//ReXcLF9++y7FLl/xdbTXN7Te//mmUvv9prRxn2h5ts8/5apvbL1oQ6iPAXn+/1R2F86wJSnz680reqTtbkx8lHAkClrgOFYmLyJGjtBgFJXe3L5q4WWe+ziMkCMhZZVE3Y329MgGtbXp+05fLnqq7oXUCmuDfJmvw4+UgASCQqUtc1TGb/4qsfnW/bgAxA4XX1UP1PfPeHvYEpTJcPuh0TEW721a+ZNstc6/a/MBYVVuLYSafstfkil0VSRyYueLOXBaoLSH1wk81N3mtDAkDyNtza6M29u3y7JwDZxvKD7PU21qTqqfhzaRlrQKZc5qaVk2F1WdB/Vdri486iyG2R8gDSxcQmW9dCa0gBSN4v2xJV1a/XdUl3AakKcLW3gLTz4U3dhGuTLDwGct3i3EvDVf2PCas8zV8FSPyxFcWm2WuDbHxNstrFMaQAJO/UT6YvU5f94bttndgQIEni8u8fbjqARN+rr/Sfxqyt6k7cBgGVTT9l/QOA3GRtFasNKQDJOvY33VLX9bjO2xAg7f3UAyQQfXet2+RRJchWW+vfB6RaSjdZ5eIakgBSpVvlQCRy8bXI2/ttF1DN0gFU2nZn1wUkA36rW5x+G+QH5F5fqA2Sw49NK1nl6RqSAFKNxsbkpDM07AAqTD9uS9DVXmfaBSTy+5e9rGe2BHV6MT8gPQT7aPtx0Yvlbi/mDBR7ybqGJIBU+rcNK3cE0gFU2oHivevcqWLv3CHRxX1zgXocFABkxkHq27yTi9ajhpusys9/HUMaQI/bBojqxXQX5AEkByeXD24vdvm2aGqOqScqjk8fqqa/UyPpr6tqAFD18Y2o6vpx9PDvbfbydehh1UlW58cxpAH03AVAhACIEAARAiBCAEQIgAgBECEAIgRAhACIEAARAiBCAEQIgAgBECEAIgRAhACIEAARAiBCAEQIgAgBECEAIgRAhACIEAARAiBCAEQIgAgdGVB2NkoD6PETtcyoCK5MP5sCmQbQ041ah1XIxY9+QusGVNqNETu5vM67qG/VgMpsV9Yr+RYLiNnGJGqDGkBFq4qNb/uSqpMLX+YSAypDWwFbeTgZNE+KXVPyEnS4q/dnHCUfR9USAAW6MQCqAT1ubz0OywDk0ZyANJrSuzQ9IaDjtmhJS1Au98jfeVvptCXmmLjTpGmKjdwv4qtgcwMK9ZOMXCQCNHd0jNiZtpnaoNHRce/5tNhH2M4F0PGTXCegEQVtWYD0fJDatkc/aoSiZyaZtqClnA863KkNyKyneQagEV3R0rt5Mx+kB4oFa6AYkYvzA9SaD2I+i0XkggloaitHu4x0VHIA5alLENNtWY10A6hstdLBCbPj1p0zAlQG9pmvfRzUzCgGTnKYv5tfJKCCfdLFysZBBlAReJRn5mN697Twbr6qJ8zCZ/etehxUGUD2MAyMgyZo7eOg0dGtbRw0Orq1jYNGR7e2cdDo6NY2DjILqJwPE/LxjMdBZgGV8yEquuFgZwiotCenlVnoyK9Vj4PqCbP2zNnU6CYEW34330x3tAFhPsgoBGhUPp7zOGhOQGc5DlogIKbOHNAz6eYnAIrIBQARwc5vHFTNC4jptqxGenR0axsHjY5ubeOg0dGtbRw0Orq1jYPMNFDJ3lC3snGQmQaSK1+Ou6FuAqDJeyzDKcY5VvU0kF5Dng+uMPNlf+CSqCv3mb5oK+QWSDQFIDsNxFlh5s0+85LooJFuqQBV9tXzH++r0FBxNKBTuKUGpJufViPUL8EANLxGcdWAUMUoQMdspJ8jIFY3v2ZAnIHiugExzu5YKyB2dABERAdARHQzAKKfUs4OEPMpNMINgM4E0MRe7LiAfCksBFCufl3Su9q+Tv+4l3Tcpio1IHUCVVUMnv5yIkC88pgaEOtpHoBKd0dLP336KmcC5HFLDcg8zXu3/GCgKKUbaQAaIJRlL376EjOKg/IvJQegWkQ3zwV0is4uNaCnm02olx89kvZFQUNb+qOG/PHyCY8aEfJcOdPtnB5WB5L05aIVJFz2Wp9XC4jpBkCEGwARbosBJDeGDx8TONRjzQvoBA+r6qjk0k/oyAVypHjjCO3KjnN8NvRZ28NvVhcvACK0lCp2pMZlnBsnaNpGurfKNVjt1wlIHl8WOAIGgKpxxwSuEhBn0n4o+mcPaMyJ5IvVUtqgxSptLyZPJOcddrtYnfphdSj6Z98GjYkOgCbkA4CI6AGIiP7ZAyrNYxfnp2sWq+QlSL0cO150sys5IP+jGABZhc5zRRtk1DmwHfNBHelliozo1goosHQBgKxy79qXTnTed2J9G9PN9wNsMW7DFzjB0VWwhqEX0wqsIp8a3SmUFlCwCQIgrcAocWp0p9DJHlbPRicCxIl+Ed08WwBECIAIARAhACIEQIQAiBAAEQIgQmfzbHAqARAhACIEQIQAiBAAEQIgQgBECIAIARChmQHVW/G68/6+Q87dt0v6VZxcmuQG1AuSnaBPN75XmnmWbQ53WWgX4KDmBST3CemVRQ4gkf3d42d78YXJv9x2bmXdFCB1mEHtZvjIk2WdoH/qrzuRwXL58+bhF55hpQPkuUx9yOnhTmS4ASRfY+eqCNSHOZgfI++WIIOhdlPXq/etWZs0FZ2feleplu0U+EpYgg533ddn9kYLJkVTNG77BwzrmtIFZP4v3aCaeW2ra6LDSKVgDlkNvtALKmUV01sTWwZTxPNN0ao71eG3qn1/u+ek+UqQKj1FrwQpOT/DuNgSVPW3JRb1jW7a2g6q2lU0TC1Asqpu2m7yVxqcy+/FYaOybVB9g0Zo9l5MX7Nb/Qp9dV2aj9uX7TUSgpHaC+pepFlxu2l8fAjOpxcbJXFJEy7o2JoL0OGb3oIQjynCLSbokACIEAARAiBCAEQIgAgBECEAIgRAhJICejLHmToP0R5ThFtMUKZmAVT0s194sj/eLSYoUwkB5c2E2S5sinCLCcrWLCVo2BThFhOUqQU/zS9DaQGpSZj2mwiPKcItJihPaQHl/Qx5TBFuMUF5StsG9ef3fFN+091igjKVGFBvStBjinCLCcpUUkC91xpeU4RbTFCm0rZBntMWfQcwTneLCcpT4iqWdTsPjynCLSYoUxgHEQIgQokBibJ9tc93hCnCLSYoS4kb6Yv74mrfGoV4TBFuMUF5StzN79SLcWeWwWOKcIsJylTygaLMlzud1TdFuMUEZWqWEuQc9eExRbjFBGVqjjao9VNkHlOEW0xQntL3Yt3VmR5ThFtMUJYwDiIEQIQAiFDKZcC77jOixxThFhOUL5QgQnizSghvVgnhzSohvFklhEaaUFpAtnv1TA+7rcF0t5igPCV+syoeEDftYxc9pgi3mKA8JX+zWsp5qmaWwWOKcIsJylTyCTP5m9DO70J7TBFuMUGZSj5hJjsQJ18eU4RbTFCm0rZBcmCW71ol22OKcIsJylPibj7ftDbPBUwRbjFBWcI4iBAAEZpjTrpd8T2mCLeYoCwlf6tR1ZtSg6YIt5igPCXu5ut98gOmCLeYoExhwozQLCWo6N/fwnN/x7vFBGVqjiV47feefVOEW0xQnuZZgue+duiZItxigjKFcRAhACIEQIQAiBAAEQIgQgBECIAIARAhACIEQIQAiBAAEQIgQv8H3ugjZgXQed0AAAAASUVORK5CYII=" /><!-- --><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAAt1BMVEUAAAAAADoAAGYAAP8AOjoAOmYAOpAAZrY6AAA6ADo6AGY6OgA6Ojo6OmY6ZmY6ZpA6ZrY6kLY6kNtmAABmADpmOgBmOjpmZjpmZmZmkLZmkNtmtrZmtttmtv+QOgCQZgCQZjqQkDqQkGaQttuQtv+Q2/+ZmZm2ZgC2Zjq2ZpC2kDq225C227a229u22/+2///bkDrbkGbbtmbbtpDb25Db27bb////tmb/25D/27b//7b//9v///8yjGR8AAAACXBIWXMAAA7DAAAOwwHHb6hkAAASzklEQVR4nO2dDXvbthHH6aCJ3bVbO7ndW+ds6xao67ZG6zbLsfH9P9cIHO5wIEEeSIKR8jz3byNTJASCPx0OLzxQnVPNqrt0Aa5dCkiQAhKkgAQpIEEKSJACEqSABCkgQQpIkAISpIAEKSBBCkiQAhKkgAQpIEEKSJACEqSABCkgQQpIkAISpIAEKSBBCkiQAhKkgAQpIEEKSJACEqSABDUHdOy67s0jbt2Gfeeue/U+JTl1oF/+5NzL25g66eXv791YLz/cdd3XYestfPzGf15WPEM5V1m7ALp512883xMgv+8hJUFAPt0Y0IdvX5Uu5ew/cPBbCCieRxKcYSJXWbsAChfydIeAGKogAtSXfAzo2BUv5UQ4EiCe6aTgDBO5ytoD0Jfhmk90Befu5re8fCd48+G+v2YE9J/fdN1nv3+MgClxvhsQ4Uf6byDUnR/6A18HyP/+ot/86ieHlPtv5gDJMdeUolZ7APrVXag7r/4SAR27N/+9g+oRFAEFo4hXe0KLygENdueAfL6PYJ5d9/o9pfYfngDEUtRqD0B/fNuX6+nuzY8AyBczq0kR0P/u+79wAIzhFOomqwx8d1bFmAX1yX/ymwd/nttH9zPLJAGCHTxFrfYA9OC/2nN3OAGgs7+0E/OoYx8UoYQ/DBDfXfRBB2AAzHtKr/+VfXIIiKeo1S6Azn1pjt1DBHSMNkJf26gVQ5sIEBKgbHcJ0O0jNAWQF+wPHmsCEE9Rqx0A3bzrS/3d/c27cwBEl0B1LAIKzjIU35u+P3D2vYEEKNs9BvT6ry62/hH2h29h628IKHwr3EZZilrtAihcgq9l/gJThcKu0Il5yXUW1H/kRyDeM2A9rA/ff4E+vQiIpajVLoBiJzoAKvRaRoAW+6DY8N0mKyO9/CmlPncjQJSiVvsAOgcH2n+Bt/DiwmWlxnsIqNxcCa3YMdhkf74/+i5VMNhf+y4zfebD2wToFItFKWq1AyBoLvryBDYnrFq0UQDEOjwB7rgfVG7moenGLtIP5I2S40NAkCtLUat9AIG5eEDJcGLH1xUBQZf5D2FPfxWvsa+bdo8BeRPqjfPDn3sY0Dn+2fsXGMI+9e74q3+mVizmylJUSqc7BCkgQQpIkAISpIAEKSBBCkiQAhKkgAQpIEEKSJACEqSABCkgQQpIkAISpIAEKSBBCkiQAhKkgAQpIEEKSJACEqSABCkgQQpIkAISpIAEKSBBCkhQY0DdJ6NLAWqb3X66MCCb9thSMlP+9MTuPbQG0LOPhDwuiNKfzM4mLLYEyJRJTOzeRSsAnW/enXxM9yZCY0CZCRn8U0QxsXsXLQf08vbgTj5eduW6KpaddVSzbFbHDNiIKVemid37aDmg5/uHAOjcAFCQo810GAAYY3htws3B7n213oKOC4KsN5w36SNSYVrtg0586cOO5026CJ+1rdiyIPQt5yVdxoAu3Q9aoovwuWpAc0SK3co9tKYVo2GK5IWe7iZTVJx3ok6FnbbYr9xDa5x0WDN57h4m+4qM4dSKq9UWFLHxftOudW9NMw8L03xLNtXSA8MtFjR90XiEjGhf772uo+jVdxSn+4rP97AMcgiodhah5qJLFtQe1bqOope3oJnO9DEslFtvQZyQ5dc9x2CG61p0W3wQVrayTt1hKSALrsX4ymNcGqfx6563rWk+a+vh6o6iX1U7Pxp7uvtsGSAYmnkc4F7AyxiyIP46lHj5H9GCqvXydrojUG1B+NWbWSPYz1FfU0cxmxAygQjNbIT3sEUvXLs1ZCsA4QL37dMdufLOn79ik+aFyE/bAC0ZzN4DkBWABNez/rx2PIIAPt4pocOGDcOOjdI31Zp+0LbZ6KnzRhfN5l5py7AZ6+i3DZt1TNrBFa3vKDY5b7IYmFFESs5mk4w27wnBlGKplqVJxwZl5AVdkHC+97MsO5pwdcjDEih8MaFhQ3OhuWr032hKRCQ2fa0QreooNjChzIKIDPx1cBOI9YqiBRmbQBlq6dCfO8zLtiS0YbqjXSsW7cURIUsCCzLROXlQxuLls14A8z5YMav4yIku2w9KdzQYkxyQl8E+tg0mYx0DhOJDt9LOoirM7KKAsDpNyKANpT0m8Ilumlcjbk0u2zmnimq4AhDWsQZVjN0TY5ZjcoMyoWZFCzLY+0mAopWlfmWSyGcXQMc3j6fbuamMZdkNvU4crCY28b9oP+GiAkPvjiz0IWNDttg1VyRe11E8+1vPTW4ckqthBmNsQZwQdaytIZdETX7jruK6juLTL96Hf5vPO/bLsVnP2RjYHTuIxmC/0catubHZRmDrZhSfv3nXBpCzIydthmwAmmcE/wd/ZKA/aLM+YomF3WhSK3yQn2g9HhpVMS80n0J7ZiKb+AZrGWFDqwoqjc6wMg5VfddoTTN/vPUt2apGrDxpX/I5HFH01SbVM4NvC2196jLSy0hT3Aolrr602oQLsxvXsSKg+I9shqrceEIfRx/5oYF2taAW4q2YQCg18+iUeGsGEEKH07gwVWtxhNakRbv0jOIsG/I8iArfxcoFWy6OdrECWiBk2gTqXXxGcdJ+Cq09Z2aQFQxzXW5B1CHafAv/0jOKQv2SsWEthexCHxydtYum1aSg9Qnbzihul4t10IU+T9zlsDXDQ1sLuiBhyxnFtYSgjuE2jvoJGeSM7pumbenkS4it8EEbh6lZdlZqw2YgMVj03oXZEcyZhBUxnXRpQWsTVkT+LDsvhEpLEOZJUcMGfAwSolPYOGPp2EmXFrRdwsXZzRnJFKHsiMlhGrgrgnYS7apApYrT5QHVGImgIka6D5lumeSyVTa1aiwWnkvdYjGLVwNAQzLsL51l1oLG9ErXXXtFEGL/fL+pMdsDUEaGpgjCOfKTj3G0tCAWglf72fnztiOEgNgID8/Az23tgmZsQwjedQJKPQcCMeKxYzPvBas0nu4arBfzasvHupxQvCGZzra+oPUJwy8JzK/VEEf8e1YxpJPd0qZzTZa5fGSfZv6EbdxkY9f97lPRHoDQTTk3OXN96cuu1x6A2IB/0NbROOXSl12vPQCpBUmi5YjTPoi/ae+f+Vbe8bF8KTErwObrXjYWw1H/5L2zPQElPpTzAMEISLOedPPpDq/mXEbv07ADZ89WFbQyYVyrsW3R6kexoCK6EZx5WOunXBveet4Rh6Vo0CxSIj97XrT87eUHq/uwcWxKLRHC2bPcIeUbA2DrB6uNHizQDgqfNnMQlIbjDT7/ml4HQjJbLSiGAZ9a+aDVs/ZDPMliBnGyePG2HM5gRxv87ZpmPjRlm+ynlZOemLLOYrDwxgDOD9mBz7FFw6H9l5+TdlU36CvRZHIw0ZEgJV7jqOEhty0W1EKDVmypERm615OFNnA6jhPBDYSQB36YQud68XXvC2gWRukNxpzRXcNBbCP5NjgDjTnwnIPFMdfdk5bigyy7fToO8bSDQL0UZ02LYHI7GmsujugaLEgkRGGJJd+DoVRmDA+DzIjL4HY0ai7S6ioATVtOMhv0OgV/PUSTIvXx5mHBbjJjamxBFXPSS7Jz/DZNAZRhtYxgsIiFof9hXavCfXm2nnFFQasSnuMv2zd7AhXVh5IV5XfhY22igAWMYDTMTcEiYKIznA9aGC604b5Yq8Fq2W64cRAnpEPR0hSoGFCk5QqDibEJ3zOxY6qglVe0abBajJMe2A/GRSdOuJdqWVy4QnvZKMOZuFTBzrTds6P5iesWLw03mluQGxoRBUaT/dhUjWi9Bosqh4RYk+JSBWyaCuvIciatLai9D3JjQsnlkDERMoxhBReTBU5bdq8ZH/iRxeLj6VYXtCph+1bMsRE4RtEzJrHuoNsOFhQsw0Z/jJZUmlEdrt3cVtAGCZdml6aMyT+zSHETl/qAqcQocb7kMC464GH3ucy2R78uHmo8+KVQ2zWcMEuMqFqRFcVmiWoSrZyLy8LRzQRCGSRbsUBcqG6LAR0aA3IDPmnwVVi6kqjEbdqEhWPgn0O2SExakSA0Your2GmP2z7wMmrIsi20nQGY+HSBABlcN/lmBwukJzxQ9qSGyoLWJWxdxeIrsyKDJmXcoKKlJiltwktMk07AliIW+dT67Us7abBwbjwRDfM5sHwnPVKA7AZfoT6Z0rTOpPHU+u1LA3LsjlXWm3ZpniM21HHgaQgJc9bx0R7kT3ClT8lQljX6FwfkhXzYGBNXN8fGO3YC0b/EjxEpk5DEFHG8OvsMneUFbZJwTXaZh8aen8sX0CUnY6m/TMt5i0+i4vOrq/uK1wHIZfeF48MowKekHqLFx+LgfQl8yAl7AlxR2FdqU1Ax4YKnAdefN80ZWxpqhuG4pREqOSD4BLnptDZsUusPbonumHkaMKr6SZxZW+bwWRMW5ixiTcPn3+FnGJ+Kp7hMJpDq3qbojqkZj6WPS05OJevzUFNvsY+Y50ENmOxg8hRjdzWtTRNmk3NmSx+XnPcTQ38H5rusMcnFUkeHahluyw5mcB9MTD9R0JqEVU8DnnpccvG8lvcUHT6Qy/JReBxngKmlznRt4QfauR9U9zTg+sclW5sRgjkOrF0hhUmE0uhi+lE4LZbLlwpanbDuacD1j0umWS7WnXbM+xj2CAFuO2On5Iq7N2nHflDpccnCE8nBV1NcQpxMDdOGZDCsWS847cLubdqzo7jiccmRBPZ9wjs8aliLVRpBNCYzV1AhYe2cNAu4rzov9hLjANXGpwLGo8U+4dLqtBzhnnc1lgKCcCfsNKc5HzYF5lw+w7wQ0Ar3tOd9sRWAYitOfNI4lDqP+VUuvOCPYUH1d1YXA6JHlNBsT/yL72jOlWu156n64FVZEBIyw45OarOYIcWPrG7V6z54JXdWQRanBBOL9Jea9sHjf6/Ogna5sxpEfLzCXVOa0wh78tR7tOkFXcmEWRqJmZZit6XZyRLykT/bcN37A8I5+TjTQ0/aqlOcxk4T+Q464I7fAALR/GLNROOVAMKoVOfw8gDPAkL4GUdzrDA7m80KgGhHxV37qwgDdi4uPPGiu15uUN/YLGvawbZZXzIN+cuAzHCjvqDbE67MzvJeEEVkZLcHh4TyiWney2ZO3bpCPSoP5eoKujnhRbLbTxcD9MnoQoAGGQq5X0NaSQpIkAISpIAEKSBBCkiQAhKkgAQpIEGfzNjgUlJAghSQIAUkSAEJUkCCFJAgBSRIAQlSQIIUkCAFJEgBCdoACG7HlmOqzylOZPGvSlble9ociFKpDYCePp8soX/gaXzo6fP90nvZNfn6xQAbn6paqQ2ApoP1IAbt6I3gvPxmf0W+YUFSm9+zkrQB0InK5xeVQ1gefPsQih++5e6w+HEyNfl6XT2g45ex/KewfgMWAsGFhNdIZjGgyny3PjS0UusBwcqf4yH++B8UO17CTXpdDKgy33O38WeIKrW1mffLy26i+Ycgx14PmwDV5ttXsY3PVa3SVkB9+c8xXiL8sEuDKlaZ7+aHg9dpM6DP3/FyFpzpSkBivo1+dFDSekBQvP7y+e9rwoWwZn45oJp8Kc3q0ldrSyvmAfTOFJqTI7d31lFc0YpV5OsfCz4X8d9OW6rYMTqI0F/JMbCRwPLvuSZfSrO3dLAqSAEJUkCCFJAgBSRIAQlSQIIUkCAFJEgBCVJAghSQIAUkSAEJUkCCFJAgBSRIAQlSQIIUkCAFJEgBCVJAghSQIAUkSAEJUkCCLgnotDxIzP9K0/3UPfnqeKHzgrv6FwQkPju/+JmZa6sFNJvJUBcFtDw845MF9HT33X3nH8UNQSkUvxujVJ7v/WEs1jkc9WlDbEs6CAfc8zffd6/+kedIGYUqBj9mdWAn8nEx3yOgU4zUx/zu8ZPhRD7LN48h8LECVDNAfeFO/oJ9gBzF7/pYKP/m+d4fxoiqvlzP97f0RdLBdODN4yDHlBH5oPxEISY7ngASHvIThU/Cifz7EKN2riDUDNCBXh4ofjf8JJcvStgTQwohMOwM5QRAcJAdOAxzTBkhIMrWnwiyPuLiB/Bt+YnCJ7F8D/VhXc0APaQXit/1R85d+B4fqOrD/lhOuCA4mB/IcuQZwcdCDDCdCC43+iCM7czzw0/Gv9VLSHYBRPG7vTPonUkJEKsrOSA8kANiGcHHQn2lE50yQHExTJ7f8O/L24/sg0YWxPdutSCeUTx6cKzZWm5BQceKZm8PQFSAUO7zoIoVfBAczA/kyFlG/mgMsqcTxQDqSR90GINy/PMfGRDF78J33h0yQIVWLB7MDowsCDMKzTxrr4IdQCPI9vqlQJifx+l/yoE762BtNR2nXQBR/G6I2j2mdsRFQql7wgFlB0Y+CDMKtsacHHSm5vpBYQ3jd6mK9Z2qN4/+WE3HUgerghSQIAUkSAEJUkCCFJAgBSRIAQlSQIIUkCAFJEgBCVJAghSQIAUkSAEJUkCCFJAgBSRIAQlSQIIUkCAFJEgBCfo/Z/rDTwEfZS4AAAAASUVORK5CYII=" /><!-- --></p>
<pre><code>#&gt; NULL
#&gt; [33mPlotting gene expression for gene:[0m LCP1</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABLFBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZpA6ZrY6kNtHk69NTU1NTW5NTY5Nbm5NbqtNjshmAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkNtmtpBmtttmtv9uTU1uTW5uTY5ubqtujshuq+SOTU2OTW6OTY6OyP+QOgCQOjqQOmaQZgCQZjqQZraQkDqQttuQ27aQ2/+rbk2rbm6rbo6rq26ryKur5OSr5P+2ZgC2Zjq2kGa2tpC225C229u22/+2///Ijk3I/8jI///bkDrbkGbbtmbbtpDb25Db27bb29vb2//b///kq27k5Kvk/8jk///r6+v/AAD/tmb/yI7/25D/27b/5Kv//7b//8j//9v//+T///+YfIzDAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAKGUlEQVR4nO2dC3vbthWG6TZxvXRtEyuprTbdrfbS2o2dbu1md5dot8az16vVJbYUyTL//38oAAIkSJH4QFEyZOp7nzbSAcE8wptDgARFIYqJkyj0B1h2KAhAQQAKAlAQgIIAFASgIEB9QZNvfhlF0Tv/qbPP2Rsvsv3/+XYUvfmxKo4Ud/6kN/XWa3+ahVNb0Phx0qroQY2dLEHDjWT3+3EqKIp2klpRCwRNDqP3RPJMvt1YO/LfKxM07kb3f4zj7x5LKbr4m+juufgr/xG1QdClacRwQ755LfLpna+lt7v/ejta+21slYm6O3qvTNCZzrzhhpBiinviddxd+123BYJ6Vm9ijheRSyKxzLGSlpUJmhyadz9YxUrQF+fjFgjKt2FyuPZxHH8rckkIWj9XnUhWZpEKyu+fHmLrJRuXhNkEqd717vlwQx0vop1JZsitWZm1W5Ug3Ufr7qx9gi7TBoo+SPazh+vxZb7RCW5BqsOa2rgk1BSU9SFSSX1B2f7/e+/rQpq1QpBo//3kjVQy3DC9cCYoK7OYHsXGXVHURkGiN1Yn0d9/JA4x0SGLk+DXh7IPMoKyMmuvsvOgB8WOqh2CsjPpO2og1yfVmaCszB7ms7NvcyYtXbRSkBjCP5Ida3L9NBS65HWVJSgtKxcUv/7jRmSuxdopaLWgIAAFASgIQEEACgJQEICCABQEoCBATUGvNOkbZ+BXqzxYTHPrQ0EACgJQEICdNGAOGXRxccEMMpS06eIiZ4iCCm26uMgboiAKckFBgJI2sQ+yKWsTRzGLeq2lIAoqQkEACgJQEAA2sHLIpyDjp+KkkYJSP5khCqKgIhQEQA1kH4QauFqj2OjJaRwPOp1Hp6akXmvbLmggzUhJ/W1TREEWJw//9iRJndETk0IUlMOISTLoLcENfKIlw0PQaPfhc1NSKx2qZ9LalkFXnxlDdVrrmIttm6D45ECX1Gitaza/TYIGWy9ny6BVERT3O52Z+qBVEDRNndauQB80Ta3Wtn8Um6Zea1t/ojgNBQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQKcAMAlAQgIIAswnynExdWUG+0/GrKsj7hg4FUVARCgK8ygx5tHZlBXEUq6ReaymIgopQECC0oFflNPdQCQUBKAhAQQAKAlAQAH+Jc7Tb6ZhvAVNQDvUwi/wK8Oj3s3yRvGWCxl2zElGkfyc8eZhlIJ/TmOWL5G0TNPUD6nHxUYRleJglc5KfTzCMP3AvyTS892LqnWRyWLKWg8I+xHryd+lttKDr473iBwyeQYUZKcPlb3xX9ZpF0BSJoKv91M/SCCo+PWza+ezPvzpXCzPIo0G/iM5Dv6wdSS1yXYYdVZhtufN4dkGj3YOsZMkFDd897+3E4w9fxGfr5iXuPYgv5RogO+Llu3sv5FEoPElVYoupEM0sKOdn2QWdiaaum47IvAhP4m1ySOkDS5QNE1XinazgdYj10vHLFtTvSJZuFCvtg9QqZ2+og0guLZS8SCeTZ0ciuYwg1VApqKvWIErKsKCetWRcJUsjqHQUUxKSpl4mA454KWbQuLujDzG5JU5SzCODVJXqeksnKIfeeiY//eW6lCP+1y9WHzS89++k8xn+4kj3QaYC7oOkWCEIjJJLLWjyucwIkS26s+iVj2JnkRy05EJeqkgdmR6jWAsELQQKAlAQwHGx6vqAKyjIDwoCrLQg3QsVL+lLP+AqCtIryBYWK674gKsoKF3s3bXq+woLUgeY4pKjmAUFAayLVXOOeObopldYUDrV4TybXmVBcU+tXz3uOsf5wAQVlKwynFvlu/IDhs+gzc3NKUG+dyqqq+T/hqIgD5ZG0Oambai0eQsTdBtGsc3NnKGsecN3/xDJ2UF9KyeZLBt/+KU6MpKotEpSqOfWWi1oIzeNmtzYkf2qbFUSlVbRhSuQQXLG+Z65lWNu7IhzvMmzIx2VVtGFrRFU2QfZDe3t6Bs76g5ZGpVWaZ2gqlGskB76xo4QJDMoiUqrtE9QjilBVgcjb411160bQKVVsKA2TLlagvStnOTGzviDX5tRTN9qnaqiC+XNoApBfiy3oCrQt4ZcUBBgJQQ1gYIAFAS4ZYJuHgoCUBCAggAUBPD5TftsWQ0KyjFI1qwZcOmacvTKLNkCLTEFFcgfYsvwMMvNwz4IQEEACgJQEICCADyTBlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCcOkaADMIQEEACgJQEICCABQEoCAABQEoCEBBAAoChBGUe56SgqaC/BO5FFQMCs90l6fTYppbn/CCKtJpMc2tT3BBhXSioFRKmS0KyqzkZJV0SItpbn3wlziv9jtbL03JnARVDmlZsPim+wEfZrk+Poj726ZoAYIq0ukG2u4FfJjl6ump9T3g2QV5nBveNkF6VYRPXs5jZRbVcL9aHvVuEChosGUESWbNoKqxqhjcrj6okEGSRQu6haPYfPogb0G36zxIipErH81hFPO5Pr2dguZ1HuQ1it02QdPMLogTZhQkoSAABXnyllfgV8sRLAEUBKAgAAUB+B1FAAUBKAhAQYD5CBrtdh4+j9XUiAr2TjrWrzLp2RK5VVz56kvf0e7B9bEO7LWA1f7bfXv/oMxFkJzYl9MiSpAMTh7pmbb9ToJsrdiqKow+1bvINchlLbHlr9mS9tdfPVfzK2pLeGYVVGy6KDrZVq/yj8GeLrzaf3RqZZDWkdSSJtIgPjEpI4O+2T88M2eQtYS4SgdB//0np2lgMuBqf+snLWj/4d//cmo2iPSRGtVy9okL4Vw6akkGxdZEddqJ9B8lDT1I30tGu+at6HS244GOTlQCqrnKfLLIPqxlfVCbmYOgwYFP4FfLEQSCggAUBKAgAAUBKAhAQYAGJ4rJ6e/TU2fgV8sRhKbxtZi8HK8M/Go5gvA0zSAU+NVyBKHhpQaggaBBOt3hCvxqOYLANDjE9g88Ar9ajiA07IMADQ6xkz2fwK+WIwhMk0PM6ikqA79ajiA0HMUAFARoIOj6uLP1fzMxXRn41XIEgZld0PXx3uiTl4PkgqAy8KvlCELTaJgXzciuL8sDv1qOIDRNM6hv/6OXBH61HEFomvVB2SV3ZeBXyxEEhqMYgIIAM0+YPf1vdr5bGfjVcgThYQYBOCcN4Jw0gPNBAPZBgAaC+vZYUxn41XIEgWn0DbPBtnmctTLwq+UIQtOoDzLfW3UEfrUcQWgaXKx+9Vz8p7/BWhn41XIEoWnQB4kWDDqdPRD41XIEgeEoBqAgQNP5oG0U+NVyBIFpNKMo/tSDcWXgV8sRhIYXq4AmZ9LbcfbvXBn41XIEgWl861nPclUFfrUcQWg4igEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBgJ8BAZ9yoS3a5ZoAAAAASUVORK5CYII=" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m HLA-DMB</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABJlBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZmY6ZpA6ZrY6kNtHk69NTU1NTW5NTY5Nbm5NbqtNjshmAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkJBmkNtmtttmtv9uTU1uTY5ubqtujshuq+SOTU2OTW6OTY6OyP+QOgCQOjqQOmaQZgCQkDqQttuQ27aQ2/+rbk2rbm6rbo6rq26ryKur5OSr5P+2ZgC2Zjq2Zma2kDq2tpC225C229u22/+2/9u2///Ijk3I/8jI///bkDrbkGbbtmbbtpDb25Db27bb///kq27k5Kvk/8jk///r6+v/AAD/tmb/yI7/25D/27b/5Kv//7b//8j//9v//+T///+/kNHvAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAKgUlEQVR4nO2djV/bxhnHBS0upU3XAskCabpuK2kHC87WDbqXqNtYB8NrN9gSbBdj9P//E7tX6SSk+50QRjb6fT8p8iOdXd+X514kmXOUEC9R229g3qEgAAUBKAhAQQAKAlAQgIIA9QVNv/8wiqJHf6/znMHysdpOttezQDGMdtxykeLRdzKIH/3O7P7hs/X02Mff1X7HjagtaPJcv9FovcaTqgXFUc8tZ15bWYujpQO1d7waZYIi99n3QF1B0370sUie6Q+r5t0HUSlovPqh+zr60PRf6sXjpQ/0L2Hwzmr6tOm3uZSbPXUFDe1vfLwqH/z4XDeIaX/lbx9ES79MnH1O86kUNFg+WnVS0R4aypSJl79eOUvka/8iEyT+v3VStzl1BcW5DBfJLxC/bpFYtmmk+wIETbZ7Qu1Z+nr2kNoZLx+9L7NrvHbkZNBf62TuHVBTkKiRE037S1+ILlTkkhDUOxPdRM/Z55D2LVFekFTodtPpIfl7iJf/0ZfZEvdU1gxu0ffdAbcTpN7syplJd1GtaV9WTR7N9jlPqxIkPbjOC4KOY5FIk+2dnKBPzpL7pJGgoXnPSwe6oUz7vSTb5zytoonp1ijHpYGWVxQ0FK8yXD4eO33Q97lhb/bUFKQzRT9qLihrNQVBpg86luVjk5X2WL4XnDn1R7FP9ANZh/Gq7T4yQdk+h3JBpn+ebKfddGEUk1l0JF5vkQSJ3lhNov/zmWhiokMWk90f+7IPsoKyfc6zygUNTYcbp9mWzoPkVpoYLn0ufuab2Mq9dkK3n0m/qwZy00QyQdm+ymHeNiwrJp1bFWbSqgeXx3Kd9JxPFBN5ZiRPl/R50ljoeucLt4ll+5Cgn9ipQNax6UNL+nxLtaVYT62ypz2632kQz+YRFASgIAAFASgIQEEACgJQEICCABQEqCnorSF94A3CSpUHs6lufSgIQEEACgJQEOAOBF1cXFCQpaROFxc5QxRUqNPFRd4QBfkFVWYTBWURBVX2QYV0oiBrJSeLgqorSEGoguyDUAU5itWobfUUm4KMnwbpNJvq1md2gnxTbAqiIAoqp05t2QehgKMYCjgPoqAiFASgIAAFASiIFGAGASgIcDtBgXPkzgoKPcvqqqDg83QKoqAiFAR4mxkKqG1nBXEUq6RebSmIgopQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgQKcFxVG0Lpdd8y2i1mVBsVxyK9pJpn3PSoUdFjTt7yRyST7xI10k7PLLU2cj6bCgyXYqyC6YNdp8cpptFBTkCDp5/CeZOmajoSA3gwpN7D1BC++wZWoIknQ6g9IFaSnIwTtRpCAKgiyYoLfl3KWRAgt2LtaiILsqvf+LGTosSK7ii9fZ7bKgRJ3PA0cdF6Qbms9R5wUl7urOnjfYWUEig7xf7NFtQchO0m1BQ2gnmSdB+U9JWibP/HUYrx3feCRRlwv9ghZsHlT4nK1l+PPQb7apLSiMeRFUXGbG1vPV7392pqZ06rqN3ky27WbpQGqRNyZ21M7syLvPuyFo/NFZvJNMPj1OBj27SeJ1eZld5shw5d9rx7IVCk9SlThiC1ROblxBQ/y1FXMuaCCq2rMdkd0IT+KhblKmYYl9Y61KPJIFQpqYmf84XyQzx4LK+yDVj6pvclEDjt5IJ9NXByK5rKBYFpKCttV3EOl9WNDAiJnr+2JZubJRTEnQVTX3rsSmmEHy2rJuYvJIolMsaBRLv0bFk0LzIyiHfeeyCsOelCP+MxunDxqvHenOZ/z+gemDbAHcB6mL9orhIgzzZYKmv1F9xLODWA9fcfkoNojkoCV7FLVLtcyAUewBCJoJFASgIID3vlj1G+ygoDAoCEBBvklQQkFQUNvMvSBmEAUVKOuk/8JhPuOGIHPmUkHnBQ3AFbNuC1KTaf93cXdZ0EB9w3e8KII2NjZuCAq9U1FdJP8KOUHmZHVRBG1suIbuQ5C6ZL+zKII2NnKGsuqNP/qt6kXNrRw95Ew+/VpdGdNRaRG9s2SEKtx6XpQ+qFrQau4yqr6xI29EyIs4OiotYnb6MkgxXJBRrFqQvOK8Zm/l2Bs7ovuYvjowUWkRsxMKUmm0CPOgqj7IrWi8Y27sqDtkaVRaJFyQzKLqDwDMjaCqUayQHubGjhAkM0hHpUVqCVqEDMpzQ5DTwchbY9s95wZQaZHOCTK3cvSNncmzz+0oZm613ihidt78eN2DElQF+tSQDwoCdEJQE3jbB8C7GoAFE3T/LFgTu3/yGQRv+lAQBRWgIAAFAbyC1B+rXu1tPn1j91CQK0gt2nF9uJ+cb9ldFOQI0ot2XL08df7sucOCqlZeuPzqTXL169dJ59fuKEMKGj21giQdzqAy8hkkoaAcl+yDsKDrw12OYlVwHsTLHRAKAlAQgIIAFASgIAAFASgIQEEACgLMm6D0c1GzqW595kxQ9sm62VS3PvMlyPls5myqWx8KAlAQYL4EsQ+CAUex0GA21a0PBQEoCEBBAAoCUBCAggA1BXUPZhDg3gTl/gCOgm48zv8JZYC62VS3PvckqPBHuAHqZlPd+rQuqOrIbKpbHwoCtN4HUVAmopa62VS3PnMwD+IodqtgNtWtDwUBKAjQjqCA847ZVLc+rQjynHdQ0FvveQcFURAFNRfEPgge4ChWpxQFURAFpVBQIO8FBWGlPMEcQEEACgJQEID35gEUBKAgAAUB7kbQ5YvNx68TtYyFCnZPNuXaQ/aoXtlCHr3a2zTLFFy+2L8+NMHV3m7+xbbO3ee3yp0IkoswySUslCAZnDwxq6IIHwpZW3FUFbj8lXnKyX6iSokjf9zcT1/sm9dqLQx1pH1uK6hYdbHrZEtt5Y/Rrtl5tffk1Mkgo0OXkibSIDmxKSODc/v89rl1BsmMyR6r3//5T788TQObAVd7T/9rBO09/vMfTu0BkT5S40ium6JdCOfS0QPJoMRZVCjtRM6f6Irup48lly/sQ9HpbCUjE52oBFTryuSTRfZhD6wPesjcgaDRfkgQVsoTtAQFASgIQEEACgJQEICCAA0minr6+/LUG4SV8gRt0/hcTJ6OVwZhpTxB+zTNIBSElfIEbcNTDUADQaP0cocvCCvlCVqmQRPb2w8Iwkp5grZhHwRo0MROdkOCsFKeoGWaNDGnp6gMwkp5grbhKAagIEADQdeHm0//Zy9MVwZhpTxBy9xe0PXh7uVXb0b6hKAyCCvlCdqm0TAvqpGdX5YHYaU8Qds0zaBz95deEoSV8gRt06wPyk65K4OwUp6gZTiKASgIcOsLZi//mc13K4OwUp6gfZhBAF6TBvCaNIDXgwDsgwANBJ27Y01lEFbKE7RMo0+YjbbsV49VBmGlPEHbNOqD7OdWPUFYKU/QNg1OVr95Lf6ZT7BWBmGlPEHbNOiDRA1Gm5u7IAgr5QlahqMYgIIATa8HbaEgrJQnaJlGVxTFTzMYVwZhpTxB2/BkFdBkJr2VZL/nyiCslCdomca3ns1VrqogrJQnaBuOYgAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQ4P8R0pi99WdMAQAAAABJRU5ErkJggg==" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m LYPD3</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABIFBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZpA6ZrY6kNtHk69NTU1NTW5NTY5Nbm5NbqtNjshmAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkNtmtttmtv9uTU1uTW5uTY5ubqtujshuq+SOTU2OTW6OTY6OyP+QOgCQOjqQOmaQZgCQkDqQttuQtv+Q27aQ2/+rbk2rbm6rbo6rq26ryKur5OSr5P+2ZgC2Zjq2tpC225C229u22/+2///Ijk3I/8jI///bkDrbkGbbtmbbtpDb25Db27bb///kq27k5Kvk/8jk///r6+v/AAD/tmb/yI7/25D/27b/5Kv//7b//8j//9v//+T///9Ix7sjAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAKcklEQVR4nO2djX/bRh2HldHghZatEDc06boBTbaR0LjAIBkvMS9tiGkhhja2G8fx//9fcG+STrZ035MV++To+3y2Sr/T+eWe3JskS4qmxEkU+gvUHQoCUBCAggAUBKAgAAUBKAhQXtDk7Y+jKPrsH2Ve0/vkjV7pRvvpSi9SPPhO5tDrn71WW98+iqLHl6W/2jIoLWj8XBcl+mmJFyWCxnt6bRC1EimRlGavmyB+TVjKCpp0osei8kzebW2c+L8qEaTNiHeRCSb1bbR5adYn/5JvO97bfD2d/DWubGEpK0gXUDDakisfn+tmMels/u1RtPHLqZUm8sZFTAXpRtbTFcWkdsUyXh8kNXO0VaaOLo2ygrqZij/akm1B/NFFxYobSJKWL0g2Mi03V5AwrfueSbdMFV0eJQWN91pWNOlsvJhO34niCkEt0U6ilpVmYQkS2n7Y0WHSxFp2N67bWifa+P0Cxbl7FhOketHNS9MKROF0pyK3pmnWyzJRNzI1K+6YZVWZETR+9iha5xpkBA2SEuqWMem0pgOr1AkZQUkttIf2GUGCj3ubdRjoSwqadOLvL5VUFWSlzvVBIqkWVaj8KPYTvSJLMtqKe+FUUJpm4S9IjmJxH76WgkTvqSbR//lSNDHRIYtZ8MeO7INiQWma9SpfQWIeJJbiM16Y6VF4Fp9JP1ADuZlUp4LSNHuYt2ff+YLsmfRga66RBmOBndV3X8qe9Tu1PhK6fvDCbmJp2gKCNh6/Tt7W7JaFhnvzAAoCUBCAggAUBKAgAAUBKAhAQQAKApQU9MGQrDgDv1z5wXKKWx4KAlAQgIIAFARYmaDt7W0Kcgja3s4YoqCZ9e3trCEKoiAKymVRQY3pgxZH+FnVR90pnAcBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgKEEeRx/HU5xS1PEEE+xz6WU9zyhBDkdfRsOcUtDwUBKAjAPgjAUQxQt3lQom45xS1PzQSljW85xS2PU9D1Nxfin4N2+zhOWbIgq/teftH9cAkatncupje/Pptef31mkijI4vzJn0UNGu7K1bgKUVAG1cQEshZNp58Klv11lJ9lf0gpfATdnh7GKUuuQes2imlBN0eJHx7uyGJGseM0hYIySEEZPxSURQrqtyUrGsXWTdA8FASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKIjMsGY16EM+1T0UkgqSNwE/6YHHHTRZUFfe6ljeYLbrMNRgQeO9feNm4HigRdMF9eR9einIxmpi8f2tu62CvNNmCzKGRlsOP80W5AMFAZotaJDcVL6Q+gi6urrKETR+5r4R/ujhm7k1yaRTVGx7oqifieJ83kdtBF1d2YaS7YNf+D4xqbygXvzUoc4aTBSvrjKGknK++sNXl6opyD+2WYz34sXGidQin860rxLTLQ+eQ0Gpw56jCtVc0Ojzy+7+dPzFGzmjMws5+x3IZ4Dsi8W/H76RrVB4kqrEljhDYc+SnSgq1mGiWCCoJ4raijuieCE8iVXdpEzDEmkjrUqsyQweTWy9BOX3QeopZ5+oRiR3KvVCOpm8OhGVKxbUlZmkoD31eBOddt8E5Y5iSoIu6kD3E2IxW4NkQXUTk1umuordvxqUxWztqefetKQc8b9ZWH3Q6OHfdecz+tGJ6YPiDD59UPx0HdfjKWstaPIb/ey2k64uQzd/FOtFctCSsxqVpFqmxyjmR60FLQUKAth78x7Pfm2yoG7yOFgHDRakBrqZ0U5d7XNz1H76fvYLNlCQGuaz+2HqmtXb0+Npf3f2C1LQNL5m9eblRXJpJgXN7Mmr68W+fb/Ca1YRNRQ0fBoLsr9gI2vQ/Ew6W4MaLiiPa/ZBWJC8JpyjmGD81fxxRM6Dsn2Q83xG9gs2UJB8WC7cFauRoOw9rDS+ZyqKs2TfYUaQPC6CTprURtDMXdBWIyg+dbgGB8xm7hOXFm/0+W/VuU9zKkcfLBt/8Tt1ZExHuVl0ojm2ViRIbF6XGlQsaCtzGFWf2JEnQ+WBZB3lZjGJzhrUXaM+qFiQPOL8MD6VE5/YEXsJk1cnJsrNYhIdgtZsFCvqg+yCdvfNiR11hiyJcrN4CMqZB9VYUNEoNlM9zIkdIUjWIB3lZsGCUtb3tE8qyOpg5KmxvZZ1Aig3S+MEmVM5+sTO+NnP41HMnGqdy2ISzU9c7qugItCvhlxQEKAGgvJv7Or+/HKCqhBeUMGtgd2fH0BQqHPzRTeXdn9+AEF+hBa0eigIEFxQuT5o9YQXVGoUWz28JBNQgxpUZqK4eigIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAWND1gbzw0EBBc8hLofrJxRoUNIe6XuxlXS6HWj0lalAdLslcPbgPqtXlUKsHN7Gvz6bDHTaxQup1Ue/qYQ0C4D5o2G4/iStQ/O0zt39quqAMiZ/MDbIoKCH1Y99izSqfn7nmCio2R0E5W5oqqLgPoqBURG4BKQgVkH1QTpky1Ymj2FyZHJMiCvrgnhRREAXNQkGAnDKxD7LJK1Pxrj0FgYCCKGiWUqX1PKzWWEG+h9WaKsg1/lNQYwWVwAha2vuvCPZBAI5iAM6DABQEoCAABQEWHOY/9Qr8cjmCGkBBAAoCUBCAt8cBUBCAggAUBLgbQdcH6neM+hf5Ijg8b1sXeJjnlsitN0dt86vr64Pj21MT3BwdZt9st2+/Pih3Ikg+Yks+oEQJksH5jnnmjfChkKUVW1WG61+Zl5wf6ysdxJY/tY+TN/v+TD3pRG0Jz6KCZosuks531VL+Mzw0iTdHOxdWDTI6dC5pIgmm53GVkUE/fn14Fq5Bssak6+rv3//ZNxdJENeAm6On/zWCjp785Y8X8QZRfaTGoXwqjnYhnEtH96QGTa2f3yedSH9HF/Q4WZdcH8SrotPZTX6Vfq4qoHpqULayyD7snvVB95k7EDQ89gn8cjmCQFAQgIIAFASgIAAFASgIUGGiqKe/Ly+cgV8uRxCayvticne8MPDL5QjCU7UGocAvlyMIDXc1ABUEDZPDHa7AL5cjCEyFJnZ07BH45XIEoWEfBKjQxM4PfQK/XI4gMFWamNVTFAZ+uRxBaDiKASgIUEHQ7Wn76f/iA9OFgV8uRxCYxQXdnh5ef/t+qHcICgO/XI4gNJWGeVGMdP8yP/DL5QhCU7UG9e0/ek7gl8sRhKZaH5TuchcGfrkcQWA4igEoCLDwAbOX/0znu4WBXy5HEB7WIACPSQN4TBrA40EA9kGACoL69lhTGPjlcgSBqfQLs+Gu+rGcK/DL5QhCU6kPin+36gj8cjmC0FTYWf3+TPxnfsFaGPjlcgShqdAHiRIM2+1DEPjlcgSB4SgGoCBA1eNBuyjwy+UIAlPpiKL41wzGhYFfLkcQGu6sAqrMpHen6d+5MPDL5QgCU/nUsznKVRT45XIEoeEoBqAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUB/g9cCG8QXduSEgAAAABJRU5ErkJggg==" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m SLAMF7</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABMlBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZmY6ZpA6ZrY6kNtHk69NTU1NTW5NTY5Nbm5NbqtNjshmAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkJBmkLZmkNtmtttmtv9uTU1uTY5ubqtujshuq8huq+SOTU2OTW6OTY6OyP+QOgCQOjqQOmaQZgCQZjqQkDqQtraQttuQ27aQ2/+rbk2rbm6rbo6rq26ryKur5OSr5P+2ZgC2Zjq2Zma2kGa2tpC225C229u22/+2/9u2///Ijk3I/8jI///bkDrbtmbbtpDb25Db27bb2//b///kq27k5Kvk/8jk///r6+v/AAD/tmb/yI7/25D/27b/5Kv//7b//8j//9v//+T///9dH8N/AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAKzElEQVR4nO2dC3vbthWGaSdRFafpxXay2G3Ty1qn3ezFyra0tbO1ZS9Za9dq1tlbYkmxLPP//4XhACAJUiQOaEqCJH7v04Y8JKhHeI0bQVIMImAl8P0F5h0IYoAgBghigCAGCGKAIAYIYqguaPTbm0EQ3P1XtYN+vBMEN/54Kla7q0fp9l6wI5fh3a/0lhcP10WaQLF6NGirtZWDyl90MlQWNPxAf/n1CgeNOuqYW6c5QWHQ0kttQAhZbEEiq++IwjN60a7ylXvBrZ+j6HWHyospaNB+U31MuHJHCe/eaK/nJMrjq/w5JkpVQT39Jxd5o5XXojzd/Zm83frpTrDyaWRsS+qPyLDSMNxuZTPfXX3ellkPV59S6RKf80mBoOG23OmFqoLCzFdXFUDkPq5DO8Y2Q5BZAozMC2HC7Kn82Oe3yeFg7XmBoNBbBassSJaBhFFn5TPRrIqyJAS1TkXT0TK2mXwfrLz95X/kqpF5Mqgshqu/dMhh2Bq00zZIpxzkPmymXE9QV7W4A1U/RJZHHcoM7U23ZQ78/e+iYJE6cw8VR/WJYi0URWm4vVMgyGcBqieoF8RdjKopo04rSrfljx29eBhkqo/uoSgWgnriiB51W/kqli21s6aiIFVS1FpVQeoYI/NxZ74uBQ2310UNi8YFpU2ZD6r3Yu+qFcrsoB1/9VRQui0l6YWoTnVNxfFO2iEaanHsuKAwV1tnyzXGQXIQ/ftDUcVEg/wVDW+oDYoFpduMo8LgHRoL/NY2S1DctVETQxJ6Kx+Jf8cEaY++uP5I+qbsyHUdSQWl24y6MdxO611SseK2l4ZWqrmmMdaYIL9N0HVOVkVbK0aC6txpIHTd+MysYuk2s/GQ52IrVIwSQW/HnTe1a7IahWoYtZ4fbXsbRRM4m2eAIAYIYoAgBghigCAGCGKAIAYIYoAghoqCXmmSFWvglqo4mE52qwNBDBDEAEEMEMQwTUHn5+cQZAnOzzOGICgXnJ9nDTVKkEvlabIgp8rTYEGOWW9uG+RaNhrbizlXnsaOg1wrT2MFuVae5grCqUYZEMQAQQxFeSpvkCBI+ynr0iDolX1QBEEQlAeCGAryhDbIpChP6MUaDQaKDBDEAEEMkxZUZ5a1CYJqzdM3QFBu1AhBEJQHghi4DKIN4jKIXqxKbiEIgvJAEAMEMUAQAwQxQBADBDFAEAMEMUAQAwQxQBADBGW4+PzEWBAQZNLfvH+SLiQQZHB87xsqOnqhgKAM2Sr2hmAWX2m+QBvEAEEMEMQAQQyJoKJf3IWgjKBPHH4iutGCduSPHtt/FrThgogw+RHeIiCIAYIYfAt6VcyEZBSxYNMdEDS/glDFiskK6nG/bQ1BEJQDghggiAGCGAxBQcwcn2qk6bJ3iya5eM9+vj1YOxpbI0adsj58UcdBufuNY3ofu/4A/pILyt+xHufzydcfnsoXM1At0AtRN/Ri5YC00AthduTGdM/ND5ohaPDWabgTDd8/irqteBGF61GP3gGyIxb/XjuiWig8kSqxJ05QOs2zXIK6IqutuCGKF8KTWFVVSlcssW2gVIk1SuBQxeI3hC1II13YBsk8yLe7yFecqAU5GT05EIUrFiSnvEjQtnwXitrGl6AeN5s4V4IKezEpQWW1p2bYxSJfgmg8o6oY7YlUEXNspEPW0fwIyqD3dunb91okR/yvF0YbNFh7rhqfwe0D3QbFCVzbIH5W2i9WQaO/UokQpUVPG4fFvVg3oE6LXpgjN8lMV+nFRp1FaIMKS9A0GC9B1ms/zRbE2YmaLajn8jbcBgtasHHQ7AW5AUHEDyhBKWOC9KihhMYL6gb2gWKzBcmTN/srKZssqCvf8hlCUJb8lVUIypGZ7hDNDwTlyJ1qoA3Kk+vFegvTi21sbIwJcr1SUZ4k+wkFgmQxWoRx0MaGaWiWgqgUlZ+0zougjY2MoTR7g7f+JuuAvpSjhr3D95/KM3EVFSZRGwtGyUWCFqAElQtqZ6ZR1YUdugeccqWiwiR6o1sJWmhBNOO8Fl/KiS/s0C3OTw50VJhEb1waQaVtkJnRcEdf2JFXyJKoMMnSCSrrxXLFQ1/YEYKoBKmoMMnyCcowJshoYOjS2HbLuABUmIQXtGi3vzCC9KUcdWFn+N5HcS+mL7WOJdEbxy/pNGJGkbtryAb/vNjl3uaDl/kv2EhBXSqMoqIltyDJ58GvDvejs63FFlQH42xe+Bl1aFSlDannwS8fnxhP1DVY0KizTjeMiCF4N3n0kMRcfPEyuvzLd9F8PBbuUZC6y5Xa+rSbJ0H9B7Eg8wv6KkGzJytIFp6soLQEEQ0WRGf+1ASJ4Xemis1XGzR7jEl7MZykJsi4l4jEXB3uzlEvNnuMbj6kHn7UMW5hmL9x0OxZsJH07IEgBghigCAGCGKAIAYIYoAghnkTlMzFTye71ZmZoMxViNJD0qs508ludWYlKHsdq+wQ43rgdLJbnRkJyl0JhSAImo4gtEHsIejFXI+fTnarM2/jIAiCIAiCIAsQxABBIAdKEAMEMUAQAwQxQBADBDFAEAMEMUAQAwQxQBADBDFAEAMEMUAQAwQxQBDDHAgqvqQ4nexWx7+gkovS08ludbwLKrutYTrZrQ4EMUAQg3dBaIPYYNF7sYtHk39buMPNQlPNdQVYQZd7+9FZ8kTdZAS53G425Xw7wwqawkO9TjcsTjnfzlQQNLHn5rWgiXzW1HGrYvdQgsoRjfSfn032ufmlaoMIenZeMRlBy9WLifYHz83b6G/iuXl3IIgBghx5wylwS2UJ5gAIYoAgBghiwH3SDBDEAEEMEMQwGUHijJ9mRNRJrQh2jzeNeVr9G3G093IvPnG5eLR/daiDy73d7IdtnZnHe2UigujnTOnH4KQgCo7v698XFD4klFuxVya4+JM+5HhfzceJPf/c3E8+7Jk6O5Z7/HNdQfmsi03HW3JJ//R39cbLvfsnRgnSOlQqMpEE0XFcZCg42zXnWLxy7RJEJSZdl3//sz98fpIEcQm43HvwXy1o7963/ziJd4jiQxr7NJOiXAjn5GhJSlBkzOQnjcjZfZXR/WSduHgUr4pGZyvq6+hYFkA505QtLNSGLVkbtMxMQFB/3yVwS2UJPAFBDBDEAEEMEMQAQQwQxFBjoKiGv49PrIFbKkvgm9rnYnQ6Xhq4pbIE/qlbgrjALZUl8A1ONRhqCOon0x22wC2VJfBMjSq2t+8QuKWyBL5BG8RQo4od77oEbqksgWfqVDGjpSgN3FJZAt+gF2OAIIYagq4ONx/8L56YLg3cUlkCz1xf0NXh7sUXL/vqhKA0cEtlCXxTq5sX2UjPL4sDt1SWwDd1S9CZ+UcvCNxSWQLf1GuD0lPu0sAtlSXwDHoxBghiuPaE2eNf0/FuaeCWyhL4ByWIAXPSDJiTZsB8EAPaIIYags7MvqY0cEtlCTxT6w6z/lb8MGJp4JbKEvimVhsU37dqCdxSWQLf1DhZffad+E/fwVoauKWyBL6p0QaJHPQ3N3eZwC2VJfAMejEGCGKoOx+0xQVuqSyBZ2rNKIp/dWdcGrilsgS+wckqQ52R9FaU/p1LA7dUlsAztS8961mussAtlSXwDXoxBghigCAGCGKAIAYIYoAgBghigCAGCGKAIAYIYoAgBghigCAGCGKAIAYIYoAgBghigCAGCGKAIAYIYoAgBghigCCG/wPo26fVbZAQhgAAAABJRU5ErkJggg==" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m OMA1</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABNVBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZpA6ZrY6kNtHk69NTU1NTW5NTY5Nbm5NbqtNjshmAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkJBmkLZmkNtmtpBmtttmtv9uTU1uTW5uTY5ubqtujshuq8huq+SOTU2OTW6OTY6OyP+QOgCQOjqQOmaQZgCQZjqQZraQkDqQttuQ27aQ2/+rbk2rbm6rbo6rq26ryKur5OSr5P+2ZgC2Zjq2Zma2kGa2tpC225C229u22/+2/9u2///Ijk3I/8jI///bkDrbtmbbtpDb25Db27bb///kq27k5Kvk/8jk///r6+v/AAD/tmb/trb/yI7/25D/27b/5Kv//7b//8j//9v//+T///9mxKDnAAAACXBIWXMAAA7DAAAOwwHHb6hkAAALPklEQVR4nO2dC3vbthWGZSdRXafpZsuZrbRdd6u8drNXO7t0c7Ju4S5ZZ89aN3tLrEmxLPP//4ThACAJXqBDirAhit/7JBEEQY7w+hAAQQjshGAuHd8fYNmBIAYIYoAgBghigCAGCGKAIIY6gmb//KDT6Tz5W8V3/eVxp/PgRxeUDp78Vud++8mWfBx1BnHJoFvjw7mihqDpxx3FVpV3vdPveviNeBJ01k5k7mRD/5Sg041KDpOkRxYXNDvufEcEz+zbDV3J0u8Sat79uvNIxFCw9lh5GT7YkInJxgf6p83+1Gm4oFH0+ScblKDIePINGXj018edtR+HRp5x4AyjeJOJYP0r8iTe9EMlaLj+WiWm/bWf9LsLfzh3LC4oWD81noljRCB++yJEJAMjLxE0O47eNe0LNcH66/cpYCabysu03xWCSdn0NxfTZgtKf/zZ8dpnoqUVsSQEdS9k+5HkFb6LBAfrfz8mMUF3IgWRyFjmaggaUpA8uphER8ipihF6NcnLvYtQgk4DETDT/kAVpry4yEoJGukObe1EHSKz426Y5OXeRWhBI/HyaP1UClLHZEcbbbigpDUhJWUF5dqg02l/iwY8UtAwNW5ouCBR/++qBCmZbAzC5JkSlOQZZHsxiqLXoiAJitrnvnxovCDRGstB9L8/EYeYaJDFmPjdMbVBkaAkL/0ucxwkXhutfSr+JUGjaLCoYq7pgpKR9EPZkeuDIxGU5JknENPUSFo2yjSgIkFaTDTCarwgOoGiczF1OjURFX/wmXmIJXmpM6zUuRhFV6DGTFuTaECgG6oVENQGIIgBghggiAGCGCCIAYIYIIgBghggiGFhQW8V0WMmbcmuUMRlJesAQQxzBV1/fh6GN4e93Tfy6WWv13t6rl+DoDAck47bF0fh5Z58fnZkvAhB4dnOH0QE3Xx5riIpvH35yngVgkJ1iF1/8Sa8+QWpEcdaryeD6D1BvvTV1dWdfES/sILGu5Gg65++MqIo97u/IlodQZK4HcpW7erKNNQiQUkbJIEgExJz++JA92J0sN3+3tbNt1aQHgdRUoyDduJjLVe1NrZB88hXzfADQRgHsUAQAwQxQBADBDFAEAMEMUAQQ+nam+NHCMonU2cgEJRLps9hIQiCYiDIEdLPXf8ndwh6MYa7EWSTAkGxH37yrMWC7A0zBEFQDAQxzKsm2qAQvRhLrdpDEARBEAcEMUAQAwQxQBADBIW5ddJJKowELTjZsyKCMuukjRXToRa06CWL1RCUXSedXq0Y+1lkwnk1BGVXuSapaJ20FnQfn9MbFdZJJykCERTaI4h4GxtaoParJGheG9T2XiyzTjpJEbVqv0qCjHXSBeOglguaBwQxQBADBDFAEAMEMUAQAwQxQBADBDFAEAMEMUAQgx9Bb4tZtBIlgCCGhu1A1SBBiCCG1gma9jsRqR2ybbROUOb2BSwtFBQGai/5crRRUCUgiMG3oPQVuYjps/k3YppsnuZSxOx4YHlHkaA/N6CRzlzTjRj9oOy9vBYWJHqzBvRi2VUBUT2f/+77F/J2DFQH/aBrJB7WTkgL3f9lIDOTVx5+XFLQUN3XqamCJh9eBINw+tFpOOxGD2GwFY7ozh8D8fCvzVM6CoUnUiVeiQpYa20KkkOhbik9SypoKKrajRqi6EF4Ekl1SOkDS+RNlCqRogJlDrGhvAtP6fsK+hVU3AbJe5vJ27zIGwqpB3Iye34igisSFFAhEtSXdx5SeaygaV8WaYqgwl5MSlBVHakxnXjIRhBVVB1i9EqoQqxMBI1k89MYQSn0q0Oq5ahLcsRf/WC0QZPN16rxmbx/otugqECpNkhGaIMFzX4p73jz7CRQ3VdQ3IuJtkR0WnR/HJkl6126Fxs1pBcrFHQn5MZBQmcDxkEeBQn+lxJ0vR/tsr0EG24vh6AUN4dH4aVaWLYEG24voaBkbesybLjtUVB0A9nsjGL5DbfvA58RNCruv+QhtlNiw+1VjyAiKHIkGumfv+Q33G6FIHWgDXKFaAm5pu2CQvNesRJqf8psuN0SQSKCsnePH/fKbbjdAkEFduy0T9Cogp2whYJs4yAbrRNUFd+Ctre3c4LKXqmwF0n/BKKpgra3TUOF1XMviK6HlG6I/Ara3k4ZSqo3+fBXciSnL+WoybLpR1/JeqlnhUVUZsE1L0OQull3fEvqhgraSE2jqgs7dAf70Xp0maewiM6cF0Ej7U7P3jdVEM04b0aXcqILO6JKs+cn+llhEZ05R9DsOLpoOyy1yMOvIGsbZFY0GOgLO/IKWfyssAgvKAmcUSO6eUsvlgkPfWFHCKIIUs8Ki6yeoBQ5QUYDQ5fG+l3jAlBhkZU7xDhB+lKOurAzffZp1IvpS625Ijozc6YeNraRLhZkg1s1NI+Cbr7cCpsWClIDxUorOe+dxQTVoamnGhA093+FoDutcyUgiAGCGCCIAYIYIIgBghggiAGCGCCIAYIYIIgBghggiIEXlKyTxkaTRSTrpAs23Iag1DLggs1uIWjuhtttoNwhtlO84TYiiIjXSRdsuA1BCrVOGm1QIck6aWy4XUyyThrjoCpAEAMEMUAQAwQxQBADBDFAEMNdCYpWr7qsZB2WTVC8/tllJevgTVB6nXPKj3zFZSXr4EtQZqU8BGXS2e9aQFBJQWiDOEHoxbKRYivispJ18LYjufDj67+uxLKNg1YngiCIAYIYIIgBghggiAGCGCCIAYIYIIgBghhcCiqaXXRZyTosg6DCM3uXlazD/QpipllbL4ibZm27oLSJlRHkbsNti6CGt0FyCZ6bDbdtgprdi8m1rXIRZ/0Nt20T9Y0eByUR5GDD7aZMRBvwbVC8dHMZNty+f/hDTGgZP12W/aTvH1ZQ+gsIEJQjiaBl2HD7/im1kHznVX7Dbc17xWlLdsUiy0DdK6sQxABBDBDUdiCIAYIYIIjBraDrfTlSkl/e3O8dnBnzR/rbnPQSnfXS6d31/tHtC5W8OTxI/ZS9S/OtPnEqiLYeoC9uCguUPBNVlLMlNA3QU7Nt4iU5eXL9M1mazlzUhMrXvegkhk6I6auflO+f2oIytRc5Z3vawvhAfx/45vDpeRJBsuKqiJwckEEl/p7pmKH05UHqlkseqR9BFDJxUkbB5fc+P9fJKAxuDnf/owQd7vyRzucoX4QPORzvxd+rPiRHKxZBoXmur5uSS4qmwyOdUlzv66Rodvb06e+ZDD36InUqWqj5Ws02aBVxKWh8VJi2ZFcs4gsIYoAgBghigCAGCGKAIAYXA8V4d6HUv5bsikW84+5cbPeNmbZkVyziooY1cRZBubQlu2IR7+BUg8GFoHE03ZFOW7IrFvGNi0Ps8KgobcmuWMQ7aIMYXBxiZweFaUt2xSK+cXKIJS2GkbZkVyziHfRiDBDE4ELQ7Yve7n/1xLSRtmRXLOIbB4JuXxxcf/FmrDfljtOW7IpFvOOmmxfViU8wo7Qlu2IR7ziLoEvjd3+ZhEcuu2IR7zhqg+JTbyNtya5YxDfoxRggiKH+hNmX/4jHvUbakl2xiIsa1gQRxIA5aQbMSTNgPogBbRCDC0GXRp9jpC3ZFYv4xs0Ks/GevuOGkbZkVyziHTdtULR21UxbsisW8Y6Lk9WXr8QftYjVSFuyKxbxjos2SNRk3OsdZNOW7IpFfINejAGCGJzNB+3l0pbsikV842ZGUfyb3GJLpy3ZFYt4ByerDE5G0nth/Ps20pbsikV84+7Ss5rsStKW7IpF6n+8mqAXY4AgBghigCAGCGKAIAYIYoAgBghigCAGCGKAIAYIYoAgBghigCAGCGKAIAYIYoAgBghigCAGCGKAIAYIYoAgBghi+D9NR/OQOA4OrwAAAABJRU5ErkJggg==" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m VOPP1</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABLFBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZpA6ZrY6kNtHk69NTU1NTW5NTY5Nbm5NbqtNjshmAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmZrZmkLZmkNtmtpBmtttmtv9uTU1uTY5ubqtujshuq8huq+SOTU2OTW6OTY6OyP+QOgCQOjqQOmaQZgCQZjqQZraQkDqQttuQ27aQ2/+rbk2rbm6rbo6rq26ryKur5OSr5P+2ZgC2Zjq2kGa2tpC225C229u22/+2///Ijk3I/8jI///bkDrbtmbbtpDb25Db27bb///kq27k5Kvk/8jk///r6+v/AAD/tmb/yI7/25D/27b/5Kv//7b//8j//9v//+T///83fWbaAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAKHElEQVR4nO2dC3vbthWG6TaZ66ZLN1vJYve2a512sxc762521q3arfGsdLO3xJIiWeb//w/DAUARpEh8kGiZlPi9T2vykJBCvAYPwJsZxcRLVPcGNB0KAlAQgIIAFASgIAAFASgIML+gyavvR1H08J/hnxjv3b8wHz1+56X6+bcPoujdn+lFvUhz7w+5eaG7Ofe2LYG5BY0/MfWItsM/04v2zWf3VJ3f2i+49208lRJJAXder1pJQZPj6Aeq8Uy+29o4Cf7QcMvUta/qLl+g1Lz9bSTNqidNKo5fSeDOx5O/RqspqJ9stqm0NIeH34q3+3//INr4eews0z4MXVN3mfSSpqdnrBSzxpkf7238Ym8lBdmqWoZbskuotqTaRbJ3TJc5gvraiuxhJg3Fsc1MpYJ+dzFeSUHZrZ4cb3wex99tSb2jzQudNtJlmYKy24gw5wscKa+kXbrzM/9UbSwmSOfT+xfDLb27qJqZhiFr02Xu53rSonrKUl6Qzct6tTMfr4eg/rRKpolMjjfjfraa089tq7XbcYmgh06PpufjVRWUphBREi5IbAzfP4nLcpCQa3SrKUjV/4dmRpQMt5IsnApKl2VQi3t6uFjci+kFayFIZWM9iP7Pp2oXUwlZDXvfHksOSgSly3Kf+95X2/YLZsdBmvUQlI6k7+mO3A6qU0HpMqebjyVIkm9mJL1+glQX/qmkUnPENFS1ffdzdxdLl2UFTQ/Issdi6yioXVAQgIIAFASgIAAFASgIQEEACgJQEGBOQW8s0xlvEFaqOFhOdeeHggAUBKAgAAUBbkHQ1dUVBSUU1OnqKmOo5YJmuTLcxrY0ksotyApiCzJQEKCgTuU5qDx7t0pQqQdP9m6XoJLAt+9REAVRUDHz1JY5CAXsxQqCKo2mDYIqpZ0WCMolZgqioDwUBEAVZA5CFWQvNk9tKYiC8lAQgIIAFASgoAyjL86diUBBLoPO4/N0oqEgh7NHf5amYycGCsqQ3cXeU9gV63wpNcdCOSj0ZGrLWpCQ+gk5JqUgCqIg5iDAYiPpwAs67RXEcVAZFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAWRHGxBAAoCrJigN8VU91DKVNB4L3kpQZT7O7VFG9hCQTN/S9W7ga0UFHeTP0dbTrsFBUBBAAoCtFpQ19t/ZTewjYK6uZcUeDewhYImx/v2h4/mCMre45Yw/sj/Sqbhg5czc9Pa+wWN97Qg8FKsxgjK3SWZ0P9J6Fu91lxQ/unhpJ7P//iZfomQzqZ2oo4R7GTjRLTI25n29cJ0zb1P2iFo+OFFdz8ef/wy7m0mk7i7HfflHSD7avLvBy9lL1SeRJVakxSI2iGop6q6mSSiZKI8qVmzS9kdSy0bGlVqTgoE7WIrdbBamIP0W87e0TuRvOXETMTJ5PmJalyJID2eEUF7+h1EZhkUFEZjBBX2YlqCqWrfHFeqSb4Fya5idjFZE5smFtCCVk1QBru2J7Xsb4oc9b+dODlo+OAfJvkM3z+xOSgpgHPQGoykJ7+RzVetxdakW9yL9SLptOQ9XnqR3jMDejGOpItZ2ZH0nQsq6ub1UyzXh50nr/MbSEGCfpr35vQovtzNbyAFxcnTvNfPzp3noShoZhcbffk6vv71N3Hmod7aqFXQ7EhaBA2eJILcDWxhCyoi24LcDaQgzYg5KCNIv8bafYGsiLk5PWAvZuhqN0PnNdYcB7mC+jY36+6sjBYLSnv4nucKdIsFpQ2nvwonzCjI/++/ebOzszMjKPRKRXmR7DdkBK3YLraz4xq6C0GrlaR3djKG0uoNP/wqkrOD9lKOOVk2/vj3evhiosIiZqE9t1YsaNrN+y5sNF/QVuY0qrmwIy/ilt+/iQqL2IW+FmQHiv5zis0XJGecHySXcpILO2qnmDw/sVFhEbvQLyiApggqzUFuRbv79sKOvkI2jQqLYEHjn66UoLJeLNc87IUdJUhakIkKiwQIms1PjRaUYUaQk2Dk0tjepnMBqLBIyC4mjQ11jCsjyF7KMRd2xh/9OOnF7KXWmSJ2oVwMKhWk6EWR/+J8swWVge4a8pFP0sqn72ZgCopX5VBjHkFVaEULqkIrclAVVqwXu3vccdAGTmVtFvQZfJIlbrWgMCiI5GALAlAQgIIAFASgIAAFAeoRlDlfSkEzQfaMOwUlUjJ+3GsSFJRtNBQ0O59xQkFAEHMQEsRebHY+pNG0WlBIo2m3II6kKUigIAAFASgIQEEACgJQ0Cyjp3xTr4/rw6P4cvo8FAXN0KxHMu+eOQQ14annuydsF3vEFlSOStK/ekFBXuTJZwMFzSD5pzlPPd89uAUNOg166vnu4Uga0ABBxSdjl1Pd+alfUMnp/OVUd35qF1R2FXE51Z0fCgJQEKB2QcxBMGAvtlCwnOrODwUBKAhAQQAKAlAQgIIATRM0HRQtp7rz0zBB6bB6OdWdn2YJcg7MllPd+aEgAAUBGvZQr/ZT90ZkaFYLYi8WHiynuvNDQQAKAlAQYMFe7L2gIKyUJ2gAFASgIAAFARo2km4eFASgIAAFAW5H0OhpR+4UNjd7quDgrOM8v2DffCNrrw+TG/pGT49uTm1wfXiQ/bLdS/fztXIrguQlbfKKGy1IgrPH9q1JyodGaqvW6gKjX9qPnB2Z+9TVmq87R9Mve2HuGtVr6mdRQfmqq0Vnu3oqPwYHduH14eNzpwVZHaaUmJgG8VnSZCS4PHDvPa6VhVuQtJh0Xv/+L3/0xfk0SFrA9eGT/1pBh4/+8qfzZIVqPqJxIHcYGxfKuThakxYUO0+4TJPI5WNT0aPpvDB6msyqpLMbD2x0phugvgM721gkh61ZDlpnbkHQ4CgkCCvlCWqCggAUBKAgAAUBKAhAQYAKA0Uz/H127g3CSnmCuql8LCaH46VBWClPUD9VWxAKwkp5grrhoQaggqDB9HSHLwgr5QlqpsIudngUEISV8gR1wxwEqLCLnR2EBGGlPEHNVNnFnExRGoSV8gR1w14MQEGACoJuTjtP/pecmC4Nwkp5gppZXNDN6cHoy9cDc0BQGoSV8gR1U6mbV9VIjy+Lg7BSnqBuqragS/eXXhCElfIEdVMtB6WH3KVBWClPUDPsxQAUBFj4hNmzf6Xj3dIgrJQnqB+2IADPSQN4ThrA80EA5iBABUGXbl9TGoSV8gQ1U+kOs8Fu8kc6S4OwUp6gbirloOS+VU8QVsoT1E2Fg9UX36j/7B2spUFYKU9QNxVykKrBoNM5AEFYKU9QM+zFABQEqHo+aBcFYaU8Qc1UOqOoftrOuDQIK+UJ6oYHq4AqI+ndOP09lwZhpTxBzVS+9GzPcpUFYaU8Qd2wFwNQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCAP8HULdkz9kxN3cAAAAASUVORK5CYII=" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m COL5A3</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABO1BMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZpA6ZrY6kNtHk69NTU1NTW5NTY5Nbm5NbqtNjshmAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkLZmkNtmtttmtv9uTU1uTW5uTY5ubqtujshuq8huq+SOTU2OTW6OTY6OyP+QOgCQOjqQOmaQZgCQZjqQkDqQtmaQttuQtv+Q27aQ2/+rbk2rbm6rbo6rq26ryKur5OSr5P+2ZgC2Zjq2Zma2kDq2kGa2tpC225C229u22/+2///Ijk3I/8jI///bkDrbkGbbtmbbtpDb25Db27bb29vb2//b///kq27k5Kvk/8jk///r6+v/AAD/tmb/trb/yI7/25D/27b/5Kv//7b//8j//9v//+T////bgfeSAAAACXBIWXMAAA7DAAAOwwHHb6hkAAALJElEQVR4nO2djX/bOBnH3bHmwrbbS7vRdrcDjuvYQcva8XbQjpcFDo6uNNwOaGFtSJY09f//F6BHkm05tfTIcRI59e/7uVv82HI+0bePJVmOnSgGTqLQH6DuQBADBDFAEAMEMUAQAwQxQBBDeUHjb+5FUfTx30ru9Ze7UfSdz83gs1Na7t56m5TpRMQjudyLttXKb0TRB6elP+XMKC1o9CyKjIp4MmirnR5S8EG/xe2j2BQ03jfetxO15GtXrsskLpyygkQtHojkGb9rrxz47zXaih7+O46/fUZ5QW8h1Hz4ZbR6agoatLfTHQbte/L9R1urR/H4T9F24dsugrKCevovK6pAC5QMHx9RpVf/ejda+VFsrMsOE5EHj/ROwkkSqIVMUM9Q3r31dTtL0UG7VLrOlLKCOrlsV0eOqJg+PMhIui4TNN5P9vqXGYj0ODUEdW/9uh3dPlBbWkJ50vKMO2WydcaUFCQ+uRGN91dEs/tO5JIQ1KLcaBnrLHsZAdnOBHVSx1Jtole89cpvyn3KWTKdINl0rp7q1Bd1VGlBW7N11/a6FuQECbNfik5LNky0ISk3eno3WroM0oJ6ukNbOVAHxHi/FWfrru11LchnkF4ndtR9Xrrlw9ZqsI6+pKCsASElvoKyvf7x4MjaBim6Ysfu5EiiGy6FyvdiD9UCKck65kyQ2VmnJB3XaItyprgX08dmh45XmTDkT/WVyyRINJlyEP3PT8UhppqND/tJnUhQts7YKxsHPbKOg2TrPv4qoiTUg0XKzOjzpGEKw/Qj6duyI9eHQiYoW2eMg9KRdMt8CzWSTg+nXnJoJr06jbl66aAhEFOcrL77lM7FvpTLg2fqDMsQlK4zBYl8aVvOxbL2hna8f6SHoLFuugbJsDMQOJtngCAGCGKAIAYIYoAgBghigCAGCGKAIIaSgs416YIz8CtVHMynuuWBIAYIYoAgBghiqJugtbU1CHIEa2uJoflUtzz1ErS2lhqaT3XLA0EMEMRQL0Fog9gAvZhvMJ/qlgeCGCCIAfNBDMggBghigCAGCGKAIAYIYoAgBghigCAGCGKAIIaFCUoneiCocDmbKoSgomVjshmCIAiC7EwrCG0QJwi9GCcI46CpBRXn1nyqW57wgiyt03yqW57ggmz923yqWx4IYoAghuCClroNGr44iePL3fWN98maOQha4l6sv/7kJL463IvPNpNV8xC0tOOg48d/EBl0+fJEZZIEgnKQmOEX7+PLn70R0UeCBX2qGsEK6m8kgghkUI58BhEQlGOINogXdHW4g17MxtzGQR6TQ/Ovuh9BRtI+04vzqW55QgjymqCeT3XLA0EMEMSANoghzHQHerEypSAIgpohiB4qe9BlnqTdZEEdeoomPbCw4zDUYEGjrW3tpud4QHrTBXXpEY8QZGIcYsnzUjstS9k4vKDzYmYoZBKjF1OGBm2Hn2YL8gGCiD/Xtw2qgaDRlutnPhovqKt/lsBGswWJ5IkiVxPdbEFd9TRwCMqTHyi6B0FxowXJR8pvc4JCkzm5uLgoEDR66n76/eDO22tLxHjf1vJGuVLL0wZdXJiG0u29H/r+TNI0guIkjezURdDFRc5QWs9Xv/3+qawFDVX0ix64iJeVA9JCv9CwLVdmW24/8xQk02gJxkEWQYP7p53tePTJWzrt1i80RdGj3wDZFi/f3nlLR6HwRKrElqSANS2KRtL/W1pBXVHVVtIQJS/Ck1hUh5Q+sMS6gVIllqiAxyEmezE5ElqGQ6y4DZKN6C15ENHMn3ohJ+NXByK5EkEdKkSCtuRvmqh1XoKU8zrPKGblinoxKUFVtacmb8TLZAZRRdUhRltilWKeGdSV71rnCbPzYvTWrvzxthbJEf/rF6MNGtz5WjU+g+8e6DYoKcC3QSSot9yCxj+nDy6ypaO6r05xLyZOGUSnRb+LI1fJI9OjF6MDcuWePMQwo2hg9mKiUWtN/Iqh7QM2U5AHEBQvcRs0FyCIAYIYIIgBghggiCE3UNRAkAHGQQwQxDBxqsH95mujBakfyMW3O/IYl31026wvkLk/YHhB+a+iK3yvVNiL5N8hJ2i8n0wkdh2/PF0bQRM3MyxAUJY4E+Og4XO6O7xmgiZu98iqN7j/Czmrri/lqMmy0Se/km2rigqLqJUFX21hBV3u7sVn6R119RfUzk2jqgs79Mv1VCsVFRbRK6c4xOp1Uy8viGac7ySXcpILO+KPP351oKPCInqlQ5Ctka7XffPnOUPnVkH6Uo6+sCOv1qRRYREPQWk3n7/qIw+xx3XLIGsvNpEe+sKOEEQZpKLCIj6C1EDx2omYaKR/+rp+gnJcE2Q0MHRpbKtlXAAqLOIlyArdO79kgvSlHHVhZ/T0B0kvpi+1XiuiV9LFoJKCqP2pz33zbkE2uG8NuTAEydZ5vD85TOyvz/n5QcsiKGmdO47poCUVVIXc7VATC44P2EBB1lONwg8IQRZCC1o8S3Y2v3iML5LrxMlMFdBkQfpUg8587cUbLWgZ5qQXz5Jd1Vg8EMQAQQwQxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMRQUlDzmC6DcvdDIoMMUj+5O0YhKCXzY95zDEEpEMQAQQznmSGP2jZWEHoxK0V1stuCIO3HdrxB0Lm7xYYgCJoEghgK6oQ2yKSoTujFDMrVFoIgaBIIYoAgBghigCAGCGKAIAYIYoAgBraCfnOxN0pQqQcLeM7m3yRB8pZM3wcLTD7FuAmC5H3zvjf1NlFQlkEeDxbQgmbz0eoB3waV+jHsBrZBwx+/iftPvO+bb14v1t+o08NNFs+sM2jieGuAIHqwQPpsk1KCfK/gL7ugHCVq6/0dEAiCIAhKKVNbtEFc0JBeLEe52jZiHJQHghggiAGCPPnIK/Ar5QhqAAQxQBADBDHgfjEGCGKAIAYIYpiNoOFzOeuorp+JYOd43bgcO3whF2nr5W7y4NPh872rQx1c7u7k32zzzNw/KDMRdHW4J/7fUYIoOBa1k1cchQ8J1VZslQWGP9G7HO+p65Jiy+/X99I3e62eriu3hGdaQZNVF6uON+Ur/dPf0Ssvd0lVmkFahypFJtIgPk5ShoKzHfMZzUGZOoMoY7Jl+fc/+96LkzRIMuByd+M/WtDu4z/+7iTZINKHNPbpSczKhXBOjm5IBsXGxbK0ETl7oiq6ly4Tw+fJomh0NtNrSMcyAeWTqvPJQm3YDWuDbjIzENTf8wn8SjmCQEAQAwQxQBADBDFAEAMEMVQYKKrh78sTZ+BXyhGEpvK5GJ2OWwO/Uo4gPFUziAv8SjmC0OBUg6GCoH463eEK/Eo5gsBUOMR29zwCv1KOIDRogxgqHGLHOz6BXylHEJgqh5jRUlgDv1KOIDToxRggiKGCoKvD9Y3/JhPT1sCvlCMIzPSCrg53hl+876sTAmvgV8oRhKZSNy+qkZ1fFgd+pRxBaKpm0Jn5Ry8I/Eo5gtBUa4OyU25r4FfKEQQGvRgDBDFMPWH28u/ZeNca+JVyBOFBBjFgTpoBc9IMmA9iQBvEUEHQmdnXWAO/Uo4gMJW+YdbflF+WcwV+pRxBaCq1Qcn3Vh2BXylHEJoKJ6uv34j/9DdYrYFfKUcQmgptkKhBf319hwn8SjmCwKAXY4AghqrzQZtc4FfKEQSm0oyi+Fd3xtbAr5QjCA1OVhmqjKQ34+zvbA38SjmCwFS+9KxnuWyBXylHEBr0YgwQxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMQAQQwQxABBDP8HdYQFbNSsarcAAAAASUVORK5CYII=" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m MMP10</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABKVBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZpA6ZrY6kLY6kNtHk69NTU1NTW5NTY5Nbm5NbqtNjshmAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkJBmkNtmtpBmtttmtv9uTU1uTW5uTY5ubqtujshuq+SOTU2OTW6OTY6OyP+QOgCQOjqQOmaQZgCQZraQkDqQttuQ27aQ2/+rbk2rbm6rbo6rq26ryKur5OSr5P+2ZgC2Zjq2kDq2tpC225C229u22/+2/9u2///Ijk3I/8jI///bkDrbtmbbtpDb25Db27bb///kq27k5Kvk/8jk///r6+v/AAD/tmb/yI7/25D/27b/5Kv//7b//8j//9v//+T///+MgjNvAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAKeUlEQVR4nO2dC1vjxhWGDdl1CGnSFrx0Idk0vQSaFLqQtmmXtM06ybaFQi7QbrAdjNH//xGdq26W5htZNiNb3/sk2Gc0XjwvZ2Z0VyciTjqhv0DToSAABQEoCEBBAAoCUBCAggDVBU2++Umn03nnn1U+03/nL+bdtx9sTYWXHcUjW9jvqpcfn4lf86/K32/OVBY0fqab09mq8KF+Z+2FejPakJ/LhUZQp7OvCi87StB4Txatn1X9gnOmqqDJSeenInkm326YNnrRX3tb+7x8Y2NrKrzUFr7pPL4S//KXHS3osvPzq8lXlf4Oi6CqoIH+9vKPL9/YbjA5efz1252130SpMlF333yqv/6ZbLyo9mslKBsaQaL4TOTN2m/3umqZ0qV+hqSqoH4m50UXEYhcEollO0lclhH06i2ZcKPNV1pQJswI+vPVWAnSP3O/LwAVBZmvbZicrH0kBlqRS0JQ90qNHklZiv76v09kX+l3R1pQJoy7WDf1O9QSsaxKT14EswlSw+rjK9uI9bPJiWykXJqUpT4m8qAv+sp4b98IyoR2kDYuVkjQIG6ZHTG60SDbWo0wMhAFg/UzKygdGkF2Sl9qQTpT9LtqgsZ7W3IFxwpKh9lsi+wYtGU+Wad59ak+i/1Mv5FKRht2FE4EJWUpZDPFyCwWWUHpsFDQss5iYjRWK9HffyC6mBiQxdrvjydyDLKCkrLUp6SRwdqH4mcsKBUWChJrk2I96MulWw9K1qQfqYncrFQngpKyzDQv13DkLBULSoXFgpZ0TTqS209yRNUbTiOh642P0l0sKcsJEimxnxKUCosFqRXOR8u3LdY2KAhAQQAKAlAQgIIAFASgIAAFASgIUFHQD4b4jTPwq1UcLKa51aEgAAUBKAhAQYAwgra3tynIsWB7O2OIgnLB9nbWEAVREAVZ5iKIYxBcwFmsSi0KoiAKajHMIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgwIMJ8jiQ0WpBPofC2izI62Dq8gm6/fhC/Djo9Y5tCQWlGfZ2LqK7P7yMbn/30hRRUIrzJ/8QGTTclW9tCs0qaEXHINXFBDKLouhNwey/SPiZ/cMB8RF0f3poS2bOoNVcD9KC7o5iPxSUxcxix0kJBWWQgjJ+KCiLFHTdk9SexVZT0DQUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEKDVguxNeQtuNRrTZkH2trLZW0bnaLGg8Z7NnEHw24OW80MxC/yNFQWFzqCAgqK+uTFxo7tYSEHxrbKbPEgHFeQDBQEoCEBBgBYL0ndI74DbpLdY0NSd051fMLygm5ubAkHj99yP4Rhtnk29k0xOyubudBfr40c0NEbQzU3aULx88CvfpyjMIsiDpgi6uckYitv56V9/eaXW6GRvMC9i8DAvay+kFvl0pn1VmCx59KwdgkbvXvX3o/H7Z9Fl175E/a1oIJ8Bsi9evts8k71QeJKqxBJboXT1eLUEXYqmdu1AZF+EJ/FWdynTsUTZSKsS72QFry7W93jMR1MEFY9B6iln6nkv6tFC+kU6mXz6QiSXFaQaKgXtqWcQ6TIsSA7RcCJrjKDCWUxJ0E0d6AlHvOQzSO620F1MLol0inlkkKpSXq9xgjKYpZfy2w+6Uo7437ykxqDR5is9+IzeemHGIFsBj0Fqf9DkBMySjRY0+aPMCJEtZrDoF89ilx05ackHgaki1TM9ZrEVELQQKAhAQQDnxqo60/7uqPf0NQUVoa4Xuz89jq5381+QgiJ7vdjd84v4sigKyqGu1fjk9VyuF5sPDRQ0fGoFpb9gCzPIPq05P0gnGdRyQXLvSX5t8pZjULaL9XOOpBh5PSZnsRjZ0RJHXA8qGqQnJzyqkWI6g5yP5263IGQnapKg7H0KNL5HKsqrZP+FrKABtNMkQbk7XTyAoKL1oOYKyt0LJGne6N0/qWnGHMrRO8vG73+m/vo6KqyiC82+tUJBfjRf0EZmN6o+sDPee3wlz5vTUWEVU+jqYisiSO5x3rSHcuyBnb19eWDDRIVVTOHKCCodg9IN7e+bAzvqCFkcFVZZOUFls1guPcyBHSFIZpCOCqtgQcs1SOeYEpQaYOShsb1u6gBQYRWPDCrYWF1iQeZQjj6wM37vQzuLmUOtU1VM4fRmhHNjddkElYHOGnLh3Fgt/4KtFRRxYzUHN1YBS7ux+vCClmJj9eFZsvWgh4fPFwMs2abGw0NBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgI0TVB80sZimludBghKn8iSnPazmOZWBwu6PZCXjRkWICh9KlTqxLGFtroCUJC8kOU6PtV+/oIyJ9MtoyB1tc/zxV3MsvSCkgxazAV1xkk6mv8vqQEegxZ9MUv2dMzlm8XkAyCHO4u8Xqz4EbULbrc3UFCzLsl8eBqQQUsuKBr2ek9sAlEQgoIAFASgIAAFASgIQEEACgJQEICCALCB2fs/UlA+yN1BlIJyQf4etBREQTkoCIAayDEINZCzGHDC9aCpNmV7VdsFTWPG5Xl8l0YyWwalciY3ceVyq3UZlJICBTn63ooLyjop8+BIrXYJKutJFARa22JBnqNLe8cg3/mpvbMY9yiWQkEACgJQkCdvegV+tRxBA6AgAAUBKAjAGywBKAhAQQAKAsxH0O2BOlVYX/QigsPzXuoaKvNgF7n07qhnLmy4PTi+PzXB3dFh9h/bvU5/PihzESSfQSaf4KIEyeB8xzwUSPhQyNaKparC7e/NR86P9cVEYsnfe8fxP/b5S/UoGLUkPLMKyjddFJ3vqlf5Y3hoCu+Odi5SGWR06FrSRBxE5zZlZHBtPx+emTNIZkzyXv39r3/x8UUc2Ay4O3r6XyPo6MkXf7uwC0T6SI1D+dgg7UI4l45WJIOi1BUu8SByvaMbehy/l9we2Ldi0NmNL/w4VwmoHquUTRY5hq3YGLTKzEHQ8Ngn8KvlCAJBQQAKAlAQgIIAFASgIECNFUW9+vv8whn41XIEoam9LSY3x0sDv1qOIDx1MwgFfrUcQWi4qQGoIWgY7+5wBX61HEFganSxo2OPwK+WIwgNxyBAjS52fugT+NVyBIGp08VSI0Vp4FfLEYSGsxiAggA1BN2f9p7+z+6YLg38ajmCwMwu6P708PaT10O9QVAa+NVyBKGpNc2LZiTbl8WBXy1HEJq6GXSd/qMXBH61HEFo6o1BySZ3aeBXyxEEhrMYgIIAM+8we/6fZH23NPCr5QjCwwwCcJ80gPukAdwfBOAYBKgh6Do915QGfrUcQWBqnWE23FUny7kCv1qOIDS1xiB73qoj8KvlCEJTY2P185fiP3MGa2ngV8sRhKbGGCRaMOz1DkHgV8sRBIazGICCAHX3B+2iwK+WIwhMrT2K4qeZjEsDv1qOIDTcWAXUWZPejZK/c2ngV8sRBKb2oWezl6ss8KvlCELDWQxAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAvwfp6sKU7NmWjUAAAAASUVORK5CYII=" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m CDCA4</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABL1BMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZpA6ZrY6kNtHk69NTU1NTW5NTY5Nbm5NbqtNjshmAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkNtmtrZmtttmtv9uTU1uTW5uTY5ubqtujshuq8huq+SOTU2OTW6OTY6OyP+QOgCQOjqQOmaQZgCQZjqQkDqQttuQ27aQ2/+rbk2rbm6rbo6rq26ryKur5OSr5P+2ZgC2Zma2kGa2tpC225C229u22/+2///Ijk3I/8jI///bkDrbkGbbtmbbtpDb25Db27bb29vb2//b///kq27k5Kvk/8jk///r6+v/AAD/tmb/trb/yI7/25D/27b/5Kv//7b//8j//9v//+T///+NPxfbAAAACXBIWXMAAA7DAAAOwwHHb6hkAAALM0lEQVR4nO2djV/jthnHTXek7PpyhXADWrruJazdyAq3lw5uL5dt3R0jXbuNbFeypITg//9vmB5JtmTHymPHutiOf99Pe1FkGaJvZL1YQg5CsJSg6g9QdyCIAYIYIIgBghggiAGCGCCIoYyg+T/eDYLgvb8VPOsv7wTB935KwWEgee/lwhHBOOjF58yOOiU+ZylKCJodq/wFu0XOmu6okz4IY0GBUmEfEQwCI8UOr5nVBc3Pgyei8My/3tm6yH/W7Cj44N9h+M0xSRm+9Yp+0FfyJySOkK53458rRDZQ0Dj60NMdCnx3rK6V+fn2X98Jtn4cWnHW5TLU5W26s32jBdHh3dQRkvflzm70C37SxEtsoHOnUNeH+M5FwYqumjjOCJqfR2f9KwxjQULqTfII1ToUq4NNrIOSn3l+viUq169FWRKCOqJsBB0rznVWJIhkpxyQUq11sH3TYEGynt2+marLQeRYlQQ6auIWztIsEUTlU0WNRRFsvqCxbo62LtSFMT/vhCZu4SyNW5Bu0cRxqbmJgkydQUryCjJnffXkpaMOkkfiHsQw46eskzKtmO6wUPamO73QvFOCTJxF1FbNjoSPzFaMjuj6eXa0fdNcQaI2lp3of34c0Pe/9YVo1s+jvJEgE2edZXo7u1Y/SFVb5shYyxpoLU28xKye9CPZkOtLwggycfawIeovU46ze9IdIybqazVTkGjCP6aR1BcyPD1W4yhLUByXGFd995ud5Fhs68nL1JFp1DWIKqamCmoDEMQAQQwQxABBDBDEAEEMEMQAQQwQxLCyoG8V0Wsq7IgukMRnJssAQQwQxABBDBDEUDtBe3t7ELQkem9PG/KZyTLUTNDeXmTIZybLAEEMEMRQM0Gog9hotGI5k/jMZBkwmmdACWKAIAYIYoAgBghigCAGCGKAIAYIYlivoGigBUHMUB2CmJs9ENQSQXefXofhfb978Fq+HXW73afX+hgEheGEdDxcnoWjQ/n+6sw6uIKgTauDrvb/KErQ/efXqiSFD89fWEdXEbRxrRiJufvsdXj/S1IjrrVuVxaitwVr+XQ1gBU0OYgE3f3shVWKVipBm9ZRTJYgSVwPQVCoBJk6SAJBNiTm4fJUt2J0sT38vkQzv5mCdD+IgqIftB9faxDEAEEMEMQAQQwQxABBDBDEAEEMEMQAQQwQxABBQIISxABBDBDEAEEMEMQAQQy1EGTPJ0LQYjAxIw1BC8HkmgYIgiAIkvgThDqITYJWbIUkPjNZhrSg+bm1Q31qCZ4JhS0UNDuK9g0Nos1XU0vwrMV4YVWCvs1mHYL0BrNWCUovwUsuhGmfILVx6sIlZhZQmZCPJXiiYl7hrEoF0eazvQVBZgmeCVkfddUSlNm017wEhVQRPTrOUYKsj7qioOzOYf0F0TNQUoLeTB3UXEHEn6MtxJNL8EzI+qjtEySa+7dsQdYSPJ/9oIbWQXKP8F6uE0sKyhxe1F2Q7Cvm3Ru9rKCy/aDb29sMQbMPlz88Yfr41UKImJ/3HGcYQUP5aIZBQwTd3tqG4uPjH+V9lldhQbMjmaQhgm5vE4bifD773Q9v5OMYqB7VL7pWFS9bF6SFnr7Qk5HmyKNjVpB8yEOv4YKm798MeuHso1fhsBO9hIPdcExP/uiJl28ev6KrUHgiVeJIlMBZ89p1kHzyVaMFDUVWO1FFFL0ITyKoLil9YYm4qVIlQpQgTx0kGa+tFXsTdZD8huUDk+STbtQLOZk/uxCFKxI0oEQk6Eg+EkfF5RQkf0niWTM1FZTZikkJKqtj9fA28ZIuQVTZqkuMjoSqiOUuQZL/NUFQAn10SLkcd0iO+F+/WHXQ9PGXqvKZfv9C10FRgjx1kGzHoufmNVHQ/Ff0zYrSMlDN1yC7FRP9GdFo0c0dGSUvmjytmH6gXOrZhU0S9EYwguIGPndLXwEVCtIdRcG4EZU0BC37rVUIMvdah7mq6dYJsp7ImGu81z5B+rl59ODFPCe2UJB6yGDep1G2UVAhIIihjYLG5pGePO0TpB/ZiUo6idXMazFo5hNYHcWeDqGjaNPYoUZyXk2Rd6bCnST5Ewhe0N1JtLNb2U3efApKzcxWKOi+fxaO1Ixz6U3e/AlKz+3H2Zu+/2vZEOupHHWzbPbRb2UHWL3LTKIi7Xn3vILMohcPm7ytQdBO4jaqmtihlplypd5lJtGRS0vQwhrFlKA6bPLGC6I7zo+jqZxoYkd8+fNnF/pdZhIduUSQC3mJ7ddlk7c4lasOsjM66OmJHTlDFr/LTLK6IKqkf/G8Lpu8mWSOVixVPPTEjhBEJUi9y0ySR5B7VoPWltVNUIIFQVYFQ1NjRx1rAigzSQ5BjlkNqn/qs8lbbkF6KkdN7Mw+/CRqxfRU60ISHanHW5mCXLMak26NNnlbLsgFt2poGY3tSUPQst9aUFAZMKvBgFkNhobNaqyfhs1qrJ+G3bRfPxDEAEEMEMQAQQzY5I0BJYgBghggiAGCGCCIAYIYIIgBghggiAGCGCCIAYIYIIgh1+IFvays3XuYuTArzOqwh9n6YQVZC6hqsIfZ+ikgyO8eZk2hwAozr3uYbU4JMivMfO5htkmCwmiFGeqgTMwKM597mG2QIGuFGfpBRYAgBghigCAGCGKAIAYIYoAgBghiyJ375G5sEJQOpvbza7OgTBHpHSFbLChbRHsFpdEmckY3B28lKFlUUAflFoRWLLOooB+UtxWDoJVyD0EQBEEcEMSwNJuupr19grJNODuHrROUbcI9/mqbIIcJCCIgKGLJJm8uE62qg+TiBdcmby4TbWrF5Kogufwla5O3FYfqmyTIlKA6bPK2fvg6KF70UodN3tYPf4kJLZOnddnDbP2wgpJLNyFoAVOCkpu8ad7ODjuiCyapA7mW4O2/WNzkTQNBDBDEAEFtB4IYIIgBghj8Cro7kR0B+VcdJ93TK+v2iPXQehrU0ejl7uTs4VIF7/uniZ9yOLJPrRKvguhvEukvOoQFCl6JLMqbATTK7aqbSeKQvDdw93OZmjrm6n7BH7pRH53Ge/Q3IRRfPaUFpXIvYq4OtYXJqf5Dofv+02tTgmTGVRI59pWFSvx/pcsMhUeniV2sK6R8CaIiEwdlKRj94NNrHYyKwX3/4D9KUH//TzRcoXhRfMjh5DD+g6s+OdqwEhTaQ1ldlYyoNPXPdEhxd6KDoto51KO7K1n06C+sEqWFqq/NrIM2EZ+CJmeZYUd0wSRVAUEMEMQAQQwQxABBDBDE4KOjGG87kPjXEV0wSeX4G4sdvLbDjuiCSXzksCTeStBC2BFdMEnlYKjB4EPQJLrdkQw7ogsmqRofl1j/LCvsiC6YpHJQBzH4uMSuTjPDjuiCSarGyyVmagwr7IgumKRy0IoxQBCDD0EPl92D/+ob01bYEV0wSdV4EPRweXr32euJ3q0zDjuiCyapHD/NvMhOPMCMwo7ogkkqx1sJGlnf/cgUj4Xogkkqx1MdFA+9rbAjumCSqkErxgBBDOVvmH3+97jfa4Ud0QWT+MhhSVCCGHBPmgH3pBlwP4gBdRCDD0Ejq82xwo7ogkmqxs8Ks8mh3orbCjuiCyapHD91ULR21Q47ogsmqRwfg9XnL8R/ahGrFXZEF0xSOT7qIJGTSbd7mg47ogsmqRq0YgwQxODtftDhQtgRXTBJ1fi5oyj+Nc/e0GFHdMEklYPBKoOXnvRhGH/fVtgRXTBJ1fibelY3u0zYEV0wSfmPVxK0YgwQxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMQAQQwQxABBDP8HIi5UAIelqWUAAAAASUVORK5CYII=" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m TACSTD2

head(results)
#&gt; log2 fold change (MLE): HPV_status Associated vs Independent 
#&gt; Wald test p-value: HPV_status Associated vs Independent 
#&gt; DataFrame with 6 rows and 6 columns
#&gt;         baseMean log2FoldChange     lfcSE      stat    pvalue      padj
#&gt;        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
#&gt; A4GALT   438.986      -1.168114  0.681001 -1.715290 0.0862921  0.359863
#&gt; AAAS     485.090       0.116862  0.677368  0.172524 0.8630256  0.956720
#&gt; AACS     412.853      -1.235676  0.703735 -1.755881 0.0791088  0.344576
#&gt; AAGAB   1116.256      -0.588544  0.392074 -1.501104 0.1333288  0.447441
#&gt; AAK1     744.302       1.111357  0.653029  1.701849 0.0887838  0.365756
#&gt; AAMDC    212.438      -1.218844  0.649871 -1.875516 0.0607217  0.294944</code></pre>
</div>
<div id="gsea-gsea-for-htg" class="section level4">
<h4>GSEA: GSEA() for HTG:</h4>
<p>A GSEA analysis can be performed using the results from the DEA
contrast. HTG_GSEA will generate plots and Excel files for easy
interpretation.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="fu">HTG_GSEA</span>(res_tutorial)</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a><span class="co">#&gt; [32mPerforming gseGO  analysis[0m</span></span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a><span class="co">#&gt; [32mPreparing gene list for GSEA[0m</span></span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a><span class="co">#&gt; [32mPerforming gseGO Analysis[0m</span></span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a><span class="co">#&gt; preparing geneSet collections...</span></span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a><span class="co">#&gt; GSEA analysis...</span></span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a><span class="co">#&gt; leading edge analysis...</span></span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a><span class="co">#&gt; done...</span></span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a><span class="co">#&gt; [32mCreating Dotplot for gseGO [0m</span></span>
<span id="cb21-10"><a href="#cb21-10" tabindex="-1"></a><span class="co">#&gt; [32mCreating Emaplot for gseGO [0m</span></span>
<span id="cb21-11"><a href="#cb21-11" tabindex="-1"></a><span class="co">#&gt; [32mCreating Ridgeplot for gseGO [0m</span></span>
<span id="cb21-12"><a href="#cb21-12" tabindex="-1"></a><span class="co">#&gt; [32mCreating Heatplot for gseGO [0m</span></span>
<span id="cb21-13"><a href="#cb21-13" tabindex="-1"></a><span class="co">#&gt; [32mCreating Treeplot for gseGO [0m</span></span>
<span id="cb21-14"><a href="#cb21-14" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-15"><a href="#cb21-15" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-16"><a href="#cb21-16" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-17"><a href="#cb21-17" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-18"><a href="#cb21-18" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-19"><a href="#cb21-19" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-20"><a href="#cb21-20" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-21"><a href="#cb21-21" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-22"><a href="#cb21-22" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-23"><a href="#cb21-23" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-24"><a href="#cb21-24" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-25"><a href="#cb21-25" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-26"><a href="#cb21-26" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-27"><a href="#cb21-27" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-28"><a href="#cb21-28" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-29"><a href="#cb21-29" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-30"><a href="#cb21-30" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-31"><a href="#cb21-31" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-32"><a href="#cb21-32" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-33"><a href="#cb21-33" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-34"><a href="#cb21-34" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-35"><a href="#cb21-35" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-36"><a href="#cb21-36" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-37"><a href="#cb21-37" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-38"><a href="#cb21-38" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-39"><a href="#cb21-39" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-40"><a href="#cb21-40" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-41"><a href="#cb21-41" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-42"><a href="#cb21-42" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-43"><a href="#cb21-43" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-44"><a href="#cb21-44" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-45"><a href="#cb21-45" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-46"><a href="#cb21-46" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-47"><a href="#cb21-47" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-48"><a href="#cb21-48" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-49"><a href="#cb21-49" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-50"><a href="#cb21-50" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-51"><a href="#cb21-51" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-52"><a href="#cb21-52" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-53"><a href="#cb21-53" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-54"><a href="#cb21-54" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-55"><a href="#cb21-55" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-56"><a href="#cb21-56" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-57"><a href="#cb21-57" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-58"><a href="#cb21-58" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-59"><a href="#cb21-59" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-60"><a href="#cb21-60" tabindex="-1"></a><span class="co">#&gt; [32mPerforming KEGG Analysis[0m</span></span>
<span id="cb21-61"><a href="#cb21-61" tabindex="-1"></a><span class="co">#&gt; &#39;select()&#39; returned 1:1 mapping between keys and columns</span></span>
<span id="cb21-62"><a href="#cb21-62" tabindex="-1"></a><span class="co">#&gt; Warning in clusterProfiler::bitr(names(original_gene_list), fromType =</span></span>
<span id="cb21-63"><a href="#cb21-63" tabindex="-1"></a><span class="co">#&gt; &quot;SYMBOL&quot;, : 2.9% of input gene IDs are fail to map...</span></span>
<span id="cb21-64"><a href="#cb21-64" tabindex="-1"></a><span class="co">#&gt; Reading KEGG annotation online: &quot;https://rest.kegg.jp/link/hsa/pathway&quot;...</span></span>
<span id="cb21-65"><a href="#cb21-65" tabindex="-1"></a><span class="co">#&gt; Reading KEGG annotation online: &quot;https://rest.kegg.jp/list/pathway/hsa&quot;...</span></span>
<span id="cb21-66"><a href="#cb21-66" tabindex="-1"></a><span class="co">#&gt; Reading KEGG annotation online: &quot;https://rest.kegg.jp/conv/ncbi-geneid/hsa&quot;...</span></span>
<span id="cb21-67"><a href="#cb21-67" tabindex="-1"></a><span class="co">#&gt; preparing geneSet collections...</span></span>
<span id="cb21-68"><a href="#cb21-68" tabindex="-1"></a><span class="co">#&gt; GSEA analysis...</span></span>
<span id="cb21-69"><a href="#cb21-69" tabindex="-1"></a><span class="co">#&gt; Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =</span></span>
<span id="cb21-70"><a href="#cb21-70" tabindex="-1"></a><span class="co">#&gt; minSize, : For some pathways, in reality P-values are less than 1e-10. You can</span></span>
<span id="cb21-71"><a href="#cb21-71" tabindex="-1"></a><span class="co">#&gt; set the `eps` argument to zero for better estimation.</span></span>
<span id="cb21-72"><a href="#cb21-72" tabindex="-1"></a><span class="co">#&gt; leading edge analysis...</span></span>
<span id="cb21-73"><a href="#cb21-73" tabindex="-1"></a><span class="co">#&gt; done...</span></span>
<span id="cb21-74"><a href="#cb21-74" tabindex="-1"></a><span class="co">#&gt; [32mCreating Dotplot for KEGG[0m</span></span>
<span id="cb21-75"><a href="#cb21-75" tabindex="-1"></a><span class="co">#&gt; [32mCreating Emaplot for KEGG[0m</span></span>
<span id="cb21-76"><a href="#cb21-76" tabindex="-1"></a><span class="co">#&gt; [32mCreating Ridgeplot for KEGG[0m</span></span>
<span id="cb21-77"><a href="#cb21-77" tabindex="-1"></a><span class="co">#&gt; [32mCreating Heatplot for KEGG [0m</span></span>
<span id="cb21-78"><a href="#cb21-78" tabindex="-1"></a><span class="co">#&gt; [32mCreating Treeplot for KEGG [0m</span></span>
<span id="cb21-79"><a href="#cb21-79" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-80"><a href="#cb21-80" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-81"><a href="#cb21-81" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-82"><a href="#cb21-82" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-83"><a href="#cb21-83" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-84"><a href="#cb21-84" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-85"><a href="#cb21-85" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-86"><a href="#cb21-86" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-87"><a href="#cb21-87" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-88"><a href="#cb21-88" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-89"><a href="#cb21-89" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-90"><a href="#cb21-90" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-91"><a href="#cb21-91" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-92"><a href="#cb21-92" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-93"><a href="#cb21-93" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-94"><a href="#cb21-94" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-95"><a href="#cb21-95" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-96"><a href="#cb21-96" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-97"><a href="#cb21-97" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-98"><a href="#cb21-98" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-99"><a href="#cb21-99" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-100"><a href="#cb21-100" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-101"><a href="#cb21-101" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-102"><a href="#cb21-102" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-103"><a href="#cb21-103" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-104"><a href="#cb21-104" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-105"><a href="#cb21-105" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-106"><a href="#cb21-106" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-107"><a href="#cb21-107" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-108"><a href="#cb21-108" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-109"><a href="#cb21-109" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-110"><a href="#cb21-110" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-111"><a href="#cb21-111" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-112"><a href="#cb21-112" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-113"><a href="#cb21-113" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-114"><a href="#cb21-114" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-115"><a href="#cb21-115" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-116"><a href="#cb21-116" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-117"><a href="#cb21-117" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-118"><a href="#cb21-118" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-119"><a href="#cb21-119" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-120"><a href="#cb21-120" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-121"><a href="#cb21-121" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-122"><a href="#cb21-122" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb21-123"><a href="#cb21-123" tabindex="-1"></a><span class="co">#&gt; [32mPerforming GO Enrichment Analysis[0m</span></span>
<span id="cb21-124"><a href="#cb21-124" tabindex="-1"></a><span class="co">#&gt; [32mPerforming GO Enrichment Analysis[0m</span></span>
<span id="cb21-125"><a href="#cb21-125" tabindex="-1"></a><span class="co">#&gt; [32mCreating Bar Plot for GO Enrichment[0m</span></span>
<span id="cb21-126"><a href="#cb21-126" tabindex="-1"></a><span class="co">#&gt; [32mGO enrichment results saved to &#39;go_results.csv&#39;[0m</span></span>
<span id="cb21-127"><a href="#cb21-127" tabindex="-1"></a><span class="co">#&gt; Picking joint bandwidth of 0.336</span></span>
<span id="cb21-128"><a href="#cb21-128" tabindex="-1"></a><span class="co">#&gt; Picking joint bandwidth of 0.384</span></span></code></pre></div>
</div>
<div id="tme-htg_tme-for-htg" class="section level4">
<h4>TME: HTG_TME () for HTG:</h4>
<p>This function performs Tumor Microenvironment (TME) analysis, which
will be stored in <code>TME_results</code>. You can access each method
using <code>$</code>. This object contains results from three
methodologies.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a></span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>TME_results<span class="ot">&lt;-</span> <span class="fu">HTG_TME</span>(<span class="at">outliers=</span> outliers_tutorial, </span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>                      <span class="at">pattern=</span> <span class="st">&quot;^NC-|^POS-|^GDNA-|^ERCC-&quot;</span>,</span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a>                      <span class="at">counts_data=</span> counts_data_tutorial, </span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a>                      <span class="at">AnnotData=</span> AnnotData_tutorial, </span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a>                      <span class="at">design_formula=</span> <span class="st">&quot;HPV_status&quot;</span>, </span>
<span id="cb22-7"><a href="#cb22-7" tabindex="-1"></a>                      <span class="at">remove_outliers =</span> <span class="cn">TRUE</span>, </span>
<span id="cb22-8"><a href="#cb22-8" tabindex="-1"></a>                      <span class="at">DEA =</span> <span class="cn">NULL</span>)</span>
<span id="cb22-9"><a href="#cb22-9" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Inicial gene number: 19394&quot;</span></span>
<span id="cb22-10"><a href="#cb22-10" tabindex="-1"></a><span class="co">#&gt; [32mTPM normalization performed and stored on tpm_counts.csv[0m</span></span>
<span id="cb22-11"><a href="#cb22-11" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt;--- This function is being optimised and we strongly recommend that you should set `source` as `local`....</span></span>
<span id="cb22-12"><a href="#cb22-12" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Number of genes omitted during TPM normalization due to their length not being available in Biomart:   1086&quot;</span></span>
<span id="cb22-13"><a href="#cb22-13" tabindex="-1"></a><span class="co">#&gt; Performing normalization.</span></span>
<span id="cb22-14"><a href="#cb22-14" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-15"><a href="#cb22-15" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt; Running epic</span></span>
<span id="cb22-16"><a href="#cb22-16" tabindex="-1"></a><span class="co">#&gt; Warning in (function (bulk, reference = NULL, mRNA_cell = NULL, mRNA_cell_sub = NULL, : The optimization didn&#39;t fully converge for some samples:</span></span>
<span id="cb22-17"><a href="#cb22-17" tabindex="-1"></a><span class="co">#&gt; patient_3</span></span>
<span id="cb22-18"><a href="#cb22-18" tabindex="-1"></a><span class="co">#&gt;  - check fit.gof for the convergeCode and convergeMessage</span></span>
<span id="cb22-19"><a href="#cb22-19" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-20"><a href="#cb22-20" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt; Running quantiseq</span></span>
<span id="cb22-21"><a href="#cb22-21" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-22"><a href="#cb22-22" tabindex="-1"></a><span class="co">#&gt; Running quanTIseq deconvolution module</span></span>
<span id="cb22-23"><a href="#cb22-23" tabindex="-1"></a><span class="co">#&gt; Gene expression normalization and re-annotation (arrays: FALSE)</span></span>
<span id="cb22-24"><a href="#cb22-24" tabindex="-1"></a><span class="co">#&gt; Removing 17 noisy genes</span></span>
<span id="cb22-25"><a href="#cb22-25" tabindex="-1"></a><span class="co">#&gt; Removing 15 genes with high expression in tumors</span></span>
<span id="cb22-26"><a href="#cb22-26" tabindex="-1"></a><span class="co">#&gt; Signature genes found in data set: 132/138 (95.65%)</span></span>
<span id="cb22-27"><a href="#cb22-27" tabindex="-1"></a><span class="co">#&gt; Mixture deconvolution (method: lsei)</span></span>
<span id="cb22-28"><a href="#cb22-28" tabindex="-1"></a><span class="co">#&gt; Deconvolution successful!</span></span>
<span id="cb22-29"><a href="#cb22-29" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-30"><a href="#cb22-30" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt; Running xcell</span></span>
<span id="cb22-31"><a href="#cb22-31" tabindex="-1"></a><span class="co">#&gt; [32mresults of the devonvolution will be stored in imm_epic.csv, imm_qti.csv and imm_xcell.csv [0m</span></span>
<span id="cb22-32"><a href="#cb22-32" tabindex="-1"></a><span class="co">#&gt; [32mAre they in the same order?[0m</span></span>
<span id="cb22-33"><a href="#cb22-33" tabindex="-1"></a><span class="co">#&gt; [32mHave to be true.[0m</span></span>
<span id="cb22-34"><a href="#cb22-34" tabindex="-1"></a><span class="co">#&gt; [32mEPIC[0m</span></span>
<span id="cb22-35"><a href="#cb22-35" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb22-36"><a href="#cb22-36" tabindex="-1"></a><span class="co">#&gt; [32mqti[0m</span></span>
<span id="cb22-37"><a href="#cb22-37" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb22-38"><a href="#cb22-38" tabindex="-1"></a><span class="co">#&gt; [32mxcell[0m</span></span>
<span id="cb22-39"><a href="#cb22-39" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb22-40"><a href="#cb22-40" tabindex="-1"></a><span class="co">#&gt; [33mAttention: plot_cell_fraction_Average_cell_fraction will work for all methods. For xCell, it will display Enrichment Scores instead of percentages.[0m</span></span>
<span id="cb22-41"><a href="#cb22-41" tabindex="-1"></a><span class="co">#&gt; [33mApply the filtering and normality test for each dataset[0m</span></span>
<span id="cb22-42"><a href="#cb22-42" tabindex="-1"></a><span class="co">#&gt; [32mChecking if All Values in Each Column Are Equal in dataset: quanTIseq [0m</span></span>
<span id="cb22-43"><a href="#cb22-43" tabindex="-1"></a><span class="co">#&gt; [33mIn dataset quanTIseq the following columns have the same value in all rows and will be excluded from the analysis:[0m</span></span>
<span id="cb22-44"><a href="#cb22-44" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Neutrophil&quot;</span></span>
<span id="cb22-45"><a href="#cb22-45" tabindex="-1"></a><span class="co">#&gt; [32mChecking if All Values in Each Column Are Equal in dataset: EPIC [0m</span></span>
<span id="cb22-46"><a href="#cb22-46" tabindex="-1"></a><span class="co">#&gt; [33mIn dataset EPIC the following columns have the same value in all rows and will be excluded from the analysis:[0m</span></span>
<span id="cb22-47"><a href="#cb22-47" tabindex="-1"></a><span class="co">#&gt; character(0)</span></span>
<span id="cb22-48"><a href="#cb22-48" tabindex="-1"></a><span class="co">#&gt; [32mChecking if All Values in Each Column Are Equal in dataset: xcell [0m</span></span>
<span id="cb22-49"><a href="#cb22-49" tabindex="-1"></a><span class="co">#&gt; [33mIn dataset xcell the following columns have the same value in all rows and will be excluded from the analysis:[0m</span></span>
<span id="cb22-50"><a href="#cb22-50" tabindex="-1"></a><span class="co">#&gt; character(0)</span></span>
<span id="cb22-51"><a href="#cb22-51" tabindex="-1"></a><span class="co">#&gt; [32mPerforming Shapiro-Wilk test for quanTIseq[0m</span></span>
<span id="cb22-52"><a href="#cb22-52" tabindex="-1"></a><span class="co">#&gt; # A tibble: 10 × 3</span></span>
<span id="cb22-53"><a href="#cb22-53" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                    shapiro_test    p.value</span></span>
<span id="cb22-54"><a href="#cb22-54" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                        &lt;list&gt;            &lt;dbl&gt;</span></span>
<span id="cb22-55"><a href="#cb22-55" tabindex="-1"></a><span class="co">#&gt;  1 B_cell                       &lt;htest&gt;      0.247     </span></span>
<span id="cb22-56"><a href="#cb22-56" tabindex="-1"></a><span class="co">#&gt;  2 Macrophage_M1                &lt;htest&gt;      0.00361   </span></span>
<span id="cb22-57"><a href="#cb22-57" tabindex="-1"></a><span class="co">#&gt;  3 Macrophage_M2                &lt;htest&gt;      0.266     </span></span>
<span id="cb22-58"><a href="#cb22-58" tabindex="-1"></a><span class="co">#&gt;  4 Monocyte                     &lt;htest&gt;      0.00178   </span></span>
<span id="cb22-59"><a href="#cb22-59" tabindex="-1"></a><span class="co">#&gt;  5 Myeloid_dendritic_cell       &lt;htest&gt;      0.00000105</span></span>
<span id="cb22-60"><a href="#cb22-60" tabindex="-1"></a><span class="co">#&gt;  6 NK_cell                      &lt;htest&gt;      0.498     </span></span>
<span id="cb22-61"><a href="#cb22-61" tabindex="-1"></a><span class="co">#&gt;  7 T_cell_CD4+_(non-regulatory) &lt;htest&gt;      0.000202  </span></span>
<span id="cb22-62"><a href="#cb22-62" tabindex="-1"></a><span class="co">#&gt;  8 T_cell_CD8+                  &lt;htest&gt;      0.0212    </span></span>
<span id="cb22-63"><a href="#cb22-63" tabindex="-1"></a><span class="co">#&gt;  9 T_cell_regulatory_(Tregs)    &lt;htest&gt;      0.0284    </span></span>
<span id="cb22-64"><a href="#cb22-64" tabindex="-1"></a><span class="co">#&gt; 10 uncharacterized_cell         &lt;htest&gt;      0.162     </span></span>
<span id="cb22-65"><a href="#cb22-65" tabindex="-1"></a><span class="co">#&gt; [32mPerforming parametric tests for quanTIseq[0m</span></span>
<span id="cb22-66"><a href="#cb22-66" tabindex="-1"></a><span class="co">#&gt; # A tibble: 10 × 3</span></span>
<span id="cb22-67"><a href="#cb22-67" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                    Test    p.value</span></span>
<span id="cb22-68"><a href="#cb22-68" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                        &lt;chr&gt;     &lt;dbl&gt;</span></span>
<span id="cb22-69"><a href="#cb22-69" tabindex="-1"></a><span class="co">#&gt;  1 B_cell                       t-test 0.0117  </span></span>
<span id="cb22-70"><a href="#cb22-70" tabindex="-1"></a><span class="co">#&gt;  2 Macrophage_M1                t-test 0.538   </span></span>
<span id="cb22-71"><a href="#cb22-71" tabindex="-1"></a><span class="co">#&gt;  3 Macrophage_M2                t-test 0.138   </span></span>
<span id="cb22-72"><a href="#cb22-72" tabindex="-1"></a><span class="co">#&gt;  4 Monocyte                     t-test 0.139   </span></span>
<span id="cb22-73"><a href="#cb22-73" tabindex="-1"></a><span class="co">#&gt;  5 NK_cell                      t-test 0.887   </span></span>
<span id="cb22-74"><a href="#cb22-74" tabindex="-1"></a><span class="co">#&gt;  6 T_cell_CD4+_(non-regulatory) t-test 0.393   </span></span>
<span id="cb22-75"><a href="#cb22-75" tabindex="-1"></a><span class="co">#&gt;  7 T_cell_CD8+                  t-test 0.0735  </span></span>
<span id="cb22-76"><a href="#cb22-76" tabindex="-1"></a><span class="co">#&gt;  8 T_cell_regulatory_(Tregs)    t-test 0.345   </span></span>
<span id="cb22-77"><a href="#cb22-77" tabindex="-1"></a><span class="co">#&gt;  9 Myeloid_dendritic_cell       t-test 0.391   </span></span>
<span id="cb22-78"><a href="#cb22-78" tabindex="-1"></a><span class="co">#&gt; 10 uncharacterized_cell         t-test 0.000286</span></span>
<span id="cb22-79"><a href="#cb22-79" tabindex="-1"></a><span class="co">#&gt; [32mPerforming Shapiro-Wilk test for EPIC[0m</span></span>
<span id="cb22-80"><a href="#cb22-80" tabindex="-1"></a><span class="co">#&gt; # A tibble: 8 × 3</span></span>
<span id="cb22-81"><a href="#cb22-81" tabindex="-1"></a><span class="co">#&gt;   Cell_Type                    shapiro_test    p.value</span></span>
<span id="cb22-82"><a href="#cb22-82" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                        &lt;list&gt;            &lt;dbl&gt;</span></span>
<span id="cb22-83"><a href="#cb22-83" tabindex="-1"></a><span class="co">#&gt; 1 B_cell                       &lt;htest&gt;      0.0225    </span></span>
<span id="cb22-84"><a href="#cb22-84" tabindex="-1"></a><span class="co">#&gt; 2 Cancer_associated_fibroblast &lt;htest&gt;      0.382     </span></span>
<span id="cb22-85"><a href="#cb22-85" tabindex="-1"></a><span class="co">#&gt; 3 Endothelial_cell             &lt;htest&gt;      0.292     </span></span>
<span id="cb22-86"><a href="#cb22-86" tabindex="-1"></a><span class="co">#&gt; 4 Macrophage                   &lt;htest&gt;      0.00703   </span></span>
<span id="cb22-87"><a href="#cb22-87" tabindex="-1"></a><span class="co">#&gt; 5 NK_cell                      &lt;htest&gt;      0.000335  </span></span>
<span id="cb22-88"><a href="#cb22-88" tabindex="-1"></a><span class="co">#&gt; 6 T_cell_CD4+                  &lt;htest&gt;      0.0902    </span></span>
<span id="cb22-89"><a href="#cb22-89" tabindex="-1"></a><span class="co">#&gt; 7 T_cell_CD8+                  &lt;htest&gt;      0.00000114</span></span>
<span id="cb22-90"><a href="#cb22-90" tabindex="-1"></a><span class="co">#&gt; 8 uncharacterized_cell         &lt;htest&gt;      0.116     </span></span>
<span id="cb22-91"><a href="#cb22-91" tabindex="-1"></a><span class="co">#&gt; [32mPerforming parametric tests for EPIC[0m</span></span>
<span id="cb22-92"><a href="#cb22-92" tabindex="-1"></a><span class="co">#&gt; # A tibble: 8 × 3</span></span>
<span id="cb22-93"><a href="#cb22-93" tabindex="-1"></a><span class="co">#&gt;   Cell_Type                    Test   p.value</span></span>
<span id="cb22-94"><a href="#cb22-94" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                        &lt;chr&gt;    &lt;dbl&gt;</span></span>
<span id="cb22-95"><a href="#cb22-95" tabindex="-1"></a><span class="co">#&gt; 1 B_cell                       t-test  0.0806</span></span>
<span id="cb22-96"><a href="#cb22-96" tabindex="-1"></a><span class="co">#&gt; 2 Cancer_associated_fibroblast t-test  0.453 </span></span>
<span id="cb22-97"><a href="#cb22-97" tabindex="-1"></a><span class="co">#&gt; 3 T_cell_CD4+                  t-test  0.761 </span></span>
<span id="cb22-98"><a href="#cb22-98" tabindex="-1"></a><span class="co">#&gt; 4 T_cell_CD8+                  t-test  0.389 </span></span>
<span id="cb22-99"><a href="#cb22-99" tabindex="-1"></a><span class="co">#&gt; 5 Endothelial_cell             t-test  0.590 </span></span>
<span id="cb22-100"><a href="#cb22-100" tabindex="-1"></a><span class="co">#&gt; 6 Macrophage                   t-test  0.837 </span></span>
<span id="cb22-101"><a href="#cb22-101" tabindex="-1"></a><span class="co">#&gt; 7 NK_cell                      t-test  0.745 </span></span>
<span id="cb22-102"><a href="#cb22-102" tabindex="-1"></a><span class="co">#&gt; 8 uncharacterized_cell         t-test  0.528 </span></span>
<span id="cb22-103"><a href="#cb22-103" tabindex="-1"></a><span class="co">#&gt; [32mPerforming Shapiro-Wilk test for xcell[0m</span></span>
<span id="cb22-104"><a href="#cb22-104" tabindex="-1"></a><span class="co">#&gt; # A tibble: 39 × 3</span></span>
<span id="cb22-105"><a href="#cb22-105" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                    shapiro_test    p.value</span></span>
<span id="cb22-106"><a href="#cb22-106" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                        &lt;list&gt;            &lt;dbl&gt;</span></span>
<span id="cb22-107"><a href="#cb22-107" tabindex="-1"></a><span class="co">#&gt;  1 B_cell                       &lt;htest&gt;      0.494     </span></span>
<span id="cb22-108"><a href="#cb22-108" tabindex="-1"></a><span class="co">#&gt;  2 B_cell_memory                &lt;htest&gt;      0.180     </span></span>
<span id="cb22-109"><a href="#cb22-109" tabindex="-1"></a><span class="co">#&gt;  3 B_cell_naive                 &lt;htest&gt;      0.108     </span></span>
<span id="cb22-110"><a href="#cb22-110" tabindex="-1"></a><span class="co">#&gt;  4 B_cell_plasma                &lt;htest&gt;      0.0104    </span></span>
<span id="cb22-111"><a href="#cb22-111" tabindex="-1"></a><span class="co">#&gt;  5 Cancer_associated_fibroblast &lt;htest&gt;      0.128     </span></span>
<span id="cb22-112"><a href="#cb22-112" tabindex="-1"></a><span class="co">#&gt;  6 Class-switched_memory_B_cell &lt;htest&gt;      0.200     </span></span>
<span id="cb22-113"><a href="#cb22-113" tabindex="-1"></a><span class="co">#&gt;  7 Common_lymphoid_progenitor   &lt;htest&gt;      0.0145    </span></span>
<span id="cb22-114"><a href="#cb22-114" tabindex="-1"></a><span class="co">#&gt;  8 Common_myeloid_progenitor    &lt;htest&gt;      0.00000811</span></span>
<span id="cb22-115"><a href="#cb22-115" tabindex="-1"></a><span class="co">#&gt;  9 Endothelial_cell             &lt;htest&gt;      0.00539   </span></span>
<span id="cb22-116"><a href="#cb22-116" tabindex="-1"></a><span class="co">#&gt; 10 Eosinophil                   &lt;htest&gt;      0.0443    </span></span>
<span id="cb22-117"><a href="#cb22-117" tabindex="-1"></a><span class="co">#&gt; # ℹ 29 more rows</span></span>
<span id="cb22-118"><a href="#cb22-118" tabindex="-1"></a><span class="co">#&gt; [32mPerforming parametric tests for xcell[0m</span></span>
<span id="cb22-119"><a href="#cb22-119" tabindex="-1"></a><span class="co">#&gt; # A tibble: 39 × 3</span></span>
<span id="cb22-120"><a href="#cb22-120" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                        Test   p.value</span></span>
<span id="cb22-121"><a href="#cb22-121" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                            &lt;chr&gt;    &lt;dbl&gt;</span></span>
<span id="cb22-122"><a href="#cb22-122" tabindex="-1"></a><span class="co">#&gt;  1 Myeloid_dendritic_cell_activated t-test  0.198 </span></span>
<span id="cb22-123"><a href="#cb22-123" tabindex="-1"></a><span class="co">#&gt;  2 B_cell                           t-test  0.0484</span></span>
<span id="cb22-124"><a href="#cb22-124" tabindex="-1"></a><span class="co">#&gt;  3 T_cell_CD4+_memory               t-test  0.220 </span></span>
<span id="cb22-125"><a href="#cb22-125" tabindex="-1"></a><span class="co">#&gt;  4 T_cell_CD4+_naive                t-test  0.185 </span></span>
<span id="cb22-126"><a href="#cb22-126" tabindex="-1"></a><span class="co">#&gt;  5 T_cell_CD4+_(non-regulatory)     t-test  0.391 </span></span>
<span id="cb22-127"><a href="#cb22-127" tabindex="-1"></a><span class="co">#&gt;  6 T_cell_CD4+_central_memory       t-test  0.776 </span></span>
<span id="cb22-128"><a href="#cb22-128" tabindex="-1"></a><span class="co">#&gt;  7 T_cell_CD4+_effector_memory      t-test  0.182 </span></span>
<span id="cb22-129"><a href="#cb22-129" tabindex="-1"></a><span class="co">#&gt;  8 T_cell_CD8+_naive                t-test  0.171 </span></span>
<span id="cb22-130"><a href="#cb22-130" tabindex="-1"></a><span class="co">#&gt;  9 T_cell_CD8+                      t-test  0.0164</span></span>
<span id="cb22-131"><a href="#cb22-131" tabindex="-1"></a><span class="co">#&gt; 10 T_cell_CD8+_central_memory       t-test  0.0146</span></span>
<span id="cb22-132"><a href="#cb22-132" tabindex="-1"></a><span class="co">#&gt; # ℹ 29 more rows</span></span>
<span id="cb22-133"><a href="#cb22-133" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING HEATMAPS[0m</span></span>
<span id="cb22-134"><a href="#cb22-134" tabindex="-1"></a><span class="co">#&gt; [32mHeatmap of qti, EPIC and xcell will be stored on plots_TME_heatmap.pdf[0m</span></span>
<span id="cb22-135"><a href="#cb22-135" tabindex="-1"></a></span>
<span id="cb22-136"><a href="#cb22-136" tabindex="-1"></a><span class="fu">head</span>(TME_results<span class="sc">$</span>EPIC)</span>
<span id="cb22-137"><a href="#cb22-137" tabindex="-1"></a><span class="co">#&gt;                 B_cell Cancer_associated_fibroblast  T_cell_CD4+  T_cell_CD8+</span></span>
<span id="cb22-138"><a href="#cb22-138" tabindex="-1"></a><span class="co">#&gt; patient_1 1.150729e-06                 2.016299e-06 9.454060e-10 2.425392e-06</span></span>
<span id="cb22-139"><a href="#cb22-139" tabindex="-1"></a><span class="co">#&gt; patient_2 1.162535e-01                 2.177641e-02 1.561392e-01 1.032210e-03</span></span>
<span id="cb22-140"><a href="#cb22-140" tabindex="-1"></a><span class="co">#&gt; patient_3 1.812948e-01                 8.256491e-03 7.200416e-02 2.920044e-06</span></span>
<span id="cb22-141"><a href="#cb22-141" tabindex="-1"></a><span class="co">#&gt; patient_4 1.517297e-01                 5.777250e-02 8.969556e-07 1.326486e-06</span></span>
<span id="cb22-142"><a href="#cb22-142" tabindex="-1"></a><span class="co">#&gt; patient_5 1.396394e-02                 4.308314e-02 6.416144e-02 2.171875e-09</span></span>
<span id="cb22-143"><a href="#cb22-143" tabindex="-1"></a><span class="co">#&gt; patient_6 1.173785e-03                 6.562770e-02 5.454055e-02 3.029674e-06</span></span>
<span id="cb22-144"><a href="#cb22-144" tabindex="-1"></a><span class="co">#&gt;           Endothelial_cell  Macrophage      NK_cell uncharacterized_cell</span></span>
<span id="cb22-145"><a href="#cb22-145" tabindex="-1"></a><span class="co">#&gt; patient_1     3.117571e-08 0.999993600 8.033374e-07         2.726286e-09</span></span>
<span id="cb22-146"><a href="#cb22-146" tabindex="-1"></a><span class="co">#&gt; patient_2     1.513580e-02 0.038318322 5.511060e-04         6.507934e-01</span></span>
<span id="cb22-147"><a href="#cb22-147" tabindex="-1"></a><span class="co">#&gt; patient_3     4.132234e-02 0.109170000 1.448324e-03         5.865009e-01</span></span>
<span id="cb22-148"><a href="#cb22-148" tabindex="-1"></a><span class="co">#&gt; patient_4     4.808822e-03 0.340783400 4.338030e-02         4.015231e-01</span></span>
<span id="cb22-149"><a href="#cb22-149" tabindex="-1"></a><span class="co">#&gt; patient_5     1.639526e-02 0.026456960 1.301272e-08         8.359392e-01</span></span>
<span id="cb22-150"><a href="#cb22-150" tabindex="-1"></a><span class="co">#&gt; patient_6     2.448703e-02 0.008433899 1.093331e-08         8.457340e-01</span></span>
<span id="cb22-151"><a href="#cb22-151" tabindex="-1"></a><span class="co">#&gt;            HPV_status</span></span>
<span id="cb22-152"><a href="#cb22-152" tabindex="-1"></a><span class="co">#&gt; patient_1  Associated</span></span>
<span id="cb22-153"><a href="#cb22-153" tabindex="-1"></a><span class="co">#&gt; patient_2  Associated</span></span>
<span id="cb22-154"><a href="#cb22-154" tabindex="-1"></a><span class="co">#&gt; patient_3  Associated</span></span>
<span id="cb22-155"><a href="#cb22-155" tabindex="-1"></a><span class="co">#&gt; patient_4  Associated</span></span>
<span id="cb22-156"><a href="#cb22-156" tabindex="-1"></a><span class="co">#&gt; patient_5 Independent</span></span>
<span id="cb22-157"><a href="#cb22-157" tabindex="-1"></a><span class="co">#&gt; patient_6 Independent</span></span>
<span id="cb22-158"><a href="#cb22-158" tabindex="-1"></a><span class="fu">head</span>(TME_results<span class="sc">$</span>QTI)</span>
<span id="cb22-159"><a href="#cb22-159" tabindex="-1"></a><span class="co">#&gt;                B_cell Macrophage_M1 Macrophage_M2    Monocyte Neutrophil</span></span>
<span id="cb22-160"><a href="#cb22-160" tabindex="-1"></a><span class="co">#&gt; patient_1 0.069908801   0.000000000    0.47458013 0.035577343          0</span></span>
<span id="cb22-161"><a href="#cb22-161" tabindex="-1"></a><span class="co">#&gt; patient_2 0.134239640   0.027649080    0.09043049 0.011559290          0</span></span>
<span id="cb22-162"><a href="#cb22-162" tabindex="-1"></a><span class="co">#&gt; patient_3 0.136095178   0.005503631    0.34483855 0.065454939          0</span></span>
<span id="cb22-163"><a href="#cb22-163" tabindex="-1"></a><span class="co">#&gt; patient_4 0.080426018   0.125520772    0.21625401 0.001473265          0</span></span>
<span id="cb22-164"><a href="#cb22-164" tabindex="-1"></a><span class="co">#&gt; patient_5 0.044623700   0.014932880    0.11713611 0.000000000          0</span></span>
<span id="cb22-165"><a href="#cb22-165" tabindex="-1"></a><span class="co">#&gt; patient_6 0.006699906   0.038054563    0.04783640 0.000000000          0</span></span>
<span id="cb22-166"><a href="#cb22-166" tabindex="-1"></a><span class="co">#&gt;               NK_cell T_cell_CD4+_(non-regulatory) T_cell_CD8+</span></span>
<span id="cb22-167"><a href="#cb22-167" tabindex="-1"></a><span class="co">#&gt; patient_1 0.009145382                   0.00000000  0.06116945</span></span>
<span id="cb22-168"><a href="#cb22-168" tabindex="-1"></a><span class="co">#&gt; patient_2 0.000000000                   0.30471172  0.03531690</span></span>
<span id="cb22-169"><a href="#cb22-169" tabindex="-1"></a><span class="co">#&gt; patient_3 0.012846664                   0.05983765  0.05960699</span></span>
<span id="cb22-170"><a href="#cb22-170" tabindex="-1"></a><span class="co">#&gt; patient_4 0.048619021                   0.00000000  0.16209930</span></span>
<span id="cb22-171"><a href="#cb22-171" tabindex="-1"></a><span class="co">#&gt; patient_5 0.024217980                   0.00000000  0.00000000</span></span>
<span id="cb22-172"><a href="#cb22-172" tabindex="-1"></a><span class="co">#&gt; patient_6 0.016230720                   0.00000000  0.00000000</span></span>
<span id="cb22-173"><a href="#cb22-173" tabindex="-1"></a><span class="co">#&gt;           T_cell_regulatory_(Tregs) Myeloid_dendritic_cell uncharacterized_cell</span></span>
<span id="cb22-174"><a href="#cb22-174" tabindex="-1"></a><span class="co">#&gt; patient_1               0.023104003                      0            0.3265149</span></span>
<span id="cb22-175"><a href="#cb22-175" tabindex="-1"></a><span class="co">#&gt; patient_2               0.125286240                      0            0.2708066</span></span>
<span id="cb22-176"><a href="#cb22-176" tabindex="-1"></a><span class="co">#&gt; patient_3               0.112300197                      0            0.2035162</span></span>
<span id="cb22-177"><a href="#cb22-177" tabindex="-1"></a><span class="co">#&gt; patient_4               0.008229871                      0            0.3573777</span></span>
<span id="cb22-178"><a href="#cb22-178" tabindex="-1"></a><span class="co">#&gt; patient_5               0.032973360                      0            0.7661160</span></span>
<span id="cb22-179"><a href="#cb22-179" tabindex="-1"></a><span class="co">#&gt; patient_6               0.033105128                      0            0.8580733</span></span>
<span id="cb22-180"><a href="#cb22-180" tabindex="-1"></a><span class="co">#&gt;            HPV_status</span></span>
<span id="cb22-181"><a href="#cb22-181" tabindex="-1"></a><span class="co">#&gt; patient_1  Associated</span></span>
<span id="cb22-182"><a href="#cb22-182" tabindex="-1"></a><span class="co">#&gt; patient_2  Associated</span></span>
<span id="cb22-183"><a href="#cb22-183" tabindex="-1"></a><span class="co">#&gt; patient_3  Associated</span></span>
<span id="cb22-184"><a href="#cb22-184" tabindex="-1"></a><span class="co">#&gt; patient_4  Associated</span></span>
<span id="cb22-185"><a href="#cb22-185" tabindex="-1"></a><span class="co">#&gt; patient_5 Independent</span></span>
<span id="cb22-186"><a href="#cb22-186" tabindex="-1"></a><span class="co">#&gt; patient_6 Independent</span></span>
<span id="cb22-187"><a href="#cb22-187" tabindex="-1"></a><span class="fu">head</span>(TME_results<span class="sc">$</span>XCELL)</span>
<span id="cb22-188"><a href="#cb22-188" tabindex="-1"></a><span class="co">#&gt;           Myeloid_dendritic_cell_activated       B_cell T_cell_CD4+_memory</span></span>
<span id="cb22-189"><a href="#cb22-189" tabindex="-1"></a><span class="co">#&gt; patient_1                     3.082705e-18 3.155324e-01       0.000000e+00</span></span>
<span id="cb22-190"><a href="#cb22-190" tabindex="-1"></a><span class="co">#&gt; patient_2                     5.147240e-19 4.665123e-01       1.328549e-01</span></span>
<span id="cb22-191"><a href="#cb22-191" tabindex="-1"></a><span class="co">#&gt; patient_3                     1.073208e-18 5.027837e-01       6.250107e-02</span></span>
<span id="cb22-192"><a href="#cb22-192" tabindex="-1"></a><span class="co">#&gt; patient_4                     0.000000e+00 1.984579e-01       1.055502e-18</span></span>
<span id="cb22-193"><a href="#cb22-193" tabindex="-1"></a><span class="co">#&gt; patient_5                     8.241773e-18 3.868377e-03       3.349103e-19</span></span>
<span id="cb22-194"><a href="#cb22-194" tabindex="-1"></a><span class="co">#&gt; patient_6                     0.000000e+00 4.558381e-18       0.000000e+00</span></span>
<span id="cb22-195"><a href="#cb22-195" tabindex="-1"></a><span class="co">#&gt;           T_cell_CD4+_naive T_cell_CD4+_(non-regulatory)</span></span>
<span id="cb22-196"><a href="#cb22-196" tabindex="-1"></a><span class="co">#&gt; patient_1      1.925551e-02                 0.000000e+00</span></span>
<span id="cb22-197"><a href="#cb22-197" tabindex="-1"></a><span class="co">#&gt; patient_2      1.130897e-01                 9.152625e-02</span></span>
<span id="cb22-198"><a href="#cb22-198" tabindex="-1"></a><span class="co">#&gt; patient_3      3.754398e-02                 0.000000e+00</span></span>
<span id="cb22-199"><a href="#cb22-199" tabindex="-1"></a><span class="co">#&gt; patient_4      0.000000e+00                 0.000000e+00</span></span>
<span id="cb22-200"><a href="#cb22-200" tabindex="-1"></a><span class="co">#&gt; patient_5      1.928309e-19                 7.599729e-18</span></span>
<span id="cb22-201"><a href="#cb22-201" tabindex="-1"></a><span class="co">#&gt; patient_6      6.646919e-18                 0.000000e+00</span></span>
<span id="cb22-202"><a href="#cb22-202" tabindex="-1"></a><span class="co">#&gt;           T_cell_CD4+_central_memory T_cell_CD4+_effector_memory</span></span>
<span id="cb22-203"><a href="#cb22-203" tabindex="-1"></a><span class="co">#&gt; patient_1               0.000000e+00                0.000000e+00</span></span>
<span id="cb22-204"><a href="#cb22-204" tabindex="-1"></a><span class="co">#&gt; patient_2               2.551973e-02                9.292065e-02</span></span>
<span id="cb22-205"><a href="#cb22-205" tabindex="-1"></a><span class="co">#&gt; patient_3               0.000000e+00                9.932477e-02</span></span>
<span id="cb22-206"><a href="#cb22-206" tabindex="-1"></a><span class="co">#&gt; patient_4               1.672849e-17                2.715178e-19</span></span>
<span id="cb22-207"><a href="#cb22-207" tabindex="-1"></a><span class="co">#&gt; patient_5               1.643340e-02                4.647476e-19</span></span>
<span id="cb22-208"><a href="#cb22-208" tabindex="-1"></a><span class="co">#&gt; patient_6               0.000000e+00                4.266006e-18</span></span>
<span id="cb22-209"><a href="#cb22-209" tabindex="-1"></a><span class="co">#&gt;           T_cell_CD8+_naive  T_cell_CD8+ T_cell_CD8+_central_memory</span></span>
<span id="cb22-210"><a href="#cb22-210" tabindex="-1"></a><span class="co">#&gt; patient_1      8.740664e-19 4.803232e-02               4.190523e-02</span></span>
<span id="cb22-211"><a href="#cb22-211" tabindex="-1"></a><span class="co">#&gt; patient_2      3.557785e-02 1.427515e-01               1.247125e-01</span></span>
<span id="cb22-212"><a href="#cb22-212" tabindex="-1"></a><span class="co">#&gt; patient_3      3.448572e-02 1.320651e-01               9.712064e-02</span></span>
<span id="cb22-213"><a href="#cb22-213" tabindex="-1"></a><span class="co">#&gt; patient_4      3.030975e-02 9.535608e-02               8.948805e-02</span></span>
<span id="cb22-214"><a href="#cb22-214" tabindex="-1"></a><span class="co">#&gt; patient_5      4.487133e-19 6.803360e-19               4.692817e-18</span></span>
<span id="cb22-215"><a href="#cb22-215" tabindex="-1"></a><span class="co">#&gt; patient_6      1.477767e-20 9.536354e-19               3.838657e-03</span></span>
<span id="cb22-216"><a href="#cb22-216" tabindex="-1"></a><span class="co">#&gt;           T_cell_CD8+_effector_memory Class-switched_memory_B_cell</span></span>
<span id="cb22-217"><a href="#cb22-217" tabindex="-1"></a><span class="co">#&gt; patient_1                1.397381e-17                 7.063573e-02</span></span>
<span id="cb22-218"><a href="#cb22-218" tabindex="-1"></a><span class="co">#&gt; patient_2                3.179827e-02                 1.564154e-01</span></span>
<span id="cb22-219"><a href="#cb22-219" tabindex="-1"></a><span class="co">#&gt; patient_3                2.609685e-03                 1.663765e-01</span></span>
<span id="cb22-220"><a href="#cb22-220" tabindex="-1"></a><span class="co">#&gt; patient_4                2.695730e-03                 6.641454e-02</span></span>
<span id="cb22-221"><a href="#cb22-221" tabindex="-1"></a><span class="co">#&gt; patient_5                2.437879e-17                 2.574549e-02</span></span>
<span id="cb22-222"><a href="#cb22-222" tabindex="-1"></a><span class="co">#&gt; patient_6                6.963297e-19                 4.373453e-19</span></span>
<span id="cb22-223"><a href="#cb22-223" tabindex="-1"></a><span class="co">#&gt;           Common_lymphoid_progenitor Common_myeloid_progenitor</span></span>
<span id="cb22-224"><a href="#cb22-224" tabindex="-1"></a><span class="co">#&gt; patient_1               4.455528e-18              0.000000e+00</span></span>
<span id="cb22-225"><a href="#cb22-225" tabindex="-1"></a><span class="co">#&gt; patient_2               2.261100e-02              2.227240e-18</span></span>
<span id="cb22-226"><a href="#cb22-226" tabindex="-1"></a><span class="co">#&gt; patient_3               9.347361e-03              0.000000e+00</span></span>
<span id="cb22-227"><a href="#cb22-227" tabindex="-1"></a><span class="co">#&gt; patient_4               5.750150e-02              1.362306e-18</span></span>
<span id="cb22-228"><a href="#cb22-228" tabindex="-1"></a><span class="co">#&gt; patient_5               0.000000e+00              3.505960e-03</span></span>
<span id="cb22-229"><a href="#cb22-229" tabindex="-1"></a><span class="co">#&gt; patient_6               4.749431e-02              1.093553e-18</span></span>
<span id="cb22-230"><a href="#cb22-230" tabindex="-1"></a><span class="co">#&gt;           Myeloid_dendritic_cell Endothelial_cell   Eosinophil</span></span>
<span id="cb22-231"><a href="#cb22-231" tabindex="-1"></a><span class="co">#&gt; patient_1           2.131066e-02     0.000000e+00 2.779892e-18</span></span>
<span id="cb22-232"><a href="#cb22-232" tabindex="-1"></a><span class="co">#&gt; patient_2           8.274088e-19     0.000000e+00 5.145732e-18</span></span>
<span id="cb22-233"><a href="#cb22-233" tabindex="-1"></a><span class="co">#&gt; patient_3           2.601932e-02     1.077469e-02 3.844183e-18</span></span>
<span id="cb22-234"><a href="#cb22-234" tabindex="-1"></a><span class="co">#&gt; patient_4           0.000000e+00     6.129541e-03 0.000000e+00</span></span>
<span id="cb22-235"><a href="#cb22-235" tabindex="-1"></a><span class="co">#&gt; patient_5           5.574760e-03     7.383512e-18 2.979015e-18</span></span>
<span id="cb22-236"><a href="#cb22-236" tabindex="-1"></a><span class="co">#&gt; patient_6           2.422973e-19     4.735047e-03 0.000000e+00</span></span>
<span id="cb22-237"><a href="#cb22-237" tabindex="-1"></a><span class="co">#&gt;           Cancer_associated_fibroblast Granulocyte-monocyte_progenitor</span></span>
<span id="cb22-238"><a href="#cb22-238" tabindex="-1"></a><span class="co">#&gt; patient_1                   0.00975768                    0.000000e+00</span></span>
<span id="cb22-239"><a href="#cb22-239" tabindex="-1"></a><span class="co">#&gt; patient_2                   0.04511596                    1.337241e-17</span></span>
<span id="cb22-240"><a href="#cb22-240" tabindex="-1"></a><span class="co">#&gt; patient_3                   0.08039560                    4.828205e-18</span></span>
<span id="cb22-241"><a href="#cb22-241" tabindex="-1"></a><span class="co">#&gt; patient_4                   0.05157214                    0.000000e+00</span></span>
<span id="cb22-242"><a href="#cb22-242" tabindex="-1"></a><span class="co">#&gt; patient_5                   0.00000000                    0.000000e+00</span></span>
<span id="cb22-243"><a href="#cb22-243" tabindex="-1"></a><span class="co">#&gt; patient_6                   0.02295305                    0.000000e+00</span></span>
<span id="cb22-244"><a href="#cb22-244" tabindex="-1"></a><span class="co">#&gt;           Hematopoietic_stem_cell Macrophage Macrophage_M1 Macrophage_M2</span></span>
<span id="cb22-245"><a href="#cb22-245" tabindex="-1"></a><span class="co">#&gt; patient_1              0.12790590 0.06367135    0.04283822  4.406785e-02</span></span>
<span id="cb22-246"><a href="#cb22-246" tabindex="-1"></a><span class="co">#&gt; patient_2              0.00000000 0.03373679    0.02194276  2.594676e-02</span></span>
<span id="cb22-247"><a href="#cb22-247" tabindex="-1"></a><span class="co">#&gt; patient_3              0.04276507 0.05029023    0.02176331  4.864833e-02</span></span>
<span id="cb22-248"><a href="#cb22-248" tabindex="-1"></a><span class="co">#&gt; patient_4              0.00000000 0.04197614    0.01875592  5.415432e-02</span></span>
<span id="cb22-249"><a href="#cb22-249" tabindex="-1"></a><span class="co">#&gt; patient_5              0.17266340 0.00000000    0.00000000  7.233774e-19</span></span>
<span id="cb22-250"><a href="#cb22-250" tabindex="-1"></a><span class="co">#&gt; patient_6              0.03582593 0.00000000    0.00000000  1.330948e-04</span></span>
<span id="cb22-251"><a href="#cb22-251" tabindex="-1"></a><span class="co">#&gt;             Mast_cell B_cell_memory     Monocyte B_cell_naive   Neutrophil</span></span>
<span id="cb22-252"><a href="#cb22-252" tabindex="-1"></a><span class="co">#&gt; patient_1 0.000000000  1.147007e-01 1.005463e-17 6.410647e-02 4.124483e-18</span></span>
<span id="cb22-253"><a href="#cb22-253" tabindex="-1"></a><span class="co">#&gt; patient_2 0.005845581  1.318827e-01 0.000000e+00 1.077412e-01 2.041566e-18</span></span>
<span id="cb22-254"><a href="#cb22-254" tabindex="-1"></a><span class="co">#&gt; patient_3 0.038543030  1.647877e-01 0.000000e+00 7.651956e-02 0.000000e+00</span></span>
<span id="cb22-255"><a href="#cb22-255" tabindex="-1"></a><span class="co">#&gt; patient_4 0.003087918  4.813989e-02 3.032488e-03 2.991675e-02 0.000000e+00</span></span>
<span id="cb22-256"><a href="#cb22-256" tabindex="-1"></a><span class="co">#&gt; patient_5 0.013839370  0.000000e+00 0.000000e+00 6.809085e-19 6.553345e-05</span></span>
<span id="cb22-257"><a href="#cb22-257" tabindex="-1"></a><span class="co">#&gt; patient_6 0.000000000  2.294100e-19 0.000000e+00 1.202937e-18 1.456983e-18</span></span>
<span id="cb22-258"><a href="#cb22-258" tabindex="-1"></a><span class="co">#&gt;                NK_cell    T_cell_NK Plasmacytoid_dendritic_cell B_cell_plasma</span></span>
<span id="cb22-259"><a href="#cb22-259" tabindex="-1"></a><span class="co">#&gt; patient_1 0.000000e+00 1.539263e-02                0.000000e+00  0.000000e+00</span></span>
<span id="cb22-260"><a href="#cb22-260" tabindex="-1"></a><span class="co">#&gt; patient_2 0.000000e+00 2.413304e-18                3.485713e-02  4.093003e-19</span></span>
<span id="cb22-261"><a href="#cb22-261" tabindex="-1"></a><span class="co">#&gt; patient_3 1.581443e-17 0.000000e+00                5.901268e-02  7.147441e-02</span></span>
<span id="cb22-262"><a href="#cb22-262" tabindex="-1"></a><span class="co">#&gt; patient_4 5.810506e-19 6.726025e-02                1.209433e-01  6.553465e-02</span></span>
<span id="cb22-263"><a href="#cb22-263" tabindex="-1"></a><span class="co">#&gt; patient_5 1.381642e-17 0.000000e+00                6.238780e-18  3.928843e-03</span></span>
<span id="cb22-264"><a href="#cb22-264" tabindex="-1"></a><span class="co">#&gt; patient_6 0.000000e+00 4.581630e-18                1.589566e-04  5.896487e-20</span></span>
<span id="cb22-265"><a href="#cb22-265" tabindex="-1"></a><span class="co">#&gt;           T_cell_gamma_delta T_cell_CD4+_Th1 T_cell_CD4+_Th2</span></span>
<span id="cb22-266"><a href="#cb22-266" tabindex="-1"></a><span class="co">#&gt; patient_1       0.000000e+00     0.000000000    0.000000e+00</span></span>
<span id="cb22-267"><a href="#cb22-267" tabindex="-1"></a><span class="co">#&gt; patient_2       1.421500e-02     0.184227500    6.938894e-18</span></span>
<span id="cb22-268"><a href="#cb22-268" tabindex="-1"></a><span class="co">#&gt; patient_3       0.000000e+00     0.121270800    2.690319e-17</span></span>
<span id="cb22-269"><a href="#cb22-269" tabindex="-1"></a><span class="co">#&gt; patient_4       1.438027e-17     0.146781600    1.336720e-17</span></span>
<span id="cb22-270"><a href="#cb22-270" tabindex="-1"></a><span class="co">#&gt; patient_5       1.781321e-18     0.002130667    5.795291e-03</span></span>
<span id="cb22-271"><a href="#cb22-271" tabindex="-1"></a><span class="co">#&gt; patient_6       0.000000e+00     0.095582720    1.264443e-17</span></span>
<span id="cb22-272"><a href="#cb22-272" tabindex="-1"></a><span class="co">#&gt;           T_cell_regulatory_(Tregs) immune_score stroma_score</span></span>
<span id="cb22-273"><a href="#cb22-273" tabindex="-1"></a><span class="co">#&gt; patient_1              1.779964e-17 2.990312e-01 4.878840e-03</span></span>
<span id="cb22-274"><a href="#cb22-274" tabindex="-1"></a><span class="co">#&gt; patient_2              0.000000e+00 4.935816e-01 2.255798e-02</span></span>
<span id="cb22-275"><a href="#cb22-275" tabindex="-1"></a><span class="co">#&gt; patient_3              0.000000e+00 4.998010e-01 4.558515e-02</span></span>
<span id="cb22-276"><a href="#cb22-276" tabindex="-1"></a><span class="co">#&gt; patient_4              3.759361e-18 2.279404e-01 2.903094e-02</span></span>
<span id="cb22-277"><a href="#cb22-277" tabindex="-1"></a><span class="co">#&gt; patient_5              1.762743e-03 1.556536e-02 3.691756e-18</span></span>
<span id="cb22-278"><a href="#cb22-278" tabindex="-1"></a><span class="co">#&gt; patient_6              0.000000e+00 4.807531e-18 1.384405e-02</span></span>
<span id="cb22-279"><a href="#cb22-279" tabindex="-1"></a><span class="co">#&gt;           microenvironment_score  HPV_status</span></span>
<span id="cb22-280"><a href="#cb22-280" tabindex="-1"></a><span class="co">#&gt; patient_1             0.30391000  Associated</span></span>
<span id="cb22-281"><a href="#cb22-281" tabindex="-1"></a><span class="co">#&gt; patient_2             0.51613950  Associated</span></span>
<span id="cb22-282"><a href="#cb22-282" tabindex="-1"></a><span class="co">#&gt; patient_3             0.54538610  Associated</span></span>
<span id="cb22-283"><a href="#cb22-283" tabindex="-1"></a><span class="co">#&gt; patient_4             0.25697130  Associated</span></span>
<span id="cb22-284"><a href="#cb22-284" tabindex="-1"></a><span class="co">#&gt; patient_5             0.01556536 Independent</span></span>
<span id="cb22-285"><a href="#cb22-285" tabindex="-1"></a><span class="co">#&gt; patient_6             0.01384405 Independent</span></span></code></pre></div>
</div>
<div id="survival-analysis-htg_survival-for-htg" class="section level4">
<h4>Survival analysis: HTG_survival() for HTG:</h4>
<p>Survival analysis can be performed using the results from DEA (res),
specific genes, or TME results. The function prioritizes inputs: first
genes_to_use, then DEA results, and finally TME results if none are
provided.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="do">############# survival using res</span></span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a> <span class="fu">HTG_survival</span>(<span class="at">variable_01 =</span> <span class="st">&quot;Recurrence_01&quot;</span>,</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>              <span class="at">time =</span> <span class="st">&quot;Time_to_death_surv&quot;</span>,</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a>              <span class="at">col_data =</span> AnnotData_tutorial,</span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a>              <span class="at">counts_data =</span> counts_data_tutorial,</span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a>              <span class="at">res =</span> res_tutorial,</span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a>              <span class="at">method =</span> <span class="st">&quot;median&quot;</span>,</span>
<span id="cb23-8"><a href="#cb23-8" tabindex="-1"></a>              <span class="at">genes_to_use =</span> <span class="cn">NULL</span>,</span>
<span id="cb23-9"><a href="#cb23-9" tabindex="-1"></a>              <span class="at">TME =</span> <span class="cn">NULL</span>,</span>
<span id="cb23-10"><a href="#cb23-10" tabindex="-1"></a>              <span class="at">outliers =</span> outliers_tutorial,</span>
<span id="cb23-11"><a href="#cb23-11" tabindex="-1"></a>              <span class="at">pattern =</span> <span class="st">&quot;^NC-|^POS-|^GDNA-|^ERCC-&quot;</span>,</span>
<span id="cb23-12"><a href="#cb23-12" tabindex="-1"></a>              <span class="at">remove_outliers =</span> <span class="cn">TRUE</span>)</span>
<span id="cb23-13"><a href="#cb23-13" tabindex="-1"></a><span class="co">#&gt; [33mFILTERING THE COUNT DATA. DELETING THE PROVES...[0m</span></span>
<span id="cb23-14"><a href="#cb23-14" tabindex="-1"></a><span class="co">#&gt; [33mREMOVING OUTLIERS...[0m</span></span>
<span id="cb23-15"><a href="#cb23-15" tabindex="-1"></a><span class="co">#&gt; [32mNormalizing data...[0m</span></span>
<span id="cb23-16"><a href="#cb23-16" tabindex="-1"></a><span class="co">#&gt; [33mOBTAINING NORMALIEZ COUNTS...[0m</span></span>
<span id="cb23-17"><a href="#cb23-17" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb23-18"><a href="#cb23-18" tabindex="-1"></a><span class="co">#&gt; [32mChecking for duplicate columns[0m</span></span>
<span id="cb23-19"><a href="#cb23-19" tabindex="-1"></a><span class="co">#&gt; [1] &quot;No duplicate columns.&quot;</span></span>
<span id="cb23-20"><a href="#cb23-20" tabindex="-1"></a><span class="co">#&gt; [32mFiltering data.[0m</span></span>
<span id="cb23-21"><a href="#cb23-21" tabindex="-1"></a><span class="co">#&gt; [32mSelecting TOP 10 genes with the lowest padj[0m</span></span>
<span id="cb23-22"><a href="#cb23-22" tabindex="-1"></a><span class="co">#&gt; [32mStarting survival analysis.[0m</span></span>
<span id="cb23-23"><a href="#cb23-23" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-24"><a href="#cb23-24" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  LCP1 </span></span>
<span id="cb23-25"><a href="#cb23-25" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb23-26"><a href="#cb23-26" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb23-27"><a href="#cb23-27" tabindex="-1"></a><span class="co">#&gt; [1] 2939.029</span></span>
<span id="cb23-28"><a href="#cb23-28" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-29"><a href="#cb23-29" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-30"><a href="#cb23-30" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-31"><a href="#cb23-31" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-32"><a href="#cb23-32" tabindex="-1"></a><span class="co">#&gt;                 LCP1_mRNA_expression=High </span></span>
<span id="cb23-33"><a href="#cb23-33" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-34"><a href="#cb23-34" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-35"><a href="#cb23-35" tabindex="-1"></a><span class="co">#&gt;  1287      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-36"><a href="#cb23-36" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-37"><a href="#cb23-37" tabindex="-1"></a><span class="co">#&gt;                 LCP1_mRNA_expression=Low </span></span>
<span id="cb23-38"><a href="#cb23-38" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-39"><a href="#cb23-39" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-40"><a href="#cb23-40" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-41"><a href="#cb23-41" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-42"><a href="#cb23-42" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-43"><a href="#cb23-43" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-44"><a href="#cb23-44" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-45"><a href="#cb23-45" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-46"><a href="#cb23-46" tabindex="-1"></a><span class="co">#&gt;                           N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-47"><a href="#cb23-47" tabindex="-1"></a><span class="co">#&gt; LCP1_mRNA_expression=High 4        2    0.533     4.033      5.63</span></span>
<span id="cb23-48"><a href="#cb23-48" tabindex="-1"></a><span class="co">#&gt; LCP1_mRNA_expression=Low  4        2    3.467     0.621      5.63</span></span>
<span id="cb23-49"><a href="#cb23-49" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-50"><a href="#cb23-50" tabindex="-1"></a><span class="co">#&gt;  Chisq= 5.6  on 1 degrees of freedom, p= 0.02 </span></span>
<span id="cb23-51"><a href="#cb23-51" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-52"><a href="#cb23-52" tabindex="-1"></a><span class="co">#&gt; [1] 0.01767673</span></span>
<span id="cb23-53"><a href="#cb23-53" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-54"><a href="#cb23-54" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb23-55"><a href="#cb23-55" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-56"><a href="#cb23-56" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  LYPD3 </span></span>
<span id="cb23-57"><a href="#cb23-57" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb23-58"><a href="#cb23-58" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb23-59"><a href="#cb23-59" tabindex="-1"></a><span class="co">#&gt; [1] 202.6128</span></span>
<span id="cb23-60"><a href="#cb23-60" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-61"><a href="#cb23-61" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-62"><a href="#cb23-62" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-63"><a href="#cb23-63" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-64"><a href="#cb23-64" tabindex="-1"></a><span class="co">#&gt;                 LYPD3_mRNA_expression=High </span></span>
<span id="cb23-65"><a href="#cb23-65" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-66"><a href="#cb23-66" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-67"><a href="#cb23-67" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-68"><a href="#cb23-68" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-69"><a href="#cb23-69" tabindex="-1"></a><span class="co">#&gt;                 LYPD3_mRNA_expression=Low </span></span>
<span id="cb23-70"><a href="#cb23-70" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-71"><a href="#cb23-71" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-72"><a href="#cb23-72" tabindex="-1"></a><span class="co">#&gt;  1287      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-73"><a href="#cb23-73" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-74"><a href="#cb23-74" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-75"><a href="#cb23-75" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-76"><a href="#cb23-76" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-77"><a href="#cb23-77" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-78"><a href="#cb23-78" tabindex="-1"></a><span class="co">#&gt;                            N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-79"><a href="#cb23-79" tabindex="-1"></a><span class="co">#&gt; LYPD3_mRNA_expression=High 4        2    3.467     0.621      5.63</span></span>
<span id="cb23-80"><a href="#cb23-80" tabindex="-1"></a><span class="co">#&gt; LYPD3_mRNA_expression=Low  4        2    0.533     4.033      5.63</span></span>
<span id="cb23-81"><a href="#cb23-81" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-82"><a href="#cb23-82" tabindex="-1"></a><span class="co">#&gt;  Chisq= 5.6  on 1 degrees of freedom, p= 0.02 </span></span>
<span id="cb23-83"><a href="#cb23-83" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-84"><a href="#cb23-84" tabindex="-1"></a><span class="co">#&gt; [1] 0.01767673</span></span>
<span id="cb23-85"><a href="#cb23-85" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-86"><a href="#cb23-86" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb23-87"><a href="#cb23-87" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-88"><a href="#cb23-88" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  SLAMF7 </span></span>
<span id="cb23-89"><a href="#cb23-89" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb23-90"><a href="#cb23-90" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb23-91"><a href="#cb23-91" tabindex="-1"></a><span class="co">#&gt; [1] 1523.957</span></span>
<span id="cb23-92"><a href="#cb23-92" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-93"><a href="#cb23-93" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-94"><a href="#cb23-94" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-95"><a href="#cb23-95" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-96"><a href="#cb23-96" tabindex="-1"></a><span class="co">#&gt;                 SLAMF7_mRNA_expression=High </span></span>
<span id="cb23-97"><a href="#cb23-97" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-98"><a href="#cb23-98" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-99"><a href="#cb23-99" tabindex="-1"></a><span class="co">#&gt;  1287      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-100"><a href="#cb23-100" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-101"><a href="#cb23-101" tabindex="-1"></a><span class="co">#&gt;                 SLAMF7_mRNA_expression=Low </span></span>
<span id="cb23-102"><a href="#cb23-102" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-103"><a href="#cb23-103" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-104"><a href="#cb23-104" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-105"><a href="#cb23-105" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-106"><a href="#cb23-106" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-107"><a href="#cb23-107" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-108"><a href="#cb23-108" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-109"><a href="#cb23-109" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-110"><a href="#cb23-110" tabindex="-1"></a><span class="co">#&gt;                             N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-111"><a href="#cb23-111" tabindex="-1"></a><span class="co">#&gt; SLAMF7_mRNA_expression=High 4        2    0.533     4.033      5.63</span></span>
<span id="cb23-112"><a href="#cb23-112" tabindex="-1"></a><span class="co">#&gt; SLAMF7_mRNA_expression=Low  4        2    3.467     0.621      5.63</span></span>
<span id="cb23-113"><a href="#cb23-113" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-114"><a href="#cb23-114" tabindex="-1"></a><span class="co">#&gt;  Chisq= 5.6  on 1 degrees of freedom, p= 0.02 </span></span>
<span id="cb23-115"><a href="#cb23-115" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-116"><a href="#cb23-116" tabindex="-1"></a><span class="co">#&gt; [1] 0.01767673</span></span>
<span id="cb23-117"><a href="#cb23-117" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-118"><a href="#cb23-118" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb23-119"><a href="#cb23-119" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-120"><a href="#cb23-120" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  OMA1 </span></span>
<span id="cb23-121"><a href="#cb23-121" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb23-122"><a href="#cb23-122" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb23-123"><a href="#cb23-123" tabindex="-1"></a><span class="co">#&gt; [1] 528.9339</span></span>
<span id="cb23-124"><a href="#cb23-124" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-125"><a href="#cb23-125" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-126"><a href="#cb23-126" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-127"><a href="#cb23-127" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-128"><a href="#cb23-128" tabindex="-1"></a><span class="co">#&gt;                 OMA1_mRNA_expression=High </span></span>
<span id="cb23-129"><a href="#cb23-129" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-130"><a href="#cb23-130" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-131"><a href="#cb23-131" tabindex="-1"></a><span class="co">#&gt;  1287      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-132"><a href="#cb23-132" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-133"><a href="#cb23-133" tabindex="-1"></a><span class="co">#&gt;                 OMA1_mRNA_expression=Low </span></span>
<span id="cb23-134"><a href="#cb23-134" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-135"><a href="#cb23-135" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-136"><a href="#cb23-136" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-137"><a href="#cb23-137" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-138"><a href="#cb23-138" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-139"><a href="#cb23-139" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-140"><a href="#cb23-140" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-141"><a href="#cb23-141" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-142"><a href="#cb23-142" tabindex="-1"></a><span class="co">#&gt;                           N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-143"><a href="#cb23-143" tabindex="-1"></a><span class="co">#&gt; OMA1_mRNA_expression=High 4        2    0.533     4.033      5.63</span></span>
<span id="cb23-144"><a href="#cb23-144" tabindex="-1"></a><span class="co">#&gt; OMA1_mRNA_expression=Low  4        2    3.467     0.621      5.63</span></span>
<span id="cb23-145"><a href="#cb23-145" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-146"><a href="#cb23-146" tabindex="-1"></a><span class="co">#&gt;  Chisq= 5.6  on 1 degrees of freedom, p= 0.02 </span></span>
<span id="cb23-147"><a href="#cb23-147" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-148"><a href="#cb23-148" tabindex="-1"></a><span class="co">#&gt; [1] 0.01767673</span></span>
<span id="cb23-149"><a href="#cb23-149" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-150"><a href="#cb23-150" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb23-151"><a href="#cb23-151" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-152"><a href="#cb23-152" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  VOPP1 </span></span>
<span id="cb23-153"><a href="#cb23-153" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb23-154"><a href="#cb23-154" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb23-155"><a href="#cb23-155" tabindex="-1"></a><span class="co">#&gt; [1] 1098.125</span></span>
<span id="cb23-156"><a href="#cb23-156" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-157"><a href="#cb23-157" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-158"><a href="#cb23-158" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-159"><a href="#cb23-159" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-160"><a href="#cb23-160" tabindex="-1"></a><span class="co">#&gt;                 VOPP1_mRNA_expression=High </span></span>
<span id="cb23-161"><a href="#cb23-161" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-162"><a href="#cb23-162" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-163"><a href="#cb23-163" tabindex="-1"></a><span class="co">#&gt;  1287      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-164"><a href="#cb23-164" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-165"><a href="#cb23-165" tabindex="-1"></a><span class="co">#&gt;                 VOPP1_mRNA_expression=Low </span></span>
<span id="cb23-166"><a href="#cb23-166" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-167"><a href="#cb23-167" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-168"><a href="#cb23-168" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-169"><a href="#cb23-169" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-170"><a href="#cb23-170" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-171"><a href="#cb23-171" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-172"><a href="#cb23-172" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-173"><a href="#cb23-173" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-174"><a href="#cb23-174" tabindex="-1"></a><span class="co">#&gt;                            N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-175"><a href="#cb23-175" tabindex="-1"></a><span class="co">#&gt; VOPP1_mRNA_expression=High 4        2    0.533     4.033      5.63</span></span>
<span id="cb23-176"><a href="#cb23-176" tabindex="-1"></a><span class="co">#&gt; VOPP1_mRNA_expression=Low  4        2    3.467     0.621      5.63</span></span>
<span id="cb23-177"><a href="#cb23-177" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-178"><a href="#cb23-178" tabindex="-1"></a><span class="co">#&gt;  Chisq= 5.6  on 1 degrees of freedom, p= 0.02 </span></span>
<span id="cb23-179"><a href="#cb23-179" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-180"><a href="#cb23-180" tabindex="-1"></a><span class="co">#&gt; [1] 0.01767673</span></span>
<span id="cb23-181"><a href="#cb23-181" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-182"><a href="#cb23-182" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb23-183"><a href="#cb23-183" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-184"><a href="#cb23-184" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  COL5A3 </span></span>
<span id="cb23-185"><a href="#cb23-185" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb23-186"><a href="#cb23-186" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb23-187"><a href="#cb23-187" tabindex="-1"></a><span class="co">#&gt; [1] 434.5902</span></span>
<span id="cb23-188"><a href="#cb23-188" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-189"><a href="#cb23-189" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-190"><a href="#cb23-190" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-191"><a href="#cb23-191" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-192"><a href="#cb23-192" tabindex="-1"></a><span class="co">#&gt;                 COL5A3_mRNA_expression=High </span></span>
<span id="cb23-193"><a href="#cb23-193" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-194"><a href="#cb23-194" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-195"><a href="#cb23-195" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-196"><a href="#cb23-196" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-197"><a href="#cb23-197" tabindex="-1"></a><span class="co">#&gt;                 COL5A3_mRNA_expression=Low </span></span>
<span id="cb23-198"><a href="#cb23-198" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-199"><a href="#cb23-199" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-200"><a href="#cb23-200" tabindex="-1"></a><span class="co">#&gt;  1287      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-201"><a href="#cb23-201" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-202"><a href="#cb23-202" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-203"><a href="#cb23-203" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-204"><a href="#cb23-204" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-205"><a href="#cb23-205" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-206"><a href="#cb23-206" tabindex="-1"></a><span class="co">#&gt;                             N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-207"><a href="#cb23-207" tabindex="-1"></a><span class="co">#&gt; COL5A3_mRNA_expression=High 4        2    3.467     0.621      5.63</span></span>
<span id="cb23-208"><a href="#cb23-208" tabindex="-1"></a><span class="co">#&gt; COL5A3_mRNA_expression=Low  4        2    0.533     4.033      5.63</span></span>
<span id="cb23-209"><a href="#cb23-209" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-210"><a href="#cb23-210" tabindex="-1"></a><span class="co">#&gt;  Chisq= 5.6  on 1 degrees of freedom, p= 0.02 </span></span>
<span id="cb23-211"><a href="#cb23-211" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-212"><a href="#cb23-212" tabindex="-1"></a><span class="co">#&gt; [1] 0.01767673</span></span>
<span id="cb23-213"><a href="#cb23-213" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-214"><a href="#cb23-214" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb23-215"><a href="#cb23-215" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-216"><a href="#cb23-216" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  MMP10 </span></span>
<span id="cb23-217"><a href="#cb23-217" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb23-218"><a href="#cb23-218" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb23-219"><a href="#cb23-219" tabindex="-1"></a><span class="co">#&gt; [1] 45.83965</span></span>
<span id="cb23-220"><a href="#cb23-220" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-221"><a href="#cb23-221" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-222"><a href="#cb23-222" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-223"><a href="#cb23-223" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-224"><a href="#cb23-224" tabindex="-1"></a><span class="co">#&gt;                 MMP10_mRNA_expression=High </span></span>
<span id="cb23-225"><a href="#cb23-225" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-226"><a href="#cb23-226" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-227"><a href="#cb23-227" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-228"><a href="#cb23-228" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-229"><a href="#cb23-229" tabindex="-1"></a><span class="co">#&gt;                 MMP10_mRNA_expression=Low </span></span>
<span id="cb23-230"><a href="#cb23-230" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-231"><a href="#cb23-231" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-232"><a href="#cb23-232" tabindex="-1"></a><span class="co">#&gt;  1287      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-233"><a href="#cb23-233" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-234"><a href="#cb23-234" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-235"><a href="#cb23-235" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-236"><a href="#cb23-236" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-237"><a href="#cb23-237" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-238"><a href="#cb23-238" tabindex="-1"></a><span class="co">#&gt;                            N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-239"><a href="#cb23-239" tabindex="-1"></a><span class="co">#&gt; MMP10_mRNA_expression=High 4        2    3.467     0.621      5.63</span></span>
<span id="cb23-240"><a href="#cb23-240" tabindex="-1"></a><span class="co">#&gt; MMP10_mRNA_expression=Low  4        2    0.533     4.033      5.63</span></span>
<span id="cb23-241"><a href="#cb23-241" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-242"><a href="#cb23-242" tabindex="-1"></a><span class="co">#&gt;  Chisq= 5.6  on 1 degrees of freedom, p= 0.02 </span></span>
<span id="cb23-243"><a href="#cb23-243" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-244"><a href="#cb23-244" tabindex="-1"></a><span class="co">#&gt; [1] 0.01767673</span></span>
<span id="cb23-245"><a href="#cb23-245" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-246"><a href="#cb23-246" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb23-247"><a href="#cb23-247" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-248"><a href="#cb23-248" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  CDCA4 </span></span>
<span id="cb23-249"><a href="#cb23-249" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb23-250"><a href="#cb23-250" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb23-251"><a href="#cb23-251" tabindex="-1"></a><span class="co">#&gt; [1] 510.1244</span></span>
<span id="cb23-252"><a href="#cb23-252" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-253"><a href="#cb23-253" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-254"><a href="#cb23-254" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-255"><a href="#cb23-255" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-256"><a href="#cb23-256" tabindex="-1"></a><span class="co">#&gt;                 CDCA4_mRNA_expression=High </span></span>
<span id="cb23-257"><a href="#cb23-257" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-258"><a href="#cb23-258" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-259"><a href="#cb23-259" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-260"><a href="#cb23-260" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-261"><a href="#cb23-261" tabindex="-1"></a><span class="co">#&gt;                 CDCA4_mRNA_expression=Low </span></span>
<span id="cb23-262"><a href="#cb23-262" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-263"><a href="#cb23-263" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-264"><a href="#cb23-264" tabindex="-1"></a><span class="co">#&gt;  1287      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-265"><a href="#cb23-265" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-266"><a href="#cb23-266" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-267"><a href="#cb23-267" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-268"><a href="#cb23-268" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-269"><a href="#cb23-269" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-270"><a href="#cb23-270" tabindex="-1"></a><span class="co">#&gt;                            N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-271"><a href="#cb23-271" tabindex="-1"></a><span class="co">#&gt; CDCA4_mRNA_expression=High 4        2    3.467     0.621      5.63</span></span>
<span id="cb23-272"><a href="#cb23-272" tabindex="-1"></a><span class="co">#&gt; CDCA4_mRNA_expression=Low  4        2    0.533     4.033      5.63</span></span>
<span id="cb23-273"><a href="#cb23-273" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-274"><a href="#cb23-274" tabindex="-1"></a><span class="co">#&gt;  Chisq= 5.6  on 1 degrees of freedom, p= 0.02 </span></span>
<span id="cb23-275"><a href="#cb23-275" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-276"><a href="#cb23-276" tabindex="-1"></a><span class="co">#&gt; [1] 0.01767673</span></span>
<span id="cb23-277"><a href="#cb23-277" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-278"><a href="#cb23-278" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb23-279"><a href="#cb23-279" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-280"><a href="#cb23-280" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  TACSTD2 </span></span>
<span id="cb23-281"><a href="#cb23-281" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb23-282"><a href="#cb23-282" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb23-283"><a href="#cb23-283" tabindex="-1"></a><span class="co">#&gt; [1] 24.58751</span></span>
<span id="cb23-284"><a href="#cb23-284" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-285"><a href="#cb23-285" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-286"><a href="#cb23-286" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-287"><a href="#cb23-287" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-288"><a href="#cb23-288" tabindex="-1"></a><span class="co">#&gt;                 TACSTD2_mRNA_expression=High </span></span>
<span id="cb23-289"><a href="#cb23-289" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-290"><a href="#cb23-290" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-291"><a href="#cb23-291" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-292"><a href="#cb23-292" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-293"><a href="#cb23-293" tabindex="-1"></a><span class="co">#&gt;                 TACSTD2_mRNA_expression=Low </span></span>
<span id="cb23-294"><a href="#cb23-294" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-295"><a href="#cb23-295" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-296"><a href="#cb23-296" tabindex="-1"></a><span class="co">#&gt;  1287      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-297"><a href="#cb23-297" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-298"><a href="#cb23-298" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-299"><a href="#cb23-299" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-300"><a href="#cb23-300" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-301"><a href="#cb23-301" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-302"><a href="#cb23-302" tabindex="-1"></a><span class="co">#&gt;                              N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-303"><a href="#cb23-303" tabindex="-1"></a><span class="co">#&gt; TACSTD2_mRNA_expression=High 4        2    3.467     0.621      5.63</span></span>
<span id="cb23-304"><a href="#cb23-304" tabindex="-1"></a><span class="co">#&gt; TACSTD2_mRNA_expression=Low  4        2    0.533     4.033      5.63</span></span>
<span id="cb23-305"><a href="#cb23-305" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-306"><a href="#cb23-306" tabindex="-1"></a><span class="co">#&gt;  Chisq= 5.6  on 1 degrees of freedom, p= 0.02 </span></span>
<span id="cb23-307"><a href="#cb23-307" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-308"><a href="#cb23-308" tabindex="-1"></a><span class="co">#&gt; [1] 0.01767673</span></span>
<span id="cb23-309"><a href="#cb23-309" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-310"><a href="#cb23-310" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb23-311"><a href="#cb23-311" tabindex="-1"></a><span class="co">#&gt; png </span></span>
<span id="cb23-312"><a href="#cb23-312" tabindex="-1"></a><span class="co">#&gt;   2</span></span>
<span id="cb23-313"><a href="#cb23-313" tabindex="-1"></a></span>
<span id="cb23-314"><a href="#cb23-314" tabindex="-1"></a> <span class="co"># USING SPECIFIC GENES</span></span>
<span id="cb23-315"><a href="#cb23-315" tabindex="-1"></a> <span class="fu">HTG_survival</span>(<span class="at">variable_01 =</span> <span class="st">&quot;Recurrence_01&quot;</span>,</span>
<span id="cb23-316"><a href="#cb23-316" tabindex="-1"></a>              <span class="at">time =</span> <span class="st">&quot;Time_to_death_surv&quot;</span>,</span>
<span id="cb23-317"><a href="#cb23-317" tabindex="-1"></a>              <span class="at">col_data =</span> AnnotData_tutorial,</span>
<span id="cb23-318"><a href="#cb23-318" tabindex="-1"></a>              <span class="at">counts_data =</span> counts_data_tutorial,</span>
<span id="cb23-319"><a href="#cb23-319" tabindex="-1"></a>              <span class="at">res =</span> <span class="cn">NULL</span>,</span>
<span id="cb23-320"><a href="#cb23-320" tabindex="-1"></a>              <span class="at">method =</span> <span class="st">&quot;quartiles&quot;</span>,</span>
<span id="cb23-321"><a href="#cb23-321" tabindex="-1"></a>              <span class="at">genes_to_use =</span> <span class="fu">c</span>(<span class="st">&quot;LCP1&quot;</span>, <span class="st">&quot;OMA1&quot;</span>),</span>
<span id="cb23-322"><a href="#cb23-322" tabindex="-1"></a>              <span class="at">TME =</span> <span class="cn">NULL</span>,</span>
<span id="cb23-323"><a href="#cb23-323" tabindex="-1"></a>              <span class="at">outliers =</span> outliers_tutorial,</span>
<span id="cb23-324"><a href="#cb23-324" tabindex="-1"></a>              <span class="at">pattern =</span> <span class="st">&quot;^NC-|^POS-|^GDNA-|^ERCC-&quot;</span>,</span>
<span id="cb23-325"><a href="#cb23-325" tabindex="-1"></a>              <span class="at">remove_outliers =</span> <span class="cn">TRUE</span>)</span>
<span id="cb23-326"><a href="#cb23-326" tabindex="-1"></a><span class="co">#&gt; [33mFILTERING THE COUNT DATA. DELETING THE PROVES...[0m</span></span>
<span id="cb23-327"><a href="#cb23-327" tabindex="-1"></a><span class="co">#&gt; [33mREMOVING OUTLIERS...[0m</span></span>
<span id="cb23-328"><a href="#cb23-328" tabindex="-1"></a><span class="co">#&gt; [32mNormalizing data...[0m</span></span>
<span id="cb23-329"><a href="#cb23-329" tabindex="-1"></a><span class="co">#&gt; [33mOBTAINING NORMALIEZ COUNTS...[0m</span></span>
<span id="cb23-330"><a href="#cb23-330" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb23-331"><a href="#cb23-331" tabindex="-1"></a><span class="co">#&gt; [32mChecking for duplicate columns[0m</span></span>
<span id="cb23-332"><a href="#cb23-332" tabindex="-1"></a><span class="co">#&gt; [1] &quot;No duplicate columns.&quot;</span></span>
<span id="cb23-333"><a href="#cb23-333" tabindex="-1"></a><span class="co">#&gt; [32mFiltering data.[0m</span></span>
<span id="cb23-334"><a href="#cb23-334" tabindex="-1"></a><span class="co">#&gt; [32mUsing provided genes[0m</span></span>
<span id="cb23-335"><a href="#cb23-335" tabindex="-1"></a><span class="co">#&gt; [32mStarting survival analysis.[0m</span></span>
<span id="cb23-336"><a href="#cb23-336" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-337"><a href="#cb23-337" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  LCP1 </span></span>
<span id="cb23-338"><a href="#cb23-338" tabindex="-1"></a><span class="co">#&gt; [32mUsing Quartiles to determine cutoff...[0m</span></span>
<span id="cb23-339"><a href="#cb23-339" tabindex="-1"></a><span class="co">#&gt; [32mQuartile 1 (Q1):[0m  868.7876 </span></span>
<span id="cb23-340"><a href="#cb23-340" tabindex="-1"></a><span class="co">#&gt; [32mQuartile 3 (Q3):[0m  7834.828 </span></span>
<span id="cb23-341"><a href="#cb23-341" tabindex="-1"></a><span class="co">#&gt; [32mUsing Q3 Quartile to determine cutoff...[0m</span></span>
<span id="cb23-342"><a href="#cb23-342" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-343"><a href="#cb23-343" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-344"><a href="#cb23-344" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-345"><a href="#cb23-345" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-346"><a href="#cb23-346" tabindex="-1"></a><span class="co">#&gt;                 LCP1_mRNA_expression=High </span></span>
<span id="cb23-347"><a href="#cb23-347" tabindex="-1"></a><span class="co">#&gt;      time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-348"><a href="#cb23-348" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-349"><a href="#cb23-349" tabindex="-1"></a><span class="co">#&gt;                 LCP1_mRNA_expression=Low </span></span>
<span id="cb23-350"><a href="#cb23-350" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-351"><a href="#cb23-351" tabindex="-1"></a><span class="co">#&gt;  1164      6       1    0.833   0.152       0.5827            1</span></span>
<span id="cb23-352"><a href="#cb23-352" tabindex="-1"></a><span class="co">#&gt;  1287      5       1    0.667   0.192       0.3786            1</span></span>
<span id="cb23-353"><a href="#cb23-353" tabindex="-1"></a><span class="co">#&gt;  2458      2       1    0.333   0.255       0.0746            1</span></span>
<span id="cb23-354"><a href="#cb23-354" tabindex="-1"></a><span class="co">#&gt;  5844      1       1    0.000     NaN           NA           NA</span></span>
<span id="cb23-355"><a href="#cb23-355" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-356"><a href="#cb23-356" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-357"><a href="#cb23-357" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-358"><a href="#cb23-358" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-359"><a href="#cb23-359" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-360"><a href="#cb23-360" tabindex="-1"></a><span class="co">#&gt;                           N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-361"><a href="#cb23-361" tabindex="-1"></a><span class="co">#&gt; LCP1_mRNA_expression=High 2        0        0       NaN       NaN</span></span>
<span id="cb23-362"><a href="#cb23-362" tabindex="-1"></a><span class="co">#&gt; LCP1_mRNA_expression=Low  6        4        4         0       NaN</span></span>
<span id="cb23-363"><a href="#cb23-363" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-364"><a href="#cb23-364" tabindex="-1"></a><span class="co">#&gt;  Chisq= 0  on 0 degrees of freedom, p= 1 </span></span>
<span id="cb23-365"><a href="#cb23-365" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-366"><a href="#cb23-366" tabindex="-1"></a><span class="co">#&gt; [1] 1</span></span>
<span id="cb23-367"><a href="#cb23-367" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-368"><a href="#cb23-368" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_SELECTED_GENES.pdf[0m</span></span>
<span id="cb23-369"><a href="#cb23-369" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-370"><a href="#cb23-370" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  OMA1 </span></span>
<span id="cb23-371"><a href="#cb23-371" tabindex="-1"></a><span class="co">#&gt; [32mUsing Quartiles to determine cutoff...[0m</span></span>
<span id="cb23-372"><a href="#cb23-372" tabindex="-1"></a><span class="co">#&gt; [32mQuartile 1 (Q1):[0m  229.0703 </span></span>
<span id="cb23-373"><a href="#cb23-373" tabindex="-1"></a><span class="co">#&gt; [32mQuartile 3 (Q3):[0m  978.9518 </span></span>
<span id="cb23-374"><a href="#cb23-374" tabindex="-1"></a><span class="co">#&gt; [32mUsing Q3 Quartile to determine cutoff...[0m</span></span>
<span id="cb23-375"><a href="#cb23-375" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-376"><a href="#cb23-376" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-377"><a href="#cb23-377" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-378"><a href="#cb23-378" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-379"><a href="#cb23-379" tabindex="-1"></a><span class="co">#&gt;                 OMA1_mRNA_expression=High </span></span>
<span id="cb23-380"><a href="#cb23-380" tabindex="-1"></a><span class="co">#&gt;         time       n.risk      n.event     survival      std.err lower 95% CI </span></span>
<span id="cb23-381"><a href="#cb23-381" tabindex="-1"></a><span class="co">#&gt;         1164            1            1            0          NaN           NA </span></span>
<span id="cb23-382"><a href="#cb23-382" tabindex="-1"></a><span class="co">#&gt; upper 95% CI </span></span>
<span id="cb23-383"><a href="#cb23-383" tabindex="-1"></a><span class="co">#&gt;           NA </span></span>
<span id="cb23-384"><a href="#cb23-384" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-385"><a href="#cb23-385" tabindex="-1"></a><span class="co">#&gt;                 OMA1_mRNA_expression=Low </span></span>
<span id="cb23-386"><a href="#cb23-386" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-387"><a href="#cb23-387" tabindex="-1"></a><span class="co">#&gt;  1287      5       1      0.8   0.179       0.5161            1</span></span>
<span id="cb23-388"><a href="#cb23-388" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.4   0.297       0.0935            1</span></span>
<span id="cb23-389"><a href="#cb23-389" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-390"><a href="#cb23-390" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-391"><a href="#cb23-391" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-392"><a href="#cb23-392" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-393"><a href="#cb23-393" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-394"><a href="#cb23-394" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-395"><a href="#cb23-395" tabindex="-1"></a><span class="co">#&gt;                           N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-396"><a href="#cb23-396" tabindex="-1"></a><span class="co">#&gt; OMA1_mRNA_expression=High 2        1    0.167     4.167         5</span></span>
<span id="cb23-397"><a href="#cb23-397" tabindex="-1"></a><span class="co">#&gt; OMA1_mRNA_expression=Low  6        3    3.833     0.181         5</span></span>
<span id="cb23-398"><a href="#cb23-398" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-399"><a href="#cb23-399" tabindex="-1"></a><span class="co">#&gt;  Chisq= 5  on 1 degrees of freedom, p= 0.03 </span></span>
<span id="cb23-400"><a href="#cb23-400" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-401"><a href="#cb23-401" tabindex="-1"></a><span class="co">#&gt; [1] 0.02534732</span></span>
<span id="cb23-402"><a href="#cb23-402" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-403"><a href="#cb23-403" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_SELECTED_GENES.pdf[0m</span></span>
<span id="cb23-404"><a href="#cb23-404" tabindex="-1"></a><span class="co">#&gt; png </span></span>
<span id="cb23-405"><a href="#cb23-405" tabindex="-1"></a><span class="co">#&gt;   2</span></span>
<span id="cb23-406"><a href="#cb23-406" tabindex="-1"></a></span>
<span id="cb23-407"><a href="#cb23-407" tabindex="-1"></a> <span class="co"># USING TME DATA</span></span>
<span id="cb23-408"><a href="#cb23-408" tabindex="-1"></a> <span class="fu">HTG_survival</span>(<span class="at">variable_01 =</span> <span class="st">&quot;Recurrence_01&quot;</span>,</span>
<span id="cb23-409"><a href="#cb23-409" tabindex="-1"></a>              <span class="at">time =</span> <span class="st">&quot;Time_to_death_surv&quot;</span>,</span>
<span id="cb23-410"><a href="#cb23-410" tabindex="-1"></a>              <span class="at">col_data =</span> AnnotData_tutorial,</span>
<span id="cb23-411"><a href="#cb23-411" tabindex="-1"></a>              <span class="at">counts_data =</span> counts_data_tutorial,</span>
<span id="cb23-412"><a href="#cb23-412" tabindex="-1"></a>              <span class="at">res =</span> <span class="cn">NULL</span>,</span>
<span id="cb23-413"><a href="#cb23-413" tabindex="-1"></a>              <span class="at">method =</span> <span class="st">&quot;maxstat&quot;</span>,</span>
<span id="cb23-414"><a href="#cb23-414" tabindex="-1"></a>              <span class="at">genes_to_use =</span> <span class="cn">NULL</span>,</span>
<span id="cb23-415"><a href="#cb23-415" tabindex="-1"></a>              <span class="at">TME =</span> TME_data_tutorial<span class="sc">$</span>EPIC,</span>
<span id="cb23-416"><a href="#cb23-416" tabindex="-1"></a>              <span class="at">outliers =</span> outliers_tutorial,</span>
<span id="cb23-417"><a href="#cb23-417" tabindex="-1"></a>              <span class="at">pattern =</span> <span class="st">&quot;^NC-|^POS-|^GDNA-|^ERCC-&quot;</span>,</span>
<span id="cb23-418"><a href="#cb23-418" tabindex="-1"></a>              <span class="at">remove_outliers =</span> <span class="cn">TRUE</span>)</span>
<span id="cb23-419"><a href="#cb23-419" tabindex="-1"></a><span class="co">#&gt; [33mFILTERING THE COUNT DATA. DELETING THE PROVES...[0m</span></span>
<span id="cb23-420"><a href="#cb23-420" tabindex="-1"></a><span class="co">#&gt; [33mREMOVING OUTLIERS...[0m</span></span>
<span id="cb23-421"><a href="#cb23-421" tabindex="-1"></a><span class="co">#&gt; [32mNormalizing data...[0m</span></span>
<span id="cb23-422"><a href="#cb23-422" tabindex="-1"></a><span class="co">#&gt; [33mOBTAINING NORMALIEZ COUNTS...[0m</span></span>
<span id="cb23-423"><a href="#cb23-423" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb23-424"><a href="#cb23-424" tabindex="-1"></a><span class="co">#&gt; [32mChecking for duplicate columns[0m</span></span>
<span id="cb23-425"><a href="#cb23-425" tabindex="-1"></a><span class="co">#&gt; [1] &quot;No duplicate columns.&quot;</span></span>
<span id="cb23-426"><a href="#cb23-426" tabindex="-1"></a><span class="co">#&gt; [32mFiltering data.[0m</span></span>
<span id="cb23-427"><a href="#cb23-427" tabindex="-1"></a><span class="co">#&gt; [32mUsing rownames from TME[0m</span></span>
<span id="cb23-428"><a href="#cb23-428" tabindex="-1"></a><span class="co">#&gt; [32mStarting survival analysis.[0m</span></span>
<span id="cb23-429"><a href="#cb23-429" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-430"><a href="#cb23-430" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  B_cell </span></span>
<span id="cb23-431"><a href="#cb23-431" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb23-432"><a href="#cb23-432" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb23-433"><a href="#cb23-433" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb23-434"><a href="#cb23-434" tabindex="-1"></a><span class="co">#&gt;         0.01396394 </span></span>
<span id="cb23-435"><a href="#cb23-435" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-436"><a href="#cb23-436" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-437"><a href="#cb23-437" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-438"><a href="#cb23-438" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-439"><a href="#cb23-439" tabindex="-1"></a><span class="co">#&gt;                 B_cell=High </span></span>
<span id="cb23-440"><a href="#cb23-440" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-441"><a href="#cb23-441" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-442"><a href="#cb23-442" tabindex="-1"></a><span class="co">#&gt;  2458      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-443"><a href="#cb23-443" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-444"><a href="#cb23-444" tabindex="-1"></a><span class="co">#&gt;                 B_cell=Low </span></span>
<span id="cb23-445"><a href="#cb23-445" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-446"><a href="#cb23-446" tabindex="-1"></a><span class="co">#&gt;  1287      4       1     0.75   0.217        0.426            1</span></span>
<span id="cb23-447"><a href="#cb23-447" tabindex="-1"></a><span class="co">#&gt;  5844      1       1     0.00     NaN           NA           NA</span></span>
<span id="cb23-448"><a href="#cb23-448" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-449"><a href="#cb23-449" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-450"><a href="#cb23-450" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-451"><a href="#cb23-451" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-452"><a href="#cb23-452" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-453"><a href="#cb23-453" tabindex="-1"></a><span class="co">#&gt;             N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-454"><a href="#cb23-454" tabindex="-1"></a><span class="co">#&gt; B_cell=High 4        2     1.03     0.904      1.48</span></span>
<span id="cb23-455"><a href="#cb23-455" tabindex="-1"></a><span class="co">#&gt; B_cell=Low  4        2     2.97     0.315      1.48</span></span>
<span id="cb23-456"><a href="#cb23-456" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-457"><a href="#cb23-457" tabindex="-1"></a><span class="co">#&gt;  Chisq= 1.5  on 1 degrees of freedom, p= 0.2 </span></span>
<span id="cb23-458"><a href="#cb23-458" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-459"><a href="#cb23-459" tabindex="-1"></a><span class="co">#&gt; [1] 0.2240827</span></span>
<span id="cb23-460"><a href="#cb23-460" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-461"><a href="#cb23-461" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb23-462"><a href="#cb23-462" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-463"><a href="#cb23-463" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  Cancer_associated_fibroblast </span></span>
<span id="cb23-464"><a href="#cb23-464" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb23-465"><a href="#cb23-465" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb23-466"><a href="#cb23-466" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb23-467"><a href="#cb23-467" tabindex="-1"></a><span class="co">#&gt;        0.008256491 </span></span>
<span id="cb23-468"><a href="#cb23-468" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-469"><a href="#cb23-469" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-470"><a href="#cb23-470" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-471"><a href="#cb23-471" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-472"><a href="#cb23-472" tabindex="-1"></a><span class="co">#&gt;                 Cancer_associated_fibroblast=High </span></span>
<span id="cb23-473"><a href="#cb23-473" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-474"><a href="#cb23-474" tabindex="-1"></a><span class="co">#&gt;  1164      4       1    0.750   0.217       0.4259            1</span></span>
<span id="cb23-475"><a href="#cb23-475" tabindex="-1"></a><span class="co">#&gt;  2458      2       1    0.375   0.286       0.0839            1</span></span>
<span id="cb23-476"><a href="#cb23-476" tabindex="-1"></a><span class="co">#&gt;  5844      1       1    0.000     NaN           NA           NA</span></span>
<span id="cb23-477"><a href="#cb23-477" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-478"><a href="#cb23-478" tabindex="-1"></a><span class="co">#&gt;                 Cancer_associated_fibroblast=Low </span></span>
<span id="cb23-479"><a href="#cb23-479" tabindex="-1"></a><span class="co">#&gt;         time       n.risk      n.event     survival      std.err lower 95% CI </span></span>
<span id="cb23-480"><a href="#cb23-480" tabindex="-1"></a><span class="co">#&gt;     1287.000        2.000        1.000        0.500        0.354        0.125 </span></span>
<span id="cb23-481"><a href="#cb23-481" tabindex="-1"></a><span class="co">#&gt; upper 95% CI </span></span>
<span id="cb23-482"><a href="#cb23-482" tabindex="-1"></a><span class="co">#&gt;        1.000 </span></span>
<span id="cb23-483"><a href="#cb23-483" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-484"><a href="#cb23-484" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-485"><a href="#cb23-485" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-486"><a href="#cb23-486" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-487"><a href="#cb23-487" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-488"><a href="#cb23-488" tabindex="-1"></a><span class="co">#&gt;                                   N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-489"><a href="#cb23-489" tabindex="-1"></a><span class="co">#&gt; Cancer_associated_fibroblast=High 5        3    3.267    0.0218     0.154</span></span>
<span id="cb23-490"><a href="#cb23-490" tabindex="-1"></a><span class="co">#&gt; Cancer_associated_fibroblast=Low  3        1    0.733    0.0970     0.154</span></span>
<span id="cb23-491"><a href="#cb23-491" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-492"><a href="#cb23-492" tabindex="-1"></a><span class="co">#&gt;  Chisq= 0.2  on 1 degrees of freedom, p= 0.7 </span></span>
<span id="cb23-493"><a href="#cb23-493" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-494"><a href="#cb23-494" tabindex="-1"></a><span class="co">#&gt; [1] 0.6948866</span></span>
<span id="cb23-495"><a href="#cb23-495" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-496"><a href="#cb23-496" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb23-497"><a href="#cb23-497" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-498"><a href="#cb23-498" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  T_cell_CD4_ </span></span>
<span id="cb23-499"><a href="#cb23-499" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb23-500"><a href="#cb23-500" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb23-501"><a href="#cb23-501" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb23-502"><a href="#cb23-502" tabindex="-1"></a><span class="co">#&gt;       8.969556e-07 </span></span>
<span id="cb23-503"><a href="#cb23-503" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-504"><a href="#cb23-504" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-505"><a href="#cb23-505" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-506"><a href="#cb23-506" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-507"><a href="#cb23-507" tabindex="-1"></a><span class="co">#&gt;                 T_cell_CD4_=High </span></span>
<span id="cb23-508"><a href="#cb23-508" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-509"><a href="#cb23-509" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-510"><a href="#cb23-510" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-511"><a href="#cb23-511" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-512"><a href="#cb23-512" tabindex="-1"></a><span class="co">#&gt;                 T_cell_CD4_=Low </span></span>
<span id="cb23-513"><a href="#cb23-513" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-514"><a href="#cb23-514" tabindex="-1"></a><span class="co">#&gt;  1164      3       1    0.667   0.272       0.2995            1</span></span>
<span id="cb23-515"><a href="#cb23-515" tabindex="-1"></a><span class="co">#&gt;  1287      2       1    0.333   0.272       0.0673            1</span></span>
<span id="cb23-516"><a href="#cb23-516" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-517"><a href="#cb23-517" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-518"><a href="#cb23-518" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-519"><a href="#cb23-519" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-520"><a href="#cb23-520" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-521"><a href="#cb23-521" tabindex="-1"></a><span class="co">#&gt;                  N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-522"><a href="#cb23-522" tabindex="-1"></a><span class="co">#&gt; T_cell_CD4_=High 5        2      3.1      0.39      2.47</span></span>
<span id="cb23-523"><a href="#cb23-523" tabindex="-1"></a><span class="co">#&gt; T_cell_CD4_=Low  3        2      0.9      1.34      2.47</span></span>
<span id="cb23-524"><a href="#cb23-524" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-525"><a href="#cb23-525" tabindex="-1"></a><span class="co">#&gt;  Chisq= 2.5  on 1 degrees of freedom, p= 0.1 </span></span>
<span id="cb23-526"><a href="#cb23-526" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-527"><a href="#cb23-527" tabindex="-1"></a><span class="co">#&gt; [1] 0.1160831</span></span>
<span id="cb23-528"><a href="#cb23-528" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-529"><a href="#cb23-529" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb23-530"><a href="#cb23-530" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-531"><a href="#cb23-531" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  T_cell_CD8_ </span></span>
<span id="cb23-532"><a href="#cb23-532" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb23-533"><a href="#cb23-533" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb23-534"><a href="#cb23-534" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb23-535"><a href="#cb23-535" tabindex="-1"></a><span class="co">#&gt;       5.286423e-07 </span></span>
<span id="cb23-536"><a href="#cb23-536" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-537"><a href="#cb23-537" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-538"><a href="#cb23-538" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-539"><a href="#cb23-539" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-540"><a href="#cb23-540" tabindex="-1"></a><span class="co">#&gt;                 T_cell_CD8_=High </span></span>
<span id="cb23-541"><a href="#cb23-541" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-542"><a href="#cb23-542" tabindex="-1"></a><span class="co">#&gt;  1164      3       1    0.667   0.272       0.2995            1</span></span>
<span id="cb23-543"><a href="#cb23-543" tabindex="-1"></a><span class="co">#&gt;  1287      2       1    0.333   0.272       0.0673            1</span></span>
<span id="cb23-544"><a href="#cb23-544" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-545"><a href="#cb23-545" tabindex="-1"></a><span class="co">#&gt;                 T_cell_CD8_=Low </span></span>
<span id="cb23-546"><a href="#cb23-546" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-547"><a href="#cb23-547" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-548"><a href="#cb23-548" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-549"><a href="#cb23-549" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-550"><a href="#cb23-550" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-551"><a href="#cb23-551" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-552"><a href="#cb23-552" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-553"><a href="#cb23-553" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-554"><a href="#cb23-554" tabindex="-1"></a><span class="co">#&gt;                  N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-555"><a href="#cb23-555" tabindex="-1"></a><span class="co">#&gt; T_cell_CD8_=High 5        2      0.9      1.34      2.47</span></span>
<span id="cb23-556"><a href="#cb23-556" tabindex="-1"></a><span class="co">#&gt; T_cell_CD8_=Low  3        2      3.1      0.39      2.47</span></span>
<span id="cb23-557"><a href="#cb23-557" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-558"><a href="#cb23-558" tabindex="-1"></a><span class="co">#&gt;  Chisq= 2.5  on 1 degrees of freedom, p= 0.1 </span></span>
<span id="cb23-559"><a href="#cb23-559" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-560"><a href="#cb23-560" tabindex="-1"></a><span class="co">#&gt; [1] 0.1160831</span></span>
<span id="cb23-561"><a href="#cb23-561" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-562"><a href="#cb23-562" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb23-563"><a href="#cb23-563" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-564"><a href="#cb23-564" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  Endothelial_cell </span></span>
<span id="cb23-565"><a href="#cb23-565" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb23-566"><a href="#cb23-566" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb23-567"><a href="#cb23-567" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb23-568"><a href="#cb23-568" tabindex="-1"></a><span class="co">#&gt;        0.004808822 </span></span>
<span id="cb23-569"><a href="#cb23-569" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-570"><a href="#cb23-570" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-571"><a href="#cb23-571" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-572"><a href="#cb23-572" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-573"><a href="#cb23-573" tabindex="-1"></a><span class="co">#&gt;                 Endothelial_cell=High </span></span>
<span id="cb23-574"><a href="#cb23-574" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-575"><a href="#cb23-575" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-576"><a href="#cb23-576" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-577"><a href="#cb23-577" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-578"><a href="#cb23-578" tabindex="-1"></a><span class="co">#&gt;                 Endothelial_cell=Low </span></span>
<span id="cb23-579"><a href="#cb23-579" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-580"><a href="#cb23-580" tabindex="-1"></a><span class="co">#&gt;  1164      3       1    0.667   0.272       0.2995            1</span></span>
<span id="cb23-581"><a href="#cb23-581" tabindex="-1"></a><span class="co">#&gt;  1287      2       1    0.333   0.272       0.0673            1</span></span>
<span id="cb23-582"><a href="#cb23-582" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-583"><a href="#cb23-583" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-584"><a href="#cb23-584" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-585"><a href="#cb23-585" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-586"><a href="#cb23-586" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-587"><a href="#cb23-587" tabindex="-1"></a><span class="co">#&gt;                       N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-588"><a href="#cb23-588" tabindex="-1"></a><span class="co">#&gt; Endothelial_cell=High 5        2      3.1      0.39      2.47</span></span>
<span id="cb23-589"><a href="#cb23-589" tabindex="-1"></a><span class="co">#&gt; Endothelial_cell=Low  3        2      0.9      1.34      2.47</span></span>
<span id="cb23-590"><a href="#cb23-590" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-591"><a href="#cb23-591" tabindex="-1"></a><span class="co">#&gt;  Chisq= 2.5  on 1 degrees of freedom, p= 0.1 </span></span>
<span id="cb23-592"><a href="#cb23-592" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-593"><a href="#cb23-593" tabindex="-1"></a><span class="co">#&gt; [1] 0.1160831</span></span>
<span id="cb23-594"><a href="#cb23-594" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-595"><a href="#cb23-595" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb23-596"><a href="#cb23-596" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-597"><a href="#cb23-597" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  Macrophage </span></span>
<span id="cb23-598"><a href="#cb23-598" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb23-599"><a href="#cb23-599" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb23-600"><a href="#cb23-600" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb23-601"><a href="#cb23-601" tabindex="-1"></a><span class="co">#&gt;         0.03831832 </span></span>
<span id="cb23-602"><a href="#cb23-602" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-603"><a href="#cb23-603" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-604"><a href="#cb23-604" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-605"><a href="#cb23-605" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-606"><a href="#cb23-606" tabindex="-1"></a><span class="co">#&gt;                 Macrophage=High </span></span>
<span id="cb23-607"><a href="#cb23-607" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-608"><a href="#cb23-608" tabindex="-1"></a><span class="co">#&gt;  1164      4       1     0.75   0.217        0.426            1</span></span>
<span id="cb23-609"><a href="#cb23-609" tabindex="-1"></a><span class="co">#&gt;  1287      3       1     0.50   0.250        0.188            1</span></span>
<span id="cb23-610"><a href="#cb23-610" tabindex="-1"></a><span class="co">#&gt;  2458      1       1     0.00     NaN           NA           NA</span></span>
<span id="cb23-611"><a href="#cb23-611" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-612"><a href="#cb23-612" tabindex="-1"></a><span class="co">#&gt;                 Macrophage=Low </span></span>
<span id="cb23-613"><a href="#cb23-613" tabindex="-1"></a><span class="co">#&gt;         time       n.risk      n.event     survival      std.err lower 95% CI </span></span>
<span id="cb23-614"><a href="#cb23-614" tabindex="-1"></a><span class="co">#&gt;         5844            1            1            0          NaN           NA </span></span>
<span id="cb23-615"><a href="#cb23-615" tabindex="-1"></a><span class="co">#&gt; upper 95% CI </span></span>
<span id="cb23-616"><a href="#cb23-616" tabindex="-1"></a><span class="co">#&gt;           NA </span></span>
<span id="cb23-617"><a href="#cb23-617" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-618"><a href="#cb23-618" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-619"><a href="#cb23-619" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-620"><a href="#cb23-620" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-621"><a href="#cb23-621" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-622"><a href="#cb23-622" tabindex="-1"></a><span class="co">#&gt;                 N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-623"><a href="#cb23-623" tabindex="-1"></a><span class="co">#&gt; Macrophage=High 5        3     1.77     0.861      2.14</span></span>
<span id="cb23-624"><a href="#cb23-624" tabindex="-1"></a><span class="co">#&gt; Macrophage=Low  3        1     2.23     0.681      2.14</span></span>
<span id="cb23-625"><a href="#cb23-625" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-626"><a href="#cb23-626" tabindex="-1"></a><span class="co">#&gt;  Chisq= 2.1  on 1 degrees of freedom, p= 0.1 </span></span>
<span id="cb23-627"><a href="#cb23-627" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-628"><a href="#cb23-628" tabindex="-1"></a><span class="co">#&gt; [1] 0.1439024</span></span>
<span id="cb23-629"><a href="#cb23-629" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-630"><a href="#cb23-630" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb23-631"><a href="#cb23-631" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-632"><a href="#cb23-632" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  NK_cell </span></span>
<span id="cb23-633"><a href="#cb23-633" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb23-634"><a href="#cb23-634" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb23-635"><a href="#cb23-635" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb23-636"><a href="#cb23-636" tabindex="-1"></a><span class="co">#&gt;       1.486227e-07 </span></span>
<span id="cb23-637"><a href="#cb23-637" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-638"><a href="#cb23-638" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-639"><a href="#cb23-639" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-640"><a href="#cb23-640" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-641"><a href="#cb23-641" tabindex="-1"></a><span class="co">#&gt;                 NK_cell=High </span></span>
<span id="cb23-642"><a href="#cb23-642" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-643"><a href="#cb23-643" tabindex="-1"></a><span class="co">#&gt;  1164      3       1    0.667   0.272       0.2995            1</span></span>
<span id="cb23-644"><a href="#cb23-644" tabindex="-1"></a><span class="co">#&gt;  1287      2       1    0.333   0.272       0.0673            1</span></span>
<span id="cb23-645"><a href="#cb23-645" tabindex="-1"></a><span class="co">#&gt;  2458      1       1    0.000     NaN           NA           NA</span></span>
<span id="cb23-646"><a href="#cb23-646" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-647"><a href="#cb23-647" tabindex="-1"></a><span class="co">#&gt;                 NK_cell=Low </span></span>
<span id="cb23-648"><a href="#cb23-648" tabindex="-1"></a><span class="co">#&gt;         time       n.risk      n.event     survival      std.err lower 95% CI </span></span>
<span id="cb23-649"><a href="#cb23-649" tabindex="-1"></a><span class="co">#&gt;         5844            1            1            0          NaN           NA </span></span>
<span id="cb23-650"><a href="#cb23-650" tabindex="-1"></a><span class="co">#&gt; upper 95% CI </span></span>
<span id="cb23-651"><a href="#cb23-651" tabindex="-1"></a><span class="co">#&gt;           NA </span></span>
<span id="cb23-652"><a href="#cb23-652" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-653"><a href="#cb23-653" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-654"><a href="#cb23-654" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-655"><a href="#cb23-655" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-656"><a href="#cb23-656" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-657"><a href="#cb23-657" tabindex="-1"></a><span class="co">#&gt;              N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-658"><a href="#cb23-658" tabindex="-1"></a><span class="co">#&gt; NK_cell=High 5        3      1.4     1.829      3.46</span></span>
<span id="cb23-659"><a href="#cb23-659" tabindex="-1"></a><span class="co">#&gt; NK_cell=Low  3        1      2.6     0.985      3.46</span></span>
<span id="cb23-660"><a href="#cb23-660" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-661"><a href="#cb23-661" tabindex="-1"></a><span class="co">#&gt;  Chisq= 3.5  on 1 degrees of freedom, p= 0.06 </span></span>
<span id="cb23-662"><a href="#cb23-662" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-663"><a href="#cb23-663" tabindex="-1"></a><span class="co">#&gt; [1] 0.06289087</span></span>
<span id="cb23-664"><a href="#cb23-664" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-665"><a href="#cb23-665" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb23-666"><a href="#cb23-666" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-667"><a href="#cb23-667" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  uncharacterized_cell </span></span>
<span id="cb23-668"><a href="#cb23-668" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb23-669"><a href="#cb23-669" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb23-670"><a href="#cb23-670" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb23-671"><a href="#cb23-671" tabindex="-1"></a><span class="co">#&gt;          0.4015231 </span></span>
<span id="cb23-672"><a href="#cb23-672" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb23-673"><a href="#cb23-673" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb23-674"><a href="#cb23-674" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-675"><a href="#cb23-675" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-676"><a href="#cb23-676" tabindex="-1"></a><span class="co">#&gt;                 uncharacterized_cell=High </span></span>
<span id="cb23-677"><a href="#cb23-677" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-678"><a href="#cb23-678" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb23-679"><a href="#cb23-679" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb23-680"><a href="#cb23-680" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-681"><a href="#cb23-681" tabindex="-1"></a><span class="co">#&gt;                 uncharacterized_cell=Low </span></span>
<span id="cb23-682"><a href="#cb23-682" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb23-683"><a href="#cb23-683" tabindex="-1"></a><span class="co">#&gt;  1164      3       1    0.667   0.272       0.2995            1</span></span>
<span id="cb23-684"><a href="#cb23-684" tabindex="-1"></a><span class="co">#&gt;  1287      2       1    0.333   0.272       0.0673            1</span></span>
<span id="cb23-685"><a href="#cb23-685" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-686"><a href="#cb23-686" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb23-687"><a href="#cb23-687" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb23-688"><a href="#cb23-688" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb23-689"><a href="#cb23-689" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-690"><a href="#cb23-690" tabindex="-1"></a><span class="co">#&gt;                           N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb23-691"><a href="#cb23-691" tabindex="-1"></a><span class="co">#&gt; uncharacterized_cell=High 5        2      3.1      0.39      2.47</span></span>
<span id="cb23-692"><a href="#cb23-692" tabindex="-1"></a><span class="co">#&gt; uncharacterized_cell=Low  3        2      0.9      1.34      2.47</span></span>
<span id="cb23-693"><a href="#cb23-693" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-694"><a href="#cb23-694" tabindex="-1"></a><span class="co">#&gt;  Chisq= 2.5  on 1 degrees of freedom, p= 0.1 </span></span>
<span id="cb23-695"><a href="#cb23-695" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb23-696"><a href="#cb23-696" tabindex="-1"></a><span class="co">#&gt; [1] 0.1160831</span></span>
<span id="cb23-697"><a href="#cb23-697" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb23-698"><a href="#cb23-698" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb23-699"><a href="#cb23-699" tabindex="-1"></a><span class="co">#&gt; png </span></span>
<span id="cb23-700"><a href="#cb23-700" tabindex="-1"></a><span class="co">#&gt;   2</span></span></code></pre></div>
</div>
</div>
</div>
<div id="rnaseq" class="section level2">
<h2>RNAseq:</h2>
<p>Now let’s show how it works with RNAseq You can download the example
Excel files from the following link: <a href="https://github.com/ISGLOBAL-Rakislova-Lab/HTGAnalyzer/tree/main/SUPLEMENTARY_OUTPUT" class="uri">https://github.com/ISGLOBAL-Rakislova-Lab/HTGAnalyzer/tree/main/SUPLEMENTARY_OUTPUT</a></p>
<div id="htg_auto-for-rnaseq" class="section level3">
<h3>HTG_auto() for RNAseq:</h3>
<div id="htg_auto-all-the-analysis-for-rnaseq" class="section level4">
<h4>HTG_auto(): All the analysis for RNAseq:</h4>
<p>Imagine you have RNAseq data and aim to perform a full suite of
analyses, including Differential Expression Analysis (DEA), Tumor
Microenvironment (TME) analysis, Survival Analysis, and Gene Set
Enrichment Analysis (GSEA).</p>
<p>The following code shows how to run each of these analyses. We’ve
highlighted the variables that can be adjusted in the functions, so
you’re aware of them. However, you can leave them as they are if the
default settings meet your requirements.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">HTG_auto</span>(</span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>  <span class="at">counts_file_path =</span> <span class="st">&quot;~/HTGAnalyzer/counts_RNAseq.xlsx&quot;</span>,  <span class="co"># path to counts data</span></span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>  <span class="at">file_type =</span> <span class="st">&quot;RNAseq&quot;</span>,  <span class="co"># Tipe of file</span></span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>  <span class="at">AnnotData_file_path =</span> <span class="st">&quot;~/HTGAnalyzer/annotdata.xlsx&quot;</span>,  <span class="co"># path to annotdata.</span></span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a>  <span class="at">pattern =</span> <span class="cn">NULL</span>,</span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a>  <span class="at">save_csv =</span> <span class="cn">TRUE</span>,</span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a>  <span class="at">csv_file =</span> <span class="st">&quot;QC_results.csv&quot;</span>,</span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a>  <span class="at">design_formula =</span> <span class="st">&quot;HPV_status&quot;</span>,</span>
<span id="cb24-9"><a href="#cb24-9" tabindex="-1"></a>  <span class="at">percentage_gene =</span> <span class="fl">0.2</span>,</span>
<span id="cb24-10"><a href="#cb24-10" tabindex="-1"></a>  <span class="at">threshold_gene =</span> <span class="dv">200</span>,</span>
<span id="cb24-11"><a href="#cb24-11" tabindex="-1"></a>  <span class="at">threshold_subject =</span> <span class="dv">10</span>,</span>
<span id="cb24-12"><a href="#cb24-12" tabindex="-1"></a>  <span class="at">genes_to_use =</span> <span class="fu">c</span>(<span class="st">&quot;CCND1&quot;</span>, <span class="st">&quot;MMP10&quot;</span>, <span class="st">&quot;CTTN&quot;</span>), <span class="co"># Genes used for survival analysis (optional)</span></span>
<span id="cb24-13"><a href="#cb24-13" tabindex="-1"></a>  <span class="at">heatmap_columns =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>, <span class="st">&quot;Ciclina_D1&quot;</span>),  <span class="co"># Columns for the heatmap</span></span>
<span id="cb24-14"><a href="#cb24-14" tabindex="-1"></a>  <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>, <span class="st">&quot;Associated&quot;</span>, <span class="st">&quot;Independent&quot;</span>),  <span class="co"># Contrast for differential analysis</span></span>
<span id="cb24-15"><a href="#cb24-15" tabindex="-1"></a>  <span class="at">pCutoff =</span> <span class="fl">5e-2</span>,</span>
<span id="cb24-16"><a href="#cb24-16" tabindex="-1"></a>  <span class="at">variable_01 =</span> <span class="st">&quot;Recurrence_01&quot;</span>,   <span class="co"># Variable for analysis</span></span>
<span id="cb24-17"><a href="#cb24-17" tabindex="-1"></a>  <span class="at">time =</span> <span class="st">&quot;Time_to_death_surv&quot;</span>,  <span class="co"># Time for survival analysis</span></span>
<span id="cb24-18"><a href="#cb24-18" tabindex="-1"></a>  <span class="at">DEA =</span> <span class="cn">TRUE</span>,</span>
<span id="cb24-19"><a href="#cb24-19" tabindex="-1"></a>  <span class="at">remove_outliers =</span> <span class="cn">TRUE</span>,</span>
<span id="cb24-20"><a href="#cb24-20" tabindex="-1"></a>  <span class="at">GSEA =</span> <span class="cn">TRUE</span>,</span>
<span id="cb24-21"><a href="#cb24-21" tabindex="-1"></a>  <span class="at">generate_heatmap =</span> <span class="cn">TRUE</span>,</span>
<span id="cb24-22"><a href="#cb24-22" tabindex="-1"></a>  <span class="at">TME =</span> <span class="cn">TRUE</span>,</span>
<span id="cb24-23"><a href="#cb24-23" tabindex="-1"></a>  <span class="at">survival_analysis =</span> <span class="cn">TRUE</span></span>
<span id="cb24-24"><a href="#cb24-24" tabindex="-1"></a>)</span></code></pre></div>
<p><strong>NOTE:</strong> remember that RNAseq data doesn’t have probes
and doesn’t pass for <code>HTG_QC()</code>as we did with HTG data.</p>
<p>The code above is an example. Some of the parameters have default
values in the function, but they are included here for clarity. Refer to
the function documentation for more details on each parameter.</p>
</div>
<div id="htg_auto-skipping-an-analysis-for-rnaseq" class="section level4">
<h4>HTG_auto(): skipping an analysis for RNAseq:</h4>
<p>If you’d like to run some analyses but skip others (for example,
leaving out the survival analysis and GSEA), simply set the
corresponding parameter to FALSE. Here’s an example where the survival
and GSEA analyses are not included:</p>
<p>In cases where a specific analysis requires certain data columns that
you don’t have (such as the survival analysis needing variable_01 and
time columns), you can set those variables to NULL to bypass that
analysis.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">HTG_auto</span>(</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a>  <span class="at">counts_file_path =</span> <span class="st">&quot;~/HTGAnalyzer/counts_RNAseq.xlsx&quot;</span>,  <span class="co"># path to counts data</span></span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>  <span class="at">file_type =</span> <span class="st">&quot;RNAseq&quot;</span>,  <span class="co"># Tipe of file</span></span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a>  <span class="at">AnnotData_file_path =</span> <span class="st">&quot;~/HTGAnalyzer/annotdata_RNAseq.xlsx&quot;</span>,  <span class="co"># path to annotdata.</span></span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a>  <span class="at">pattern =</span> <span class="cn">NULL</span>,</span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a>  <span class="at">save_csv =</span> <span class="cn">TRUE</span>,  </span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a>  <span class="at">csv_file =</span> <span class="st">&quot;QC_results.csv&quot;</span>, </span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a>  <span class="at">design_formula =</span> <span class="st">&quot;HPV_status&quot;</span>, </span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a>  <span class="at">percentage_gene =</span> <span class="fl">0.2</span>,  </span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a>  <span class="at">threshold_gene =</span> <span class="dv">200</span>,  </span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a>  <span class="at">threshold_subject =</span> <span class="dv">10</span>,  </span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a>  <span class="at">genes_to_use =</span> <span class="fu">c</span>(<span class="st">&quot;CCND1&quot;</span>, <span class="st">&quot;MMP10&quot;</span>, <span class="st">&quot;CTTN&quot;</span>),  <span class="co"># Genes used for survival analysis (optional)</span></span>
<span id="cb25-13"><a href="#cb25-13" tabindex="-1"></a>  <span class="at">heatmap_columns =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>, <span class="st">&quot;Ciclina_D1&quot;</span>),  </span>
<span id="cb25-14"><a href="#cb25-14" tabindex="-1"></a>  <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>, <span class="st">&quot;Associated&quot;</span>, <span class="st">&quot;Independent&quot;</span>),  <span class="co"># Contrast for differential analysis</span></span>
<span id="cb25-15"><a href="#cb25-15" tabindex="-1"></a>  <span class="at">pCutoff =</span> <span class="fl">5e-2</span>,  </span>
<span id="cb25-16"><a href="#cb25-16" tabindex="-1"></a>  <span class="at">variable_01 =</span> <span class="st">&quot;Recurrence_01&quot;</span>,  <span class="co"># Variable for survival analysis</span></span>
<span id="cb25-17"><a href="#cb25-17" tabindex="-1"></a>  <span class="at">time =</span> <span class="st">&quot;Time_to_death_surv&quot;</span>,  <span class="co"># Time for survival analysis</span></span>
<span id="cb25-18"><a href="#cb25-18" tabindex="-1"></a>  <span class="at">DEA =</span> <span class="cn">TRUE</span>,  </span>
<span id="cb25-19"><a href="#cb25-19" tabindex="-1"></a>  <span class="at">remove_outliers =</span> <span class="cn">TRUE</span>,  </span>
<span id="cb25-20"><a href="#cb25-20" tabindex="-1"></a>  <span class="at">GSEA =</span> <span class="cn">FALSE</span>,  </span>
<span id="cb25-21"><a href="#cb25-21" tabindex="-1"></a>  <span class="at">generate_heatmap =</span> <span class="cn">TRUE</span>,  </span>
<span id="cb25-22"><a href="#cb25-22" tabindex="-1"></a>  <span class="at">TME =</span> <span class="cn">TRUE</span>,  </span>
<span id="cb25-23"><a href="#cb25-23" tabindex="-1"></a>  <span class="at">survival_analysis =</span> <span class="cn">FALSE</span>  <span class="co"># Skip survival analysis by setting this parameter to FALSE</span></span>
<span id="cb25-24"><a href="#cb25-24" tabindex="-1"></a>)</span></code></pre></div>
<p>In this example, survival analysis is skipped by setting
<code>survival_analysis = FALSE</code> and GSA by setting
<code>GSEA = FALSE</code>. The output will indicate that the survival
analysis step has been skipped.</p>
</div>
</div>
<div id="step-by-step-for-rnaseq" class="section level3">
<h3>STEP BY STEP for RNAseq:</h3>
<p>What if you already had some analysis or you want to get more control
over it? Don’t worry we have the solution.Let’s go
<code>step-by-step</code>, starting with data import:</p>
<div id="load-data-htg_import_counts-for-rnaseq" class="section level4">
<h4>Load data: HTG_import_counts() for RNAseq:</h4>
<p>You can upload data directly from HTG Edge Reveal using this
function.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="do">## LOAD COUNTS DATA </span></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>htg_data <span class="ot">&lt;-</span> <span class="fu">HTG_import_counts</span>(<span class="st">&quot;~/HTGAnalyzer/counts.xlsx&quot;</span>, <span class="st">&quot;RNAseq&quot;</span>)</span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a><span class="fu">head</span>(htg_data)</span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a></span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a><span class="do">## LOAD ANNOTDATA</span></span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a>annotdata<span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_excel</span>(<span class="st">&quot;~/HTGAnalyzer/annotdata.xlsx&quot;</span>)</span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a><span class="fu">head</span>(annotdata)</span></code></pre></div>
<p><strong>NOTE:</strong> For the following functions, we’ll use
tutorial data from the package. If you want to try with your data,
replace <code>counts_data_tutorial</code> with
<code>htg_data</code>.</p>
</div>
<div id="htg_quant_to_qual-for-rnaseq" class="section level4">
<h4>HTG_quant_to_qual() for RNAseq:</h4>
<p>If you need to transform a quantitative column to a qualitative one,
you can use this function.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="fu">head</span>(AnnotData_tutorial)</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a><span class="co">#&gt;                  id  HPV_status Ciclina_D1 FIGO_2021_STAGE Recurrence</span></span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a><span class="co">#&gt; patient_1 patient_1  Associated         10            IIIB        yes</span></span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a><span class="co">#&gt; patient_2 patient_2  Associated         50              II         no</span></span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a><span class="co">#&gt; patient_3 patient_3  Associated         NA            IIIC         no</span></span>
<span id="cb27-6"><a href="#cb27-6" tabindex="-1"></a><span class="co">#&gt; patient_4 patient_4  Associated          0              IB        yes</span></span>
<span id="cb27-7"><a href="#cb27-7" tabindex="-1"></a><span class="co">#&gt; patient_5 patient_5 Independent         50              IB        yes</span></span>
<span id="cb27-8"><a href="#cb27-8" tabindex="-1"></a><span class="co">#&gt; patient_6 patient_6 Independent         80              IB         no</span></span>
<span id="cb27-9"><a href="#cb27-9" tabindex="-1"></a><span class="co">#&gt;           Recurrence_01 Time_to_death_surv Ciclina_D12</span></span>
<span id="cb27-10"><a href="#cb27-10" tabindex="-1"></a><span class="co">#&gt; patient_1             1               1287         low</span></span>
<span id="cb27-11"><a href="#cb27-11" tabindex="-1"></a><span class="co">#&gt; patient_2             0                510         low</span></span>
<span id="cb27-12"><a href="#cb27-12" tabindex="-1"></a><span class="co">#&gt; patient_3             0                762        &lt;NA&gt;</span></span>
<span id="cb27-13"><a href="#cb27-13" tabindex="-1"></a><span class="co">#&gt; patient_4             1               1164         low</span></span>
<span id="cb27-14"><a href="#cb27-14" tabindex="-1"></a><span class="co">#&gt; patient_5             1               5844         low</span></span>
<span id="cb27-15"><a href="#cb27-15" tabindex="-1"></a><span class="co">#&gt; patient_6             0               1436        high</span></span>
<span id="cb27-16"><a href="#cb27-16" tabindex="-1"></a></span>
<span id="cb27-17"><a href="#cb27-17" tabindex="-1"></a>AnnotData_tutorial <span class="ot">&lt;-</span> <span class="fu">HTG_quant_to_qual</span>(AnnotData_tutorial,  <span class="co"># Dataset to modify</span></span>
<span id="cb27-18"><a href="#cb27-18" tabindex="-1"></a>                                        <span class="st">&quot;Ciclina_D1&quot;</span>,  <span class="co"># Column to transform</span></span>
<span id="cb27-19"><a href="#cb27-19" tabindex="-1"></a>                                        <span class="dv">60</span>,  <span class="co"># Threshold to categorize the data</span></span>
<span id="cb27-20"><a href="#cb27-20" tabindex="-1"></a>                                        <span class="st">&quot;high&quot;</span>,  <span class="co"># Value for entries greater than 60</span></span>
<span id="cb27-21"><a href="#cb27-21" tabindex="-1"></a>                                        <span class="st">&quot;low&quot;</span>)  <span class="co"># Value for entries less than 60</span></span>
<span id="cb27-22"><a href="#cb27-22" tabindex="-1"></a><span class="fu">head</span>(AnnotData_tutorial)</span>
<span id="cb27-23"><a href="#cb27-23" tabindex="-1"></a><span class="co">#&gt;                  id  HPV_status Ciclina_D1 FIGO_2021_STAGE Recurrence</span></span>
<span id="cb27-24"><a href="#cb27-24" tabindex="-1"></a><span class="co">#&gt; patient_1 patient_1  Associated         10            IIIB        yes</span></span>
<span id="cb27-25"><a href="#cb27-25" tabindex="-1"></a><span class="co">#&gt; patient_2 patient_2  Associated         50              II         no</span></span>
<span id="cb27-26"><a href="#cb27-26" tabindex="-1"></a><span class="co">#&gt; patient_3 patient_3  Associated         NA            IIIC         no</span></span>
<span id="cb27-27"><a href="#cb27-27" tabindex="-1"></a><span class="co">#&gt; patient_4 patient_4  Associated          0              IB        yes</span></span>
<span id="cb27-28"><a href="#cb27-28" tabindex="-1"></a><span class="co">#&gt; patient_5 patient_5 Independent         50              IB        yes</span></span>
<span id="cb27-29"><a href="#cb27-29" tabindex="-1"></a><span class="co">#&gt; patient_6 patient_6 Independent         80              IB         no</span></span>
<span id="cb27-30"><a href="#cb27-30" tabindex="-1"></a><span class="co">#&gt;           Recurrence_01 Time_to_death_surv Ciclina_D12</span></span>
<span id="cb27-31"><a href="#cb27-31" tabindex="-1"></a><span class="co">#&gt; patient_1             1               1287         low</span></span>
<span id="cb27-32"><a href="#cb27-32" tabindex="-1"></a><span class="co">#&gt; patient_2             0                510         low</span></span>
<span id="cb27-33"><a href="#cb27-33" tabindex="-1"></a><span class="co">#&gt; patient_3             0                762        &lt;NA&gt;</span></span>
<span id="cb27-34"><a href="#cb27-34" tabindex="-1"></a><span class="co">#&gt; patient_4             1               1164         low</span></span>
<span id="cb27-35"><a href="#cb27-35" tabindex="-1"></a><span class="co">#&gt; patient_5             1               5844         low</span></span>
<span id="cb27-36"><a href="#cb27-36" tabindex="-1"></a><span class="co">#&gt; patient_6             0               1436        high</span></span></code></pre></div>
</div>
<div id="htg_subset-for-rnaseq" class="section level4">
<h4>HTG_subset() for RNAseq:</h4>
<p>Now that you have the counts data loaded, you can search for certain
genes of interest, look for specific probes, or filter results from DEA.
This function can also normalize the data by TPM (Transcripts Per
Million).</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a> <span class="co"># Subset rows with the prefix &quot;ERCC&quot;</span></span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a> cts_ERCC <span class="ot">&lt;-</span> <span class="fu">HTG_subset</span>(counts_data_RNAseq, <span class="st">&quot;ERCC&quot;</span>)</span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a><span class="co">#&gt; [32mSubsetting without normalization...[0m</span></span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a><span class="co">#&gt; Dimensions: 9 11</span></span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a></span>
<span id="cb28-6"><a href="#cb28-6" tabindex="-1"></a> <span class="co"># Subset rows with the prefix &quot;POS&quot;</span></span>
<span id="cb28-7"><a href="#cb28-7" tabindex="-1"></a> cts_POS <span class="ot">&lt;-</span> <span class="fu">HTG_subset</span>(counts_data_RNAseq, <span class="st">&quot;POS&quot;</span>)</span>
<span id="cb28-8"><a href="#cb28-8" tabindex="-1"></a><span class="co">#&gt; [32mSubsetting without normalization...[0m</span></span>
<span id="cb28-9"><a href="#cb28-9" tabindex="-1"></a><span class="co">#&gt; Dimensions: 1 11</span></span>
<span id="cb28-10"><a href="#cb28-10" tabindex="-1"></a></span>
<span id="cb28-11"><a href="#cb28-11" tabindex="-1"></a> <span class="co"># Subset rows with the prefix &quot;NC&quot;</span></span>
<span id="cb28-12"><a href="#cb28-12" tabindex="-1"></a> cts_NC <span class="ot">&lt;-</span> <span class="fu">HTG_subset</span>(counts_data_RNAseq, <span class="st">&quot;NC&quot;</span>)</span>
<span id="cb28-13"><a href="#cb28-13" tabindex="-1"></a><span class="co">#&gt; [32mSubsetting without normalization...[0m</span></span>
<span id="cb28-14"><a href="#cb28-14" tabindex="-1"></a><span class="co">#&gt; Dimensions: 46 11</span></span>
<span id="cb28-15"><a href="#cb28-15" tabindex="-1"></a> </span>
<span id="cb28-16"><a href="#cb28-16" tabindex="-1"></a><span class="co"># Subset rows with the prefix &quot;GDNA&quot;</span></span>
<span id="cb28-17"><a href="#cb28-17" tabindex="-1"></a> cts_GDNA <span class="ot">&lt;-</span> <span class="fu">HTG_subset</span>(counts_data_RNAseq, <span class="st">&quot;GDNA&quot;</span>)</span>
<span id="cb28-18"><a href="#cb28-18" tabindex="-1"></a><span class="co">#&gt; [32mSubsetting without normalization...[0m</span></span>
<span id="cb28-19"><a href="#cb28-19" tabindex="-1"></a><span class="co">#&gt; Dimensions: 0 11</span></span>
<span id="cb28-20"><a href="#cb28-20" tabindex="-1"></a></span>
<span id="cb28-21"><a href="#cb28-21" tabindex="-1"></a> <span class="co"># Subset rows with the prefix &quot;ZZZ3&quot;</span></span>
<span id="cb28-22"><a href="#cb28-22" tabindex="-1"></a> cts_ZZZ3 <span class="ot">&lt;-</span> <span class="fu">HTG_subset</span>(counts_data_RNAseq, <span class="st">&quot;ZZZ3&quot;</span>)</span>
<span id="cb28-23"><a href="#cb28-23" tabindex="-1"></a><span class="co">#&gt; [32mSubsetting without normalization...[0m</span></span>
<span id="cb28-24"><a href="#cb28-24" tabindex="-1"></a><span class="co">#&gt; Dimensions: 1 11</span></span>
<span id="cb28-25"><a href="#cb28-25" tabindex="-1"></a></span>
<span id="cb28-26"><a href="#cb28-26" tabindex="-1"></a>  <span class="co"># Subset rows with the prefix &quot;ZZZ3&quot; and normalize the data</span></span>
<span id="cb28-27"><a href="#cb28-27" tabindex="-1"></a> cts_ZZZ3 <span class="ot">&lt;-</span> <span class="fu">HTG_subset</span>(counts_data_RNAseq, <span class="st">&quot;ZZZ3&quot;</span>, <span class="at">normalize =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-28"><a href="#cb28-28" tabindex="-1"></a><span class="co">#&gt; [32mNormalizing data using TPM (Transcripts Per Million)...[0m</span></span>
<span id="cb28-29"><a href="#cb28-29" tabindex="-1"></a><span class="co">#&gt; Dimensions: 1 11</span></span>
<span id="cb28-30"><a href="#cb28-30" tabindex="-1"></a></span>
<span id="cb28-31"><a href="#cb28-31" tabindex="-1"></a> resAAAS<span class="ot">&lt;-</span> <span class="fu">HTG_subset</span>(res_tutorial, <span class="st">&quot;AAAS&quot;</span>)</span>
<span id="cb28-32"><a href="#cb28-32" tabindex="-1"></a><span class="co">#&gt; [32mSubsetting without normalization...[0m</span></span>
<span id="cb28-33"><a href="#cb28-33" tabindex="-1"></a><span class="co">#&gt; Dimensions: 1 6</span></span></code></pre></div>
</div>
<div id="htg_calculate_summary_stats-for-rnaseq" class="section level4">
<h4>HTG_calculate_summary_stats() for RNAseq:</h4>
<p>As RNaseq doen’t have a QC() if is needed you can perform a summary
stats.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a>summary <span class="ot">&lt;-</span> <span class="fu">HTG_calculate_summary_stats</span>(counts_data_RNAseq, <span class="at">pattern =</span> <span class="cn">NULL</span>)</span></code></pre></div>
</div>
<div id="perform-all-the-analysis-htg_analysis-for-rnaseq" class="section level4">
<h4>Perform all the analysis: HTG_analysis() for RNAseq:</h4>
<p>Now that you know which variables are outliers, you can perform any
of the analyses.This function will remove the outliers, perform DEA,
TME, survival analysis, and GSEA. If you want to perform each step
separately, you can use the functions <code>HTG_GSEA()</code>,
<code>HTG_DEA()</code>, <code>HTG_TME()</code>, and
<code>HTG_survival()</code>.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>ALL_done <span class="ot">&lt;-</span> <span class="fu">HTG_analysis</span>(<span class="at">outliers =</span> <span class="cn">NULL</span>, <span class="co"># RNAseq doen&#39;t have outliers. </span></span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>                         <span class="at">pattern =</span> <span class="cn">NULL</span>, </span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a>                         <span class="at">counts_data =</span> counts_data_RNAseq, </span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a>                         <span class="at">col_data =</span> AnnotData_tutorial, </span>
<span id="cb30-5"><a href="#cb30-5" tabindex="-1"></a>                         <span class="at">design_formula =</span> <span class="st">&quot;HPV_status&quot;</span>, </span>
<span id="cb30-6"><a href="#cb30-6" tabindex="-1"></a>                         <span class="at">percentage_gene =</span> <span class="fl">0.2</span>, </span>
<span id="cb30-7"><a href="#cb30-7" tabindex="-1"></a>                         <span class="at">threshold_gene =</span> <span class="dv">200</span>, </span>
<span id="cb30-8"><a href="#cb30-8" tabindex="-1"></a>                         <span class="at">threshold_subject =</span> <span class="dv">10</span>,</span>
<span id="cb30-9"><a href="#cb30-9" tabindex="-1"></a>                         <span class="at">genes_to_use =</span> <span class="fu">c</span>(<span class="st">&quot;CCND1&quot;</span>, <span class="st">&quot;MMP10&quot;</span>, <span class="st">&quot;CTTN&quot;</span>),</span>
<span id="cb30-10"><a href="#cb30-10" tabindex="-1"></a>                         <span class="at">heatmap_columns =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>,<span class="st">&quot;Ciclina_D1&quot;</span>),</span>
<span id="cb30-11"><a href="#cb30-11" tabindex="-1"></a>                         <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>, <span class="st">&quot;Associated&quot;</span>, <span class="st">&quot;Independent&quot;</span>), </span>
<span id="cb30-12"><a href="#cb30-12" tabindex="-1"></a>                         <span class="at">pCutoff =</span> <span class="fl">5e-2</span>, </span>
<span id="cb30-13"><a href="#cb30-13" tabindex="-1"></a>                         <span class="at">variable_01 =</span> <span class="st">&quot;Recurrence_01&quot;</span>,</span>
<span id="cb30-14"><a href="#cb30-14" tabindex="-1"></a>                         <span class="at">time =</span> <span class="st">&quot;Time_to_death_surv&quot;</span>, </span>
<span id="cb30-15"><a href="#cb30-15" tabindex="-1"></a>                         <span class="at">DEA =</span> <span class="cn">TRUE</span>, </span>
<span id="cb30-16"><a href="#cb30-16" tabindex="-1"></a>                         <span class="at">remove_outliers =</span> <span class="cn">FALSE</span>, <span class="co">#set false because don&#39;t have outliers. </span></span>
<span id="cb30-17"><a href="#cb30-17" tabindex="-1"></a>                         <span class="at">GSEA =</span> <span class="cn">FALSE</span>, </span>
<span id="cb30-18"><a href="#cb30-18" tabindex="-1"></a>                         <span class="at">generate_heatmap =</span> <span class="cn">TRUE</span>, </span>
<span id="cb30-19"><a href="#cb30-19" tabindex="-1"></a>                         <span class="at">TME =</span> <span class="cn">TRUE</span>, </span>
<span id="cb30-20"><a href="#cb30-20" tabindex="-1"></a>                         <span class="at">survival_analysis =</span> <span class="cn">FALSE</span>)</span>
<span id="cb30-21"><a href="#cb30-21" tabindex="-1"></a><span class="co">#&gt; [33mSTARTING THE DIFERENTIAN EXPRESSION ANALYSIS.[0m</span></span>
<span id="cb30-22"><a href="#cb30-22" tabindex="-1"></a><span class="co">#&gt; [1] &quot;HPV_status&quot;</span></span>
<span id="cb30-23"><a href="#cb30-23" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb30-24"><a href="#cb30-24" tabindex="-1"></a><span class="co">#&gt; [32m[0m</span></span>
<span id="cb30-25"><a href="#cb30-25" tabindex="-1"></a><span class="co">#&gt; [32mBEFORE FILTERING[0m</span></span>
<span id="cb30-26"><a href="#cb30-26" tabindex="-1"></a><span class="co">#&gt; class: DESeqDataSet </span></span>
<span id="cb30-27"><a href="#cb30-27" tabindex="-1"></a><span class="co">#&gt; dim: 19394 11 </span></span>
<span id="cb30-28"><a href="#cb30-28" tabindex="-1"></a><span class="co">#&gt; metadata(1): version</span></span>
<span id="cb30-29"><a href="#cb30-29" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb30-30"><a href="#cb30-30" tabindex="-1"></a><span class="co">#&gt; rownames(19394): A3GALT2 A4GALT ... ZZEF1 ZZZ3</span></span>
<span id="cb30-31"><a href="#cb30-31" tabindex="-1"></a><span class="co">#&gt; rowData names(0):</span></span>
<span id="cb30-32"><a href="#cb30-32" tabindex="-1"></a><span class="co">#&gt; colnames(11): patient_1 patient_10 ... patient_8 patient_9</span></span>
<span id="cb30-33"><a href="#cb30-33" tabindex="-1"></a><span class="co">#&gt; colData names(8): id HPV_status ... Time_to_death_surv Ciclina_D12</span></span>
<span id="cb30-34"><a href="#cb30-34" tabindex="-1"></a><span class="co">#&gt; [32m[0m</span></span>
<span id="cb30-35"><a href="#cb30-35" tabindex="-1"></a><span class="co">#&gt; [32m [0m</span></span>
<span id="cb30-36"><a href="#cb30-36" tabindex="-1"></a><span class="co">#&gt; [32m[32mAFTER FILTERING[0m</span></span>
<span id="cb30-37"><a href="#cb30-37" tabindex="-1"></a><span class="co">#&gt; class: DESeqDataSet </span></span>
<span id="cb30-38"><a href="#cb30-38" tabindex="-1"></a><span class="co">#&gt; dim: 10491 11 </span></span>
<span id="cb30-39"><a href="#cb30-39" tabindex="-1"></a><span class="co">#&gt; metadata(1): version</span></span>
<span id="cb30-40"><a href="#cb30-40" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb30-41"><a href="#cb30-41" tabindex="-1"></a><span class="co">#&gt; rownames(10491): A4GALT AAAS ... ZZEF1 ZZZ3</span></span>
<span id="cb30-42"><a href="#cb30-42" tabindex="-1"></a><span class="co">#&gt; rowData names(0):</span></span>
<span id="cb30-43"><a href="#cb30-43" tabindex="-1"></a><span class="co">#&gt; colnames(11): patient_1 patient_10 ... patient_8 patient_9</span></span>
<span id="cb30-44"><a href="#cb30-44" tabindex="-1"></a><span class="co">#&gt; colData names(8): id HPV_status ... Time_to_death_surv Ciclina_D12</span></span>
<span id="cb30-45"><a href="#cb30-45" tabindex="-1"></a><span class="co">#&gt; [32m[32m [0m</span></span>
<span id="cb30-46"><a href="#cb30-46" tabindex="-1"></a><span class="co">#&gt; -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the</span></span>
<span id="cb30-47"><a href="#cb30-47" tabindex="-1"></a><span class="co">#&gt;    function: y = a/x + b, and a local regression fit was automatically substituted.</span></span>
<span id="cb30-48"><a href="#cb30-48" tabindex="-1"></a><span class="co">#&gt;    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.</span></span>
<span id="cb30-49"><a href="#cb30-49" tabindex="-1"></a><span class="co">#&gt; [32m[32mRESULTS OF VST NORMALIZATION[0m</span></span>
<span id="cb30-50"><a href="#cb30-50" tabindex="-1"></a><span class="co">#&gt; class: DESeqTransform </span></span>
<span id="cb30-51"><a href="#cb30-51" tabindex="-1"></a><span class="co">#&gt; dim: 10491 11 </span></span>
<span id="cb30-52"><a href="#cb30-52" tabindex="-1"></a><span class="co">#&gt; metadata(1): version</span></span>
<span id="cb30-53"><a href="#cb30-53" tabindex="-1"></a><span class="co">#&gt; assays(1): &#39;&#39;</span></span>
<span id="cb30-54"><a href="#cb30-54" tabindex="-1"></a><span class="co">#&gt; rownames(10491): A4GALT AAAS ... ZZEF1 ZZZ3</span></span>
<span id="cb30-55"><a href="#cb30-55" tabindex="-1"></a><span class="co">#&gt; rowData names(4): baseMean baseVar allZero dispFit</span></span>
<span id="cb30-56"><a href="#cb30-56" tabindex="-1"></a><span class="co">#&gt; colnames(11): patient_1 patient_10 ... patient_8 patient_9</span></span>
<span id="cb30-57"><a href="#cb30-57" tabindex="-1"></a><span class="co">#&gt; colData names(9): id HPV_status ... Ciclina_D12 sizeFactor</span></span>
<span id="cb30-58"><a href="#cb30-58" tabindex="-1"></a><span class="co">#&gt; [32m[32m [0m</span></span>
<span id="cb30-59"><a href="#cb30-59" tabindex="-1"></a><span class="co">#&gt; estimating size factors</span></span>
<span id="cb30-60"><a href="#cb30-60" tabindex="-1"></a><span class="co">#&gt; estimating dispersions</span></span>
<span id="cb30-61"><a href="#cb30-61" tabindex="-1"></a><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span id="cb30-62"><a href="#cb30-62" tabindex="-1"></a><span class="co">#&gt; mean-dispersion relationship</span></span>
<span id="cb30-63"><a href="#cb30-63" tabindex="-1"></a><span class="co">#&gt; -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the</span></span>
<span id="cb30-64"><a href="#cb30-64" tabindex="-1"></a><span class="co">#&gt;    function: y = a/x + b, and a local regression fit was automatically substituted.</span></span>
<span id="cb30-65"><a href="#cb30-65" tabindex="-1"></a><span class="co">#&gt;    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.</span></span>
<span id="cb30-66"><a href="#cb30-66" tabindex="-1"></a><span class="co">#&gt; final dispersion estimates</span></span>
<span id="cb30-67"><a href="#cb30-67" tabindex="-1"></a><span class="co">#&gt; fitting model and testing</span></span>
<span id="cb30-68"><a href="#cb30-68" tabindex="-1"></a><span class="co">#&gt; -- replacing outliers and refitting for 2174 genes</span></span>
<span id="cb30-69"><a href="#cb30-69" tabindex="-1"></a><span class="co">#&gt; -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 </span></span>
<span id="cb30-70"><a href="#cb30-70" tabindex="-1"></a><span class="co">#&gt; -- original counts are preserved in counts(dds)</span></span>
<span id="cb30-71"><a href="#cb30-71" tabindex="-1"></a><span class="co">#&gt; estimating dispersions</span></span>
<span id="cb30-72"><a href="#cb30-72" tabindex="-1"></a><span class="co">#&gt; fitting model and testing</span></span>
<span id="cb30-73"><a href="#cb30-73" tabindex="-1"></a><span class="co">#&gt; [32m[32mHEATMAP[0m</span></span>
<span id="cb30-74"><a href="#cb30-74" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING HEATMAP[0m</span></span>
<span id="cb30-75"><a href="#cb30-75" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING CONTRAST RESULTS[0m</span></span>
<span id="cb30-76"><a href="#cb30-76" tabindex="-1"></a><span class="co">#&gt; [32mSUMMARY OF RESULT OF THE CONTRAST[0m</span></span>
<span id="cb30-77"><a href="#cb30-77" tabindex="-1"></a><span class="co">#&gt; [1] &quot;DESeqResults object of length 6 with 2 metadata columns&quot;</span></span>
<span id="cb30-78"><a href="#cb30-78" tabindex="-1"></a><span class="co">#&gt; [32m [0m</span></span>
<span id="cb30-79"><a href="#cb30-79" tabindex="-1"></a><span class="co">#&gt; [32mRESULTS OF THE CONTRAST (TOP 10 p-adj)[0m</span></span>
<span id="cb30-80"><a href="#cb30-80" tabindex="-1"></a><span class="co">#&gt; log2 fold change (MLE): HPV_status Associated vs Independent </span></span>
<span id="cb30-81"><a href="#cb30-81" tabindex="-1"></a><span class="co">#&gt; Wald test p-value: HPV_status Associated vs Independent </span></span>
<span id="cb30-82"><a href="#cb30-82" tabindex="-1"></a><span class="co">#&gt; DataFrame with 10 rows and 6 columns</span></span>
<span id="cb30-83"><a href="#cb30-83" tabindex="-1"></a><span class="co">#&gt;           baseMean log2FoldChange     lfcSE      stat      pvalue        padj</span></span>
<span id="cb30-84"><a href="#cb30-84" tabindex="-1"></a><span class="co">#&gt;          &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span id="cb30-85"><a href="#cb30-85" tabindex="-1"></a><span class="co">#&gt; PPP2R2C    450.322      -12.18328  1.372048  -8.87963 6.70822e-19 6.85848e-15</span></span>
<span id="cb30-86"><a href="#cb30-86" tabindex="-1"></a><span class="co">#&gt; PDZK1IP1   680.205      -10.43474  1.293814  -8.06510 7.31736e-16 3.74064e-12</span></span>
<span id="cb30-87"><a href="#cb30-87" tabindex="-1"></a><span class="co">#&gt; MMP10      898.934       -9.69845  1.246862  -7.77829 7.35138e-15 2.50535e-11</span></span>
<span id="cb30-88"><a href="#cb30-88" tabindex="-1"></a><span class="co">#&gt; LYPD3     1101.958       -6.86637  0.927191  -7.40556 1.30596e-13 3.33803e-10</span></span>
<span id="cb30-89"><a href="#cb30-89" tabindex="-1"></a><span class="co">#&gt; IRF6       564.565       -7.24378  1.075176  -6.73730 1.61356e-11 3.29941e-08</span></span>
<span id="cb30-90"><a href="#cb30-90" tabindex="-1"></a><span class="co">#&gt; IL20RB     693.154       -5.60484  0.837154  -6.69512 2.15493e-11 3.67200e-08</span></span>
<span id="cb30-91"><a href="#cb30-91" tabindex="-1"></a><span class="co">#&gt; IGSF3      473.223       -5.71455  0.880058  -6.49337 8.39346e-11 1.22593e-07</span></span>
<span id="cb30-92"><a href="#cb30-92" tabindex="-1"></a><span class="co">#&gt; KRT80      277.950       -7.75546  1.221085  -6.35128 2.13527e-10 2.53673e-07</span></span>
<span id="cb30-93"><a href="#cb30-93" tabindex="-1"></a><span class="co">#&gt; SULT2B1    589.880       -7.01528  1.105745  -6.34439 2.23304e-10 2.53673e-07</span></span>
<span id="cb30-94"><a href="#cb30-94" tabindex="-1"></a><span class="co">#&gt; MMP13      309.630       -6.16267  0.982895  -6.26992 3.61236e-10 3.69328e-07</span></span>
<span id="cb30-95"><a href="#cb30-95" tabindex="-1"></a><span class="co">#&gt; [32m [0m</span></span>
<span id="cb30-96"><a href="#cb30-96" tabindex="-1"></a><span class="co">#&gt; [32mRESULTS of DEA WILL BE SAVED IN .CSV IN YOUR CURRENT DIRECTORY[0m</span></span>
<span id="cb30-97"><a href="#cb30-97" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING POISSON DISTANCES PLOT[0m</span></span>
<span id="cb30-98"><a href="#cb30-98" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING VSD DISTANCE HEATMAP[0m</span></span>
<span id="cb30-99"><a href="#cb30-99" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING VSD BOXPLOT[0m</span></span>
<span id="cb30-100"><a href="#cb30-100" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING PCA PLOT[0m</span></span>
<span id="cb30-101"><a href="#cb30-101" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING VOLCANO PLOT[0m</span></span>
<span id="cb30-102"><a href="#cb30-102" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING MA PLOT[0m</span></span>
<span id="cb30-103"><a href="#cb30-103" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING DISPERSION PLOT[0m</span></span>
<span id="cb30-104"><a href="#cb30-104" tabindex="-1"></a><span class="co">#&gt; [33mSTARTING TME[0m</span></span>
<span id="cb30-105"><a href="#cb30-105" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Inicial gene number: 19394&quot;</span></span>
<span id="cb30-106"><a href="#cb30-106" tabindex="-1"></a><span class="co">#&gt; [32mTPM normalization performed and stored on tpm_counts.csv[0m</span></span>
<span id="cb30-107"><a href="#cb30-107" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt;--- This function is being optimised and we strongly recommend that you should set `source` as `local`....</span></span>
<span id="cb30-108"><a href="#cb30-108" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Number of genes omitted during TPM normalization due to their length not being available in Biomart:   1086&quot;</span></span>
<span id="cb30-109"><a href="#cb30-109" tabindex="-1"></a><span class="co">#&gt; [32mWe are going to use information from DEA[0m</span></span>
<span id="cb30-110"><a href="#cb30-110" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb30-111"><a href="#cb30-111" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt; Running epic</span></span>
<span id="cb30-112"><a href="#cb30-112" tabindex="-1"></a><span class="co">#&gt; Warning in (function (bulk, reference = NULL, mRNA_cell = NULL, mRNA_cell_sub = NULL, : The optimization didn&#39;t fully converge for some samples:</span></span>
<span id="cb30-113"><a href="#cb30-113" tabindex="-1"></a><span class="co">#&gt; patient_3</span></span>
<span id="cb30-114"><a href="#cb30-114" tabindex="-1"></a><span class="co">#&gt;  - check fit.gof for the convergeCode and convergeMessage</span></span>
<span id="cb30-115"><a href="#cb30-115" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb30-116"><a href="#cb30-116" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt; Running quantiseq</span></span>
<span id="cb30-117"><a href="#cb30-117" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb30-118"><a href="#cb30-118" tabindex="-1"></a><span class="co">#&gt; Running quanTIseq deconvolution module</span></span>
<span id="cb30-119"><a href="#cb30-119" tabindex="-1"></a><span class="co">#&gt; Gene expression normalization and re-annotation (arrays: FALSE)</span></span>
<span id="cb30-120"><a href="#cb30-120" tabindex="-1"></a><span class="co">#&gt; Removing 17 noisy genes</span></span>
<span id="cb30-121"><a href="#cb30-121" tabindex="-1"></a><span class="co">#&gt; Removing 15 genes with high expression in tumors</span></span>
<span id="cb30-122"><a href="#cb30-122" tabindex="-1"></a><span class="co">#&gt; Signature genes found in data set: 132/138 (95.65%)</span></span>
<span id="cb30-123"><a href="#cb30-123" tabindex="-1"></a><span class="co">#&gt; Mixture deconvolution (method: lsei)</span></span>
<span id="cb30-124"><a href="#cb30-124" tabindex="-1"></a><span class="co">#&gt; Deconvolution successful!</span></span>
<span id="cb30-125"><a href="#cb30-125" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb30-126"><a href="#cb30-126" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt; Running xcell</span></span>
<span id="cb30-127"><a href="#cb30-127" tabindex="-1"></a><span class="co">#&gt; [32mresults of the TME will be stored in imm_epic.csv, imm_qti.csv and imm_xcell.csv [0m</span></span>
<span id="cb30-128"><a href="#cb30-128" tabindex="-1"></a><span class="co">#&gt; [32mAre they in the same order?[0m</span></span>
<span id="cb30-129"><a href="#cb30-129" tabindex="-1"></a><span class="co">#&gt; [32mHave to be true.[0m</span></span>
<span id="cb30-130"><a href="#cb30-130" tabindex="-1"></a><span class="co">#&gt; [32mEPIC[0m</span></span>
<span id="cb30-131"><a href="#cb30-131" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb30-132"><a href="#cb30-132" tabindex="-1"></a><span class="co">#&gt; [32mqti[0m</span></span>
<span id="cb30-133"><a href="#cb30-133" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb30-134"><a href="#cb30-134" tabindex="-1"></a><span class="co">#&gt; [32mxcell[0m</span></span>
<span id="cb30-135"><a href="#cb30-135" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb30-136"><a href="#cb30-136" tabindex="-1"></a><span class="co">#&gt; [33mApply the filtering and normality test for each dataset[0m</span></span>
<span id="cb30-137"><a href="#cb30-137" tabindex="-1"></a><span class="co">#&gt; [32mChecking if All Values in Each Column Are Equal in dataset: quanTIseq [0m</span></span>
<span id="cb30-138"><a href="#cb30-138" tabindex="-1"></a><span class="co">#&gt; [33mIn dataset quanTIseq the following columns have the same value in all rows and will be excluded from the analysis:[0m</span></span>
<span id="cb30-139"><a href="#cb30-139" tabindex="-1"></a><span class="co">#&gt; character(0)</span></span>
<span id="cb30-140"><a href="#cb30-140" tabindex="-1"></a><span class="co">#&gt; [32mChecking if All Values in Each Column Are Equal in dataset: EPIC [0m</span></span>
<span id="cb30-141"><a href="#cb30-141" tabindex="-1"></a><span class="co">#&gt; [33mIn dataset EPIC the following columns have the same value in all rows and will be excluded from the analysis:[0m</span></span>
<span id="cb30-142"><a href="#cb30-142" tabindex="-1"></a><span class="co">#&gt; character(0)</span></span>
<span id="cb30-143"><a href="#cb30-143" tabindex="-1"></a><span class="co">#&gt; [32mChecking if All Values in Each Column Are Equal in dataset: xcell [0m</span></span>
<span id="cb30-144"><a href="#cb30-144" tabindex="-1"></a><span class="co">#&gt; [33mIn dataset xcell the following columns have the same value in all rows and will be excluded from the analysis:[0m</span></span>
<span id="cb30-145"><a href="#cb30-145" tabindex="-1"></a><span class="co">#&gt; character(0)</span></span>
<span id="cb30-146"><a href="#cb30-146" tabindex="-1"></a><span class="co">#&gt; [32mPerforming Shapiro-Wilk test for quanTIseq[0m</span></span>
<span id="cb30-147"><a href="#cb30-147" tabindex="-1"></a><span class="co">#&gt; # A tibble: 11 × 3</span></span>
<span id="cb30-148"><a href="#cb30-148" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                    shapiro_test      p.value</span></span>
<span id="cb30-149"><a href="#cb30-149" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                        &lt;list&gt;              &lt;dbl&gt;</span></span>
<span id="cb30-150"><a href="#cb30-150" tabindex="-1"></a><span class="co">#&gt;  1 B_cell                       &lt;htest&gt;      0.368       </span></span>
<span id="cb30-151"><a href="#cb30-151" tabindex="-1"></a><span class="co">#&gt;  2 Macrophage_M1                &lt;htest&gt;      0.000220    </span></span>
<span id="cb30-152"><a href="#cb30-152" tabindex="-1"></a><span class="co">#&gt;  3 Macrophage_M2                &lt;htest&gt;      0.275       </span></span>
<span id="cb30-153"><a href="#cb30-153" tabindex="-1"></a><span class="co">#&gt;  4 Monocyte                     &lt;htest&gt;      0.000420    </span></span>
<span id="cb30-154"><a href="#cb30-154" tabindex="-1"></a><span class="co">#&gt;  5 Myeloid_dendritic_cell       &lt;htest&gt;      0.000000124 </span></span>
<span id="cb30-155"><a href="#cb30-155" tabindex="-1"></a><span class="co">#&gt;  6 NK_cell                      &lt;htest&gt;      0.0837      </span></span>
<span id="cb30-156"><a href="#cb30-156" tabindex="-1"></a><span class="co">#&gt;  7 Neutrophil                   &lt;htest&gt;      0.0000000224</span></span>
<span id="cb30-157"><a href="#cb30-157" tabindex="-1"></a><span class="co">#&gt;  8 T_cell_CD4+_(non-regulatory) &lt;htest&gt;      0.00000221  </span></span>
<span id="cb30-158"><a href="#cb30-158" tabindex="-1"></a><span class="co">#&gt;  9 T_cell_CD8+                  &lt;htest&gt;      0.000309    </span></span>
<span id="cb30-159"><a href="#cb30-159" tabindex="-1"></a><span class="co">#&gt; 10 T_cell_regulatory_(Tregs)    &lt;htest&gt;      0.255       </span></span>
<span id="cb30-160"><a href="#cb30-160" tabindex="-1"></a><span class="co">#&gt; 11 uncharacterized_cell         &lt;htest&gt;      0.207       </span></span>
<span id="cb30-161"><a href="#cb30-161" tabindex="-1"></a><span class="co">#&gt; [32mPerforming parametric tests for quanTIseq[0m</span></span>
<span id="cb30-162"><a href="#cb30-162" tabindex="-1"></a><span class="co">#&gt; # A tibble: 11 × 3</span></span>
<span id="cb30-163"><a href="#cb30-163" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                    Test    p.value</span></span>
<span id="cb30-164"><a href="#cb30-164" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                        &lt;chr&gt;     &lt;dbl&gt;</span></span>
<span id="cb30-165"><a href="#cb30-165" tabindex="-1"></a><span class="co">#&gt;  1 B_cell                       t-test 0.107   </span></span>
<span id="cb30-166"><a href="#cb30-166" tabindex="-1"></a><span class="co">#&gt;  2 Macrophage_M1                t-test 0.475   </span></span>
<span id="cb30-167"><a href="#cb30-167" tabindex="-1"></a><span class="co">#&gt;  3 Macrophage_M2                t-test 0.120   </span></span>
<span id="cb30-168"><a href="#cb30-168" tabindex="-1"></a><span class="co">#&gt;  4 Monocyte                     t-test 0.218   </span></span>
<span id="cb30-169"><a href="#cb30-169" tabindex="-1"></a><span class="co">#&gt;  5 Neutrophil                   t-test 0.356   </span></span>
<span id="cb30-170"><a href="#cb30-170" tabindex="-1"></a><span class="co">#&gt;  6 NK_cell                      t-test 0.862   </span></span>
<span id="cb30-171"><a href="#cb30-171" tabindex="-1"></a><span class="co">#&gt;  7 T_cell_CD4+_(non-regulatory) t-test 0.348   </span></span>
<span id="cb30-172"><a href="#cb30-172" tabindex="-1"></a><span class="co">#&gt;  8 T_cell_CD8+                  t-test 0.0706  </span></span>
<span id="cb30-173"><a href="#cb30-173" tabindex="-1"></a><span class="co">#&gt;  9 T_cell_regulatory_(Tregs)    t-test 0.449   </span></span>
<span id="cb30-174"><a href="#cb30-174" tabindex="-1"></a><span class="co">#&gt; 10 Myeloid_dendritic_cell       t-test 0.291   </span></span>
<span id="cb30-175"><a href="#cb30-175" tabindex="-1"></a><span class="co">#&gt; 11 uncharacterized_cell         t-test 0.000153</span></span>
<span id="cb30-176"><a href="#cb30-176" tabindex="-1"></a><span class="co">#&gt; [32mPerforming Shapiro-Wilk test for EPIC[0m</span></span>
<span id="cb30-177"><a href="#cb30-177" tabindex="-1"></a><span class="co">#&gt; # A tibble: 8 × 3</span></span>
<span id="cb30-178"><a href="#cb30-178" tabindex="-1"></a><span class="co">#&gt;   Cell_Type                    shapiro_test     p.value</span></span>
<span id="cb30-179"><a href="#cb30-179" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                        &lt;list&gt;             &lt;dbl&gt;</span></span>
<span id="cb30-180"><a href="#cb30-180" tabindex="-1"></a><span class="co">#&gt; 1 B_cell                       &lt;htest&gt;      0.00112    </span></span>
<span id="cb30-181"><a href="#cb30-181" tabindex="-1"></a><span class="co">#&gt; 2 Cancer_associated_fibroblast &lt;htest&gt;      0.0204     </span></span>
<span id="cb30-182"><a href="#cb30-182" tabindex="-1"></a><span class="co">#&gt; 3 Endothelial_cell             &lt;htest&gt;      0.138      </span></span>
<span id="cb30-183"><a href="#cb30-183" tabindex="-1"></a><span class="co">#&gt; 4 Macrophage                   &lt;htest&gt;      0.000139   </span></span>
<span id="cb30-184"><a href="#cb30-184" tabindex="-1"></a><span class="co">#&gt; 5 NK_cell                      &lt;htest&gt;      0.0000287  </span></span>
<span id="cb30-185"><a href="#cb30-185" tabindex="-1"></a><span class="co">#&gt; 6 T_cell_CD4+                  &lt;htest&gt;      0.0491     </span></span>
<span id="cb30-186"><a href="#cb30-186" tabindex="-1"></a><span class="co">#&gt; 7 T_cell_CD8+                  &lt;htest&gt;      0.000000248</span></span>
<span id="cb30-187"><a href="#cb30-187" tabindex="-1"></a><span class="co">#&gt; 8 uncharacterized_cell         &lt;htest&gt;      0.0177     </span></span>
<span id="cb30-188"><a href="#cb30-188" tabindex="-1"></a><span class="co">#&gt; [32mPerforming parametric tests for EPIC[0m</span></span>
<span id="cb30-189"><a href="#cb30-189" tabindex="-1"></a><span class="co">#&gt; # A tibble: 8 × 3</span></span>
<span id="cb30-190"><a href="#cb30-190" tabindex="-1"></a><span class="co">#&gt;   Cell_Type                    Test   p.value</span></span>
<span id="cb30-191"><a href="#cb30-191" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                        &lt;chr&gt;    &lt;dbl&gt;</span></span>
<span id="cb30-192"><a href="#cb30-192" tabindex="-1"></a><span class="co">#&gt; 1 B_cell                       t-test  0.0879</span></span>
<span id="cb30-193"><a href="#cb30-193" tabindex="-1"></a><span class="co">#&gt; 2 Cancer_associated_fibroblast t-test  0.998 </span></span>
<span id="cb30-194"><a href="#cb30-194" tabindex="-1"></a><span class="co">#&gt; 3 T_cell_CD4+                  t-test  0.726 </span></span>
<span id="cb30-195"><a href="#cb30-195" tabindex="-1"></a><span class="co">#&gt; 4 T_cell_CD8+                  t-test  0.300 </span></span>
<span id="cb30-196"><a href="#cb30-196" tabindex="-1"></a><span class="co">#&gt; 5 Endothelial_cell             t-test  0.493 </span></span>
<span id="cb30-197"><a href="#cb30-197" tabindex="-1"></a><span class="co">#&gt; 6 Macrophage                   t-test  0.489 </span></span>
<span id="cb30-198"><a href="#cb30-198" tabindex="-1"></a><span class="co">#&gt; 7 NK_cell                      t-test  0.627 </span></span>
<span id="cb30-199"><a href="#cb30-199" tabindex="-1"></a><span class="co">#&gt; 8 uncharacterized_cell         t-test  0.164 </span></span>
<span id="cb30-200"><a href="#cb30-200" tabindex="-1"></a><span class="co">#&gt; [32mPerforming Shapiro-Wilk test for xcell[0m</span></span>
<span id="cb30-201"><a href="#cb30-201" tabindex="-1"></a><span class="co">#&gt; # A tibble: 39 × 3</span></span>
<span id="cb30-202"><a href="#cb30-202" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                    shapiro_test     p.value</span></span>
<span id="cb30-203"><a href="#cb30-203" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                        &lt;list&gt;             &lt;dbl&gt;</span></span>
<span id="cb30-204"><a href="#cb30-204" tabindex="-1"></a><span class="co">#&gt;  1 B_cell                       &lt;htest&gt;      0.0703     </span></span>
<span id="cb30-205"><a href="#cb30-205" tabindex="-1"></a><span class="co">#&gt;  2 B_cell_memory                &lt;htest&gt;      0.0999     </span></span>
<span id="cb30-206"><a href="#cb30-206" tabindex="-1"></a><span class="co">#&gt;  3 B_cell_naive                 &lt;htest&gt;      0.00428    </span></span>
<span id="cb30-207"><a href="#cb30-207" tabindex="-1"></a><span class="co">#&gt;  4 B_cell_plasma                &lt;htest&gt;      0.0141     </span></span>
<span id="cb30-208"><a href="#cb30-208" tabindex="-1"></a><span class="co">#&gt;  5 Cancer_associated_fibroblast &lt;htest&gt;      0.00000117 </span></span>
<span id="cb30-209"><a href="#cb30-209" tabindex="-1"></a><span class="co">#&gt;  6 Class-switched_memory_B_cell &lt;htest&gt;      0.107      </span></span>
<span id="cb30-210"><a href="#cb30-210" tabindex="-1"></a><span class="co">#&gt;  7 Common_lymphoid_progenitor   &lt;htest&gt;      0.0184     </span></span>
<span id="cb30-211"><a href="#cb30-211" tabindex="-1"></a><span class="co">#&gt;  8 Common_myeloid_progenitor    &lt;htest&gt;      0.000000151</span></span>
<span id="cb30-212"><a href="#cb30-212" tabindex="-1"></a><span class="co">#&gt;  9 Endothelial_cell             &lt;htest&gt;      0.000189   </span></span>
<span id="cb30-213"><a href="#cb30-213" tabindex="-1"></a><span class="co">#&gt; 10 Eosinophil                   &lt;htest&gt;      0.0000242  </span></span>
<span id="cb30-214"><a href="#cb30-214" tabindex="-1"></a><span class="co">#&gt; # ℹ 29 more rows</span></span>
<span id="cb30-215"><a href="#cb30-215" tabindex="-1"></a><span class="co">#&gt; [32mPerforming parametric tests for xcell[0m</span></span>
<span id="cb30-216"><a href="#cb30-216" tabindex="-1"></a><span class="co">#&gt; # A tibble: 39 × 3</span></span>
<span id="cb30-217"><a href="#cb30-217" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                        Test   p.value</span></span>
<span id="cb30-218"><a href="#cb30-218" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                            &lt;chr&gt;    &lt;dbl&gt;</span></span>
<span id="cb30-219"><a href="#cb30-219" tabindex="-1"></a><span class="co">#&gt;  1 Myeloid_dendritic_cell_activated t-test 0.247  </span></span>
<span id="cb30-220"><a href="#cb30-220" tabindex="-1"></a><span class="co">#&gt;  2 B_cell                           t-test 0.0116 </span></span>
<span id="cb30-221"><a href="#cb30-221" tabindex="-1"></a><span class="co">#&gt;  3 T_cell_CD4+_memory               t-test 0.0826 </span></span>
<span id="cb30-222"><a href="#cb30-222" tabindex="-1"></a><span class="co">#&gt;  4 T_cell_CD4+_naive                t-test 0.142  </span></span>
<span id="cb30-223"><a href="#cb30-223" tabindex="-1"></a><span class="co">#&gt;  5 T_cell_CD4+_(non-regulatory)     t-test 0.391  </span></span>
<span id="cb30-224"><a href="#cb30-224" tabindex="-1"></a><span class="co">#&gt;  6 T_cell_CD4+_central_memory       t-test 0.656  </span></span>
<span id="cb30-225"><a href="#cb30-225" tabindex="-1"></a><span class="co">#&gt;  7 T_cell_CD4+_effector_memory      t-test 0.221  </span></span>
<span id="cb30-226"><a href="#cb30-226" tabindex="-1"></a><span class="co">#&gt;  8 T_cell_CD8+_naive                t-test 0.0160 </span></span>
<span id="cb30-227"><a href="#cb30-227" tabindex="-1"></a><span class="co">#&gt;  9 T_cell_CD8+                      t-test 0.00677</span></span>
<span id="cb30-228"><a href="#cb30-228" tabindex="-1"></a><span class="co">#&gt; 10 T_cell_CD8+_central_memory       t-test 0.00848</span></span>
<span id="cb30-229"><a href="#cb30-229" tabindex="-1"></a><span class="co">#&gt; # ℹ 29 more rows</span></span>
<span id="cb30-230"><a href="#cb30-230" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING HEATMAPS[0m</span></span>
<span id="cb30-231"><a href="#cb30-231" tabindex="-1"></a><span class="co">#&gt; [32mHeatmap of qti, EPIC and xcell will be stored on plots_TME_heatmap.pdf[0m</span></span>
<span id="cb30-232"><a href="#cb30-232" tabindex="-1"></a><span class="co">#&gt; [32mSkipping survival analysis.[0m</span></span></code></pre></div>
</div>
<div id="differentia-expression-analysis-htg_dea-for-rnaseq" class="section level4">
<h4>Differentia expression analysis: HTG_DEA() for RNAseq:</h4>
<p>If you want more control, such as applying extra filtering or
performing LFC shrinkage, <code>HTG_DEA()</code> will provide more
options.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="do">### DEA:</span></span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">HTG_DEA</span>(<span class="at">outliers=</span> <span class="cn">NULL</span>, </span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a>                   <span class="at">counts_data=</span> counts_data_RNAseq, </span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a>                   AnnotData_tutorial, </span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a>                   <span class="at">design_formula =</span> <span class="st">&quot;HPV_status&quot;</span>, </span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a>                   <span class="at">heatmap_columns =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>, <span class="st">&quot;Recurrence&quot;</span>),</span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a>                   <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;HPV_status&quot;</span>, <span class="st">&quot;Associated&quot;</span>, <span class="st">&quot;Independent&quot;</span>), </span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a>                   <span class="at">pattern =</span> <span class="cn">NULL</span>, </span>
<span id="cb31-9"><a href="#cb31-9" tabindex="-1"></a>                   <span class="at">remove_outliers =</span> <span class="cn">TRUE</span>,</span>
<span id="cb31-10"><a href="#cb31-10" tabindex="-1"></a>                   <span class="at">percentage_gene =</span> <span class="fl">0.2</span>, </span>
<span id="cb31-11"><a href="#cb31-11" tabindex="-1"></a>                   <span class="at">threshold_gene =</span> <span class="dv">200</span>,</span>
<span id="cb31-12"><a href="#cb31-12" tabindex="-1"></a>                   <span class="at">threshold_subject =</span> <span class="dv">10</span>,</span>
<span id="cb31-13"><a href="#cb31-13" tabindex="-1"></a>                   <span class="at">pCutoff =</span> <span class="fl">5e-2</span>, </span>
<span id="cb31-14"><a href="#cb31-14" tabindex="-1"></a>                   <span class="at">apply_filtering =</span> <span class="cn">TRUE</span>, </span>
<span id="cb31-15"><a href="#cb31-15" tabindex="-1"></a>                   <span class="at">apply_lfc_shrinkage =</span> <span class="cn">TRUE</span>, </span>
<span id="cb31-16"><a href="#cb31-16" tabindex="-1"></a>                   <span class="at">extract_shrinkage =</span> <span class="cn">FALSE</span>)</span>
<span id="cb31-17"><a href="#cb31-17" tabindex="-1"></a><span class="co">#&gt; [33mPREPARING DATA FOR DESeq2 ANALYSIS...[0m</span></span>
<span id="cb31-18"><a href="#cb31-18" tabindex="-1"></a><span class="co">#&gt; [33mCREATING DESeqDataSet OBJECT...[0m</span></span>
<span id="cb31-19"><a href="#cb31-19" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb31-20"><a href="#cb31-20" tabindex="-1"></a><span class="co">#&gt; [33mAPPLYING GENE FILTERING BASED ON THRESHOLDS...[0m</span></span>
<span id="cb31-21"><a href="#cb31-21" tabindex="-1"></a><span class="co">#&gt; [33mPERFORMING DESeq2 ANALYSIS...[0m</span></span>
<span id="cb31-22"><a href="#cb31-22" tabindex="-1"></a><span class="co">#&gt; -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the</span></span>
<span id="cb31-23"><a href="#cb31-23" tabindex="-1"></a><span class="co">#&gt;    function: y = a/x + b, and a local regression fit was automatically substituted.</span></span>
<span id="cb31-24"><a href="#cb31-24" tabindex="-1"></a><span class="co">#&gt;    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.</span></span>
<span id="cb31-25"><a href="#cb31-25" tabindex="-1"></a><span class="co">#&gt; estimating size factors</span></span>
<span id="cb31-26"><a href="#cb31-26" tabindex="-1"></a><span class="co">#&gt; estimating dispersions</span></span>
<span id="cb31-27"><a href="#cb31-27" tabindex="-1"></a><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span id="cb31-28"><a href="#cb31-28" tabindex="-1"></a><span class="co">#&gt; mean-dispersion relationship</span></span>
<span id="cb31-29"><a href="#cb31-29" tabindex="-1"></a><span class="co">#&gt; -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the</span></span>
<span id="cb31-30"><a href="#cb31-30" tabindex="-1"></a><span class="co">#&gt;    function: y = a/x + b, and a local regression fit was automatically substituted.</span></span>
<span id="cb31-31"><a href="#cb31-31" tabindex="-1"></a><span class="co">#&gt;    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.</span></span>
<span id="cb31-32"><a href="#cb31-32" tabindex="-1"></a><span class="co">#&gt; final dispersion estimates</span></span>
<span id="cb31-33"><a href="#cb31-33" tabindex="-1"></a><span class="co">#&gt; fitting model and testing</span></span>
<span id="cb31-34"><a href="#cb31-34" tabindex="-1"></a><span class="co">#&gt; -- replacing outliers and refitting for 2174 genes</span></span>
<span id="cb31-35"><a href="#cb31-35" tabindex="-1"></a><span class="co">#&gt; -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 </span></span>
<span id="cb31-36"><a href="#cb31-36" tabindex="-1"></a><span class="co">#&gt; -- original counts are preserved in counts(dds)</span></span>
<span id="cb31-37"><a href="#cb31-37" tabindex="-1"></a><span class="co">#&gt; estimating dispersions</span></span>
<span id="cb31-38"><a href="#cb31-38" tabindex="-1"></a><span class="co">#&gt; fitting model and testing</span></span>
<span id="cb31-39"><a href="#cb31-39" tabindex="-1"></a><span class="co">#&gt; [32mResults saved as &#39;results_HTG_DEA.csv&#39;[0m</span></span>
<span id="cb31-40"><a href="#cb31-40" tabindex="-1"></a><span class="co">#&gt; [32mDisplaying top 10 differential expression results...[0m</span></span>
<span id="cb31-41"><a href="#cb31-41" tabindex="-1"></a><span class="co">#&gt; log2 fold change (MLE): HPV_status Associated vs Independent </span></span>
<span id="cb31-42"><a href="#cb31-42" tabindex="-1"></a><span class="co">#&gt; Wald test p-value: HPV_status Associated vs Independent </span></span>
<span id="cb31-43"><a href="#cb31-43" tabindex="-1"></a><span class="co">#&gt; DataFrame with 10 rows and 6 columns</span></span>
<span id="cb31-44"><a href="#cb31-44" tabindex="-1"></a><span class="co">#&gt;           baseMean log2FoldChange     lfcSE      stat      pvalue        padj</span></span>
<span id="cb31-45"><a href="#cb31-45" tabindex="-1"></a><span class="co">#&gt;          &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span id="cb31-46"><a href="#cb31-46" tabindex="-1"></a><span class="co">#&gt; PPP2R2C    450.322      -12.18328  1.372048  -8.87963 6.70822e-19 6.85848e-15</span></span>
<span id="cb31-47"><a href="#cb31-47" tabindex="-1"></a><span class="co">#&gt; PDZK1IP1   680.205      -10.43474  1.293814  -8.06510 7.31736e-16 3.74064e-12</span></span>
<span id="cb31-48"><a href="#cb31-48" tabindex="-1"></a><span class="co">#&gt; MMP10      898.934       -9.69845  1.246862  -7.77829 7.35138e-15 2.50535e-11</span></span>
<span id="cb31-49"><a href="#cb31-49" tabindex="-1"></a><span class="co">#&gt; LYPD3     1101.958       -6.86637  0.927191  -7.40556 1.30596e-13 3.33803e-10</span></span>
<span id="cb31-50"><a href="#cb31-50" tabindex="-1"></a><span class="co">#&gt; IRF6       564.565       -7.24378  1.075176  -6.73730 1.61356e-11 3.29941e-08</span></span>
<span id="cb31-51"><a href="#cb31-51" tabindex="-1"></a><span class="co">#&gt; IL20RB     693.154       -5.60484  0.837154  -6.69512 2.15493e-11 3.67200e-08</span></span>
<span id="cb31-52"><a href="#cb31-52" tabindex="-1"></a><span class="co">#&gt; IGSF3      473.223       -5.71455  0.880058  -6.49337 8.39346e-11 1.22593e-07</span></span>
<span id="cb31-53"><a href="#cb31-53" tabindex="-1"></a><span class="co">#&gt; KRT80      277.950       -7.75546  1.221085  -6.35128 2.13527e-10 2.53673e-07</span></span>
<span id="cb31-54"><a href="#cb31-54" tabindex="-1"></a><span class="co">#&gt; SULT2B1    589.880       -7.01528  1.105745  -6.34439 2.23304e-10 2.53673e-07</span></span>
<span id="cb31-55"><a href="#cb31-55" tabindex="-1"></a><span class="co">#&gt; MMP13      309.630       -6.16267  0.982895  -6.26992 3.61236e-10 3.69328e-07</span></span>
<span id="cb31-56"><a href="#cb31-56" tabindex="-1"></a><span class="co">#&gt; [33mAPPLYING LOG-FOLD CHANGE SHRINKAGE...[0m</span></span>
<span id="cb31-57"><a href="#cb31-57" tabindex="-1"></a><span class="co">#&gt; using &#39;apeglm&#39; for LFC shrinkage. If used in published research, please cite:</span></span>
<span id="cb31-58"><a href="#cb31-58" tabindex="-1"></a><span class="co">#&gt;     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for</span></span>
<span id="cb31-59"><a href="#cb31-59" tabindex="-1"></a><span class="co">#&gt;     sequence count data: removing the noise and preserving large differences.</span></span>
<span id="cb31-60"><a href="#cb31-60" tabindex="-1"></a><span class="co">#&gt;     Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</span></span>
<span id="cb31-61"><a href="#cb31-61" tabindex="-1"></a><span class="co">#&gt; log2 fold change (MAP): HPV status Independent vs Associated </span></span>
<span id="cb31-62"><a href="#cb31-62" tabindex="-1"></a><span class="co">#&gt; Wald test p-value: HPV status Independent vs Associated </span></span>
<span id="cb31-63"><a href="#cb31-63" tabindex="-1"></a><span class="co">#&gt; DataFrame with 10 rows and 5 columns</span></span>
<span id="cb31-64"><a href="#cb31-64" tabindex="-1"></a><span class="co">#&gt;           baseMean log2FoldChange     lfcSE      pvalue        padj</span></span>
<span id="cb31-65"><a href="#cb31-65" tabindex="-1"></a><span class="co">#&gt;          &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span id="cb31-66"><a href="#cb31-66" tabindex="-1"></a><span class="co">#&gt; PPP2R2C    450.322       12.07215  1.619226 6.70822e-19 6.85848e-15</span></span>
<span id="cb31-67"><a href="#cb31-67" tabindex="-1"></a><span class="co">#&gt; PDZK1IP1   680.205       15.69545  1.298237 7.31736e-16 3.74064e-12</span></span>
<span id="cb31-68"><a href="#cb31-68" tabindex="-1"></a><span class="co">#&gt; MMP10      898.934        9.36887  1.295663 7.35138e-15 2.50535e-11</span></span>
<span id="cb31-69"><a href="#cb31-69" tabindex="-1"></a><span class="co">#&gt; LYPD3     1101.958        6.60919  0.969968 1.30596e-13 3.33803e-10</span></span>
<span id="cb31-70"><a href="#cb31-70" tabindex="-1"></a><span class="co">#&gt; IRF6       564.565        6.91018  1.136743 1.61356e-11 3.29941e-08</span></span>
<span id="cb31-71"><a href="#cb31-71" tabindex="-1"></a><span class="co">#&gt; IL20RB     693.154       11.87084  0.848452 2.15493e-11 3.67200e-08</span></span>
<span id="cb31-72"><a href="#cb31-72" tabindex="-1"></a><span class="co">#&gt; IGSF3      473.223       10.90270  0.892990 8.39346e-11 1.22593e-07</span></span>
<span id="cb31-73"><a href="#cb31-73" tabindex="-1"></a><span class="co">#&gt; KRT80      277.950        7.34640  1.301983 2.13527e-10 2.53673e-07</span></span>
<span id="cb31-74"><a href="#cb31-74" tabindex="-1"></a><span class="co">#&gt; SULT2B1    589.880        6.64769  1.180208 2.23304e-10 2.53673e-07</span></span>
<span id="cb31-75"><a href="#cb31-75" tabindex="-1"></a><span class="co">#&gt; MMP13      309.630        5.82998  1.048179 3.61236e-10 3.69328e-07</span></span>
<span id="cb31-76"><a href="#cb31-76" tabindex="-1"></a><span class="co">#&gt; [32mSaving plots as &#39;DEA_plots.pdf&#39;[0m</span></span></code></pre></div>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAB7FBMVEUAAAAAAA0AACgADQ0ADSgADVEAKIENAAANAA0NACgNDQANDSgNDVENKCgNKFENKIENUYENUbwoAAAoAA0oDQAoDQ0oDSgoDVEoKA0oUYEoUbwogYEogbwogf9FdbRJebZOfbhRDQBRDQ1RDShRKA1RgYFRgbxRvP9SgrtXhr1ci79gj8JllMRpmMZunclzoct3ps58qtCAr9KBKACBKA2BUQ2BUSiBgVGBvFGBvLyBvP+B/7yB//+Fs9WKuNeOvNmTwNuYw92ZmZmcxt+hyuGmzeKr0OSw0+a01ui52em8UQ28USi8gSi8gVG8gYG8vIG8vLy8/4G8//++3OvD4O3I4+/M5vDR6fLW7PTXMCfZNSrbOy3b7/bdQDDfRjPg8/fh8/TiTDbj9PHkUTnl9e3mVzzn9eroXT/p9ubrYkLr9+PtaEXt+N/vbUjv+Nzw+djxc0vy+tX0eU70+9L2flH2+874hFT4/Mv6ilf6/cf8j1r8lF38mWH8nmT8o2f8qGv8rW78snH8/cT9t3X9vHj9wXv9xn/9y4L90IX91Yn92oz+34/+4ZL+45X+5Zj+55v+6Z7+66H+7aP+76b+8Kn+8qz+9K/+9rL++LX++rf+/Lr+/r3+/sD/gSj/vFH/vIH/vLz//4H//7z///8m5EZJAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAKZUlEQVR4nO2di78bRRXH5xbujYAW77WiYiO9il4FxEpalVoVV3wFUIwIXHyg4tuqVev7AYIFoSj4wmeiSf5RZnf2cc7ZZGfnZnazm/39Pv00zWTmnNlvzs7MTs9u1BwqlFp3B5ouALIIgCwCIIscAN3TU+q6cyUqjtWx87nC+27Il61u9LVKbR0/QvOwNwst5lQe0FCFKmNzkeclvVnN6JmotdrquzaP3nkGNN1T14/+t6d2XTuzvGxVo5ruyfn83p7aHjk2L8cmkgsgfRiDsDOz8Js7PppPeltvPaGuPpjov86FFd74ZqUP2Lifvaentm5MWg/Cb3pHh/YJpa5+mx+js30DdhidY4llXfrqfXXsTvNyLmlhmsc+TG8MpaSd9nTyZqVeKc/X0oC0Y3XdndnR6oOa9KJ/XNWL3mkPymCIPJtKSWgMkk8i9b0Y1SgP0u6llkOT+qR9v3k5n7QgzbdHBFDaLvEko7H8GBT1fOtk3LEz2rou2RnpOA//OnZee9g+NxuorYNx9JmuNOmlkRz1RlfZGekqWXyvYnSsMkCZZQ1IF98fv6Qt0uZn0uFnbOzH7RJP8txzmMVmZ3rmPJn//72vV+ZY+qabsa9+1NN++M6MvlmwxF0KD8lUXN0oBZRZ1mRCa/FL2sKcT7GPDFDWznjKD05u66BPn9AGZzfHU0/89aTHEvvaZccyMecKmTfCKnHpUY1G7egpllmOR6b4hQFKfVBASbuVAZmmYac09TfcP1xwLPzLjg+fA0q+L1p6BKORpvHsN9ajGIsgBihuEYeL8VFJBOlDun40O6sNDsPJYT8aT8Wx6PKz6XChbpwPlWUMWs2omebv67GxhANKWxi+sY8lY9CKp9gwm0+WnA0Fs1jUaCex0fdklCwUU8sc0JzOYqkP05vIYtrOwxh0z7V69g0XIfry4PhnovNEHIteSJA1B10HRUuQnZFcB61oNDIXX2qQdRAFlLSImic+TG/ulesgD4N0kdjk1GijTgIgiwDIIuwHWQRAFq0HkKpc/rrqzVKjvAJQbQ4AqDZLjfIKQLU5AKDaLDXKKwDV5gCAarPUKK8AVJsDAKrNUqO8AlBtDgCoNkuN8gpAtTkAoNosNcorANXmAIBqs9QorwBUmwMAqs1So7wCUG0OAKg2S43yCkALrVaRyLBRgKpwBkBHM1qxAGgtXgHoaFYByGIVgCxWAchiFYAsVgHIYrWhgNaQ4twyQCWqeOa2eYDqsQdAlioAtKSKn7FtkwGt4G3lZl4MA5BP3wDk3R4AWaoAkKUKAFmqdAGQ22VHOwA5XEOVALRKcVMBlXcAQBYHAGRxAEAWBwBkceAMyDIBAJDFNgABkKUGAFlqdAGQz4uHzQTk1A0AcivuIKCVVjadAFRf7TJHOrkm96ByACKa7uWf5A5AmYbqVVVGUNAWme4eUpmiT40qPcUC5vIweIYp+A9X8DOm4N9MwR+5gn8yybcPMAW3cwWnmZYDqngMahmgR6gACICcAX2eCoAAyBnQF6kAqCSgBeosoC9RVQEod5kJQJaGLQP0FSoAAiBnQI9RARAAOQP6OhUAAZAzoG9S+QFk+Z+sde+DlVZ1gIobypARAfU8l4gR8XHwd67gx0zBd5iCjzMFt3EFp5hiQN+mAiAAcgb0XSoAAiBnQN+jAiAAcgZ0kQqA1gEot2HWLkA/oqoCUE4AtFmALlEBEAA5A/opFQABkDOgn1MBEAA5A/olVT2A2iLT3V9RmaLZfvpj6pOe2jrwDUhsOouYOBQRxabZSwFb+V8M/sUljb3EFNzNlNuTXhhBCwANdsI/oaZ7/fnQwOosoF9TGSqvOUjyX8KX8C0AUUCEymy/Px9vjzoN6DdUElA4HBk+ALQQ0HRvt/Nj0G+pJCCMQYsAVTZIJxtm7QL0OJUpyqb52b7PUywubz+g6V60UBzsel4othPQE1RHOGoA6jqg31EBEAA5A3qKynLUy272L3fzfxcAFXyeB7EhgH5PVQ+gtmhtgERapthTvSRCRgQUS1t+LHiSS0aQSOL8IFNwC1fwJqYY0GUqAAIgZ0BPU3kCVDy7AZDlIT8tA8T+jwqAAMgZ0B+oKgcUbZgBUAGgeesi6FkqAAIgZ0DPUQEQADkDukIFQADkDIgZBKD1AjLXq+veByst02v29VULKLYgb1wWX/pFsakqQkYE1D+4ZP7jX5mCu5jKRRAAWQD9iQqASgEiSZz6n2q324BeoDJFWXbHfLA9mrwiTX/pJKAXqaISkh+UZOIBEAUkMsy6DujPVBLQePusUn2fgMINs00CpEfo8ZbHMShUywD9hSoHaHsU5UoDEAUkkji7DohZMEV0mtenWJbECUCmiCRxhgtFn4N0CwH9jaroqMsDstQDoM0CxO5oBKBGAGqLTHdZEl89gMQeq7gtV2Ydij1VETGHIqLYFvuVgGU/XbY9JvAOJgAqB4j5AyAAcgbE/l8EgADIGRDrDQABkDMgtqqtAFA+GRiAOKDcpy0DxO68ASAAcgb0XyoAAiBnQMw9AAGQM6BfUNUDqC1aGyCROCgeXPe82IIVb8We6hURMiKg2IM3Hpc39cotVxFQprssoRKAAMgZ0E+ofAEqfMoAAOXatRoQ6x0AAZAzoB9SmSKS5Zpl4QFQBoikv+g3xzwA4htm7QL0A6qohCRQzefjq3xEEGvYekDsSZxveV/XAX2fKgdouOtlDGozoAtUEtD0dSMAKgI06PuZxdoMiD30PSrJBunpXjTtuGa5bjogPs13PoK+RWWKSJarB0C5K9Z174OV1lJASyPhKIDSxhkgEUHipx34g+tekrVF1uFlsakqQkYE1CeZgndyyXt+TXe/QQVAAOQM6GtUAARAzoC+SgVAAOQM6MtUALQuQEUbZg0H9ChVVYCYAGizAH2BCoAAyBnQw1QABEDOgB6iAqBGAGqLTHcfpKoHkMgUFD9+erfYFxVfq7gv9wFhTDQWISMC6t1cwYeYAKgcoM9RARAAOQNi9gEIgJwBfZYKgADIGRCzsAKgsr89plRXARV8xtUyQJ+gAqBSgEiW66RX9hFdGwuI9c4UZekv071+2ecodggQyXJ1eI7ixgL6GFVUQpM45yaKAKgA0FBHUZcBfZRqAaBhyUz7rgIaxj8W7hNQW2S6y4hHJfRWhGF6V4tHQCJX8jam3K3IItGyMO8yV1tsOouIORQRxVKiL8SAPkJlirJpPn5cOwBxQFmW66B8nvTGAvowlQVA4TUoABVrYwGxWROAAMgZEFsb+QNUtGEGQMXVWwaILeQBCICcAX2ACoAAyBkQeyoVAAGQMyDWBIAqBFT62nXd+2ClZbp7mmoFQOWrB8zl6eAUk3h7Sv7YhXjLn+x3h4xOsV8r9lQviJARAQVApQC9iwqAAMgZEDMIQADkDIgtFACoPkAFG2btAsSWsf4AFQiANgvQrVQABEDOgNjVLwABkDOgd1CZIvYsVwDKA2IPees8oLdTRSXsWa5VAGqLlgESSZzeAbVMLJ6jEgBiAiCLbqKKSgCIKQ+o6kG6ZWKzpimqeJpvv+IsV66XAWSanjMnXtBwAAAAAElFTkSuQmCC" /><!-- --><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAB7FBMVEUAAAAAAA0AACgADQ0ADSgADVEAKIENAAANAA0NACgNDQANDSgNDVENKCgNKFENKIENUYENUbwoAAAoAA0oDQAoDQ0oDSgoDVEoKA0oUYEoUbwogYEogbwogf9FdbRJebZOfbhRDQBRDQ1RDShRKA1RgYFRgbxRvP9SgrtXhr1ci79gj8JllMRpmMZunclzoct3ps58qtCAr9KBKACBKA2BUQ2BUSiBgVGBvFGBvLyBvP+B/7yB//+Fs9WKuNeOvNmTwNuYw92ZmZmcxt+hyuGmzeKr0OSw0+a01ui52em8UQ28USi8gSi8gVG8gYG8vIG8vLy8/4G8//++3OvD4O3I4+/M5vDR6fLW7PTXMCfZNSrbOy3b7/bdQDDfRjPg8/fh8/TiTDbj9PHkUTnl9e3mVzzn9eroXT/p9ubrYkLr9+PtaEXt+N/vbUjv+Nzw+djxc0vy+tX0eU70+9L2flH2+874hFT4/Mv6ilf6/cf8j1r8lF38mWH8nmT8o2f8qGv8rW78snH8/cT9t3X9vHj9wXv9xn/9y4L90IX91Yn92oz+34/+4ZL+45X+5Zj+55v+6Z7+66H+7aP+76b+8Kn+8qz+9K/+9rL++LX++rf+/Lr+/r3+/sD/gSj/vFH/vIH/vLz//4H//7z///8m5EZJAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAKZUlEQVR4nO2di78bRRXH5xbujYAW77WiYiO9il4FxEpalVoVV3wFUIwIXHyg4tuqVev7AYIFoSj4wmeiSf5RZnf2cc7ZZGfnZnazm/39Pv00zWTmnNlvzs7MTs9u1BwqlFp3B5ouALIIgCwCIIscAN3TU+q6cyUqjtWx87nC+27Il61u9LVKbR0/QvOwNwst5lQe0FCFKmNzkeclvVnN6JmotdrquzaP3nkGNN1T14/+t6d2XTuzvGxVo5ruyfn83p7aHjk2L8cmkgsgfRiDsDOz8Js7PppPeltvPaGuPpjov86FFd74ZqUP2Lifvaentm5MWg/Cb3pHh/YJpa5+mx+js30DdhidY4llXfrqfXXsTvNyLmlhmsc+TG8MpaSd9nTyZqVeKc/X0oC0Y3XdndnR6oOa9KJ/XNWL3mkPymCIPJtKSWgMkk8i9b0Y1SgP0u6llkOT+qR9v3k5n7QgzbdHBFDaLvEko7H8GBT1fOtk3LEz2rou2RnpOA//OnZee9g+NxuorYNx9JmuNOmlkRz1RlfZGekqWXyvYnSsMkCZZQ1IF98fv6Qt0uZn0uFnbOzH7RJP8txzmMVmZ3rmPJn//72vV+ZY+qabsa9+1NN++M6MvlmwxF0KD8lUXN0oBZRZ1mRCa/FL2sKcT7GPDFDWznjKD05u66BPn9AGZzfHU0/89aTHEvvaZccyMecKmTfCKnHpUY1G7egpllmOR6b4hQFKfVBASbuVAZmmYac09TfcP1xwLPzLjg+fA0q+L1p6BKORpvHsN9ajGIsgBihuEYeL8VFJBOlDun40O6sNDsPJYT8aT8Wx6PKz6XChbpwPlWUMWs2omebv67GxhANKWxi+sY8lY9CKp9gwm0+WnA0Fs1jUaCex0fdklCwUU8sc0JzOYqkP05vIYtrOwxh0z7V69g0XIfry4PhnovNEHIteSJA1B10HRUuQnZFcB61oNDIXX2qQdRAFlLSImic+TG/ulesgD4N0kdjk1GijTgIgiwDIIuwHWQRAFq0HkKpc/rrqzVKjvAJQbQ4AqDZLjfIKQLU5AKDaLDXKKwDV5gCAarPUKK8AVJsDAKrNUqO8AlBtDgCoNkuN8gpAtTkAoNosNcorANXmAIBqs9QorwBUmwMAqs1So7wCUG0OAKg2S43yCkALrVaRyLBRgKpwBkBHM1qxAGgtXgHoaFYByGIVgCxWAchiFYAsVgHIYrWhgNaQ4twyQCWqeOa2eYDqsQdAlioAtKSKn7FtkwGt4G3lZl4MA5BP3wDk3R4AWaoAkKUKAFmqdAGQ22VHOwA5XEOVALRKcVMBlXcAQBYHAGRxAEAWBwBkceAMyDIBAJDFNgABkKUGAFlqdAGQz4uHzQTk1A0AcivuIKCVVjadAFRf7TJHOrkm96ByACKa7uWf5A5AmYbqVVVGUNAWme4eUpmiT40qPcUC5vIweIYp+A9X8DOm4N9MwR+5gn8yybcPMAW3cwWnmZYDqngMahmgR6gACICcAX2eCoAAyBnQF6kAqCSgBeosoC9RVQEod5kJQJaGLQP0FSoAAiBnQI9RARAAOQP6OhUAAZAzoG9S+QFk+Z+sde+DlVZ1gIobypARAfU8l4gR8XHwd67gx0zBd5iCjzMFt3EFp5hiQN+mAiAAcgb0XSoAAiBnQN+jAiAAcgZ0kQqA1gEot2HWLkA/oqoCUE4AtFmALlEBEAA5A/opFQABkDOgn1MBEAA5A/olVT2A2iLT3V9RmaLZfvpj6pOe2jrwDUhsOouYOBQRxabZSwFb+V8M/sUljb3EFNzNlNuTXhhBCwANdsI/oaZ7/fnQwOosoF9TGSqvOUjyX8KX8C0AUUCEymy/Px9vjzoN6DdUElA4HBk+ALQQ0HRvt/Nj0G+pJCCMQYsAVTZIJxtm7QL0OJUpyqb52b7PUywubz+g6V60UBzsel4othPQE1RHOGoA6jqg31EBEAA5A3qKynLUy272L3fzfxcAFXyeB7EhgH5PVQ+gtmhtgERapthTvSRCRgQUS1t+LHiSS0aQSOL8IFNwC1fwJqYY0GUqAAIgZ0BPU3kCVDy7AZDlIT8tA8T+jwqAAMgZ0B+oKgcUbZgBUAGgeesi6FkqAAIgZ0DPUQEQADkDukIFQADkDIgZBKD1AjLXq+veByst02v29VULKLYgb1wWX/pFsakqQkYE1D+4ZP7jX5mCu5jKRRAAWQD9iQqASgEiSZz6n2q324BeoDJFWXbHfLA9mrwiTX/pJKAXqaISkh+UZOIBEAUkMsy6DujPVBLQePusUn2fgMINs00CpEfo8ZbHMShUywD9hSoHaHsU5UoDEAUkkji7DohZMEV0mtenWJbECUCmiCRxhgtFn4N0CwH9jaroqMsDstQDoM0CxO5oBKBGAGqLTHdZEl89gMQeq7gtV2Ydij1VETGHIqLYFvuVgGU/XbY9JvAOJgAqB4j5AyAAcgbE/l8EgADIGRDrDQABkDMgtqqtAFA+GRiAOKDcpy0DxO68ASAAcgb0XyoAAiBnQMw9AAGQM6BfUNUDqC1aGyCROCgeXPe82IIVb8We6hURMiKg2IM3Hpc39cotVxFQprssoRKAAMgZ0E+ofAEqfMoAAOXatRoQ6x0AAZAzoB9SmSKS5Zpl4QFQBoikv+g3xzwA4htm7QL0A6qohCRQzefjq3xEEGvYekDsSZxveV/XAX2fKgdouOtlDGozoAtUEtD0dSMAKgI06PuZxdoMiD30PSrJBunpXjTtuGa5bjogPs13PoK+RWWKSJarB0C5K9Z174OV1lJASyPhKIDSxhkgEUHipx34g+tekrVF1uFlsakqQkYE1CeZgndyyXt+TXe/QQVAAOQM6GtUAARAzoC+SgVAAOQM6MtUALQuQEUbZg0H9ChVVYCYAGizAH2BCoAAyBnQw1QABEDOgB6iAqBGAGqLTHcfpKoHkMgUFD9+erfYFxVfq7gv9wFhTDQWISMC6t1cwYeYAKgcoM9RARAAOQNi9gEIgJwBfZYKgADIGRCzsAKgsr89plRXARV8xtUyQJ+gAqBSgEiW66RX9hFdGwuI9c4UZekv071+2ecodggQyXJ1eI7ixgL6GFVUQpM45yaKAKgA0FBHUZcBfZRqAaBhyUz7rgIaxj8W7hNQW2S6y4hHJfRWhGF6V4tHQCJX8jam3K3IItGyMO8yV1tsOouIORQRxVKiL8SAPkJlirJpPn5cOwBxQFmW66B8nvTGAvowlQVA4TUoABVrYwGxWROAAMgZEFsb+QNUtGEGQMXVWwaILeQBCICcAX2ACoAAyBkQeyoVAAGQMyDWBIAqBFT62nXd+2ClZbp7mmoFQOWrB8zl6eAUk3h7Sv7YhXjLn+x3h4xOsV8r9lQviJARAQVApQC9iwqAAMgZEDMIQADkDIgtFACoPkAFG2btAsSWsf4AFQiANgvQrVQABEDOgNjVLwABkDOgd1CZIvYsVwDKA2IPees8oLdTRSXsWa5VAGqLlgESSZzeAbVMLJ6jEgBiAiCLbqKKSgCIKQ+o6kG6ZWKzpimqeJpvv+IsV66XAWSanjMnXtBwAAAAAElFTkSuQmCC" /><!-- --><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAAyVBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZpAAZrY6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZmY6ZpA6ZrY6kLY6kNtHk69mAABmADpmOgBmOjpmZmZmkLZmkNtmtttmtv+QOgCQOjqQZgCQZjqQkGaQkLaQtpCQttuQtv+Q29uQ2/+2ZgC2Zjq2ZpC2kDq2kGa227a229u22/+2/7a2/9u2///bkDrbkGbbtmbbtpDb27bb29vb2//b/7bb////tmb/25D/27b//7b//9v///+hNF+ZAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAMV0lEQVR4nO2dC3vcOBWGnW7CzG5YYJmGwm4b2IVCXW6BeLcFZtKM//+PQhfL44v0HUkzsuX6fM/TPGPL50h+rcuxJbtFzYIq5i5A7mJAhBgQIQZEiAERYkCEGBAhBkSIARFiQIQYECEGRIgBEWJAhBgQIQZEiAERYkCEGBAhBkSIARFiQIQYECEGRIgBEWJAhBgQIQZEiAERYkCEGBAhBkToHEDH++Lqnfr1tC1u9vXx77dFcfWrB7mnLFpt2uP/+hjk/v22KL6JLFtZvGgyq3QpmnL1MvApz1k1SOS9O/0QvHRR3tRWQJ9evQgCdJDGu8iijQA15erIrzxnAXq+kxVHVSWRV1sSma8F0KnMfqpM/YyRBdDQnV95zuuDSp2paGEbtfFaXBdRj5pLJbhtuscqduK0//SqePFQf3gltr9+p07h6i/vi+Jalf9H00xLc04fflsUX3y71wdK23/cFZsPtzK3H7eN1VHYX32z1z/FzrddQOrXx1f6UrXZNuXplCMBoINuA5Vi0pRJ0NpBQFI3+8PpqlbF6LfwZAA1e2RVbWz/d9cY3Jr6+qz3XD/WbUMfAmqq+ynbpjydciQAJDIVuesWJot/9e0/O6l9QA1AcdTNQ/0fkfazfQNT7Lp+0P2YcLjZ1z8Vercste7/q2aPspU4dvKg4nWTUMoqqZ2JE97sZYUaAlI/utmWTdnbHQkAycvwxrhvrt312z0ByHSWH+WotzG7VDsVf64NYw2oqZdl08tJW10X9F+Vt/i5MyhKU6UsgHQ1Mdm2rdDsSAFIXi/TE9X1T7dtTe8CUg3gZm8A6YOP37dduN4lK08Due1x3skdN+1GY6uObP4qQOKPaSgmz1EfZPydstWHdHakACSv1L+asUyd9scftm1mCJAkcf3Hj3cDQGLs1Wf652a33quuxBsnoMNpnDLHOwB1szVNrN2RApBsY7/TPXXbjtuyIUD66OcRIIHoh1vdJwfVINNszfFjQKqn7GarDunuSAJI1W5VApHJ1XeibB+2Q0Atyw6gg+l3dkNA0vAPuscZ90F2QN3zc/VBMvzY9LJVR3Z3JAGkOo1NUxKjQROrTUn1OG5q0M1eF7oLSJT313vZzkwNGoxidkA6BPtkxnExiomKbQ0UR9l2dyQBpPJ/c2LVjUAGgA4qqdMHWTrp+v0pLGmO7MVBDkBNHKRSy0Eperca3WxVef7d2ZEG0NP2BESNYnoIsgCSwcn1Q3cUu35bnVpO006Uj68f6tN4pyLp7+oaAKo/fS+aur4dPf5tW3zx2nWz2slWl6ezIw2gz10MiBADIsSACDEgQgyIEAMixIAIMSBCDIgQAyLEgAgxIEIMiBADIsSACDEgQgyIEAMixIAIMSBCDIgQAyLEgAgxIEIMiBADIsSACDEgQgyIEAMidGFAxWI0F6DLuksnBkSIARHKDlBgy0+u7ABdPqvzxIAIMSBCDIhQjoCyEgMixIAI5QgoK3gpAKk3ZdRrVQfnW8JrBnS8V+/8yfeIxI+DnVC2gIaRfAJAT1v5WlglX62W1ai0vmCVLaBh/sn6IFFzWlKB7tYBqHzx+PSleum0Cyj8OdQMmgTQQb6Erd/KtXZCawd0MH30EgH1lAbQQb8gP25iXu6ygpcEUKWjoKV20j2lAFS1r3/zMG/T09Z8nGCJgeIw/wSAzPvn+itIC7zVmCYOmtbdRcWACDGgEOUIKCt4DIgQAyLyTwTo6avmW1OF44M7Kwf0fKc/Jfll1HcrP39Ah0J/hsh+nxrsbmolByTu5TWayvU5q5UDqk3dKX9eDL5jvIgnij2lBPR8Jz+4Vlq/q5dzH9RT0ho0/OnpbmWA9GOzEHdzA5quD1Kyj/UM6NHUneU1sQlHMTnMk5309KMakeOETUx+DdTWAw3dTV5nYIYZPg/ydB4QQvUPHRkWo0P9Copy9D0wxp0/LtAwEAQiC5DjkgGNLr2/06UBik2D5xnQisGhDIg4dGmA4JEXKuN0TxS91yjGnlk0kbmH+eaJovfUs3uQTRRFzgyoeaLov3ghutdByncUM08U/Ze/RI/Pl3E6wzCvAXkvoLoQrqUBsizB83rkOgEg/xwnBuTlbg1xUGATi81quXHQWZ20vxYbB2lAKYb5CykPQBcIFIOODDDMYxTzX6O4vjgo0N364qCz3K0hDjrL3RrioLPcrSEOCnRntkI/bLTcOCjQXQvoZauij2vCCjwhIO81ig5AL5288NbAqSXR0TbpI92KAeS9RtEPkP9W70QHbmxnY88/7LxjAHmsURxUBNTEAgD1kIRUxIkBeaxR9K8IFwPkvZUe0HiNIgPqyrJGcdwHep/nZwhICz8wY0DEGsVLAXI1xiUAgmsUQwCh4QcBiu3KkgPyWaMY0klHbeUMyGeNIlnAjolfnVkSII81iqjwQxPvMxsAeul56Ex9EHaXBlAPScaRtIc7GtDpxzoBgfMcZJwGEJV/ZoB6hR9kDDsP11kvIA7C7kIAWRIdzvuA/OvMsgGRW36PyLICRK9RDIiDLMVwAIKGOQHymHpeNSCfxQv0MI+KAQE5tjIa5sOXv8BhntwKONTVeU0MKHiNIg0oQF4Njsw/JSD/NYoMiHC3UkDhaxRHBRxVt/NHMaRsO2lQQGi3cEDh31FcGaDwNYpTxEHIcPJbjdA1iqsD5OvOH1B0FjCNARFp9hsf0ot3dhFiQFHuZh/FGBBRqlz7oHbvygCduUbR384jMcth/sw1imOjwdNqy5b9to24pZsNkPd3FLMZxSYG5P0dxbUC8lijWPdaim3aJ07QibOlTjzM+6xRzF7ph/nwl3qzUnpA4d9RzEqJAFWnzif8O4pZKWUNgmsUF6OEgNAaReT87ID4ooa+irJyr1FEzlcEKM45AyKcMyDCOQMinDMgwjkDIpwzIMI5A/osxYAIMSBCDIgQAyLEgAgxIEIMiBADIsSACGUKyPrC1fPdTicVlufherbOpeO9fhLt8xy0r/wAlUWxe/rFXpzx8GwUoEpyGKeJXb/cOh+TqykYubK7dC8rcCgRIL1CRmtUJphYCgClqj3VsEpIQA2aUZpKrOxrcpqV3Wqewb68GyhVDTreg2sFEhUAx8sOkkEzXzmedGrbn4VAhypYmWJXsiZ2vAfXyp2oz/P43zqmBimZ/+y0l52uQRvXGxRA6fqgA+oR3YntGbRn3Oq0sm2cZtvVcSrbrKyXz3e5NLF4VXr4siMUjOR7EJbFNxCQGvquYvhkCCgzTQLo+HvnosZ4Q+g02nAkBkSIARGaAxAMFJGhd9rCAeEoEhn6pi0dEI4ikaFn2uIB4SgSGfqlLR/QBQwZEGG4eEDPv9HloO6fR+VFhtBptCFSekDU/bMTkM0QOo02REoFqDyFOsRy2AEgZAidRhtipa9BlJw1KNhptCHS/Hfz0T34NEoIyMwkhPZB0BA6jTYESgjIcwZhDAgZQqfRhkAJ+yD4iO+kcR8EDKHTaEOklID8ZuksgNyG0Gm0IVI6QPCGFMVtyDByrsT/9nikhH0QuiGFcRsyjJwr8b49HillE3M+E8NxGzCEafGGSDPFQdFx2+SaP1DMXCkBiWp9s3e8mQjjNmSI0uIN3UrZSV+9q272jgAExW3IEDqNNgRKOczv1BoD+5MJUFBkCJ1GGyKlDRSdC06omM5lSDqNMkRKX4NK29I4HNO5DQmnkYZIyfugyh6gwZgOGGKnsYZAiUcx1xcpcdwGDGFavKFbHAcRYkCEZgMUGbdNrmTLgHe4m3HFbcgQOo02JDRTDYqO2ybXTDOrMKbjmVUct/HMqpQzbuOZVZPsjNt4ZnVRSgjIjKzBAxUyhE6jDYFSzqyKPmbj+JxgtCF0Gm0IlHZm9SBDHduLOXgCwmWI0+INkdI+MHv66lH9u5ghdBptiJT2gZkcO+znAmdW3YbQabQhUsI+SL0junO0BhS3IUOUFm8IlHKYLzdy7LAgoOI2pyGRFm/oFs+sEuJAkVDyZ9KONk8HfMjQ1ZFEG7qVdlajbl9BHQrFbcgQOo02BEo5zOuTt85EobgNGUKn0YZIMz4wc8Z0K3lgZq6ZNfLAMZ3bkHAaaYiUfAmeYy4TxW3IEDqNNgSaYAmefaxyx23IMPqjKf7vyQ7EcRAhBkRozpnVmLhtcs0FKDZum1yzzaxGxm2Ta+a7+eC4bXLNXIOC47bJNVsfFBm3Ta65mlhs3Da5OA4ixIAIMSBCDIgQAyL0f9iNbA24qaCQAAAAAElFTkSuQmCC" /><!-- --><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAAt1BMVEUAAAAAADoAAGYAAP8AOjoAOmYAOpAAZrY6AAA6ADo6AGY6OgA6Ojo6OmY6ZmY6ZpA6ZrY6kLY6kNtmAABmADpmOgBmOjpmZjpmZmZmkLZmkNtmtrZmtttmtv+QOgCQZgCQZjqQkDqQkGaQttuQtv+Q2/+ZmZm2ZgC2Zjq2ZpC2kDq225C227a229u22/+2///bkDrbkGbbtmbbtpDb25Db27bb////tmb/25D/27b//7b//9v///8yjGR8AAAACXBIWXMAAA7DAAAOwwHHb6hkAAARQUlEQVR4nO2dCXvcthGGKaOWlSZt0lXSK5HauhXWTVubTVvRlvD/f1cJDAYHCXJAEBAlZ77H3l2RII53B4ODILZTrFV1R2fguYsBEWJAhBgQIQZEiAERYkCEGBAhBkSIARFiQIQYECEGRIgBEWJAhBgQIQZEiAERYkCEGBAhBkSIARFiQIQYECEGRIgBEWJAhBgQIQZEiAERYkCEGBCh6oDOXddd3uOnN+bY0HWvPvggfQf61XulHm9taK/Hv39Qcz2+u+q6b8ynW7j8Ql9Py6aQjpVWE0AXd+OHh2sHSB+78UEQkA43B/Tpu1epogz6gpP+hIBsOpQghYVYaTUBZAry8QoBBaiMHKAx53NA5y5ZlN7h8IDCSBcFKSzESqsFoK9MmXtXgqG7+F2Yvx7++HQ9lhkB/ee3XfeLP9xbwC5wfBgQ4SXjN2DqzrvxxDcG8r+/HD9+/V4h5fGbOUFwjNWHyFULQL++MnXn1V8soHN3+d8rqB5GFpAxClvaHi0qBjQ5HAPS8d6DeXbd6w8utL54AVAQIlctAP1wO+br49XljwBIZzOqSRbQ/67HdzgBxtCbuhlUhvBwVMUCCxqDv9cfTzqdN/fqpyASDwgOhCFy1QLQjf5qh+7UA6BBF60PPOrcB1ko5i0AFB5O+qATMADmI6XX/4qunAIKQ+SqCaBhzM25u7GAztZG3Nc2a8XQJgwEDyg6nAL05h6aAogLjhuPtQAoDJGrBoAu7sZcf399cTcYQK4Iro5ZQMZZmuxr09cnBt0b8ICiw3NAr/+qbOtvYX/6Dj79DQGZbyW00SBErpoAMkXQtUwX0Fco7Ar1gZcss6Dxkh+B+Mgg6GF9evsl+vQkoCBErpoAsp1oAyjRa5kB2uyDbMP3xluZ0+OffOihmwFyIXLVBtBgHOj4Bb6BF2WK5RvvKaB0c0W0Ymdjk2N6P+gulTHY3+gus7vm060H1NtsuRC5agAImosxP4ZNj1XLfUgACjo8Bu68H5Ru5qHpxi7SO+eNvONDQBBrECJXbQCBuWhA3nBsx1clAUGX+Y/myFiK19jX9YfngLQJjcb56c8jDOgc/6T9CwxhP47u+Ot/+lbMxhqEyBRPdxBiQIQYECEGRIgBEWJAhBgQIQZEiAERYkCEGBAhBkSIARFiQIQYECEGRIgBEWJAhBgQIQZEiAERYkCEGBAhBkSIARFiQIQYECEGRIgBEWJAhBgQocqAuhejowDVja6dGgE6w7qx5TVsnzWgB70S8ry2St/w+eJOB71Jh/icAQ0Xd71e071MyJw7mzWr/cI60c8Y0OPtyRS7X15DrO1GBxs1xKG2+76jtR2QLr0GNKwsstbW0//cLei8ssj64Vo/v3Qy1XFnuker2Af13YL7tYGgIi0+Cvk5A4I161sWoe9J92hxR5EQAyJU0oq5pnrVC1VK92iVOGnzzOTQ3az1Faule7RKmnlomnRLtuFxqtJ0j1ZZR1Fr7Ciu9RVrpZuU2HX1JpV1FLW0BR0ESIinI7THB2Flq5dubrnpcNUIFncU9VO15QaUTjdlGWUlrWdjz6MfJO27CF7tkcKSHmpB1dOVUgZ/xUye0B8nVQAIH3DfUcGWLAh0NJNIBYB2uZ49E2Zy+VQ7piX9oPL+82K6ibLPDsUVEWXQNGz3yzuKNdNNlB0ORYeTfADNc7KgPb2fxXTl/JPhg+CWAQgqwE4VdRQrmNBSuhNbQj5SJBDoI2hBzerYjumOiq1YaEGJijQWX4YIpAkPR7wFrfjwHXoG/SCpzUZi+fCj9CWWUozHBPypSQhjZ5qPsDGoJR8eqAzg8YCkBD7WKduPeNSGUNYnjWiErnD6Vej/cmRk0VB8KIBURrMDYh2rVcUkVBmpIiwOFRw0AXU9g6MjFUPIHpJxVGk9mQWdL+97vR/SLlcdputMCBkFZmVO47s0poNnTaWTofexxxcTLUFU1lEc9K3n8ulEHx225Wg8MymscLp6CeOtlXVbSkYe3ZrdCoWiSlbWUfz4yw/mf7kgOud4lFok5D+K0ELwEpXro9VTWZCeUXz49q4KICVd4Za4KP8OH4Wwbby1LvuiHKq6KvBBeqL1fKpUxVwJXVVKmZL3OpqPqWXCclEK0RU3VDkZ3RTw/Ea3ZKuNWI93zZamrTE618CrSX1KybRYQoD1YAXzFy4T2sGtTT+ov7gbGeox2zogZyHhH2uE4EU4HM55oTElE9tjWU0AwY2Px9uVZVYmOqw/WXBmUrazZHtQayP/p7UgekYRZ0TOsztDkwmzPXwCUq6HCZGqmt6oyYwi3jobvRXhg/aRmWGyUark/EmZ2swoIpZlX+6ddD1AUgaEkAwezC3nYrmzA2bNKOL6s8fbZhaU7A5AnBg1vId/bNfRM4p+LLFXsypGTtduyuiGgDuHqZPo9sNxHtq/RRazzwltBBSsnqoz3bGXjoekJn8HQ/wd+Tx8wmxhbLGRjor+klLWa8YOBiTr+J+YV9Uxa9FYzOxLve/2oXfStbAEfGaV6ml70rDEHoZaFdKtBCgm5RPa26/etQSvONV4TroeF/9BhR2g1H3bgoxmBnRL8CoBqsNnAkuFXWg5r3LzQ3RGswNCL/nj1S4n5PtBrfg4C0rBaGlB9pcEqj2rUa2nODtszWdfo3ZYM//7l6KDAB1d7Hw9LSA3Tjm62Pl6WkBORxc7XwcBatlRtBPVyRZMukR9R2C1STt6sKpmA/FKlBwtFQ1csXMUHl9t5T6T6Y4ATbhMxKXh0pVbbzCWP6uxryMUAKozYFXun3L3EjGxcGgmNw73y6dcq9x6xvtaVZWwk9CEyjKaH7DqYFVWdkDRvdqZ65kqh1X5YHVtY4H86Ny949ry0Yfv8cEcaypeBtzX8UFGdYgAFjdQTU4ExQfbWJBtynbZTzxYrTd1H8eoZgiSB3MzWiugUfbyl8UlQZvAwAvGN5tV9No+sD92+YtWhdsaMz4V5+0PXf5iVYePXyeyIae0mvSkl5e/pNMtbOvdeMItrK58y2eW0SoB1ZblL04buIgIDlqMa5/WLKiAXiMflL38RYW+g7IYQwCW1odWZK/33c5ZauEYfqMazUkvLX+ZP5LpC5jFR7rl0io84319ioI79DQWNNhftq/yKALkPcd+vEJC4d0dR2iW2o52bcd9MbKr6FzRWnSb+OinEZAPYoJYAmuSyRsZCcPKUsvBahYgd28mw2ZgFTl+FGG7Jf1HlaxmL9aCSPMRfnm0sg2YAkLK3zeNiRC3wloPNbJ90BYLWmRj4ejWyx02F1g+4QtGuF6btlW2lndWMwHlNWHmSShtNvD0Kvp2kWy1Fj7bA89hsLoxOqg5E+sR4ZMHhor10YaTeVzVHF4dlc7wbbOfgqHGjX4Uar8m6c4Aibi5giec4QM+0LK2o4DHsGpeq5KpjC4LAZ1aAEq4n6i/LC0P/dyhcJzWCruOIQeSZby1ivWVb/tAXpKEok8CKpN1QwqeFzM+qOCR+bxqVmRBo+pbUMInw0NhQW0T3nJwiwEJPggCpqpaqX0tZrROwILoEoSwjRf4FJ0JpjeoiAjp/9N9GYJYVzKQy+hZAEp2pYWtXYE7dk/K22uwy4xbMExv9qykn1fL5hmtELAouqQNBc9gWjjuBS6C/yKIJL/kL8yCgtF4UMPsph0Ca5Z1NSLaCiaoXbM+dYOM7g9YGN1kusc/Iw97dAiYKDNBJ52gtluelYzmd+wGvLiHGfJwzsd4XrAfEbdexp6kVJFn3kgpP3j56o7KuwEH5mM9sXDDL+X5SNxSAMeroI0bLG0Ivmt1R9XdgMei4zyG9T+2btl6hl0hhbu9BM5m8wZULS2o2W7AOAUmoXa5xgtMxeyIgzO0s/I9r03eWu0GjCN12EUJmnTwN25WyG7L9fPcDVigbD2zsxnCTSP6IUXYyhdnoTijZMAGuwEL6PHB/kDYORQ4CyTdDQthq5fb3i3a6a2Fnkc/KOzjKBy9w+AUmnU704GNGfpvFe0V2ETPA5BCy1C2A2RdkPJMsH84sSC8PPWxxLJmV7Sck86MTssbBHogJCT9HpN+X8Bkwec7Tm63rPkVR99Z9fmyloFdRQUjdyUdDV+vXDESW3Mfb0FZ98U2/oSf8PXL3Qec+Bqbedhu0plU+62529xZ3fgTfoGT8T4Z71rAYEOFtwn9hXGkLRA1saDNP+EX1Bg/KFXCjsuknytbXf3TpDFr4oOWf8IvI11hmyiYi1Zu4tmOU8MqNrWZ52FBOa3Yxp/wm5VX4jRz6Jv0OHXSVM0dc3VGbfpB237CT6xIrp2cBZZBH8F6tEizRGmirTqKCz/hl54wM22V9T4CqxcWOWIlA2T2ppgrvLC3pgXOybr/IuxDRWnSfuvZ9KSFyzL2plMWhKM0B0i5+Vi4gYgbKWNtNO9SzAHZNrO2BW16oC57dYfNrc2xmNSLgJGfasR7ZUFYpRROlKgYkLOhSZL55a4Z0GsDIF8OpQQalIXmR2TC3DzEgZr0BIRz3zKMRrn3lK28LEDPVocBejE6CFAq2rUkKocriY5QozrBgDZEy4CIaBkQES0DIqJlQES0DIiIlgER0TKgz1oMiBADIsSACDEgQgyIEAMixIAIMSBCDIgQAyLEgAgxIEK7AcFN2fTK6sGvFsn+bcms+PrsxZQ50a2F2Q/IrDhLS68BsetA9LqQevHpRwEy91TNiW4lTAVAy0v24P6ZWZE25D9MnRGfWemW+TxATvZWn03ZDah3+dSPlsMtRfhK4OemzLfdnbIfkMmJTysTUE50/VpMuwGdv7IJ6z0E7Y9O2ByYV0smG1BmfLlbhuZE58KktBfQw7VejHc+2Z8AhOzbtC/8azagzPiG5RJtjs6FSapOM68fMruwdmuWOo66KQKUGx9s51gvuoJlqVs0JmwX58HPu+yoYpnxbdoaPCe6pR/oqwToi7swvwmnuhEQGd+mnxzMim6hrd8LCJIZix8uvofEgmY+H1BOfC7MU0S3vxXTALSHM83KObT7oKO4oRXLiE9vCp6zSik3OgyT0v4qdu5wC+Vu2h0MRgT5VSwnPhemfXQ8WCXEgAgxIEIMiBADIsSACDEgQgyIEAMixIAIMSBCDIgQAyLEgAgxIEIMiBADIsSACDEgQgyIEAMixIAIMSBCDIgQAyLEgAgxIEJHAuozl4kF0r/StLSHY/6KoWHDbqMHAirZO3+ZjsoHtBrJVIcC2r5t7IsF9PHq++uxwugFgGbdFi4btQtLHq71aczWYM7qsGY5ij8JJ9TDt2+7V/+IY3QRmSoGP2Z1ChLSS1neIqDeLpzH+K7xSpOQjvLy3qxVzABVDdCYuV4X2Cw8xiW3ZmdK8xPj+jSupRrzpX9zHL9Id9KfuLyfxOgjcj4oTsisxrYJQMBTnJC5EhLSf5tlZUMGoWqATu7lxi25NT/JpbNijthFhbA0bIB8AiA4GZw4TWP0ESEgF61OCKI+42MP4NvihMyVmL+b/AVd1QDd+Be35FafGTrzPd64qg/HbT6hQHAyPhHFGEYEl5lVwC4hKK71Qbi6M44Pr7Tv2Q+PNAHkltyOzmB0JilAQV2JAeGJGFAQEVxm6qtLqI8A2fWqcXzT98fbJ/ZBMwsKj+61oDAie/ak1HQV+BYLMjpnNHstALkMmHwPkyqW8EFwMj4RIw8igk2bzTJ7l5Bd87zog05zUCq8/okBuSW38J13pwhQohWzJ6MTMwvCiEwzH7RXxg6gEQyO6oeBMD6Nc6xRAaATWFtOx6kJILfk1iy0Pft2RFlCvnsSAopOzHwQRmRsLXBy0Jla6weZRwq/91Vs7FRd3utzOR1LHqwSYkCEGBAhBkSIARFiQIQYECEGRIgBEWJAhBgQIQZEiAERYkCEGBAhBkSIARFiQIQYECEGRIgBEWJAhBgQIQZE6P/PsWDaoweBJwAAAABJRU5ErkJggg==" /><!-- --></p>
<pre><code>#&gt; NULL
#&gt; [33mPlotting gene expression for gene:[0m PPP2R2C</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABelBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZpA6ZrY6kNtHk69NTU1NTW5NTXlNTYNNTY5Nbm5Nbo5NbqtNjqJNjshZjuRkTU1kTW5kTY5mAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkNtmtttmtv9uTU1uTW5uTY5ubo5ubp1ubqtujshuq7Vuq+R5TU15TY6DTU2DTY6OTU2OTW6OTXmOTY6Obk2Ooo6OyP+QOgCQOjqQOmaQZgCQZjqQkDqQttuQ27aQ2/+dbm6rZE2rbk2rbm6rbo6rq26ryKur5Mir5OSr5P+1jk22ZgC2Zjq2kDq2kGa2tpC225C229u22/+2///Ijk3I5KvI/8jI/+TI///bkDrbkGbbkJDbtmbbtpDb25Db27bb29vb2//b///kq27k5Kvk/8jk///r6+v/AAD/tmb/yHn/yI7/25D/27b/29v/5Kv//7b//8j//9v//+T///+7rsv0AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAMeUlEQVR4nO2djX/bRhmAndFmpmNrG7cEugDLlvGxlI5CKAQCSRmMQDY2ghkfbWgTaAfU0Mazhz+i/51770sn6XSvJPsky3qf31afX59l3ePT3enknFoB4aRV9Q4sOiQIgQQhkCAEEoRAghBIEAIJQsgvaPrkS61W69W/5nnPWYtz6X1HOnj6Cks9CKOtVx9Ewuyj/8TSX3gn9y7PQm5B4y25+9dzvEkVubWdnu7zxMqhEX3poRkOhm0RvpZ3n2chr6DpQesqqzzTp22+zxk5g6IGwZPW5Wdp6enByvvB9JPWqs49/SP7EozweLN17d9B8OkW6CyNvIL6sKvAsA2Jz7fEkTA9uPznV1or3wmMGMurSiKLHHTZY0p62IYqybajtTEj1wMzLCvtsM2elEZeQV258wJR6VldYhVLHSo6lk+QyDg9CGvQ/7r6MIbw9EBl+lfOfZ6JnILGm6vGM1b/WYv5tA1731qF73jViBnoQ2k1PS0yRlom/WVAOPrRpVFMEC/C5Wei/kMxxdcLr4Yx422qyGYTHE8DwpSMrlxTR9IT0QTVT1Bfl483EvxI6MfKHOj8smVKS/NWmbcuXO6T9lXpR4brIShsCEBJdkFGfUpLw2F63Yj2hS0dDj/671cf5NvrWcjfi8lRCCgZtlUrHAoKYwZZBI03V96JREUbbYSlv/Fm5PD1TIFxEB9E//NtNXYJPj+ANkgJCmPGu7II6uo6J6NCRBgOx0F5xqizUnwkfelQDXP5eE4JCmOWbt6RVsNko/OHTtEI63SpbVGBk9Wnb0O7yk+fguGWODkyBOlYPkGqQzMEsdq6bYRZvfxle/HPxZoGCUIgQQgkCIEEIZAgBBKEQIIQSBACCULIKegFRz4YCUvI9VqGkJ/i5ocEIZAgBKeg0Z3TIJjsdm49VxESZDLorJ8GF0f7QW9DhUiQwcmNj1gNmuydiprEKUHQ2tpaTQSJQ2x093kw+ckxe/Yyw/8OrQH+PyYzqKDBLSUI8F6D1gS1rEEACYowKr8Nqp+gi6OdefVisvl155J+6iNofuMgVfRsGv0XPRvljaT1wZNtC36Kmx8ShECCEEo8Wc3YBjVXULZerMmClu1k1QYJQiBBCCQIgQQhkCAEEoRAghBIEAIJQiBBCCQIgQQhkCAEEkTEoBqEQIIQSBACCUIgQQgkCGFhBMWvmfkpbn4WRVDiqquf4uZnQQSZ16VJEAkiQXMXRG0QGqJezGXDkstPcfNTiSDbL6lIUJhItsgkaCkEjTe31SolqTRZkFx8LLbOXYy5CKppG9RdjScszEdQLXsxfoBx+o4qNCdBGUK+C54VXFCv04G/PBQ0WND0QK0MdhZtpk/2jScNFmQsjBVZQ+3ig2PjWZMFyfXmxpvRCjTZZYcYr0Tl/EnmomEOFPkqc/FVkEffPTZqUaNrkAPdDlUt6IWd+biwQoIQzDZIrlYY7ebhr54vPlyUbr5KQV3WOPOlIZPjoBu6I2uwIDFQhDWJF3kkXbkgWG2UBJlED7GAr7k7Z0EZzkxrIUgtTDw9mK+gLHMb9RCUhfyCMs2O1UzQP+ZZg3wJOj8/twgav+m+HcrwysNECpgepN3MJDoOgqky54roCyPo/Nw0pF/vfzvrauX5BfGbP1yHBcUdd4bJL8hPG3R+HjGky3n/1998xhdFl7cFEjezUA8rh6AFTjm3eTB85dIWKojPcvQvf9Ka+6S9j14sRdDwtWfd7WD81sPgbFU9BF32rcP6+9vs4dMrD+EoZJ5AFXtFZUitFdFxEJPrrqJFBPkYB6UIOmNFXVUNkXpgnlhSHFLywGKxoVDFUpAhwyEmBSGH8KIIsrdB/A5DL/GDCIYs4gGcTO8fssqlBPGTThC0yad3RCyjoC8i98RaGEHWXoxLEEXti5aCPcRrEBRUHGLwSiCqWOYaVB9BEeSrZ1DK/irIYf/LB6MNGl75i2h8WDllG6QyZG2Dai1o+guoEay2yImbrr0XO2tBpwUnDDzEj8wMvRg/IGN3raqZIC8szA+oSFD9BdmnXG072ExBaVOulh1spKBsU65VU7kgmnKNU8KU67II8jXlujyCskCCELKXONeagPUQ1HWeZJg7mKHE+VaVrIWgrnEb6lSyFi/nuqR1EMRnRNKnRSI72EhB8mfkc5pR9C/IaON05qxXKtKzRLfgUZD3Nsjcfi0Fee7FIjXUFDR87T1+3UpeyhHD3vFbv+IzY+KZNYsIWkbJ3gT5HQelC2pHplHFhR34YSqcQIln1iwy6KxBdZpRTBcEM85X1KUcdWGHffnT+4fymTWLDDoEZWNRBKW2QWZBu9vywg6/QqafWbMsnaC0XixWPeSFHSYIapB4Zs2SRdBcR9LeBUVICDIaGLg0BrNc+gKQNUsGQXMdSVcuSF7KERd2xm9+S/Vi8lJrIosMJqcyPI2kqxGUBvarIRc17eZJ0LKfizVJkNebQKafkNh3t1JB9pG015tAOk5p7btboaA0fN6AzTUp4rHMuUAF+bwJpBQ0343OGXQk7fMmkPWqQSkjaa83gczdBpUPOpL2exPIvL1Y+aDd/BxvAjmHcVD5VDwOWgZBEUgQCYpRsznp8qnZlGv5kCAEEoRAghBIEAIJQiBBCCQIgQQhkCAEEoRAghBIEAIJQiBBCCQIgQQh0E0gEagGIZAgBBKEQIIQSBACCUIgQQgkCIEEIZAgBBKEQIIQ5iQoXNWQBNmKZ6yLSYIsxTNXViVBJMgFCUJIKR61QYq04lEvJslR4gYL0tXFXBP8RTzUGEGJm0DqBsdcFdxsrhsmKH4TSN1lRdaV13Ur2Z0tuaDETSBJUJTETSClBCMRYgnVHfxvVhM3gaQ2SHPS6Yg/pYve45B6sQQLcxPI8kEFLdZNIMsn0zhocW4CWT50qoFQ6PdBLycSlpDrtYyhRYAEIZAgBBKEQL9RRCBBCCQIgQQhzCpodJuPsyd7pyy1c6InH2FFHRbpmNORYq0dmD9hpy+j2/sXRyz1390dY1sbPZ7/hL9xP/lx5TOjIFgCDhbQmew9ZqkTVrbRuz/qCL78M1HCQbh4DBMEjoLR9+GNcAo8evfHv5MmYFIFFuIZ3f0DC/TY6xv2Dy2VYoL4LBpw8x5Uj5ONyd4jVvDBDjh4tLvO12TiKgIRCbPDilYBz85nmSY/vQcVBmoZ5O/tyBCsfKU2UCkFaxDUGvHIv/3eV+48FimxpNfurf/cOZWvQQ1S2Se7N37/oSg9qz7gc/A6t8CMrz9WNeh7v2HmBhtm1auOooeYWtZsIpqQ3vrpZHdfJOD56Pb6qapnvHaoVdBYu8NKDiHRzmwY1QRaMd4GiWZqcOM4/qkVQL0YwsyCBvvxhC3keF/hLZQDCUIgQQgeBemxwLru5o0xIwliPdZvvyYjqpu3va/BgoLPvqE66nCx00SmJgtqchskxnfq3EqchSZDirCcn73hyO7YQmXMeC72VXVmys7KE6HwREEJ0u+zZXduoTJmq0FmwhZSxGtQ/i1URjmnGpbWpS4UFjRQAxudsIVU5lDQz13Z07dQGYUPsd39WMIWUhiH2Ncd2R1bqAxqgxAKH2InO/FEMmTpwF3ZraGqKX6IyVYinBWLh27+MNGB33qent0eqhyvvZjl4KkdNKOIUFjQxRE7dOAUVCcsIUsH7spuC1VNUUEXRzuju8/FBQuRsIQsHbgruy1UOTN08+LKlU5YQvYOPDW7LVQ5s9Wgnvqye/r7j4SSHbgzuyVUObO0QeJ0WyeSIUsH7spuDVUN9WIIJAih4ISZ+j3CzXsysf4oGUr26R/vObJbt1A5XmuQpQP3+XFe8Donbe/AU7M3b0460YE7szdxTjrZgTuzL9V8UFMoLKin+hmdsIUc7yu8hXKZ4Rdmgw34xZxOJEKvJ/v09dP07PZQ5czQBsX+s4Uc7yu8hZIpfLL6wTH7b3TnVCcsIUsH7spuC1VO4TaI7f2g09kxEvHQD5LXmeEH5GnZU0JVQ3PSCNTNI8w2H7RhJiwhSwfuym4LVc0MM4rsX9YR64QlZOnAXdltocrxfbJq6cBTsy/RyWogv174RyeSIUsH7spuDVXNjJeejQvM4Y99w9DfEh245bfBzi1UXoeoF0MgQQh+BVk68Lrhe046iHXgtaOEU43oD6jqht9DzNKB1w2/Ncj59z71gHoxBBKEQIIQSBACCUIgQQgkCIEEIZAgBBKEQIIQSBACCUIgQQgkCIEEIZAgBBKEQIIQSBACCUIgQQgkCIEEIZAgBBKEQIIQ/g93DNPFoPBB/gAAAABJRU5ErkJggg==" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m PDZK1IP1</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABelBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZpA6ZrY6kNtHk69NTU1NTW5NTXlNTYNNTY5Nbm5Nbo5NbqtNjqJNjshZjuRkTU1kTW5kTY5mAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkJBmkNtmtpBmtttmtv9uTU1uTW5uTY5ubo5ubp1ubqtujshuq7Vuq+R5TU15TY6DTU2DTY6OTU2OTW6OTXmOTY6Obk2Ooo6OyP+QOgCQOjqQOmaQZgCQZraQkDqQkNuQttuQ27aQ2/+dbm6rZE2rbk2rbm6rbo6rq26ryKur5Mir5OSr5P+1jk22ZgC2Zjq2Zma2tma2tpC225C229u22/+2///Ijk3I5KvI/8jI/+TI///bkDrbkGbbtmbbtpDb25Db27bb///kq27k5Kvk/8jk///r6+v/AAD/tmb/yHn/yI7/25D/27b/5Kv//7b//8j//9v//+T///+xUq6eAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAMeUlEQVR4nO2diX/cRhWA1ybBXQK9vAmG1BxOU85uSAmYgMFgByhgcMsRlqPUJvGSgg2Ot7vdw/rfmVsjzdM8SWtd0ft+bXb8NFrvfB7N07WjTkB46VT9AeoOCUIgQQgkCIEEIZAgBBKEQIIQsgtaPPtcp9N5+R9Z1hl2BNd+HZZf/pC/135H0Q+Gq0e86qS70hfrDNZEheunkbXNkrLILGh2VzXp9QwrDUMPkbIriPk5UKvEBYk1zJKyyCqINelV1nkWH+mGpEL1jmcd3lxRXvzLeoORiU+6qx+KX/OXjiXIWtssKYusgkb60026vPDJXbmxsIb8/bOdlW8FVozV7au1VBODAXvV5ZHpg6MuD/H4qHtN+JltrXx7Ky5IrG2WlEVWQQP1YSWTLu/5rCvobaVvxRBBou3yTUSvYXHZRxizX57OEgTpJWWRUVD0wy32V94Ogo9YX2KC1k7F6BDGLMxGshbK0q5ZnziQdd61BzZX0DPVeZsgSAyb108nXdEg1oDFPm8EXxrGrNX0MMtVxAQxtXrs7Xz6bjgu2YKstYNGCRqZTy63l8X+WjCKtkZip/aYoIHuNcPOa3xrO4r8JluQWDuouSDZU2QpiyCrP0XHoDBni3j4IzAGRZaURfYs9pos8I8+6epROBQUxiwgQSKLjfjQZcdNh2qqIDZkiJ3o/7zFNjE2ILO920/2+RikBYUxay1XENsPEmn9+mk0rofsxgoK96SviUSudqpDQWEMSPOibO0XD3SZVTeHGjq5NVQQS+Fv8RFTHhhNmK5PvW1vYmHMK2jlVT7kAoL0MNRgQe2CBCGQIAQShECCEEgQAglCIEEIJAiBBCFkFPRcoF6sAhDyLUsRKqa52SFBCCQIgQQhkCCEwgStr6+TIE+L19dDQyTIDa2vW4ZIEAkykCCE1C2mMQhrMWUxrMXt3A9qH9SDEEgQAglCIEEIJAiBBCGQIAQShECCEEgQAglCIEEIXkHT+ydBcNbr9TZOVIQE2YyFmOM9K0SCLI5v/oH1oMv3HlsxEhSBb2LzHbaJiU70EqOcD1UnUEHT7zy2ehH1oAhikOaYcYgERSBBKQSNb58Hl+9TmgfR+0E3TSIjQQgkCIEEIZAgBBKEQIIQSBACCUIgQQgkCIEEETGoByHURlD8ns9impudughy7houprnZqYkg+75qEtRMQbMtMxNLdLKVCC0WFJ+XDaYoQc0YgwbhdFBJFCaIsljWUDHNzQ4JQiBBCJaggTd/SdosaOBOpyaui813erfPdaTFghb7ffWPQdwfdHm4F5xt6lDVgp7DFGUniOwoCkHWPOPy/qD57kl4jZ4E7UcmYhe3vzw4D+Y/5tee63B/UA0F8ZsXpKDwA1IPUkR7UPgBWyrIPVid0hiE3/5yebhNWSyJ+u0H1UyQS4sFVXvCLLugi4sLQNDsjv9xKJMbR06JE91DBgWlozaCLi5sQ2b56Btpn6jzggu6uIgYMu189JuvnYpJ0flWoF7YtqFeVg64Fv5klL4Ihkuu3W2HoMkrp4N+MHvzKBiu6Zdg8How4vPv99nLv28c8a2QeeKq2BJdodMOQUPW1DU9EOkX5okV5SalNiwWm0hVrMQrpNjEGjZIg2OQeMLQqtiI+LMV5At3snh0wDqXFiROfHFBW+L5HzKWogdVelXjSrKYkCCbOpIXINhLvAfxYyq5ifElgexi6Qbpoq9qpJtSSNXCBEVQS4e8laM1Lof9r16sMWhy4wM5+Ew+c6DGIF2hDmNQukmpdK0cghY/F48uuXOgTh4P4Cw27PCkxR/CI0Jiy7yyLJYH1mD1sq6K/sq+Wv4eVAThIP0rE/vrlQ7Sukukm/fN1PL9/moEbfV14UqzmGlx0wWxTVMO/x3f0x2XEFT8GFQI9nUxbmbgfwTlMoKsLAaYuoosVgSRNL/6QTcx3dkfMNcY5A9l3g+qQhDrPti+Yg5BbndxbwZqiiDzZM9E8gjybXUNE6QeEOahxYLKPFht7BiEczWCvFmMBKUL+X5/JYLUsT9j1ITzQRUI2tK7iDFBtZo/qFJB31RPgI8JqtX8QaET6+SSqZz2SkVyleg7xAT12eEY3w+KCqrX/EG2H8tQWYLko7ujguo1f1DEj3WCUsLPpL7yC/GIW3UpR54sm735rjgElz+BVWRQnVuziZ/uGHVW/xapU6/5g3BB3chpVHlhZ7Z1/ZT/2eVPYBUVxHoQBzogq830OLggfsb5hr6Uoy/ssLYtHh2on8AqKugRFBIm/PoKShyD7IYO+urCjrhCZn4Cq2QRFKVe8weZ2klZLNY91IUdJoj3IPkTWCWVoJF+iHeEWs0f9BzGEWQNMPzS2NaadQEIrJJC0GJfPfDdd8qjMYLUpRx5YWd25+s6i6lLrU4VFVQSQEFDJSZ2I2eTBCWB3TXkI3qnvWDo6UItFmTSPB2sRiBBCCQIgQQhNOwGqvJp2CnX8iFBCBm/s9o+vN9ZdWlxDwK+swrQYkHQF+pcSBAJikGCEEgQAu1JI9COIgIJQiBBCCQIgQQhkCAEXFCtJhYoH1RQvSZYKh9UUL0mNykfVFC9Jlgqn1R3d9Rn/qDyydCDOCTIgcYghHpNsFQ+tB+EQHvSCCQIgQQhkCAEEoRAghBIEAIJQiBBCCQIoYQJlprNFfWgcMq1lveghOZZk/aRIKB59rSPJIgE+SBBCAnNozFI43hxSiToeXTLSmwxCYpMWUyCJCQIgQQhaDOhoVjzgFHJZyN5BuVimpsdXJAzwVLoxZ403fhx+pRHkGf20mKbnR5cUHyCJXCvENjqUgjyzX9bXJOzgV96jk+wRIKiOBMsKQlgZd8ygHRzcFcLfvuLM8GSb5xp2Rh03OttykJk/iBfpmphFuPUZoKl8mnYBEvlk2o/qD4TLJUPXfZByHXZ5yWnAIR8y1KG6gAJQiBBCCQIgS49I5AgBBKEQIIQlhU0vSf2s+e7J6y0fWxOPooHjd/r9ezTkfKLH/z8CTt8md7buzxkpf/tbFvvtXkm6h+LFffcX1c+Swri30fk3+aY7z5lpWPWtuk7P+xJPv9T2cJx+E0GJog7Cqbf4yvyQ+DpOz/6vTLBT6rwb4VMH/yZBc7Y8s3lPtyVkE+QOIvGufWQd4/jzfnuE9bw8TZ38GRnQ3xBSKgIZCSszr9eFYjq4izT/CcPeYfhvYzXP9tWIf41LP0GlZKzB/FeI1/FX//sC/efypL8ftnO7f/eP1HLeA/S1ec7N//0vmw96z7c5/gNYYEZ33iqe9B3f8vMjTftrlcdeTcx/R27uRxCzjZO5jt7ssB/nt7bONH9TPQO/ZU8Nu6wlvOQHGc2rW7CRzExBslhanzzcfy3VgBlMYSlBY334gUo5Fkv9zuUAwlCIEEIBQoy+wIbJs1b+4wkiGWs331JRXSah9ZrsaDg46/oRB1+896p1GZBbR6D5P6dPraSR6FuSBO28+Oveqp73qEyljwW+6I+MmVH5U4oPFDQgsx6UHXvO1TGcj3ILkAhTbwHZX+HyijnUAMYXZpCbkFjvWNjClBIVw4F/cxXPfkdKiP3JrazFytAIY21iX3ZU93zDpVBYxBC7k3seDtecENAAvdVB0NVk38TU6NEeFYsHrr1AyeB3z5Prg6HKqfQLAZsPI2Dzigi5BZ0ecg2HX4IagpACEjgvupQqGryCro83J4+OJcXLGQBCAEJ3FcdClXOEmleXrkyBSAEJ/DE6lCocpbrQWf6j31m/v6RkJvAvdWBUOUsMwbJw21TcENAAvdVB0NVQ1kMgQQh5Dxhpu9HuPVQFTaeuCE3p/9x11MdfIfKKbQHAQm8yF9XCIWek4YTeGL19p2TdhK4t3obz0m7Cdxb/YU6H9QWcgs603nGFKCQZ73c71AuS9xhNt7kd8yZghN6w83pGyfJ1eFQ5SwxBsX+g0Ke9XK/Q8nkPlh97zH7b3r/xBSAEJDAfdWhUOXkHoPYpx/3ettWIR76vnudmd9AnlQ9IVQ1dE4agdI8wnLngzbtAhACErivOhSqmiXOKLJ/WSI2BSAEJHBfdShUOUUfrAIJPLH6C3SwGqg/L//HFNwQkMB91cFQ1Sx56dm6wBze7BuG/ukkcODeYO87VN6HKIshkCCEYgUBCbxpFH1OOogl8MZRwqFG9AaqplHsJgYk8KZRbA/yft+nGVAWQyBBCCQIgQQhkCAEEoRAghBIEAIJQiBBCCQIgQQhkCAEEoRAghBIEAIJQiBBCCQIgQQhkCAEEoRAghBIEAIJQiBBCCQI4f9zxkUj1dyePgAAAABJRU5ErkJggg==" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m MMP10</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABdFBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZpA6ZrY6kLY6kNtHk69NTU1NTW5NTXlNTYNNTY5Nbm5Nbo5NbqtNjqJNjshZjuRkTU1kTW5kTY5mAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkJBmkNtmtpBmtttmtv9uTU1uTW5uTY5ubo5ubp1ubqtujshuq7Vuq+R5TU15TY6DTU2DTY6OTU2OTW6OTXmOTY6Obk2Ooo6OyP+QOgCQOjqQOmaQZgCQZraQkDqQttuQ27aQ2/+dbm6rZE2rbk2rbm6rbo6rq26ryKur5Mir5OSr5P+1jk22ZgC2Zjq2kDq2tpC225C229u22/+2/9u2///Ijk3I5KvI/8jI/+TI///bkDrbtmbbtpDb25Db27bb///kq27k5Kvk/8jk///r6+v/AAD/tmb/yHn/yI7/25D/27b/5Kv//7b//8j//9v//+T///+W4xytAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAMGklEQVR4nO2djX/bRhnHnaz1sowNlrgl0AVGto7xspSOQigEwpIBg45svAQPCiS0yQYxdPHs+SX657lX6SSd9OjF8lny7/tp49Ojs+z7+nR3kuxTywOptFy/gUUHggggiACCCCCIAIIIIIgAggjyC5p+8qVWq/XS3/M8p/vSb1Xq0zc3YosXLcENHey2xcMXd9nL/CP3+5sxuQWN78ritDZyPKnbWnkkEsM1/rzIohLUau2K4EVLCBrv8NDqad43OGPyCpoetb7MKs/00zVVxkx0V16UPi+eW9uILV5IC5+0bl6yLX/ckoIuWq9cTv+a63OogryC+vLd8w+fJ/RuMD26+bcXWyvf84wYy7urntVdfZ8XnmX7rhAUXlSCWPiU1ZuV7++0xTqhS/x1SV5B3VCdZ7sIg9UlVrH0TuLHQoIev8Ar3HD9sRQUWgwJ+s3lWAiSfyOv54CcgtTbVkyPVt5mDS2rS0xQ+1K0HkHMoLv6zyO+r3TbQykotOjvYm3jNcQati7PnlwFxQSJZvXmpS7E6un0iBeSrw1ixtNYPeiyfWW8s6sEhRZ1I61cNEhQ3y+ZbjHaXj9cWgkz0meB/uqpFmQuKkG6S6+1IFlTZCqfoPHOBh/gaEHmYri2eboN2lDPLFO88uTvxb4iE1zJcE23woGgIGbAi8laZrZKCzIXrYLq2oux1lgMov/zJtvFWIPMRr9fHPE2SAsKYsazuJH+ylvsry/IWLQKYqNJNg76uHbjoGAkfUN05GpQHQgKYqFuno9weC/lCzIW7YJqOpL2+PETb1HlgdOQ6XrubXMXC2IRQaxK7BqCjEW7IDHgvFG/Y7FlA4IIIIgAggggiACCCCCIAIIIIIgAgghyCvpMoB6MhCWUti5DqJri5geCCCCIAIIIIIjAjaDNzU0ISlm3uekbapqgmbApcfDKBXBRg5SgRtYgCCKYiaAmt0GzEYRejFqHcRAEQZAdCCKAIAIIIoAgglKCgtEPBNnWGeNnCLKsM4/AIAiCIMhOCUFog8h1zevFRvfPPW+y37nzTEdKCWrcOGjQ2Tr3ro8Pvd62DkGQwdmtP7IaNDk4lzVJAEEhuJjRg2fe5OcnbOl5xpze1QJBChrc0YI4qEEhwjWIA0EhRmiDaEHXx3voxZLAOAgnzEggiACCCCCIAIIIIIgAggggiACCCCCIAIIIIIgAggggiACCCPCTTALUIAIIIoAgAggigCACCCKYg6AMP+2phyA9Y6hlHkSfAoKy/DisFoL0nJfh+Wwj5BeU6eeFmQV9ZmeWRiL4gsY7uub0U+YuhCAPgsIYu5iaNXXGu9hs2yCXgvx5fGfcSM+0F3MqKAtFBM1yHARBEARBTRUkp29uEXM4L46gq6sri6Dx6+lT4A/XT2MpzvQoqWsyDzXsYnqdDv/G/YIJuroyDfnr+9/JOoN5AUH+SCjM2WHsDboXdHUVMuSX873ffftSDFj4h60e2L6hHlYecS38zii7IhisuXE3iyAr1x+eGEsLLmj48mV31xu/cepdtPWD193w+nz+/V328O/1U74XMk9cFVujMySO/khBk322i4lKtAg/RSAEXbCitnVDpB+YJ5aUu5TasVhsKFWxFM+QbRezMvrBiVGLFqUG2dsgcYchca8FcVsP+cCdTN97xCqXFtTlmbigHXH/DxnLIKib3H/57dDCCLL2YkKCLGpftqfsIVqD+FG53MX4Gk9WsSw1iDfRCR3ZXAWpQzdKUAi19oKXst/mcth/9WC0QcP1x7LxGb7wSLVBOgPdBgmHFpH81z7XH82vm9cH/wUETX/FP15WW9S+0LX3Yhct3mnxm/CIkNgzM/Ri4nzQ9Cg+jGDjoFt+R1a5IP/0UQFBlUALsrxBCEoEglwLKtMGVcLiHayW6MWqAKc7CCoTlGuahSUUlG+ijuUTlHOqlzoI0reSnU0j3UBB/PRS2hUx8w0upyBPHLwQjjKXuKltEN/RZnNltbm92PSoFlc1HAliNSj13sGLI8iooX7mrFcqkrOEtxAVRNlZJEFmGzcnQX3SzgIQ8mO0chJ+JvXlX4tWVF3KkSfLxm+8Lwonl6xZZFCdWzOpaBxUdQ1KFrQWOo0qL+yMd25e8q+FySVrFhVM38Uyf4KLLIifcV7Xl3L0hZ2dXX5hQy1Zs6hgYwQltkFmQbu76sKOuELmL1mzNE5QUi8WqR7qwg4TxGuQXLJmaZ6gEDFBRgPDL43ttI0LQNYstKB6NdKkIHUpR17YGb/+lu7F1KXWWBYVjI+SKzpYdSMoCepbQ2lUdbDaTEGzPFhdJEFlwMEqQW0PVl0IqtPBqgNBmQ5WXQuaPzUbB82fmo2k5w8EESyMoOhFkGqKmx/Hgnwtscto1RQ3P7SgKmcD9rXEf7hZaalzQH/TvsK7IgRaaiyoyhnJGyGoyrsiKC1+crZbnw2koErvimA0zbXtxaq9K0LyFfyKy50Zp21QWqjicmcmQy+GuyKkg7si5AGCCCCIAIIIIIgAggggiACCCCCIAIIIIIgAggiSihfMgwBBtuIZM2lAkKV45lwsEARB+VGCZrGphQNtEAF6MQKMgwggiACCCCCIAIIIIIgAggggiKCUIHOyUAgShMpiThcKQRKzCKEJZyFIAkEEEEQQKgvaoDjhsqAXixHxEi9e2rqGCordFcHW8lj2uqURFL0rgvUMvaXdXhZBsbsiQFCY2F0R0i5hNPDyRoG7IqAN8jnrdORXyMOT/qMXi7Ewt42YP5l+7TPPuyLUTtCc74pQQ0EhICgTz8cSllDauoyhRQCCCCCIAIIIZvIFqiYDQQQQRABBBGUFje6Jcfbk4Jyl9s78k4/8l+Qs0jFPR8rfmPPzJ+zwZXTv8PqYpf63v2dsa7sn8p+JJx7GX27+lBTEpz7hPxyfHDxlqTNWttE7P+1IvvpLWcJB8KNpJog78kY/4k/kh8Cjd372B2WCn1ThP0AfPfgLC/TY+m37i86VYoLEWTTO7Ye8epxtTw6esIIP9riDJ/tbYi4CocKTkSA7n8nBE9nFWabJLx7yCsNrGc/f21MhPuOD3oBTCtYgXmvko/j0e1+7/1Sm5FQW+3f+e/9creM1SGef7N/680ey9Kz6cJ+DV4UFZnzrqa5BP/yAmRtsm1XPHUV3MT2dx0Q2Ib2t88n+oUzw5dG9rXNdz0Tt0LN/sHaHlZyHZDuzbVQT3oqJNkg2U4NbJ9FXdQB6MYLSggaH0YQtlPK8wluYDxBEAEEEFQryxwJbfjdvjBkhiPVYv/+Giuhu3va8JRbkff6a7qiDSb5imZZZ0DK3QXJ8p4+t5FFoPKQJyvn5t1Kyp2zBGSWPxb6uj0zZUXksFBwoaEH+82zZU7fgjHI1yEzYQppoDcq/BWfM51DD0rrUhcKCBnpg4ydsIZ05EPRuWvbkLTij8C62fxhJ2EIaYxf7Zkr2lC04A20QQeFd7GwvmoiHLB14WnZryDXFdzHVSgRnxaKh2z+JdeB3niVnt4ecU2kvZtl5agfOKBIUFnR9zHYdfgjqJywhSweelt0Wck1RQdfHe6MHz+QFC5mwhCwdeFp2W8g5Jbp5eeXKT1hC9g48Mbst5JxyNainP+ye//mHQvEOPDW7JeScMm2QPNz2E/GQpQNPy24NuQa9GAEEERQ8Yaa/j3D7oUpsPYmH4n36nw5Sslu34JxKa5ClA6/y5Sqh0nPS9g48MfvynZOOdeCp2ZfxnHS8A0/N3qjzQctCYUE93c/4CVso5XmFtzBfSnzDbLDNvzHnJ2KhV+N9+tZ5cnZ7yDkl2qDIP1so5XmFtzBnCh+sfnjC/o3un/sJS8jSgadlt4WcU7gNYu9+0OnsGYlo6Mfx68z8C+RJ2RNCrsE5aQJ08wTlzgdtmwlLyNKBp2W3hVxT4owi+8s6Yj9hCVk68LTstpBzqj5YtXTgidkbdLDqqY+X//ET8ZClA0/Lbg25puSlZ+MCc/Bl3yD0r1gHbvlucOoWnNch9GIEEERQrSBLB143qj4n7UU68Noxh0ON8Beo6ka1u5ilA68b1dag1N/71AP0YgQQRABBBBBEAEEEEEQAQQQQRABBBBBEAEEEEEQAQQQQRABBBBBEAEEEEEQAQQQQRABBBBBEAEEEEEQAQQQQRABBBP8HeLwTrja3WAkAAAAASUVORK5CYII=" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m LYPD3</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABa1BMVEUAAAAAADoAAGYAOjoAOmYAOpAAZpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZpA6ZrY6kNtHk69NTU1NTW5NTXlNTYNNTY5Nbm5Nbo5NbqtNjqJNjshZjuRkTU1kTW5kTY5mAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkNtmtttmtv9uTU1uTW5uTY5ubo5ubp1ubqtujshuq7Vuq+R5TU15TY6DTU2DTY6OTU2OTW6OTXmOTY6Obk2Ooo6OyP+QOgCQOjqQOmaQZgCQkDqQttuQtv+Q27aQ2/+dbm6rZE2rbk2rbm6rbo6rq26ryKur5Mir5OSr5P+1jk22ZgC2Zjq2tpC225C229u22/+2///Ijk3I5KvI/8jI/+TI///bkDrbkGbbtmbbtpDb25Db27bb///kq27k5Kvk/8jk///r6+v/AAD/tmb/yHn/yI7/25D/27b/5Kv//7b//8j//9v//+T///+E2oz/AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAL20lEQVR4nO2dCX8bRxmH1yVGNQltwE4wuAbqNC0Ql5SACbiYWoEWDG45jDgSk1gkYIFrV4oO78dnzt1Z7ey8e2Qv6//82mj0arTSPJ6Zdw/trucDJ17dX6DpQBABBBFAEAEEEUAQAQQRQBBBdkGz51/1PO+Nf2R5T/+1J7LQ87bDQt8TXPuE15DlNx6LV5/f8Lybp5m/WhlkFjS5K5vifT3DmwJBky1ZuvA6gRSPSzPL6ol+T71kFTTrejdZ55m9WFk6SP+uQJA0w5bCAyr63Fs+VeXZv/hiJ1vLj/3ZX3Vnq5esgmQDGaMVXnh5Vw6LWXf5bze8pR/4RozV1U0MBclB1pcdRUV77FGXL4KeOVrJ0kdLI6ugXqTjj1b4WGB/dNax9AAJYnZBfJBJuVZBzLSce2a9LF20PDIKmmx1jGez7tI933/BmssEddg48TpGzMAQxLR9uSufBkOsY07jcqx1vaXf5GjOqyefIDGLLp+qUcAaJycV/moYM94WedbzVM/SEzPvKnOCJndueG3uQUrQRdBCOTJm3Y5/YbQ6ICIo6IVmap8TxHi5tdyERJ9R0Kyrvz9XUlSQEY3NQSzUiC6UPYt9TRZ4S0YrehYOBYUxg/SCeBbTc3grBbHZU6xE/+ddNsTYhMzWgl92+RykBYUx411pBbH1IPbIPuOeWj2qn/xr0tdEIlcr1aGgMGameXPt2y7IXJO+WIkN0trIsbH64l0+s34iyiOm60v3zCEWxnIIWrr5OFis2iyrG2zNE0AQAQQRQBABBBFAEAEEEUAQAQQRQBBBRkGfC9SDUbCEXK+lCJXT3OxAEAEEEUAQASZpgip70Orq6pXvQUUEra6GhiAo/trqqmEIgiAIgqwUEIQ5iHwNWSz1axAEQcUEWSYjCDIKtnQGQWHBTPgQBEEQZPJKBGEOIl9DFiNew3pQdkHznaqc5manKYJi01I5zc1OQwTFE1s5zc2OU9D4/onvT3fXNs50BIJMhmvrJ/7l4b4/2NQhCDI4vvVH1oOmeyeyJwnKEtTSOYiLGT8486cfHrFnrzPK+yLMT3kLLwApaLihBXFK60HtXA+K9iAOBEUYVzcHtVfQ5eFOBVmsxYKqWQ9qqaA4EEQAQQQQRABBBBBEAEEEEEQAQQQQRABBBBBEAEEEEEQAQQQQRABBBBBEkL7FmX4UfXUEpUYcCCxp2ZVSUg/KeGLG1elBEEQAQQSpW5zt1J4FFLSoWSxDi7EeVIegz+0U95AIBBGEgvgVig/6xLXYF1lQj1+HlV/9sucwtMCCJlvbys2F42r7iy6ozy8iCkEmxhDTF9/tdRLq+ostSBkarTj8LLagNEAQQXMEnZ+fWwRN7rgvQj26/iRW4sy6SXc5MQVdBFe8br6g83PTUNiE76e9W0l2Qep6/hPnzQiaIuj8PGIoaMOj3753Kv7SvC3qYbKlH5YOuBZ+Z5RtEQxfuXaXFtTXt0TptmBFMUHQ6M3T3rY/eecJX2FRD3zl7oJff3+bPfz7+hM+Cpknroq9oiskDpxAUNjJ+o4u1HBBfdbUjp6I9APzxIpySKmBxWIjqYqVeIUUQ0ysKArmVhQHa2v8tKhmCbLPQeIOQ6+JQcS3meQDdzJ7dMA6lxbU45W4oC1xawEZyy/oeD/2BRsgyJrFhATZ1As5DNjDfA/iDZVDjL/iyy5WoAddfnpkPGuOoAjq1b6450SHy2H/qwdjDhpd/7ucfEZfOVBzkK5Az0FJgqa7bIiJTlTu+WLpcAqa/UreN+mgJ9NXz57F+h5PWjxpi5AYmSmymBiQXvzeeeMfHhm9qNk9qAzSrUkH8xAE2SkkKMV1KNohqGe9MSU/JfPyswJpPs2VTFohqBfcqzIKWw+6FSSy7IJSXQtnTmgjBYk1geTVgcgXLFOQru36/FoEiTTv3A4Lv2CJgoLqrs+/UoIyzkELKChbFltEQdlCDZ6D7GvSti9Y6opiU7NYOioQ1Nj1oFQssKDJe2lu2rnIgrbS3N+9OYKMX7AFldMeqUiuEl1CRBC/k6djdi5JUPIP9QhB5m8gqxLEdxxRR5VesSDHTz3dgiK/ojUFjd78tTi0pw7lyJ1lk3c+FnvG5DNrFRlU+9aSBOlDh5WledePhfMKWonsRpUHdvixPr6fVD6zVlFBdw9i/qrtQWUI4nucr+tDOfrADlsJnj06UM+sVVTQKahX+RyUX1DiHGQ2tLetDuyII2TBM2sVWlAtWSz3HJSYxea6hzqwwwTxHiSfWaukEFTLelDuLBYlJsiYYPihsa2OcQDIWiXNENO09yd4hiB1KEce2Jnc+Z7OYupQa6yKCqpfcOQXVDduQUlQvxpycaV6UBIQlLv5NBBEAEEEzduj2FRB6ahbUPVAEAEEEUAQAQQRQBABBBFAEAEEEUAQAQQRQBABBBFAEAEEEdCCcF8NN9XcX6zFgnBvH4Kq7i/WVEhBuL8YAe4vRoA5iAD3F6PAelAWxHcPT1kPGmUJLa4g46IHulGW0OIKMi+b4Rt+oiEIgiArEEQQ6Ig2CnOQRuuYbxSymCJDiyEIgixE22JeJAuCBJG2mJfJgiCJ2YTIhdYgSAJBBBBEEGkL5qA40bYgi8XI0GIIgiDgoweRQBABBBFAEAEEEbxqQZZ1awgySratMwgKS9bDGxAEQSEQREA1D3MQ1TxksdQthiAIsgBBBBCUitdjBUvI9VrKUBOAIAIIIoAgAuy0J4AgAggigCCCooLG74s7/0z3Tlhp5zi4HSI/QYhF1swbJMpTh/gd3TbO2Iv7l4es9L/dHWNZmwNR/1i8cT/+cdVTUBA/o5WfDzTde8ZKx6xt4w9+tib5xi9lC4fhuTBMEHfkj3/M38hvyjX+4Od/UCb4bd74eUXjB39hgQF7fdP+oZWST5C4rx/n9kPePY43p3tPWcOHO9zB0911cYqZUOHLSFidn6Dni+rivnfTXzzkHYb3Ml5/sKNC/EQ+vYBaydmDeK+Rj+KvP/jm/WeyJM9Q3N347/0T9RrvQbr6dPfWnz+TrWfdh/scviUsMOPrz3QP+tHvmLnhptn16iPvENNnaU7lFDJYP5nu7ssCfz5+f/1E9zPRO/RJnWzeYS3nITnPbBrdhM9iYg6S09Tw1tH8p9YAshhBYUHD/fmCLeR4X+4lVAMEEUAQQYmCgnWB9SDNG+uMEMQy1u+/rSI6zdvet8CC/C++qxN1eO2GWKVFFrTIc5Bcv9PbVnIrNB7ShO384m1HdccSaqPgtti39JYp2yqPhcINBS0oeJ+tunMJtVGsB5kFW0gz34OyL6E2qtnUsMwubSG3oKFesQkKtpCuHAr6yFU9eQm1kXuI7e7PFWwhjTHEvuOo7lhCbWAOIsg9xI535gvxkCWBu6pbQ3WTf4ipWSLcKzYfuv3TWALfOEuubg/VTqlZzDJ4Wkc1ab7F5BZ0eciGDt8EDQqWkCWBu6rbQnWTV9Dl4c74wZk8YCELlpAlgbuq20K1UyDNyyNXQcESsifwxOq2UO0U60ED/cceBH//SCiewJ3VLaHaKTIHyc3toBAPWRK4q7o1VDfIYgQQRJBzh5n+PcLth6qw/jQeiuf0P+05qluXUDul9iBLAi/z40qh1H3S9gSeWH3x9knHEriz+iLuk44ncGf1K7U/aFHILWig80xQsIUc78u9hGrJPcQ+PBpu8l/MBYVY6K14Tl8/Sa5uD9VOgTlo7j9byPG+3EuomNwbq58esf/G90+CgiVkSeCu6rZQ7eSeg9i3H66t7RiF+dBP4seZ+Q/Ik6onhOoG+6QJkOYJiu0P2jQLlpAlgbuq20J1U2CPIvuXJeKgYAlZErirui1UO2VvrFoSeGL1K7Sx6qs/L/8nKMRDlgTuqm4N1U3BQ8/GAebwx75h6J+xBG75bbBzCbX3IWQxAggiKFeQJYG3jbL3SftzCbx1VLCpEf0BVdsod4hZEnjbKLcHOc/3aQfIYgQQRABBBBBEAEEEEEQAQQQQRABBBBBEAEEEEEQAQQQQRABBBBBEAEEEEEQAQQQQRABBBBBEAEEEEEQAQQQQRABBBP8HaLuLr18ZerIAAAAASUVORK5CYII=" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m IRF6</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABdFBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZpA6ZrY6kNtHk69NTU1NTW5NTXlNTYNNTY5Nbm5Nbo5NbqtNjqJNjshZjuRkTU1kTW5kTY5mAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkNtmtttmtv9uTU1uTW5uTY5ubo5ubp1ubqtujshuq7Vuq+R5TU15TY6DTU2DTY6OTU2OTW6OTXmOTY6Obk2Ooo6OyP+QOgCQOjqQOmaQZgCQZjqQkDqQttuQ27aQ29uQ2/+dbm6rZE2rbk2rbm6rbo6rq26ryKur5Mir5OSr5P+1jk22ZgC2Zjq2kDq2kGa2tpC225C229u22/+2///Ijk3I5KvI/8jI/+TI///bkDrbkJDbtmbbtpDb25Db27bb///kq27k5Kvk/8jk///r6+v/AAD/tmb/yHn/yI7/25D/27b/29v/5Kv//7b//8j//9v//+T///9VQmKPAAAACXBIWXMAAA7DAAAOwwHHb6hkAAALpklEQVR4nO2diX8bRxXH5RKrbkrb1EowpIbiNuFySAmYgCFgB2iLwS2HEVcbk9gkYEFqVQqS7P3nmTezszurHc3bQ6s99Pt+2mj0drT2fD07by/ttDzgpFX2L1B1IIgBghggiAGCGCCIAYIYIIghvaCLp19stVqvfZLmI/urZ95pS/HapyKi37z0yPNe/KrV+sJ3Uv8eCyK1oPEtv21vJv9MRJCUYggab8nSdtpfZEGkFXSx33rjE/HybG3lIMWHSBCJEeU/k1r/DdFtXTvzhluiRiVJK6jfaqvCcI0KL26pbUYo+NurrZVve0ZM1PW7hSlI9JiIoHFl1SjSCuqGf3qPLNHWIfqS6Fh6QwlidkH/60Z7UL+yG5cipaDxVtt4d7G/IgbXZ6IvCUFtGmbaRsysN2MMEsX+yq//utacQdoXJNu3ejZckyO16A4X+9RsWhrGwk+ZglaunXmmoNOVV9OO+Qsll6C+3ydWDqQCIaLthbHwU+Em9nTtDTnihP5OW1c+pXhk060QKQWpnqJKWQSJxXJMNsYgVfE0RVJcKOmz2DVVoEYP1/QAGwoKYyHmIC3HaEOQX78xgsRoLHei/31bdAUxIH8g0vo+jUFaUBgzPxQKGm/JQTpYrOo/Xatqss++J31FJnJ/gA0FhbEZ+0GU6swxXO0WVDbZZzhYfXabjqg+kOXhLXUcZQgKYjMEiT64HUlyL24r29UER/MMEMQAQQwQxABBDBDEAEEMEMQAQQwQxJBS0GcS/8UoWEKuZQlCxTQ3PRDEAEEMEMQAQQyLFLS+vg5BjmXr66Ghhgga3T3xvMlu5+ZzHckhaH3dMNQMQYPOxol3ebjn9TZ1CIIMjq//XvSgyYMT1ZMkEBSBxIzuPfcmPzkS714W5PlR0o9raZ6VFwYraHBTCyJy9CB3Fov1rgW0PREpehCRS5AjZG5/NRM0tzGoqYIuD3fmk8WaKmhu+0HuUD3HoDhzEmTZpZ4OFdPc9JQiyLZDVMv9oDhzERQfcCAIgiBIMRdBGIPYZZYsBkEpQ8U0Nz0QxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMSAexQZ0IMYIIgBghggiAGCGCCIAYIYIIiheoL8Kx7FNDc9lROkr5kV09z0VE1QcNW1mOamB4IYIIihaoIwBrEhZLFkoWKamx4IYoAgBghigCAGXlCv06HvjSkgKM7xnvEGgmJcfnRkvIOgGJNdsYnJTpT3C3Xz4DM7Bf5EVtDou0dGLyq7B1VQkCQYh7IISnDP71ILSnLXeJ0F0XdWLz/OnuYTfe+gzoJoP+h6kMggiAGCGNILavwYFCWDoMZkMXqmsZz9wfX4/SyCitkPOj8/twgav+N+7vLw6qNYSTV+1vOsQ0Fd8kNyuu0Zdask6PzcNBQs738r6ewKqQWNt7a94Svkv++YoKAqgs7PI4aCdj788Btn8qHo1Ab/ZbylX1YOSAs9An1bBsMlV26xgsih6qA1FjR8/awrmvHuI++0rV+87pten7aObfHyr6uPqJHCE6kSS3SFmY+MDzexvhh8TlfP1Nw7NRV0Kpra1gORfhGeRFFtUv6GJWJDpUqUqEKSMUjPz+OcAaQqguxjkGzBS3Ijolko1As5uXh4IDqXFtSlSiRoS87/oWIJBCWhMoKsWUxKUE3tq2koxMt0D5KDrdzEaImnuliiHlQvQRH8pafUyn6b5Ij//RdjDBpe/bsafEQ28scgXYEfg0hszQVd/IJ6hOgtXTVQdO1Z7LRFSYsm4ZEhuWUmyGJSkCuBVV9QIUAQAwQxQBADBDEYgvQchK5dxSUWlAwIIv5Z0ekGvZIFif0qOhOkppBz/oLL2YPkZKC00+2aLG6JBcmzif3Vv7Sckw0usSCZ5odrzISwEMSc0l16Qa8wkzFCUG0EGQ+nDionvVIxu0p0DXUWZE6xsChBdTrUiDwg3xQ0fP2Xcj/Fv5SjTpaN331fnhlT76xVVNA/t2YVlIzqC1qLnEZVF3bGW6tndByu3lmr+EFHD2qKIDrjfFVfytEXdsTwcfHwwH9nreIHGyNo5hhkNrS77V/YkVfIgnfWKrygeo1BM7PYVPfwL+wIQdSD1Dtrlcb1oCgxQcYAQ5fGttrGBSBrlaUT5F/KURd2xu98U2cx/1JrrIofpItBDRY0C+6uIRcQxLAUgvIAQQwQxABBDLygxUxdk1DQ4uG/UDevKfyaKmgx02fVWND8pvCrJ4m+DlX8DHWN6EEEBMXAGMSwmCn8aiwI+0GpgCAGCGKAIAYIYmCbF37HJJGN2dM/F9Pc9MxZkPEtpSSCzKtbSyHI/J5bAkGuKeiLaW56IIgBghgwBjEgizFgP4gBghggiAGCGOYkyDI2Q5BRsGV3CAoL1v3D5RRkxxc0j1VVDvQgBoxBDMhiDNgPYsglyHwGFARJIm0xnwIFQQqzCZHniEGQAoIYIIgh0haMQXGibUEWi5GixRAEQRYgiAGCGCCIAYIYIIgBghggiEH+7oluUJh9W0LTBMWm8Et0i4vjxpamCZqewi/RTVKuW6MaJig2hR8ERYlN4ZfoKqEvaC6/YrlkmMIPY1DAcaejvggVnaEOWSxGrjkOGy4o7xR+jReUdwq/5guKAEGJeDlWsIRcyxKGqgAEMUAQAwQxzOUexSYDQQwQxABBDHkFje7I/ezJgxNR2jkOTj7S81BEpGOejlRPSqHzJ+LwZXRn7/JQlP67u2Osa7Mn6x/LD+7Ff9ziySmIHuBFjz+ZPHgiSseibaP3ftRRfOlnqoWD8NEfQhA58kbfpw/SIfDovR//zjdBJ1XoMSqje38SgZ5Yvmn/oQslmyB5Fo24cZ+6x/Hm5MFj0fDBDjl4vLshn6gjVXgqElan5xF5sro8yzT56X3qMNTLqH5vxw/Rc4v0CkolYw+iXqNe5V+/9+W7T1RJPZBp9+Z/7p74y6gH6eqT3et//Fi1XnQf8jl4S1oQxjee6B70vd8Ic4NNs+uVR9ZNTD+UaqKGkN7GyWR3TxXo/ejOxonuZ7J36GdYiXFHtJxCapzZNLoJjWJyDFLD1OD60fRPLQFkMYbcggZ70wVbyPG5zGtYDBDEAEEMBQoK9gU2gjRv7DNCkMhYv/2qH9Fp3va5JRbkff62TtThoypjlZZZ0DKPQWr/Th9bqaPQeEgTtvPzrzuqO9ZQGjmPxb6ij0zFUXksFB4oaEHB52zVnWsojXw9yCzYQprpHpR+DaWxmEMNy+hSFzILGugdm6BgC+nKoaCfu6rPXkNpZN7EdvemCraQxtjEvuao7lhDaWAMYsi8iR3vTBfiIUsCd1W3hsom+ybmjxLhWbHp0I0fxhL4zeezq9tDpVNoFrNsPLUDZxQZMgu6PBSbDh2CBgVLyJLAXdVtobLJKujycGd077m6YKEKlpAlgbuq20KlkyPNqytXQcESsifwmdVtodLJ14N6+o/dC/7+kVA8gTurW0Klk2cMUofbQSEesiRwV3VrqGyQxRggiCHjCTN9P8KN+35h43E8FM/pf3jgqG5dQ+kU2oMsCbzIH1cIhZ6TtifwmdWX75x0LIE7qy/jOel4AndWb9T5oGUhs6CezjNBwRZyfC7zGhZLjjvMBpt0x1xQiIXeiuf0jZPZ1e2h0skxBk39Zws5Ppd5DQsm88HqR0fiv9Hdk6BgCVkSuKu6LVQ6mccg8dsPOp0dozAd+kH8OjPdQD6r+oxQ2eCcNAPSPEO+80GbZsESsiRwV3VbqGxynFEU/4pEHBQsIUsCd1W3hUqn6INVSwKfWb1BB6ue/+elf4JCPGRJ4K7q1lDZ5Lz0bFxgDm/2DUP/iCVwy73BzjWU3oeQxRggiKFYQZYEXjeKPiftTSXw2rGAQ43oDVR1o9hNzJLA60axPcj5fZ96gCzGAEEMEMQAQQwQxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMTwf8F5/BkMBtd5AAAAAElFTkSuQmCC" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m IL20RB</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABblBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZpA6ZrY6kLY6kNtHk69NTU1NTW5NTXlNTYNNTY5Nbm5Nbo5NbqtNjqJNjshZjuRkTU1kTW5kTY5mAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkNtmtttmtv9uTU1uTW5uTY5ubo5ubp1ubqtujshuq7Vuq+R5TU15TY6DTU2DTY6OTU2OTW6OTXmOTY6Obk2Ooo6OyP+QOgCQOjqQOmaQZgCQkDqQttuQ27aQ2/+dbm6rZE2rbk2rbm6rbo6rq26ryKur5Mir5OSr5P+1jk22ZgC2Zjq2kDq2tpC225C229u22/+2///Ijk3I5KvI/8jI/+TI///bkDrbkJDbtmbbtpDb25Db27bb///kq27k5Kvk/8jk///r6+v/AAD/tmb/yHn/yI7/25D/27b/29v/5Kv//7b//8j//9v//+T///8pDI2eAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAMgElEQVR4nO2djV8cRxnHjxgoprapgUQ0xRfa1JdCU6MYpUYh2hal9QXPaAsmYKKgAXrU4479751nXnZnd2fn2Rd2915+30+bm3t27o75MjvP7AtznQB46bT9A4w6EMQAQQwQxABBDBDEAEEMEMRQXNDw+dc6nc5rnxV5ydbsUXB4bc88f/Fqp3P9cyp9eU+8FZUOOwr7yc3PC/9wV09hQRf3dFNu5X9NQtCZfP3MtnizFSrRBiMo9iRS2h5FBQ23Ojc/Ew8v5qmBuV9kCxpuzXwUDJ905sjE149ESbjWW4d/jZ6I8mrBn64Gigo6o3YR5/NUMLuIUPC3VzszPwqsmKirGxgXdD5/S8dkPL71YiWypSq2TFFB3Vi3P5/XO4voWJJVK5YlyMTmhI0585566/+6Vg96UqCT1kZBQbpJGrGzvCuGXNGXhKA54aAzZ8XsemlBh8Ke7iKHQoRzDBqBDlRSkGzB7JFp4LW94RY1jLZGsehVDkHPaVdNC5p54yiIBMknLVNJ0JluyMy2GU3mgigWvSolSIy/VN8WRFufz9+USnTV5x27G7ZEQUGqp6hSeUFih5Rm5Ihsj0Fn0ltYtTsCeb54FntDFajR5/MmD0eColhEQtDFCg1T1quirXKMHmtB4pcvJ9H/fqdDaVrMaIIvt2gMMoKimP2imKBu2Lu6NA+ypj7CnRykzS422/4gVH4mfX3bzIlFAyNBUSyZ5k1qUvMAma+smbTWd6jmj51w1tA2JQ5WX7xDx0wfyfK50PWVd+1dLIplCbIPJGhSKY/KjCDRQ1dN1ddGYBqEo3kOCGKAIAYIYoAgBghigCAGCGKAIAaPoP7a4uJmEAw2Fu++bO4HGjWyBQ0+2A367+1e7mwGx8smeCLRD1bBEfJtyxFqpPU5yBbUIyv7m4NHB0H//oEOQlAc0Yv6D17KzhQErwia+aFGCa+gy531oHfXCCLQg2wGG+tiqH4AQRn010QOC8qOQQsLCxMuSPmRu1mJLLawEBmaUEHHi8RmfB6Ut3kLC5ahCRXkBIIYIIghd4unYAxykr/Fk5/FnBRo8eTPg1xAEAMEMUAQAwQxQBADBDFAEEiAHsQAQQwQxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMQAQQxeQfKuDrpGv4Q7zFz0pJj9TSsEQRb7t/8oetDlJ7tWDIJi0C422JA3wQS4R9EBCeq/t2v1IvSgGOGtd+E4BEExICiHILoN+PJTpHknZh50O0xkEMQAQQwQxABBDBDEAEEMEMQAQQwQxABBDBDEAEEgAXoQAwQxQBDDmAk6cVPdQyaWoK5e49C3gOo0C+rSypqd1WC45VmleIoFDbdWg+CMlns88ywQOsWCLlZCQZ4lZiEIghI0KajQH0L7Pn9CBRX7U3rf57ckyKxR7Pu+jwqCmMUYkvJ8n9+KoHzUJii1zff5xOnpqUPQxVv+JZbPb+ylSoTM4U5cgv7ZfA+yN+YSdHpqGwq3n/0w71dNlBDUVZNotWq6RF56LrU8jmubbwwqKuj0NGYobOfjj79/JBdFl4OpehCDh36Y2SYtdLSwKoPRluv3eEHy2zFu0buauvIWvCtcJtCTxa5I0PnrR93V4OLtveBwzjwE3Vs096U+cjb7rxt7tBcKT6RKbDEVMleHt2bStAr97JNooX11C15DywQWHIMyBB2Kps6Zgcg8CE+iqHYpvWOJ2LlSJUpUIccuJtO86H32PizvMGtomUDhJ0etk5ihk4Qg+Q1D1+RORF+8oB7IyfDxtuhcgRbUpUokaEV+/4eK5RUUG+PM/UGjswpeWNuZxaQE1VR9QCkekj2IGqp2MdoSqC6Wuwd9NZYl4z1otATF0FsPqZVncyRH/K8frDHo/Mbf1eAj2qnHIFOBH4OyBI3WUqVeQcPfUI8QvaWr0lfXncUOO5S06Et4ZEjumTmyWJagkssEttKD6sB7qHG186AxF5QPCGKYakHmO3d83zg0zYLMl3nFv8bQ+QNOpSCZxSQ45WoDQQyx62LyEbtYDGuQPsMg7QBpnmH0BOnzQr7Pb0mQPrZreZA2Z858n9+OIBqk5VF/q4LCc6++z29FkErzlMJ8gmpHC8raHDmxTnCHgvJeqciuEn8HIjkPGm7NNdCDHJc3CvYg+xJJM4LMPEifXqpVkOsCWbExKHaRzRZ0/vpv5URFX8pRrbl4+0N5Zkw9c1ZRQUfj7UFafcPwcKtuQelrPAl7VpsLC5qPnUZVF3YuVmaPaL9Qz5xVdNDXg3LRgKBc86BsQXTG+Ya5lGMu7IjhY/h4Wz9zVtHBXIJqH4OuQFDmGGQ3tLuqL+zIK2ThM2eVkRLkHYPyzqQzsliie+gLO0IQ9SD1zFlltAR5slhuQTFSgqwBhi6N0eQlvADkrDJigmo6FrME6Us56sLOxVs/MFlMX2pNVdHBdIaaKEFZcHcN+YAghuZuwWtRUBUamAflGJGnWlCenD7NgnLNCsdBUOYYVHGZwKsV1Dx8D6q4TODEC6q8TOCVjkHNwwqqvkxgvrsPRxVWEJYJzANWwWOAoGywTCAHlgksAgQxQBADBDFAEAMEMUAQAwQxQBADBDFAEAMEgQToQQwQxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMRQRlC0thEEJdF+jKHKgrLXfaunucXhBSWXx7HXV6sqyLNyYK2tLgB/h1lymcArFORbe7LeZueHv8MsuUQXBMVJLxMo/VzJhzN/wTsS5Lo/KLEK3tVlsUkYg+pdJnACsthoLRPYPDmy2CgtE9g8xedBEOQDghjY5uXLcNMiyF7ROYzkOU6bEkH2ms5BFMkxy54OQbYN/QhBGYJMAYLcghKl4jYmU9BJvAOFxRI2JlRQaMO5Z0FQ3FRqz4Igu+DYsyCIafF0Cpo+0IMYIIih1C72SqrgCPm25QyNAhDEAEEMEMSANM8AQQwQxABBDFUF9dfkwhWDRweitL4fruYlv9xubXHRXt9LXaalBYnuvhQbNy93ROm/G+vWey0fy/r78oWb6Y9rnoqC6O4huvY6ePRMlPZF2/rv/3xR8Y1fqRb2ouuOQhA5Cvo/oRfSmjL993/xB22CVimia7j9B38RgWOxfdn9oY1STpBcloq485C6x/7y4NFT0fDeOjl4urEkL+dLFYGKRNXpZohAVpfLNg1++ZA6DPUyqn+8rkN004R5g1Yp2YOo16hH+ds//ub9Z6qk7gbZuPuf+wd6G/UgU32wcfvPn6rWi+5DPntvSgvC+NIz04N+/Dthrrdsd732KLuLmTtiBmoIOV46GGxsqgI9768tHZh+JnuHuYFGjDui5RRS48yy1U1oFJNjkBqmerd3k5/aAshiDJUF9TaTBVfI87rS79AMEMQAQQw1CgrnAkthmrfmjBAkMtbvv60jJs27XjfFgoIvvmsSdXSfbKrSNAua5jFIze/MsZU6Ck2HDFE7v/iep7rnHVqj4rHYt8yRqTgqT4WiAwUjKHydq7r3HVqjWg+yC66QIdmDir9DazRzqOEYXcaF0oJ6ZmITFlwhUzkS9Gtf9ex3aI3Su9jGZqLgChmsXew7nuqed2gNjEEMpXex/fVkIR1yJHBfdWeobcrvYnqUiM6KJUN3fpZK4HdfZld3h1qn1izm2HnGjmbS/BhTWtDljth16BA0LDhCjgTuq+4KtU1ZQZc76/0HL9UFC1VwhBwJ3FfdFWqdCmleXbkKC46QO4FnVneFWqdaDzo2v+zj8PcfC6UTuLe6I9Q6VcYgdbgdFtIhRwL3VXeG2gZZjAGCGEqeMDP3I9x5qAtLT9OhdE7/0yNPdec7tE6tPciRwOv8uFqo9Zy0O4FnVp++c9KpBO6tPo3npNMJ3Ft9os4HTQulBR2bPBMWXCHP60q/Q7OU3sU+2O0t0x1zYSEVejOd05cOsqu7Q61TYQxK/OcKeV5X+h0apvTB6ie74r/+/YOw4Ag5ErivuivUOqXHIPHT9xYX161CMvTT9HVmuoE8q3pGqG1wTpoBaZ6h2vmgZbvgCDkSuK+6K9Q2Fc4oin9FIg4LjpAjgfuqu0KtU/fBqiOBZ1afoIPVQP966Z+wkA45ErivujPUNhUvPVsXmKObfaPQP1IJ3HFvsPcdWu9DyGIMEMRQryBHAh836j4nHSQS+NjRwKFG/AaqcaPeXcyRwMeNenuQ9+99xgNkMQYIYoAgBghigCAGCGKAIAYIYoAgBghigCAGCGKAIAYIYoAgBghigCAGCGKAIAYIYoAgBghigCAGCGKAIAYIYoAghv8Dh7RABVFNbqAAAAAASUVORK5CYII=" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m IGSF3</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABd1BMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZmY6ZpA6ZrY6kNtHk69NTU1NTW5NTXlNTYNNTY5Nbm5Nbo5NbqtNjqJNjshZjuRkTU1kTW5kTY5mAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkLZmkNtmtttmtv9uTU1uTW5uTY5ubo5ubp1ubqtujshuq7Vuq+R5TU15TY6DTU2DTY6OTU2OTW6OTXmOTY6Obk2Ooo6OyP+QOgCQOjqQOmaQZgCQZjqQkDqQtraQttuQtv+Q27aQ2/+dbm6rZE2rbk2rbm6rbo6rq26ryKur5Mir5OSr5P+1jk22ZgC2Zjq2kGa2tpC225C229u22/+2///Ijk3I5KvI/8jI/+TI///bkDrbtmbbtpDb25Db27bb2//b///kq27k5Kvk/8jk///r6+v/AAD/tmb/yHn/yI7/25D/27b/5Kv//7b//8j//9v//+T////14UQJAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAMVUlEQVR4nO2dCX/cRhmH123jmoSWUDvBkJrDactVh5SACRhc7HAVDG45wpajMYk3odjgeLubPawPz7wzGmkkjebVsVpprf/za7Pjd0frncdzSCNp1PGAk07dX6DpQBADBDFAEAMEMUAQAwQxQBBDfkHTp5/vdDqv/TPPJvtXTuJbTv92rdN5+TsU73UULz3yvKciev0k95eqjtyCxm/7xXkj+zZKUGTL6b5K0xuGoF5gqinkFSQKdl1UgemzlaWDHBsJD9Et+50rn3jei/3OlhAUCBlviuj0Ywo2hbyC+p1llRiuUOKFqBWvfSIV/P1aZ+m7nhETef2CSkHRLXvK73hz2RSkM+SonVWTV1A3UpjhCrUIUVbdYraMWExQdMu+0URjgqbdHJWzcnIKkn/xgOn+0rue90zUCCFo+UT0IMtGzMx35SS6ped93Fn6wm/+I5O9jtHziE9a+m2RklREMUE91b/6bUHUgOk+FY/eDWPhVoagnu6ZvU9/JaoayYwKGt++1ln8GuQXs+8PP0sHapya7i97YSzcyiqI4s/eoZYW74NebF5pzkCfU5CqKSqVT1B0S/OdhCDdgzeC/KPYF1WCijZc0eNxKCiMhfijmLHlWFcS6rpDQWpoXGhBog+Vu8KfvtOhWrH0Ae3MUB+kBYUxcyO1HxRu6XU712nv4OlKpAaJPKJTetpZ3CZm7A+/Igdyf9c4FBTGYsN8dMvxZtgSjSbWX4k3z7opcLAqelaxJ/iBTA9FoV9+12xiYSwuKLqlPBZbomoU6YOGei+zKeBongGCGCCIAYIYIIgBghggiAGCGCCIAYIYcgp6LvFfjIQl5HovQ6ia4uYHghggiAGCGCCIoTJBq6urEOQo8epqaAiCkqHVVcNQmwRlxRdUzYfPFdQgBvRBDBjFGLAfxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMTkGju8eed7q2trZ+7EcgyGQgxRztGSEIMji68SdRgy4+fGjEICgCNbHJjmhishK9KpjPl2oSrKDR9x4atQg1KILspImgH4KgCBCUQdDg1pl38RGGeSt6P+hGMJBBEAMEMUAQAwQxQBADBDFAEAMEMUAQAwQxQBADBIEYqEEMEMQAQQwQxABBDBDEAEEMEMQAQQwQxABBDBDEwJ84nOys3TrTkVKCct2+UX3Rs8FeQHVxuOedbuhQGUH5bgCaQ9kzwV5ANdk9Di9iKCMo5y1kcyl9Bvjrg+6deZOf0sn5khdQLehNiJmu7lCCiLpr0HM7M5ZikqMGhV+wvj6ogYJm1gfNZBRroKCLw+0ZjWKz2A+qUxCtdH3Qi6zmP+P9oAUX1KWVfGkV1a5jRfkWCxpvbvlu+o6nNrRdUI8Wq4UgE6OJ6UWcu8speb12C/INDVccfhok6Pz83CJofNu9FPXw6qNEipjupz3rZMGmO0w/hqHg/f63sz6z5JILOj+PGArK+eB33zyRi6JTP+q/jDf1y9IBaaEno2zJYPjOK2+zgmQnXaOg+GFIIUHD10+6W974rUc03PgvNDT3af39LfHy76uPqBUKT6RKvKMzpD5OKCrINYBVKyhxIFtIUE8UdVl3RPpFeBJJ1aT8hiViQ6VKpChDhiY2V0GJo1bzUL94HySfMPSSbES0x6teyMn0wYGoXFpQlzKRoE35gAEVa5Sg5LxHbkHWUUxKUEXtq50W8RKvQVRQ1cToHU9VsZnVoJlgmTrLOpv23I7/bo9K2V8mOeJ//8Xog4ZX/6E6n+HnDvw+SGdoVB+UrC65+yCroOkv6NuL2tJVw1fXPoqJA3IxaNGjdGRItsxMo1hHU/Whhk1Q2C35iQKCKqGW/SDb3GvEXtsFJUexIBXULtfvr0kQPZOJGpprZ31GgtJDDRbUl48qFLtZm3VOmDVX0HT/DU89HKzneHhV5YKa2wepYV4+3LLeCbOmjmJqRpEqD2YUTYwmtuU/H6xbaxNrrCCqPbILSt/r9totiPbM6Vl8zufntVpQCo1agaqJghq1AlUDBTVrBapaBdFAL/qhTvS0T7NWoAqdGJeKBIKynqlIzxL9BCIUJJ9i3V2Oz96XX4Eq/ci0RA0yLzaakyB5QpX+sRxqlFlgyTW3UVhQ5HI1U9Dw9V/Kp977p3LUZNn4rV/LmTH1kzWLCvpza1ZBquKQIMuedAlB1tmxCgWtRKZR1YkderI0lUr9ZM3iBx01KGxZUUFlV6CatyCacb6qT+XoEzuibNMHB/5P1ix+0CFIHs1LYk2s5ApU1QhK7YPMgna3/BM78gxZ8JM1Cy8oeOKy8xRrfkHV9EGpo1isevgndoQgqkHqJ2uWDIK8rjzImP3VHZWMYlESgowOhk6NbS4bJ4CsWbIIkmf1mUeZFxE0vx1FQ5B/Kked2Bnf/pYexfxTrYksfpBOBqULykCzBaXBXTXkAoIYWiGoDHWcWV1MQdmAIAYIYqhb0PypWVD6DUDVFDc/9a5AtQD3INZag1w3IVZT3PxAEAMEMdTbSZuzOhDUilFs1oKwHwRBEBQBghggiAGCGCCIAYKSNGppivnDX0A1syW6LqmgGS6PczkFuZboOj8/r+hrNQdWkGOJLvPGWtQghVmEyK3Z7RWU3gdBkCR9iS4IUqTvB6EPShIti7k4BARJ0ornqkoQZDY2iykIMrprW7cNQaEgc2CDIAiykFK8SAuDIMcohj6IKx5Gscwlbqeg9oEaxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMQAQQxFBGWb0miNoMQSXRknxVojKL5EV9Zp1bYISizRBUFREkt0+YIq/2YNgb8+KLFEF/qggKO1NXXhS3QFKoxiCcqsYXbZBZVdouvSCyq7RNflFxQBgjLxaiJhCbneyxhqAhDEAEEMEMSAc/MMEMQAQQwQxFBW0OiO3M+e7B6L1PZRMPlI97+IyJo5HanujKH5E3H4Mrqzd3EoUv/b2TY+a+NU5j+SG+4lf938KSmIbtik210mu09E6kiUbfTej9cUX3pflXAQ3uohBJEjb/QD2pAOgUfv/eSPvgmaVKHbZkb3/ioCp+L9DfsvnSvFBMlZNOLmfaoeRxuT3cei4INtcvB4Z13eQSVVeCoSZqf7zzyZXc4yTX52nyoM1TLKf7rth+g+Nf0BtVKwBlGtUa/yr3/65btPVErdgLdz6793j/33qAbp7JOdG3/5SJVeVB/yOXhTWhDG15/oGvT93wtzgw2z6tVH0Samb0KcqC7kdP14srOnEvTz6M76sa5nsnboexZFvyNKTiHVz2wY1YR6MdkHqW5qcONh/LfWAEYxhtKCBnvxhC3k2K7wJ8wHCGKAIIYKBQX7AuvBMG/sM0KQGLH+8FU/ood523YtFuR99nU9UIdLEyQytVlQm/sgtX+nj63UUWgypAnL+dk3HNkdn1AbJY/FvqKPTMVReSIUHihoQcF2tuzOT6iNcjXITNhCmngNyv8JtTGfQw1L77IoFBY00Ds2QcIW0plDQT93ZU//hNoo3MR29mIJW0hjNLGvObI7PqE20AcxFG5iR9vxRDJkGcBd2a2huinexPxeIpwVi4du/igxgN86S89uD9VOpaOYpfEsHJhRZCgs6OJQNB06BA0SlpBlAHdlt4Xqpqigi8Pt0b0zdcJCJSwhywDuym4L1U6JYV6duQoSlpB9AE/NbgvVTrkadKr/2KfB3z8SSg7gzuyWUO2U6YPU4XaQSIYsA7gruzVUNxjFGCCIoeCEmb4e4eZ9P7H+OBlKjul/3nVkt35C7VRagywDeJW/rhIqnZO2D+Cp2ds3J50YwJ3Z2zgnnRzAndkv1XxQWygs6FSPM0HCFnJsV/gT5kuJK8wGG3TFXJBIhN5Mjunrx+nZ7aHaKdEHxf6zhRzbFf6EOVP4YPXDh+K/0d3jIGEJWQZwV3ZbqHYK90Hi2w/W1raNRDz0w+R5ZrqAPC17SqhuMCfNgGGeodx80IaZsIQsA7gruy1UNyVmFMW/YiAOEpaQZQB3ZbeFaqfqg1XLAJ6a/RIdrHr+n5f+CRLJkGUAd2W3huqm5Kln4wRzeLFvGPpXYgC3XBvs/ITa6xBGMQYIYqhWkGUAXzSqnpP2YgP4wjGHQ43oBVSLRrVNzDKALxrV1iDn/T6LAUYxBghigCAGCGKAIAYIYoAgBghigCAGCGKAIAYIYoAgBghigCAGCGKAIAYIYoAgBghigCAGCGKAIAYIYoAgBghigCCG/wOUaGEt/SY2IwAAAABJRU5ErkJggg==" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m KRT80</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABX1BMVEUAAAAAADoAAGYAOjoAOmYAOpAAZpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZpA6ZrY6kLY6kNtHk69NTU1NTW5NTXlNTY5Nbm5Nbo5NbqtNjqJNjshmAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkNtmtttmtv9uTU1uTW5uTYNuTY5ubo5ubqtujshuq+RvZE15TU15TY6DTU2DTY6OTU2OTW6OTXmOTYOOTY6Obk2OyMiOyP+QOgCQOjqQOmaQZgCQkDqQkNuQttuQ27aQ2/+rbk2rbm6rbo6rq26r5OSr5P+2ZgC2Zjq2Zma2kDq2tpC225C229u22/+2///Ijk3I5KvI/8jI/+TI///bkDrbkJDbtmbbtpDb25Db27bb///kq27k5Kvk/8jk/+Tk///r6+v/AAD/tmb/yIP/yI7/25D/27b/29v/5Kv//7b//8j//9v//+T///+6Y1anAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAMDUlEQVR4nO2djX/bRhnHndDieYVtpXEXKHGX8bKUl2AYeKNb0jFGskE9xmhCF9NCzJbG2MUv0f//4d50OlknPSfZiiT79/1s8fmkU3Vf390j6Wyp5oFEakXvQNmBIAIIIoAgAggigCACCCKAIIKUgmbPvlur1V77e5oyvc0T/jJqbOx5vZrktSeeN2mpNzW+wmi3VrvxhK/4clcuLwfpBE12VZXupCgkBTE/R54WxJ2EBI0aPMFXkdlSaglIJWh2WHuDNZ7Zc1FZV4SgUWPziadb0+xzpXh2ePNcJLq1d7zZFzy3V/veuUyVglSCBrW6TIwaPOH3BVbLv32ntvFTz8hj6+6pUlzKoCG7jxLEmsm8IP46adVVjs4vnFSCuqGGr3sFa1iCPSMvLOhZTdVXCfpfd74F9VgLYn/uCEmRf6pA0ghS+66YHW6wOj1v8A+9Vj9ntasbeQa9zY/1oGWOQXIjShDrdIzbXLFYtZemE+dJBkGikjfP/ZpsnswOeXX50iDPKMbW//auqq8StHFbdSAt6OWu3OoKCRqotrBx5A8bdS/IM4r1WMNgY7RwJtQ9a7zhDzC+INb0PuEL6pUWJFuKTKURxEv15Pgu29bAH5K0IKWFjTxq+K7kGMQqdlsmeL1GDX8UDgQFeQZSihyWVefzx+h5Qb3AdiWjGBuNxUH0v99mTUD2ipeHfAzyBQV5RikpZdISB4pyyaQ1N0izpaKLsXddfhz0eTWPg4Ij6RsikKuD6kBQkDd3HOR5chjyh2/V44KWMtBHClU+kmY8f5ufSX0i0vz06VvvmF0syIsKElJ6uuXIxUFXer4rtcuAJg8rywDO5gkgiACCCCCIAIIIIIgAggggiACCCCCIII2gFwL1YqYsWQsvzK3GKYEgAggigCACCCK4RkFbW1sQlLBwayswBEHRhVtbhiEIgiAIsrCQIIxB5EJEsaUuzK3GKYEgAggigCACCCKAIAIIIoAgAggigCACTPsQoAURQBABBBFAEAEEEUAQAQQRQBABBBGUT5C6tp9bjVNSOkH+7FBuNU5J2QTp+cXcapwSCCKAIILcBdnmUzEGzbWINILWLIrZv9OB4yAIgiAbGaqZYQxaM0Gpo9jaCcq6MLcapwSCCAhB/WazuX2m3kBQlNMD4w0ERbj69LHxDoIiTDusi4lG9ArjWnaobCQLGv/ssdGK0ILs6HFoaQ4Sjx0h6EXy2UelBA3vXXhXny05zJvnZ1UXxI+D7upABkEEEESwJEGrNAaFWZagVYtimqUJWrHjIE0JBL2wsxQVdiCIwBDk38rYcrdRc+/WVlBX3yy8HrPueguatPyWM4i7TSgESSDIxOhi6q6q6GIhjEF6gEHaAsI8QWUFXV5eWgRN7iffCn906ySS4swO4/pNzViH9S7ezeKGoFIJurw0Denlg5+43sk8vaDuHXUn9W4FBunLy5AhXc9Hf/rRuRhNeUXUy6Tlv2wccS384Sh7IjNYcmOXFCTCvHBThTAfI2j0+nl3z5u8deL16v4L/+QH/Bb8e+zlX7dOeC9knrgqtsRfITY0rZagHqtq3R+I/BfmiSVll1Idi+WNpCqW4iusXBezj0HiIUPyWVz8EQzyhTuZPTpijcsX1OUrcUEt8QgQmbdqg7Q1igkJsqoDedjLXuZbEO8qsovxJZ5sYi4tiKZUgkKopT1ey0Gdy2H/qxdjDBrd+lIOPqNXj9QY5K/gOAaVnkRBs4/EE07uH3Vl+Orao1ivxoMWfw6PyBJ9xzWKuexdiVtQHkAQAQQRGIL8h8fFPldn3QWhBdnIS5DL3OA6C3KaXV5jQW7fT6iWIIN/LjxIpxcU+UJ+0i4UJKgrT8P04/Vi9i4XQdGfdCTtQjGCxANT+VlJ7GmJu6C0Y5DlR0FJu1CIIC6Hnbp9UYt/PGUKQSmjWBUEiUF61Eh6OmUaQekWVkhQ0jXv/ARVYQySgl5NmjXJUVD6KGYU0Cu7zlTErxLeQpkEpT0OMpscBEUFhQYtU9Do9T/4j7PlF8HkxbLJWx+LK2PynXUVmamurcUIqtLZfLygRugyqpzYmbRunvO5GvnOuorKTGhBDlRCEL/ifMufyvEndljvmD06Uu+sq6jMpC72R5331/K3oNgxyKxod09N7IgZMv3OuoqDIP9kdRmnGvkLiotic81DTewwQbwFyXfWVWhBXk+eYgzEcFZ+QSEigowBhk+NterGBJB1FQdBXpeb6cbPG1ZKkJrKkRM7k/s/9qOYmmqNrKIy+WRQrCCvu/llI/77ZXkISrjrQBZBcVDfGkoiFMW6kQg/7TTvXYT3bomCku5bYd/dggXNDudO5a+OD7z+Tnjvlico8c4n9t3NJmgRwsdBs8PwCDR9eOaNf5HXjQWqJch6JD3ev/Cm7/GfHObxs3AlKE2RolvQPPw3q1JQsHcVHIMWgfjd/H6ugtJHseuHuPNCrmNQ4sL8q+4Gce+O43aOUWwFBOV8HLQCgkJAEAEEEUAQAQQRQBABBBFAEAEEEUAQAQQRQBABBBFAEAEEEdDVNH9GCkHRapo/JIWgaDVDP0WGIAiKAEEEZDUxBlHVXPMotpbgQJEAggggiACCCCCIAIIIIIgAggggiACCCCCIAIIIIIgAggggiACCCCCIAIIIIIgAggggiACCCFILCqa+dMo2G7YugiIPww4mT3XKOp+6LoLmH4YdTL/rlH1Gfk0ERR6GDUFhIg/DVjbMVJC1kqR9GDbGIM1psyl/rRp+1jOiWITlPy18dQTl8jDsVRKUy8OwV0pQCAhy4pVoypK1jIWlAIIIIIgAggjwLVcCCCKAIAIIIlhM0PiBONCePvzHg2b7NLj4KO5axRY2zQuS8oZW/AoKO4EZPzi4OmaprzttY2M7fbn+qSh5EPn3CmAhQfw2i/wmVdPff3DgnbKajfe/7jQl218dywoOgzs0MUHckTf+FS/JT4LHv/zdn5UIflmF3+1qvH/Bl/TZCjv2f/ZaySBoqh085bVl9Zi+/+6ZN2xLA51ted8zoYKv/v5vjPX5bePYIpYS15lEyVPRavj6/bZaKDz5WyiULC2INxqVEB9+//s//0C2Aln9zr3/MEFXugV9468/7dz9y2dnfiPhRoc/FBKY8+2vdAv6hqsb7phtrzgydTF958CpHEH62087B+JVfuTjB6JJdPwxKLjTIBt3WMWDYWbHaCVsDGrLMUiOU8O7j73iQRQjWEzQ8CCSsmQtUDJxG9cCBBFAEEF+gvTRwJs6zJ+5lVwTQd5/f62PBtrePBDEUh/qo4H3IgEbgtZ6DJJHd+LU009ZsoL1AwcfupW0bKMoFjkXu6fPTH/w20iWeZrgC3Isad1GUSzQgkIpS1bAfAsiS5ai7Uiu5VTDMsxUhmyChvq4RqcsWcHqgSDHktFtFEW2LtY5mE9ZsgKMLuZW0rKNosAYRJCti522IylLliWSO5Y0UkWTsYvpS2F+Kpr1NBrJL9xKmqnCyTOKWXpR9cAVRYJsgq6O2fGuOAPVKUuWLZI7lgxSRZNJ0NVxe7x/Ib7A6KcsWbZI7lgySBVO1jCvpq10ypIVE8kdS1Z5Xkx9wH39Uff9dhDKskVyx5LGNoom8xikzrV1ypJlieSOJY1U0SCKEUAQQZYLZg+f+gfLfurNdyNZoZiuEo4lQ9somjxbkCWSV488r0nbI7ljSa8kZyh5XpOORHLLZef1viYdjeSOJUvRdiSIYgTZBPV1kNEpS9YCJRO3ca1k/YbZcEd8X06nolnTaCQ/cytppgon6xikwo9OWbIWKJm4jesl28nqp4/Zf/yrrDplybJFcseSQapwso1BbNeHzWbbTEWyprZr0k4lQ6miwTVpAoR5ggWuB+2EUpYsWyR3LBmkiibrFUX2l0dhnbJk2SK5Y8kgVTg5n6xaIrljSa8kQ1fGI+mdyB9LliWSO5Y0/hTNIlPP5vxy8F1fnbV9Fo3kjiWNbSy7wmlBFCOAIIJcBVkieeXI+Zq0NxfJq0f+pxrhL1BVjly7mCWSV45cW1Di730qAqIYAQQRQBABBBFAEAEEEUAQAQQRQBABBBFAEAEEEUAQAQQRQBABBBFAEAEEEUAQAQQRQBABBBFAEAEEEUAQAQQRQBDB/wFOTSFkRsINfgAAAABJRU5ErkJggg==" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m SULT2B1</code></pre>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABfVBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrYzMzM6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZmY6ZpA6ZrY6kNtHk69NTU1NTW5NTXlNTYNNTY5Nbm5Nbo5NbqtNjqJNjshZjuRkTU1kTW5kTY5mAABmADpmAGZmOgBmOjpmOpBmZjpmZmZmkLZmkNtmtpBmtttmtv9uTU1uTW5uTY5ubo5ubp1ubqtujshuq7Vuq+R5TU15TY6DTU2DTY6OTU2OTW6OTXmOTY6Obk2Ooo6OyP+QOgCQOjqQOmaQZgCQZjqQZraQkDqQtraQttuQ27aQ2/+dbm6rZE2rbk2rbm6rbo6rq26ryKur5Mir5OSr5P+1jk22ZgC2Zjq2kDq2kGa2tpC225C229u22/+2///Ijk3I5KvI/8jI/+TI///bkDrbtmbbtpDb25Db27bb2//b///kq27k5Kvk/8jk///r6+v/AAD/tmb/yHn/yI7/25D/27b/5Kv//7b//8j//9v//+T///8KsvtBAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAMY0lEQVR4nO2diX/cRhXH123impSU1naCITWH24azDikBEzAE7LQcxZCWI2zLkZjEm1BsSLzd7R7W3955c0gjaTRP0kbHZn+/T5udfRppNV+/mTczkkadAPKq0/QJtF0AxAiAGAEQIwBiBECMAIgRADEqDmj66IudTufiv4rt9PeXO50Xv3sskr0X7kvTeGs1+iK+dbTI8ljkPnePMivbq/f0cbrLhc93RhUGNH5LF2S1wE7TfbXP+eNcgPoysXQQAurobL1O6wGJor4qnGf6eIXOP6/6nfPCBz7b72xnAVLHJoD0ufR+MP2YYKjt049oR/poP6C+OcXhCiU+E/508Z4s2z9e7ix9L7BsIu+23qunaI63lvMAGq6smm96u7SMt5a+v9V6QF27ROLEdV0wdWjbslmA+laF5AGZb5EHfUyHG//meNx6QPEzFHXhbdGiCl8SgJaPZRMR2Wx91Fn60u/+K5N5AfVkfYw3eHMDqKdaXFUXqJTTfSopbY1ssR0/+bVwLEKXF9AjWZcNoNeO7Z+vUzMB6uvTXzpQZaNaEdmS+04fX+2s5gQk2uPzVp/gkW752g9IeYpKFQVk2l29RbpaBiBRY1WlMtt109d+QKL8r6kElWa4YlrhCFBkizTe0iXvyk6OytCzY751EJlfVsZgHgGJv63sRH9yVdQB2WER3RtqgwygyGbt1ZV94emjFZFruCJL/0h2pNyAuqH7hVVMbZkDQFFP+pwM5DrGRIAimxXmTT9ZFtyOTLEoFfWDwu6z2a6PNA+ARAi/SmOx92V6KHC9+LZdxSKbBUiNxZb0kMo6gBOQNb7QyYvao+YD0GIJgBgBECMAYgRAjACIEQAxAiBGAMQIgBgBEKOCgJ5K6Q8r4TD5tuUwVVPc4gIgRgDECIAYNQNobW0NgDzb1tZCQgDkMK2tRYQACIAAyNYzAYQ2iN2GKMZsQz8IgADILQBiBECMAIgRADECoJhG14+CYLK7fuWJsQCQrcH6xlFwdmcvONk0JgCydHjpz8KDJreOlCdJAVBMBGZ040kw+fld8e0loZrOqkViAQ2uGEAkeFBMcQ8iAVBMI7RBPKCzOzuIYllCPwg9aVYAxAiAGAEQIwBiBECMAIgRADECIEYAxAiAGAEQIwBiBECMAIhRDYBy3C31/AAqIXm/nUlV/mvPXJV7UHTHpnVn4vPrQTMAsu9tBSAAyg3oaayGAZAniqEN4rZFfACI2QZAMwDSflZNcYsrBBQtptpJrPBnq3pApqWqpfQ5ZHlQculDlyoHFMa6CstcSHYV69oLYbq12IByCIAYVQ6ozW1QHlUPyB/Fnro1O4dMWYC6nc4qLWHoW4mvBkDeflCTgLq0smZnO5juexZjX2BA0/3tIOhTpO97gtkCAxpvhYAa7SgC0HMH6GR9nZ7ZsE8QgOwch3upE1xQQO7B6tkHd61vCwwoS5NdUcWkE7XhYZaIyenpqQPQ+A3/61CGF+6nUiQZw51iAY1+cNfyotZ40OmpTSjc3v9O3jfqlADU1ZXLEcXCdqgtgE5PY4TCct7+/beO5aLosjFVH6Lx0B9LB4SFRgvb0hhtOfcWD4gmO+Qa4nMMaPjKcXc7GL95P+gtm4+gu0p9X/KR/vn/XLhPtVBwIlRii8nQYQHJKCbXRI8DoufFzj5sW5jPANQTRV02DZH5EJxEUlUpXbGEbahQiRRlyFHFFCBalj7dD7oUBrK2AHK3QfINQy/ISkSvF1AfxGR6+0A4lwEk2xICtCXfQ6Bs+apYENbYDLUGkDOKSQiqqHpAKT6SHkSeoKoYbQmUi+VrpNXbLKb7cwEoJr21R6XsLxMc8b/+sNqg4YV/qsZn+IUD3QaZDHwblE+tBjR9l/6ywlt0PO66o1ivQ0GL/ECaZM3MEcV0GzTPgCoRADECIEYAxKh1l55bDAge5BIAMQIgRgDEyNGTxlUNWwDEqDWAknfA+n5/EQGl7qH2/X6LAVUlfd8Um6/1gKryoPQi5b7f13ukAOW9UpGdJX6EGKBmhxqFAdm37dcDKJ+qAlS0DYo9+GEDGr7ynnxrrb6UoybLxm/+Vs6MqW/OLMromG+uDFCh5w6eFo1i2YBWYtOo6sIOvWuaGg71zZlFG/NUMW8TnbvExZ5ccZjKAqIZ5wvmUo65sCNGCdPbB/qbM4s25qlifdeL0IsCKvjsU2FAmW2QXdDutr6wI6+Qhd+cWQoACrqeuzhbAygriiXcQ1/YEYDIg9Q3Z5YigJ5FmK8eUEwpQFYDQ5fG6HpxeAHImaVuQFW3QSwgfSlHXdgZv/FtE8X0pdZUFm1MXxSsrKNYNIpVOVjl7hryqaI2qCCNuQL0TKJYuwDNonQ/yMdnkQHlEwAxWmhAPQqIoqL5boNsGlD9igBR4zzdp5Fdi5/2qV/W0z6rdNPatvCkFj/tU78SjyJQCGvzVY36FQcknScBCCuSK9HsIzVByafD8WYWI+E9sglK3M+ItyKEood6hRMletJ4MwsjvJmFEd7MwghtECO8mYVTsX5Q9AjFwgCKiSue9RAOADmKZz/GBUAA5BAAMeKKV7QNyr44VE1xi+tZT7kWi2Key4vVFLe4Gp2T9l2grqa4xQVAjACIUbOXfdAGcabFi2LlOj0ABEAApARAjAAISmgmD7JX8IEHScXKYg3dHWNUALImf1yzHAAUAXLOkwEQADkUK0u8hgEQKV4Wu41GGySVVTxEMa0CJQYgAHIIgBgBECMAYgRAjACIEQAxAiBGAMRInnu+OzgWBlDqzSw57wFaGEDJN7PkvYtsUQCl3swCQHGl3syiAVV+Zi1RiTezoA0Kdbi+rh5CiL94BFEspda8maV+5XperD1vZqlfufpB7XkzS/3CUINRqbs7XkolHCbftpymNgiAGAEQIwBihDvMGAEQIwBiBECMZgU0uib72ZNbRyK1cxhOPtJaBMKybk9HqlUKaP5EDF9G1/bO7ojU/3d3rGNtnsj8h3LHvfTP1a8ZAdHiObT0wOTWQ5E6FGUbvfPTdaUv/1KVcBA9di8AEaNg9CPakYbAo3d+9idNgiZVaAmD0Y2/CcOJ2L7p/tFaVQ6QnEUjXb5J7nG4Obn1QBR8sEMMHuxuyNUsJIpAWaLstBZIILPLWabJL26Sw5CXUf6THW2iNUPMARpVSQ8ir1Gf8q9/8pXrD1VKLYaye+V/14/0NvIgk32ye+mvH6rSC/chnoPXJQVBfOOh8aAf/kGQG2zartecylYxsyDMRDUhJxtHk909laDvo2sbR8bPpHeY9WNEuyNKTibVzmxabkKtmGyDVDM1uHQ3+asNCFGM0cyABnvJhMvk2a/0EeoRADECIEYVAgr7AhthmLf6jAAkItYfv6YtJsy79ltgQMGn3zCBOlomLpVpkQEtchuk+ndmbKVGoWmTUVTOT7/pye45QmOacSz2VTMyFaPylCkaKBhA4X6u7N4jNKbZPMhOuExGSQ8qfoTGVM9Qw9G6zItKAxqYjk2YcJlM5gjQr3zZs4/QmEpXsd29RMJlMrKq2Nc92T1HaExogxiVrmKHO8lE2uQI4L7sTlPTKl/FdCsRzYolTZd/kgrgV55kZ3ebGlelUcxReeZOmFFkVBrQ2R1RdWgIGiYcJkcA92V3mZpWWUBnd3ZGN56oCxYq4TA5Argvu8vUuGYI8+rKVZhwmNwBPDO7y9S4ZvOgE/PHPgn//jFTOoB7sztMjWuWNkgNt8NE2uQI4L7sTlPTQhRjBECMSk6YmfsRLt/UiY0HaVM6pv/llie78wiNq1IPcgTwKn+uElU6J+0O4JnZF29OOhXAvdkXcU46HcC92Z+r+aBFUWlAJybOhAmXybNf6SPUqxnuMBts0h1zYSJlej0d0zeOsrO7TY1rhjYo8Z/L5Nmv9BFqVunB6gd3xX+j60dhwmFyBHBfdpepcZVug8TZD9bXd6xE0vTj9HVmuoE8K3uGqWlhTpoRwjyj2eaDNu2Ew+QI4L7sLlPTmmFGUfwrAnGYcJgcAdyX3WVqXFUPVh0BPDP7czRYDfSfl/4JE2mTI4D7sjtNTWvGS8/WBeboZt/I9O9UAHfcG+w9QuM+hCjGCIAYVQvIEcDnTVXPSQeJAD53qmGoEb+Bat5UbRVzBPB5U7Ue5H3eZz6EKMYIgBgBECMAYgRAjACIEQAxAiBGAMQIgBgBECMAYgRAjACIEQAxAiBGAMQIgBgBECMAYgRAjACIEQAxAiBGAMQIgBgBEKPPAakxlrMYry1fAAAAAElFTkSuQmCC" /><!-- --></p>
<pre><code>#&gt; [33mPlotting gene expression for gene:[0m MMP13

head(results)
#&gt; log2 fold change (MLE): HPV_status Associated vs Independent 
#&gt; Wald test p-value: HPV_status Associated vs Independent 
#&gt; DataFrame with 6 rows and 6 columns
#&gt;         baseMean log2FoldChange     lfcSE      stat    pvalue      padj
#&gt;        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
#&gt; A4GALT  264.6559      -0.859248  0.971880 -0.884109  0.376637  0.883436
#&gt; AAAS    241.9723       0.651588  1.092950  0.596174  0.551059  0.935317
#&gt; AACS    192.2601      -0.465598  1.141494 -0.407884  0.683359  0.961809
#&gt; AAGAB   516.4116       0.222419  1.069095  0.208044  0.835194  0.979410
#&gt; AAK1    484.0285       0.746603  0.707829  1.054779  0.291527  0.836475
#&gt; AAMDC    92.2931      -0.350702  1.276492 -0.274739  0.783517  0.975221</code></pre>
</div>
<div id="gsea-gsea-for-rnaseq" class="section level4">
<h4>GSEA: GSEA() for RNAseq:</h4>
<p>A GSEA analysis can be performed using the results from the DEA
contrast. HTG_GSEA will generate plots and Excel files for easy
interpretation.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a><span class="fu">HTG_GSEA</span>(res_tutorial)</span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a><span class="co">#&gt; [32mPerforming gseGO  analysis[0m</span></span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a><span class="co">#&gt; [32mPreparing gene list for GSEA[0m</span></span>
<span id="cb42-4"><a href="#cb42-4" tabindex="-1"></a><span class="co">#&gt; [32mPerforming gseGO Analysis[0m</span></span>
<span id="cb42-5"><a href="#cb42-5" tabindex="-1"></a><span class="co">#&gt; preparing geneSet collections...</span></span>
<span id="cb42-6"><a href="#cb42-6" tabindex="-1"></a><span class="co">#&gt; GSEA analysis...</span></span>
<span id="cb42-7"><a href="#cb42-7" tabindex="-1"></a><span class="co">#&gt; leading edge analysis...</span></span>
<span id="cb42-8"><a href="#cb42-8" tabindex="-1"></a><span class="co">#&gt; done...</span></span>
<span id="cb42-9"><a href="#cb42-9" tabindex="-1"></a><span class="co">#&gt; [32mCreating Dotplot for gseGO [0m</span></span>
<span id="cb42-10"><a href="#cb42-10" tabindex="-1"></a><span class="co">#&gt; [32mCreating Emaplot for gseGO [0m</span></span>
<span id="cb42-11"><a href="#cb42-11" tabindex="-1"></a><span class="co">#&gt; [32mCreating Ridgeplot for gseGO [0m</span></span>
<span id="cb42-12"><a href="#cb42-12" tabindex="-1"></a><span class="co">#&gt; [32mCreating Heatplot for gseGO [0m</span></span>
<span id="cb42-13"><a href="#cb42-13" tabindex="-1"></a><span class="co">#&gt; [32mCreating Treeplot for gseGO [0m</span></span>
<span id="cb42-14"><a href="#cb42-14" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-15"><a href="#cb42-15" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-16"><a href="#cb42-16" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-17"><a href="#cb42-17" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-18"><a href="#cb42-18" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-19"><a href="#cb42-19" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-20"><a href="#cb42-20" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-21"><a href="#cb42-21" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-22"><a href="#cb42-22" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-23"><a href="#cb42-23" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-24"><a href="#cb42-24" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-25"><a href="#cb42-25" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-26"><a href="#cb42-26" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-27"><a href="#cb42-27" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-28"><a href="#cb42-28" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-29"><a href="#cb42-29" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-30"><a href="#cb42-30" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-31"><a href="#cb42-31" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-32"><a href="#cb42-32" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-33"><a href="#cb42-33" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-34"><a href="#cb42-34" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-35"><a href="#cb42-35" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-36"><a href="#cb42-36" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-37"><a href="#cb42-37" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-38"><a href="#cb42-38" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-39"><a href="#cb42-39" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-40"><a href="#cb42-40" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-41"><a href="#cb42-41" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-42"><a href="#cb42-42" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-43"><a href="#cb42-43" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-44"><a href="#cb42-44" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-45"><a href="#cb42-45" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-46"><a href="#cb42-46" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-47"><a href="#cb42-47" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-48"><a href="#cb42-48" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-49"><a href="#cb42-49" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-50"><a href="#cb42-50" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-51"><a href="#cb42-51" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-52"><a href="#cb42-52" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-53"><a href="#cb42-53" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-54"><a href="#cb42-54" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-55"><a href="#cb42-55" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-56"><a href="#cb42-56" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-57"><a href="#cb42-57" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-58"><a href="#cb42-58" tabindex="-1"></a><span class="co">#&gt; [32mPerforming KEGG Analysis[0m</span></span>
<span id="cb42-59"><a href="#cb42-59" tabindex="-1"></a><span class="co">#&gt; &#39;select()&#39; returned 1:1 mapping between keys and columns</span></span>
<span id="cb42-60"><a href="#cb42-60" tabindex="-1"></a><span class="co">#&gt; Warning in clusterProfiler::bitr(names(original_gene_list), fromType =</span></span>
<span id="cb42-61"><a href="#cb42-61" tabindex="-1"></a><span class="co">#&gt; &quot;SYMBOL&quot;, : 2.9% of input gene IDs are fail to map...</span></span>
<span id="cb42-62"><a href="#cb42-62" tabindex="-1"></a><span class="co">#&gt; preparing geneSet collections...</span></span>
<span id="cb42-63"><a href="#cb42-63" tabindex="-1"></a><span class="co">#&gt; GSEA analysis...</span></span>
<span id="cb42-64"><a href="#cb42-64" tabindex="-1"></a><span class="co">#&gt; Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =</span></span>
<span id="cb42-65"><a href="#cb42-65" tabindex="-1"></a><span class="co">#&gt; minSize, : For some pathways, in reality P-values are less than 1e-10. You can</span></span>
<span id="cb42-66"><a href="#cb42-66" tabindex="-1"></a><span class="co">#&gt; set the `eps` argument to zero for better estimation.</span></span>
<span id="cb42-67"><a href="#cb42-67" tabindex="-1"></a><span class="co">#&gt; leading edge analysis...</span></span>
<span id="cb42-68"><a href="#cb42-68" tabindex="-1"></a><span class="co">#&gt; done...</span></span>
<span id="cb42-69"><a href="#cb42-69" tabindex="-1"></a><span class="co">#&gt; [32mCreating Dotplot for KEGG[0m</span></span>
<span id="cb42-70"><a href="#cb42-70" tabindex="-1"></a><span class="co">#&gt; [32mCreating Emaplot for KEGG[0m</span></span>
<span id="cb42-71"><a href="#cb42-71" tabindex="-1"></a><span class="co">#&gt; [32mCreating Ridgeplot for KEGG[0m</span></span>
<span id="cb42-72"><a href="#cb42-72" tabindex="-1"></a><span class="co">#&gt; [32mCreating Heatplot for KEGG [0m</span></span>
<span id="cb42-73"><a href="#cb42-73" tabindex="-1"></a><span class="co">#&gt; [32mCreating Treeplot for KEGG [0m</span></span>
<span id="cb42-74"><a href="#cb42-74" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-75"><a href="#cb42-75" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-76"><a href="#cb42-76" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-77"><a href="#cb42-77" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-78"><a href="#cb42-78" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-79"><a href="#cb42-79" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-80"><a href="#cb42-80" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-81"><a href="#cb42-81" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-82"><a href="#cb42-82" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-83"><a href="#cb42-83" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-84"><a href="#cb42-84" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-85"><a href="#cb42-85" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-86"><a href="#cb42-86" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-87"><a href="#cb42-87" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-88"><a href="#cb42-88" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-89"><a href="#cb42-89" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-90"><a href="#cb42-90" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-91"><a href="#cb42-91" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-92"><a href="#cb42-92" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-93"><a href="#cb42-93" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-94"><a href="#cb42-94" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-95"><a href="#cb42-95" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-96"><a href="#cb42-96" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-97"><a href="#cb42-97" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-98"><a href="#cb42-98" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-99"><a href="#cb42-99" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-100"><a href="#cb42-100" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-101"><a href="#cb42-101" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-102"><a href="#cb42-102" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-103"><a href="#cb42-103" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-104"><a href="#cb42-104" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-105"><a href="#cb42-105" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-106"><a href="#cb42-106" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-107"><a href="#cb42-107" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-108"><a href="#cb42-108" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-109"><a href="#cb42-109" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-110"><a href="#cb42-110" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-111"><a href="#cb42-111" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-112"><a href="#cb42-112" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-113"><a href="#cb42-113" tabindex="-1"></a><span class="co">#&gt; ! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</span></span>
<span id="cb42-114"><a href="#cb42-114" tabindex="-1"></a><span class="co">#&gt; [32mPerforming GO Enrichment Analysis[0m</span></span>
<span id="cb42-115"><a href="#cb42-115" tabindex="-1"></a><span class="co">#&gt; [32mPerforming GO Enrichment Analysis[0m</span></span>
<span id="cb42-116"><a href="#cb42-116" tabindex="-1"></a><span class="co">#&gt; [32mCreating Bar Plot for GO Enrichment[0m</span></span>
<span id="cb42-117"><a href="#cb42-117" tabindex="-1"></a><span class="co">#&gt; [32mGO enrichment results saved to &#39;go_results.csv&#39;[0m</span></span>
<span id="cb42-118"><a href="#cb42-118" tabindex="-1"></a><span class="co">#&gt; Picking joint bandwidth of 0.334</span></span>
<span id="cb42-119"><a href="#cb42-119" tabindex="-1"></a><span class="co">#&gt; Picking joint bandwidth of 0.371</span></span></code></pre></div>
</div>
<div id="tme-htg_tme-for-rnaseq" class="section level4">
<h4>TME: HTG_TME () for RNAseq:</h4>
<p>This function performs Tumor Microenvironment (TME) analysis, which
will be stored in <code>TME_results</code>. You can access each method
using <code>$</code>. This object contains results from three
methodologies.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a></span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a>TME_results<span class="ot">&lt;-</span> <span class="fu">HTG_TME</span>(<span class="at">outliers=</span> <span class="cn">NULL</span>, </span>
<span id="cb43-3"><a href="#cb43-3" tabindex="-1"></a>                      <span class="at">pattern=</span> <span class="cn">NULL</span>,</span>
<span id="cb43-4"><a href="#cb43-4" tabindex="-1"></a>                      <span class="at">counts_data=</span> counts_data_RNAseq, </span>
<span id="cb43-5"><a href="#cb43-5" tabindex="-1"></a>                      <span class="at">AnnotData=</span> AnnotData_tutorial, </span>
<span id="cb43-6"><a href="#cb43-6" tabindex="-1"></a>                      <span class="at">design_formula=</span> <span class="st">&quot;HPV_status&quot;</span>, </span>
<span id="cb43-7"><a href="#cb43-7" tabindex="-1"></a>                      <span class="at">remove_outliers =</span> <span class="cn">FALSE</span>, </span>
<span id="cb43-8"><a href="#cb43-8" tabindex="-1"></a>                      <span class="at">DEA =</span> <span class="cn">NULL</span>)</span>
<span id="cb43-9"><a href="#cb43-9" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Inicial gene number: 19394&quot;</span></span>
<span id="cb43-10"><a href="#cb43-10" tabindex="-1"></a><span class="co">#&gt; [32mTPM normalization performed and stored on tpm_counts.csv[0m</span></span>
<span id="cb43-11"><a href="#cb43-11" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt;--- This function is being optimised and we strongly recommend that you should set `source` as `local`....</span></span>
<span id="cb43-12"><a href="#cb43-12" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Number of genes omitted during TPM normalization due to their length not being available in Biomart:   1086&quot;</span></span>
<span id="cb43-13"><a href="#cb43-13" tabindex="-1"></a><span class="co">#&gt; Performing normalization.</span></span>
<span id="cb43-14"><a href="#cb43-14" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb43-15"><a href="#cb43-15" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt; Running epic</span></span>
<span id="cb43-16"><a href="#cb43-16" tabindex="-1"></a><span class="co">#&gt; Warning in (function (bulk, reference = NULL, mRNA_cell = NULL, mRNA_cell_sub = NULL, : The optimization didn&#39;t fully converge for some samples:</span></span>
<span id="cb43-17"><a href="#cb43-17" tabindex="-1"></a><span class="co">#&gt; patient_3</span></span>
<span id="cb43-18"><a href="#cb43-18" tabindex="-1"></a><span class="co">#&gt;  - check fit.gof for the convergeCode and convergeMessage</span></span>
<span id="cb43-19"><a href="#cb43-19" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb43-20"><a href="#cb43-20" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt; Running quantiseq</span></span>
<span id="cb43-21"><a href="#cb43-21" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb43-22"><a href="#cb43-22" tabindex="-1"></a><span class="co">#&gt; Running quanTIseq deconvolution module</span></span>
<span id="cb43-23"><a href="#cb43-23" tabindex="-1"></a><span class="co">#&gt; Gene expression normalization and re-annotation (arrays: FALSE)</span></span>
<span id="cb43-24"><a href="#cb43-24" tabindex="-1"></a><span class="co">#&gt; Removing 17 noisy genes</span></span>
<span id="cb43-25"><a href="#cb43-25" tabindex="-1"></a><span class="co">#&gt; Removing 15 genes with high expression in tumors</span></span>
<span id="cb43-26"><a href="#cb43-26" tabindex="-1"></a><span class="co">#&gt; Signature genes found in data set: 132/138 (95.65%)</span></span>
<span id="cb43-27"><a href="#cb43-27" tabindex="-1"></a><span class="co">#&gt; Mixture deconvolution (method: lsei)</span></span>
<span id="cb43-28"><a href="#cb43-28" tabindex="-1"></a><span class="co">#&gt; Deconvolution successful!</span></span>
<span id="cb43-29"><a href="#cb43-29" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb43-30"><a href="#cb43-30" tabindex="-1"></a><span class="co">#&gt; &gt;&gt;&gt; Running xcell</span></span>
<span id="cb43-31"><a href="#cb43-31" tabindex="-1"></a><span class="co">#&gt; [32mresults of the devonvolution will be stored in imm_epic.csv, imm_qti.csv and imm_xcell.csv [0m</span></span>
<span id="cb43-32"><a href="#cb43-32" tabindex="-1"></a><span class="co">#&gt; [32mAre they in the same order?[0m</span></span>
<span id="cb43-33"><a href="#cb43-33" tabindex="-1"></a><span class="co">#&gt; [32mHave to be true.[0m</span></span>
<span id="cb43-34"><a href="#cb43-34" tabindex="-1"></a><span class="co">#&gt; [32mEPIC[0m</span></span>
<span id="cb43-35"><a href="#cb43-35" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb43-36"><a href="#cb43-36" tabindex="-1"></a><span class="co">#&gt; [32mqti[0m</span></span>
<span id="cb43-37"><a href="#cb43-37" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb43-38"><a href="#cb43-38" tabindex="-1"></a><span class="co">#&gt; [32mxcell[0m</span></span>
<span id="cb43-39"><a href="#cb43-39" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb43-40"><a href="#cb43-40" tabindex="-1"></a><span class="co">#&gt; [33mAttention: plot_cell_fraction_Average_cell_fraction will work for all methods. For xCell, it will display Enrichment Scores instead of percentages.[0m</span></span>
<span id="cb43-41"><a href="#cb43-41" tabindex="-1"></a><span class="co">#&gt; [33mApply the filtering and normality test for each dataset[0m</span></span>
<span id="cb43-42"><a href="#cb43-42" tabindex="-1"></a><span class="co">#&gt; [32mChecking if All Values in Each Column Are Equal in dataset: quanTIseq [0m</span></span>
<span id="cb43-43"><a href="#cb43-43" tabindex="-1"></a><span class="co">#&gt; [33mIn dataset quanTIseq the following columns have the same value in all rows and will be excluded from the analysis:[0m</span></span>
<span id="cb43-44"><a href="#cb43-44" tabindex="-1"></a><span class="co">#&gt; character(0)</span></span>
<span id="cb43-45"><a href="#cb43-45" tabindex="-1"></a><span class="co">#&gt; [32mChecking if All Values in Each Column Are Equal in dataset: EPIC [0m</span></span>
<span id="cb43-46"><a href="#cb43-46" tabindex="-1"></a><span class="co">#&gt; [33mIn dataset EPIC the following columns have the same value in all rows and will be excluded from the analysis:[0m</span></span>
<span id="cb43-47"><a href="#cb43-47" tabindex="-1"></a><span class="co">#&gt; character(0)</span></span>
<span id="cb43-48"><a href="#cb43-48" tabindex="-1"></a><span class="co">#&gt; [32mChecking if All Values in Each Column Are Equal in dataset: xcell [0m</span></span>
<span id="cb43-49"><a href="#cb43-49" tabindex="-1"></a><span class="co">#&gt; [33mIn dataset xcell the following columns have the same value in all rows and will be excluded from the analysis:[0m</span></span>
<span id="cb43-50"><a href="#cb43-50" tabindex="-1"></a><span class="co">#&gt; character(0)</span></span>
<span id="cb43-51"><a href="#cb43-51" tabindex="-1"></a><span class="co">#&gt; [32mPerforming Shapiro-Wilk test for quanTIseq[0m</span></span>
<span id="cb43-52"><a href="#cb43-52" tabindex="-1"></a><span class="co">#&gt; # A tibble: 11 × 3</span></span>
<span id="cb43-53"><a href="#cb43-53" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                    shapiro_test      p.value</span></span>
<span id="cb43-54"><a href="#cb43-54" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                        &lt;list&gt;              &lt;dbl&gt;</span></span>
<span id="cb43-55"><a href="#cb43-55" tabindex="-1"></a><span class="co">#&gt;  1 B_cell                       &lt;htest&gt;      0.368       </span></span>
<span id="cb43-56"><a href="#cb43-56" tabindex="-1"></a><span class="co">#&gt;  2 Macrophage_M1                &lt;htest&gt;      0.000220    </span></span>
<span id="cb43-57"><a href="#cb43-57" tabindex="-1"></a><span class="co">#&gt;  3 Macrophage_M2                &lt;htest&gt;      0.275       </span></span>
<span id="cb43-58"><a href="#cb43-58" tabindex="-1"></a><span class="co">#&gt;  4 Monocyte                     &lt;htest&gt;      0.000420    </span></span>
<span id="cb43-59"><a href="#cb43-59" tabindex="-1"></a><span class="co">#&gt;  5 Myeloid_dendritic_cell       &lt;htest&gt;      0.000000124 </span></span>
<span id="cb43-60"><a href="#cb43-60" tabindex="-1"></a><span class="co">#&gt;  6 NK_cell                      &lt;htest&gt;      0.0837      </span></span>
<span id="cb43-61"><a href="#cb43-61" tabindex="-1"></a><span class="co">#&gt;  7 Neutrophil                   &lt;htest&gt;      0.0000000224</span></span>
<span id="cb43-62"><a href="#cb43-62" tabindex="-1"></a><span class="co">#&gt;  8 T_cell_CD4+_(non-regulatory) &lt;htest&gt;      0.00000221  </span></span>
<span id="cb43-63"><a href="#cb43-63" tabindex="-1"></a><span class="co">#&gt;  9 T_cell_CD8+                  &lt;htest&gt;      0.000309    </span></span>
<span id="cb43-64"><a href="#cb43-64" tabindex="-1"></a><span class="co">#&gt; 10 T_cell_regulatory_(Tregs)    &lt;htest&gt;      0.255       </span></span>
<span id="cb43-65"><a href="#cb43-65" tabindex="-1"></a><span class="co">#&gt; 11 uncharacterized_cell         &lt;htest&gt;      0.207       </span></span>
<span id="cb43-66"><a href="#cb43-66" tabindex="-1"></a><span class="co">#&gt; [32mPerforming parametric tests for quanTIseq[0m</span></span>
<span id="cb43-67"><a href="#cb43-67" tabindex="-1"></a><span class="co">#&gt; # A tibble: 11 × 3</span></span>
<span id="cb43-68"><a href="#cb43-68" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                    Test    p.value</span></span>
<span id="cb43-69"><a href="#cb43-69" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                        &lt;chr&gt;     &lt;dbl&gt;</span></span>
<span id="cb43-70"><a href="#cb43-70" tabindex="-1"></a><span class="co">#&gt;  1 B_cell                       t-test 0.107   </span></span>
<span id="cb43-71"><a href="#cb43-71" tabindex="-1"></a><span class="co">#&gt;  2 Macrophage_M1                t-test 0.475   </span></span>
<span id="cb43-72"><a href="#cb43-72" tabindex="-1"></a><span class="co">#&gt;  3 Macrophage_M2                t-test 0.120   </span></span>
<span id="cb43-73"><a href="#cb43-73" tabindex="-1"></a><span class="co">#&gt;  4 Monocyte                     t-test 0.218   </span></span>
<span id="cb43-74"><a href="#cb43-74" tabindex="-1"></a><span class="co">#&gt;  5 Neutrophil                   t-test 0.356   </span></span>
<span id="cb43-75"><a href="#cb43-75" tabindex="-1"></a><span class="co">#&gt;  6 NK_cell                      t-test 0.862   </span></span>
<span id="cb43-76"><a href="#cb43-76" tabindex="-1"></a><span class="co">#&gt;  7 T_cell_CD4+_(non-regulatory) t-test 0.348   </span></span>
<span id="cb43-77"><a href="#cb43-77" tabindex="-1"></a><span class="co">#&gt;  8 T_cell_CD8+                  t-test 0.0706  </span></span>
<span id="cb43-78"><a href="#cb43-78" tabindex="-1"></a><span class="co">#&gt;  9 T_cell_regulatory_(Tregs)    t-test 0.449   </span></span>
<span id="cb43-79"><a href="#cb43-79" tabindex="-1"></a><span class="co">#&gt; 10 Myeloid_dendritic_cell       t-test 0.291   </span></span>
<span id="cb43-80"><a href="#cb43-80" tabindex="-1"></a><span class="co">#&gt; 11 uncharacterized_cell         t-test 0.000153</span></span>
<span id="cb43-81"><a href="#cb43-81" tabindex="-1"></a><span class="co">#&gt; [32mPerforming Shapiro-Wilk test for EPIC[0m</span></span>
<span id="cb43-82"><a href="#cb43-82" tabindex="-1"></a><span class="co">#&gt; # A tibble: 8 × 3</span></span>
<span id="cb43-83"><a href="#cb43-83" tabindex="-1"></a><span class="co">#&gt;   Cell_Type                    shapiro_test     p.value</span></span>
<span id="cb43-84"><a href="#cb43-84" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                        &lt;list&gt;             &lt;dbl&gt;</span></span>
<span id="cb43-85"><a href="#cb43-85" tabindex="-1"></a><span class="co">#&gt; 1 B_cell                       &lt;htest&gt;      0.00112    </span></span>
<span id="cb43-86"><a href="#cb43-86" tabindex="-1"></a><span class="co">#&gt; 2 Cancer_associated_fibroblast &lt;htest&gt;      0.0204     </span></span>
<span id="cb43-87"><a href="#cb43-87" tabindex="-1"></a><span class="co">#&gt; 3 Endothelial_cell             &lt;htest&gt;      0.138      </span></span>
<span id="cb43-88"><a href="#cb43-88" tabindex="-1"></a><span class="co">#&gt; 4 Macrophage                   &lt;htest&gt;      0.000139   </span></span>
<span id="cb43-89"><a href="#cb43-89" tabindex="-1"></a><span class="co">#&gt; 5 NK_cell                      &lt;htest&gt;      0.0000287  </span></span>
<span id="cb43-90"><a href="#cb43-90" tabindex="-1"></a><span class="co">#&gt; 6 T_cell_CD4+                  &lt;htest&gt;      0.0491     </span></span>
<span id="cb43-91"><a href="#cb43-91" tabindex="-1"></a><span class="co">#&gt; 7 T_cell_CD8+                  &lt;htest&gt;      0.000000248</span></span>
<span id="cb43-92"><a href="#cb43-92" tabindex="-1"></a><span class="co">#&gt; 8 uncharacterized_cell         &lt;htest&gt;      0.0177     </span></span>
<span id="cb43-93"><a href="#cb43-93" tabindex="-1"></a><span class="co">#&gt; [32mPerforming parametric tests for EPIC[0m</span></span>
<span id="cb43-94"><a href="#cb43-94" tabindex="-1"></a><span class="co">#&gt; # A tibble: 8 × 3</span></span>
<span id="cb43-95"><a href="#cb43-95" tabindex="-1"></a><span class="co">#&gt;   Cell_Type                    Test   p.value</span></span>
<span id="cb43-96"><a href="#cb43-96" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                        &lt;chr&gt;    &lt;dbl&gt;</span></span>
<span id="cb43-97"><a href="#cb43-97" tabindex="-1"></a><span class="co">#&gt; 1 B_cell                       t-test  0.0879</span></span>
<span id="cb43-98"><a href="#cb43-98" tabindex="-1"></a><span class="co">#&gt; 2 Cancer_associated_fibroblast t-test  0.998 </span></span>
<span id="cb43-99"><a href="#cb43-99" tabindex="-1"></a><span class="co">#&gt; 3 T_cell_CD4+                  t-test  0.726 </span></span>
<span id="cb43-100"><a href="#cb43-100" tabindex="-1"></a><span class="co">#&gt; 4 T_cell_CD8+                  t-test  0.300 </span></span>
<span id="cb43-101"><a href="#cb43-101" tabindex="-1"></a><span class="co">#&gt; 5 Endothelial_cell             t-test  0.493 </span></span>
<span id="cb43-102"><a href="#cb43-102" tabindex="-1"></a><span class="co">#&gt; 6 Macrophage                   t-test  0.489 </span></span>
<span id="cb43-103"><a href="#cb43-103" tabindex="-1"></a><span class="co">#&gt; 7 NK_cell                      t-test  0.627 </span></span>
<span id="cb43-104"><a href="#cb43-104" tabindex="-1"></a><span class="co">#&gt; 8 uncharacterized_cell         t-test  0.164 </span></span>
<span id="cb43-105"><a href="#cb43-105" tabindex="-1"></a><span class="co">#&gt; [32mPerforming Shapiro-Wilk test for xcell[0m</span></span>
<span id="cb43-106"><a href="#cb43-106" tabindex="-1"></a><span class="co">#&gt; # A tibble: 39 × 3</span></span>
<span id="cb43-107"><a href="#cb43-107" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                    shapiro_test     p.value</span></span>
<span id="cb43-108"><a href="#cb43-108" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                        &lt;list&gt;             &lt;dbl&gt;</span></span>
<span id="cb43-109"><a href="#cb43-109" tabindex="-1"></a><span class="co">#&gt;  1 B_cell                       &lt;htest&gt;      0.0703     </span></span>
<span id="cb43-110"><a href="#cb43-110" tabindex="-1"></a><span class="co">#&gt;  2 B_cell_memory                &lt;htest&gt;      0.0999     </span></span>
<span id="cb43-111"><a href="#cb43-111" tabindex="-1"></a><span class="co">#&gt;  3 B_cell_naive                 &lt;htest&gt;      0.00428    </span></span>
<span id="cb43-112"><a href="#cb43-112" tabindex="-1"></a><span class="co">#&gt;  4 B_cell_plasma                &lt;htest&gt;      0.0141     </span></span>
<span id="cb43-113"><a href="#cb43-113" tabindex="-1"></a><span class="co">#&gt;  5 Cancer_associated_fibroblast &lt;htest&gt;      0.00000117 </span></span>
<span id="cb43-114"><a href="#cb43-114" tabindex="-1"></a><span class="co">#&gt;  6 Class-switched_memory_B_cell &lt;htest&gt;      0.107      </span></span>
<span id="cb43-115"><a href="#cb43-115" tabindex="-1"></a><span class="co">#&gt;  7 Common_lymphoid_progenitor   &lt;htest&gt;      0.0184     </span></span>
<span id="cb43-116"><a href="#cb43-116" tabindex="-1"></a><span class="co">#&gt;  8 Common_myeloid_progenitor    &lt;htest&gt;      0.000000151</span></span>
<span id="cb43-117"><a href="#cb43-117" tabindex="-1"></a><span class="co">#&gt;  9 Endothelial_cell             &lt;htest&gt;      0.000189   </span></span>
<span id="cb43-118"><a href="#cb43-118" tabindex="-1"></a><span class="co">#&gt; 10 Eosinophil                   &lt;htest&gt;      0.0000242  </span></span>
<span id="cb43-119"><a href="#cb43-119" tabindex="-1"></a><span class="co">#&gt; # ℹ 29 more rows</span></span>
<span id="cb43-120"><a href="#cb43-120" tabindex="-1"></a><span class="co">#&gt; [32mPerforming parametric tests for xcell[0m</span></span>
<span id="cb43-121"><a href="#cb43-121" tabindex="-1"></a><span class="co">#&gt; # A tibble: 39 × 3</span></span>
<span id="cb43-122"><a href="#cb43-122" tabindex="-1"></a><span class="co">#&gt;    Cell_Type                        Test   p.value</span></span>
<span id="cb43-123"><a href="#cb43-123" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                            &lt;chr&gt;    &lt;dbl&gt;</span></span>
<span id="cb43-124"><a href="#cb43-124" tabindex="-1"></a><span class="co">#&gt;  1 Myeloid_dendritic_cell_activated t-test 0.247  </span></span>
<span id="cb43-125"><a href="#cb43-125" tabindex="-1"></a><span class="co">#&gt;  2 B_cell                           t-test 0.0116 </span></span>
<span id="cb43-126"><a href="#cb43-126" tabindex="-1"></a><span class="co">#&gt;  3 T_cell_CD4+_memory               t-test 0.0826 </span></span>
<span id="cb43-127"><a href="#cb43-127" tabindex="-1"></a><span class="co">#&gt;  4 T_cell_CD4+_naive                t-test 0.142  </span></span>
<span id="cb43-128"><a href="#cb43-128" tabindex="-1"></a><span class="co">#&gt;  5 T_cell_CD4+_(non-regulatory)     t-test 0.391  </span></span>
<span id="cb43-129"><a href="#cb43-129" tabindex="-1"></a><span class="co">#&gt;  6 T_cell_CD4+_central_memory       t-test 0.656  </span></span>
<span id="cb43-130"><a href="#cb43-130" tabindex="-1"></a><span class="co">#&gt;  7 T_cell_CD4+_effector_memory      t-test 0.221  </span></span>
<span id="cb43-131"><a href="#cb43-131" tabindex="-1"></a><span class="co">#&gt;  8 T_cell_CD8+_naive                t-test 0.0160 </span></span>
<span id="cb43-132"><a href="#cb43-132" tabindex="-1"></a><span class="co">#&gt;  9 T_cell_CD8+                      t-test 0.00677</span></span>
<span id="cb43-133"><a href="#cb43-133" tabindex="-1"></a><span class="co">#&gt; 10 T_cell_CD8+_central_memory       t-test 0.00848</span></span>
<span id="cb43-134"><a href="#cb43-134" tabindex="-1"></a><span class="co">#&gt; # ℹ 29 more rows</span></span>
<span id="cb43-135"><a href="#cb43-135" tabindex="-1"></a><span class="co">#&gt; [33mGENERATING HEATMAPS[0m</span></span>
<span id="cb43-136"><a href="#cb43-136" tabindex="-1"></a><span class="co">#&gt; [32mHeatmap of qti, EPIC and xcell will be stored on plots_TME_heatmap.pdf[0m</span></span>
<span id="cb43-137"><a href="#cb43-137" tabindex="-1"></a></span>
<span id="cb43-138"><a href="#cb43-138" tabindex="-1"></a></span>
<span id="cb43-139"><a href="#cb43-139" tabindex="-1"></a><span class="fu">head</span>(TME_results<span class="sc">$</span>EPIC)</span>
<span id="cb43-140"><a href="#cb43-140" tabindex="-1"></a><span class="co">#&gt;                  B_cell Cancer_associated_fibroblast  T_cell_CD4+  T_cell_CD8+</span></span>
<span id="cb43-141"><a href="#cb43-141" tabindex="-1"></a><span class="co">#&gt; patient_1  1.150729e-06                 2.016299e-06 9.454060e-10 2.425392e-06</span></span>
<span id="cb43-142"><a href="#cb43-142" tabindex="-1"></a><span class="co">#&gt; patient_10 1.865684e-02                 1.951744e-03 3.278105e-02 1.210909e-08</span></span>
<span id="cb43-143"><a href="#cb43-143" tabindex="-1"></a><span class="co">#&gt; patient_11 2.155160e-02                 2.694712e-03 3.723145e-02 3.357117e-02</span></span>
<span id="cb43-144"><a href="#cb43-144" tabindex="-1"></a><span class="co">#&gt; patient_2  1.162535e-01                 2.177641e-02 1.561392e-01 1.032210e-03</span></span>
<span id="cb43-145"><a href="#cb43-145" tabindex="-1"></a><span class="co">#&gt; patient_3  1.812948e-01                 8.256491e-03 7.200416e-02 2.920044e-06</span></span>
<span id="cb43-146"><a href="#cb43-146" tabindex="-1"></a><span class="co">#&gt; patient_4  1.517297e-01                 5.777250e-02 8.969556e-07 1.326486e-06</span></span>
<span id="cb43-147"><a href="#cb43-147" tabindex="-1"></a><span class="co">#&gt;            Endothelial_cell  Macrophage      NK_cell uncharacterized_cell</span></span>
<span id="cb43-148"><a href="#cb43-148" tabindex="-1"></a><span class="co">#&gt; patient_1      3.117571e-08 0.999993600 8.033374e-07         2.726286e-09</span></span>
<span id="cb43-149"><a href="#cb43-149" tabindex="-1"></a><span class="co">#&gt; patient_10     6.217918e-03 0.016243630 7.745980e-03         9.164028e-01</span></span>
<span id="cb43-150"><a href="#cb43-150" tabindex="-1"></a><span class="co">#&gt; patient_11     4.903101e-02 0.005872793 1.749065e-03         8.482982e-01</span></span>
<span id="cb43-151"><a href="#cb43-151" tabindex="-1"></a><span class="co">#&gt; patient_2      1.513580e-02 0.038318322 5.511060e-04         6.507934e-01</span></span>
<span id="cb43-152"><a href="#cb43-152" tabindex="-1"></a><span class="co">#&gt; patient_3      4.132234e-02 0.109170000 1.448324e-03         5.865009e-01</span></span>
<span id="cb43-153"><a href="#cb43-153" tabindex="-1"></a><span class="co">#&gt; patient_4      4.808822e-03 0.340783400 4.338030e-02         4.015231e-01</span></span>
<span id="cb43-154"><a href="#cb43-154" tabindex="-1"></a><span class="co">#&gt;             HPV_status</span></span>
<span id="cb43-155"><a href="#cb43-155" tabindex="-1"></a><span class="co">#&gt; patient_1   Associated</span></span>
<span id="cb43-156"><a href="#cb43-156" tabindex="-1"></a><span class="co">#&gt; patient_10 Independent</span></span>
<span id="cb43-157"><a href="#cb43-157" tabindex="-1"></a><span class="co">#&gt; patient_11 Independent</span></span>
<span id="cb43-158"><a href="#cb43-158" tabindex="-1"></a><span class="co">#&gt; patient_2   Associated</span></span>
<span id="cb43-159"><a href="#cb43-159" tabindex="-1"></a><span class="co">#&gt; patient_3   Associated</span></span>
<span id="cb43-160"><a href="#cb43-160" tabindex="-1"></a><span class="co">#&gt; patient_4   Associated</span></span>
<span id="cb43-161"><a href="#cb43-161" tabindex="-1"></a><span class="fu">head</span>(TME_results<span class="sc">$</span>QTI)</span>
<span id="cb43-162"><a href="#cb43-162" tabindex="-1"></a><span class="co">#&gt;                B_cell Macrophage_M1 Macrophage_M2    Monocyte Neutrophil</span></span>
<span id="cb43-163"><a href="#cb43-163" tabindex="-1"></a><span class="co">#&gt; patient_1  0.06990880  0.000000e+00   0.474580130 0.035577343          0</span></span>
<span id="cb43-164"><a href="#cb43-164" tabindex="-1"></a><span class="co">#&gt; patient_10 0.00000000  5.268426e-06   0.008705942 0.000000000          0</span></span>
<span id="cb43-165"><a href="#cb43-165" tabindex="-1"></a><span class="co">#&gt; patient_11 0.11031970  2.848635e-02   0.258826850 0.042369490          0</span></span>
<span id="cb43-166"><a href="#cb43-166" tabindex="-1"></a><span class="co">#&gt; patient_2  0.13423964  2.764908e-02   0.090430490 0.011559290          0</span></span>
<span id="cb43-167"><a href="#cb43-167" tabindex="-1"></a><span class="co">#&gt; patient_3  0.13609518  5.503631e-03   0.344838553 0.065454939          0</span></span>
<span id="cb43-168"><a href="#cb43-168" tabindex="-1"></a><span class="co">#&gt; patient_4  0.08042602  1.255208e-01   0.216254015 0.001473265          0</span></span>
<span id="cb43-169"><a href="#cb43-169" tabindex="-1"></a><span class="co">#&gt;                NK_cell T_cell_CD4+_(non-regulatory) T_cell_CD8+</span></span>
<span id="cb43-170"><a href="#cb43-170" tabindex="-1"></a><span class="co">#&gt; patient_1  0.009145382                   0.00000000  0.06116945</span></span>
<span id="cb43-171"><a href="#cb43-171" tabindex="-1"></a><span class="co">#&gt; patient_10 0.000000000                   0.00000000  0.00000000</span></span>
<span id="cb43-172"><a href="#cb43-172" tabindex="-1"></a><span class="co">#&gt; patient_11 0.055631630                   0.00000000  0.00000000</span></span>
<span id="cb43-173"><a href="#cb43-173" tabindex="-1"></a><span class="co">#&gt; patient_2  0.000000000                   0.30471172  0.03531690</span></span>
<span id="cb43-174"><a href="#cb43-174" tabindex="-1"></a><span class="co">#&gt; patient_3  0.012846664                   0.05983765  0.05960699</span></span>
<span id="cb43-175"><a href="#cb43-175" tabindex="-1"></a><span class="co">#&gt; patient_4  0.048619021                   0.00000000  0.16209930</span></span>
<span id="cb43-176"><a href="#cb43-176" tabindex="-1"></a><span class="co">#&gt;            T_cell_regulatory_(Tregs) Myeloid_dendritic_cell</span></span>
<span id="cb43-177"><a href="#cb43-177" tabindex="-1"></a><span class="co">#&gt; patient_1                0.023104003                      0</span></span>
<span id="cb43-178"><a href="#cb43-178" tabindex="-1"></a><span class="co">#&gt; patient_10               0.074409150                      0</span></span>
<span id="cb43-179"><a href="#cb43-179" tabindex="-1"></a><span class="co">#&gt; patient_11               0.074404590                      0</span></span>
<span id="cb43-180"><a href="#cb43-180" tabindex="-1"></a><span class="co">#&gt; patient_2                0.125286240                      0</span></span>
<span id="cb43-181"><a href="#cb43-181" tabindex="-1"></a><span class="co">#&gt; patient_3                0.112300197                      0</span></span>
<span id="cb43-182"><a href="#cb43-182" tabindex="-1"></a><span class="co">#&gt; patient_4                0.008229871                      0</span></span>
<span id="cb43-183"><a href="#cb43-183" tabindex="-1"></a><span class="co">#&gt;            uncharacterized_cell  HPV_status</span></span>
<span id="cb43-184"><a href="#cb43-184" tabindex="-1"></a><span class="co">#&gt; patient_1             0.3265149  Associated</span></span>
<span id="cb43-185"><a href="#cb43-185" tabindex="-1"></a><span class="co">#&gt; patient_10            0.9168796 Independent</span></span>
<span id="cb43-186"><a href="#cb43-186" tabindex="-1"></a><span class="co">#&gt; patient_11            0.4299614 Independent</span></span>
<span id="cb43-187"><a href="#cb43-187" tabindex="-1"></a><span class="co">#&gt; patient_2             0.2708066  Associated</span></span>
<span id="cb43-188"><a href="#cb43-188" tabindex="-1"></a><span class="co">#&gt; patient_3             0.2035162  Associated</span></span>
<span id="cb43-189"><a href="#cb43-189" tabindex="-1"></a><span class="co">#&gt; patient_4             0.3573777  Associated</span></span>
<span id="cb43-190"><a href="#cb43-190" tabindex="-1"></a><span class="fu">head</span>(TME_results<span class="sc">$</span>XCELL)</span>
<span id="cb43-191"><a href="#cb43-191" tabindex="-1"></a><span class="co">#&gt;            Myeloid_dendritic_cell_activated      B_cell T_cell_CD4+_memory</span></span>
<span id="cb43-192"><a href="#cb43-192" tabindex="-1"></a><span class="co">#&gt; patient_1                      3.704441e-02 0.325428700       1.142668e-01</span></span>
<span id="cb43-193"><a href="#cb43-193" tabindex="-1"></a><span class="co">#&gt; patient_10                     0.000000e+00 0.069585870       0.000000e+00</span></span>
<span id="cb43-194"><a href="#cb43-194" tabindex="-1"></a><span class="co">#&gt; patient_11                     1.825679e-17 0.005666945       1.658565e-19</span></span>
<span id="cb43-195"><a href="#cb43-195" tabindex="-1"></a><span class="co">#&gt; patient_2                      7.274085e-19 0.465249000       3.324771e-01</span></span>
<span id="cb43-196"><a href="#cb43-196" tabindex="-1"></a><span class="co">#&gt; patient_3                      1.298727e-02 0.502043800       2.299917e-01</span></span>
<span id="cb43-197"><a href="#cb43-197" tabindex="-1"></a><span class="co">#&gt; patient_4                      7.658966e-18 0.209639900       5.325605e-02</span></span>
<span id="cb43-198"><a href="#cb43-198" tabindex="-1"></a><span class="co">#&gt;            T_cell_CD4+_naive T_cell_CD4+_(non-regulatory)</span></span>
<span id="cb43-199"><a href="#cb43-199" tabindex="-1"></a><span class="co">#&gt; patient_1       4.735101e-02                 5.191434e-18</span></span>
<span id="cb43-200"><a href="#cb43-200" tabindex="-1"></a><span class="co">#&gt; patient_10      2.721425e-17                 0.000000e+00</span></span>
<span id="cb43-201"><a href="#cb43-201" tabindex="-1"></a><span class="co">#&gt; patient_11      0.000000e+00                 0.000000e+00</span></span>
<span id="cb43-202"><a href="#cb43-202" tabindex="-1"></a><span class="co">#&gt; patient_2       1.129919e-01                 8.307792e-02</span></span>
<span id="cb43-203"><a href="#cb43-203" tabindex="-1"></a><span class="co">#&gt; patient_3       3.391430e-02                 1.631098e-19</span></span>
<span id="cb43-204"><a href="#cb43-204" tabindex="-1"></a><span class="co">#&gt; patient_4       2.384884e-18                 1.818445e-17</span></span>
<span id="cb43-205"><a href="#cb43-205" tabindex="-1"></a><span class="co">#&gt;            T_cell_CD4+_central_memory T_cell_CD4+_effector_memory</span></span>
<span id="cb43-206"><a href="#cb43-206" tabindex="-1"></a><span class="co">#&gt; patient_1                1.358802e-17                1.042981e-17</span></span>
<span id="cb43-207"><a href="#cb43-207" tabindex="-1"></a><span class="co">#&gt; patient_10               5.542262e-18                1.330074e-18</span></span>
<span id="cb43-208"><a href="#cb43-208" tabindex="-1"></a><span class="co">#&gt; patient_11               4.254897e-17                5.249508e-17</span></span>
<span id="cb43-209"><a href="#cb43-209" tabindex="-1"></a><span class="co">#&gt; patient_2                2.889525e-02                1.845534e-01</span></span>
<span id="cb43-210"><a href="#cb43-210" tabindex="-1"></a><span class="co">#&gt; patient_3                4.304898e-03                2.041590e-01</span></span>
<span id="cb43-211"><a href="#cb43-211" tabindex="-1"></a><span class="co">#&gt; patient_4                1.440364e-17                1.399956e-18</span></span>
<span id="cb43-212"><a href="#cb43-212" tabindex="-1"></a><span class="co">#&gt;            T_cell_CD8+_naive T_cell_CD8+ T_cell_CD8+_central_memory</span></span>
<span id="cb43-213"><a href="#cb43-213" tabindex="-1"></a><span class="co">#&gt; patient_1         0.03974244  0.08203368               5.154995e-02</span></span>
<span id="cb43-214"><a href="#cb43-214" tabindex="-1"></a><span class="co">#&gt; patient_10        0.00000000  0.00000000               0.000000e+00</span></span>
<span id="cb43-215"><a href="#cb43-215" tabindex="-1"></a><span class="co">#&gt; patient_11        0.00000000  0.00000000               2.310697e-17</span></span>
<span id="cb43-216"><a href="#cb43-216" tabindex="-1"></a><span class="co">#&gt; patient_2         0.08857475  0.17772240               1.113198e-01</span></span>
<span id="cb43-217"><a href="#cb43-217" tabindex="-1"></a><span class="co">#&gt; patient_3         0.08616049  0.16266110               8.185110e-02</span></span>
<span id="cb43-218"><a href="#cb43-218" tabindex="-1"></a><span class="co">#&gt; patient_4         0.07952526  0.14421760               6.882800e-02</span></span>
<span id="cb43-219"><a href="#cb43-219" tabindex="-1"></a><span class="co">#&gt;            T_cell_CD8+_effector_memory Class-switched_memory_B_cell</span></span>
<span id="cb43-220"><a href="#cb43-220" tabindex="-1"></a><span class="co">#&gt; patient_1                 4.787293e-02                   0.06874662</span></span>
<span id="cb43-221"><a href="#cb43-221" tabindex="-1"></a><span class="co">#&gt; patient_10                2.857536e-18                   0.00000000</span></span>
<span id="cb43-222"><a href="#cb43-222" tabindex="-1"></a><span class="co">#&gt; patient_11                0.000000e+00                   0.00000000</span></span>
<span id="cb43-223"><a href="#cb43-223" tabindex="-1"></a><span class="co">#&gt; patient_2                 1.482160e-01                   0.15564710</span></span>
<span id="cb43-224"><a href="#cb43-224" tabindex="-1"></a><span class="co">#&gt; patient_3                 1.165878e-01                   0.16540470</span></span>
<span id="cb43-225"><a href="#cb43-225" tabindex="-1"></a><span class="co">#&gt; patient_4                 1.325500e-01                   0.06442109</span></span>
<span id="cb43-226"><a href="#cb43-226" tabindex="-1"></a><span class="co">#&gt;            Common_lymphoid_progenitor Common_myeloid_progenitor</span></span>
<span id="cb43-227"><a href="#cb43-227" tabindex="-1"></a><span class="co">#&gt; patient_1                9.028999e-03              6.235078e-19</span></span>
<span id="cb43-228"><a href="#cb43-228" tabindex="-1"></a><span class="co">#&gt; patient_10               4.590643e-18              8.178118e-18</span></span>
<span id="cb43-229"><a href="#cb43-229" tabindex="-1"></a><span class="co">#&gt; patient_11               2.307430e-02              3.095361e-01</span></span>
<span id="cb43-230"><a href="#cb43-230" tabindex="-1"></a><span class="co">#&gt; patient_2                3.899960e-02              0.000000e+00</span></span>
<span id="cb43-231"><a href="#cb43-231" tabindex="-1"></a><span class="co">#&gt; patient_3                2.846220e-02              2.027926e-19</span></span>
<span id="cb43-232"><a href="#cb43-232" tabindex="-1"></a><span class="co">#&gt; patient_4                8.795192e-02              2.508084e-19</span></span>
<span id="cb43-233"><a href="#cb43-233" tabindex="-1"></a><span class="co">#&gt;            Myeloid_dendritic_cell Endothelial_cell Eosinophil</span></span>
<span id="cb43-234"><a href="#cb43-234" tabindex="-1"></a><span class="co">#&gt; patient_1            2.014373e-02     1.193218e-03 0.00000000</span></span>
<span id="cb43-235"><a href="#cb43-235" tabindex="-1"></a><span class="co">#&gt; patient_10           0.000000e+00     0.000000e+00 0.06886608</span></span>
<span id="cb43-236"><a href="#cb43-236" tabindex="-1"></a><span class="co">#&gt; patient_11           7.089901e-19     4.049257e-18 0.08245618</span></span>
<span id="cb43-237"><a href="#cb43-237" tabindex="-1"></a><span class="co">#&gt; patient_2            2.511137e-19     0.000000e+00 0.00000000</span></span>
<span id="cb43-238"><a href="#cb43-238" tabindex="-1"></a><span class="co">#&gt; patient_3            2.118263e-02     9.108657e-03 0.00000000</span></span>
<span id="cb43-239"><a href="#cb43-239" tabindex="-1"></a><span class="co">#&gt; patient_4            0.000000e+00     5.052323e-03 0.00000000</span></span>
<span id="cb43-240"><a href="#cb43-240" tabindex="-1"></a><span class="co">#&gt;            Cancer_associated_fibroblast Granulocyte-monocyte_progenitor</span></span>
<span id="cb43-241"><a href="#cb43-241" tabindex="-1"></a><span class="co">#&gt; patient_1                  0.000000e+00                    0.000000e+00</span></span>
<span id="cb43-242"><a href="#cb43-242" tabindex="-1"></a><span class="co">#&gt; patient_10                 3.831830e-17                    7.604252e-18</span></span>
<span id="cb43-243"><a href="#cb43-243" tabindex="-1"></a><span class="co">#&gt; patient_11                 3.340841e-18                    8.283345e-03</span></span>
<span id="cb43-244"><a href="#cb43-244" tabindex="-1"></a><span class="co">#&gt; patient_2                  2.699647e-03                    4.258166e-18</span></span>
<span id="cb43-245"><a href="#cb43-245" tabindex="-1"></a><span class="co">#&gt; patient_3                  4.566704e-02                    0.000000e+00</span></span>
<span id="cb43-246"><a href="#cb43-246" tabindex="-1"></a><span class="co">#&gt; patient_4                  2.935790e-02                    0.000000e+00</span></span>
<span id="cb43-247"><a href="#cb43-247" tabindex="-1"></a><span class="co">#&gt;            Hematopoietic_stem_cell Macrophage Macrophage_M1 Macrophage_M2</span></span>
<span id="cb43-248"><a href="#cb43-248" tabindex="-1"></a><span class="co">#&gt; patient_1             1.262376e-01 0.10488450  9.811489e-02    0.03181943</span></span>
<span id="cb43-249"><a href="#cb43-249" tabindex="-1"></a><span class="co">#&gt; patient_10            4.029082e-02 0.00000000  1.635625e-02    0.03774417</span></span>
<span id="cb43-250"><a href="#cb43-250" tabindex="-1"></a><span class="co">#&gt; patient_11            1.574903e-01 0.00000000  3.966871e-18    0.00000000</span></span>
<span id="cb43-251"><a href="#cb43-251" tabindex="-1"></a><span class="co">#&gt; patient_2             0.000000e+00 0.06121064  6.162512e-02    0.01733281</span></span>
<span id="cb43-252"><a href="#cb43-252" tabindex="-1"></a><span class="co">#&gt; patient_3             3.944384e-02 0.08813288  6.769867e-02    0.03912408</span></span>
<span id="cb43-253"><a href="#cb43-253" tabindex="-1"></a><span class="co">#&gt; patient_4             1.736266e-18 0.07883394  6.367928e-02    0.04452152</span></span>
<span id="cb43-254"><a href="#cb43-254" tabindex="-1"></a><span class="co">#&gt;               Mast_cell B_cell_memory     Monocyte B_cell_naive   Neutrophil</span></span>
<span id="cb43-255"><a href="#cb43-255" tabindex="-1"></a><span class="co">#&gt; patient_1  7.499468e-19    0.11418660 1.272603e-18   0.06461067 0.000000e+00</span></span>
<span id="cb43-256"><a href="#cb43-256" tabindex="-1"></a><span class="co">#&gt; patient_10 1.603632e-03    0.15814950 7.272301e-17   0.01422468 7.053283e-18</span></span>
<span id="cb43-257"><a href="#cb43-257" tabindex="-1"></a><span class="co">#&gt; patient_11 1.640472e-19    0.19786370 3.439483e-17   0.00000000 1.586729e-03</span></span>
<span id="cb43-258"><a href="#cb43-258" tabindex="-1"></a><span class="co">#&gt; patient_2  5.512014e-03    0.13157300 2.385468e-18   0.10798930 0.000000e+00</span></span>
<span id="cb43-259"><a href="#cb43-259" tabindex="-1"></a><span class="co">#&gt; patient_3  3.724377e-02    0.16500350 1.097886e-17   0.07702582 4.327200e-18</span></span>
<span id="cb43-260"><a href="#cb43-260" tabindex="-1"></a><span class="co">#&gt; patient_4  4.087144e-03    0.04751746 0.000000e+00   0.03052735 2.480742e-18</span></span>
<span id="cb43-261"><a href="#cb43-261" tabindex="-1"></a><span class="co">#&gt;                 NK_cell    T_cell_NK Plasmacytoid_dendritic_cell B_cell_plasma</span></span>
<span id="cb43-262"><a href="#cb43-262" tabindex="-1"></a><span class="co">#&gt; patient_1  4.282241e-17 2.334648e-02                  0.00000000  1.248448e-19</span></span>
<span id="cb43-263"><a href="#cb43-263" tabindex="-1"></a><span class="co">#&gt; patient_10 1.296309e-17 0.000000e+00                  0.02112072  3.445545e-02</span></span>
<span id="cb43-264"><a href="#cb43-264" tabindex="-1"></a><span class="co">#&gt; patient_11 0.000000e+00 2.165840e-01                  0.18403760  4.849640e-02</span></span>
<span id="cb43-265"><a href="#cb43-265" tabindex="-1"></a><span class="co">#&gt; patient_2  3.201773e-17 6.525717e-18                  0.03433043  0.000000e+00</span></span>
<span id="cb43-266"><a href="#cb43-266" tabindex="-1"></a><span class="co">#&gt; patient_3  0.000000e+00 4.526615e-19                  0.05893872  7.131881e-02</span></span>
<span id="cb43-267"><a href="#cb43-267" tabindex="-1"></a><span class="co">#&gt; patient_4  0.000000e+00 7.194266e-02                  0.11994490  6.530570e-02</span></span>
<span id="cb43-268"><a href="#cb43-268" tabindex="-1"></a><span class="co">#&gt;            T_cell_gamma_delta T_cell_CD4+_Th1 T_cell_CD4+_Th2</span></span>
<span id="cb43-269"><a href="#cb43-269" tabindex="-1"></a><span class="co">#&gt; patient_1        2.239689e-02    6.769285e-02    1.699192e-02</span></span>
<span id="cb43-270"><a href="#cb43-270" tabindex="-1"></a><span class="co">#&gt; patient_10       0.000000e+00    2.200404e-19    0.000000e+00</span></span>
<span id="cb43-271"><a href="#cb43-271" tabindex="-1"></a><span class="co">#&gt; patient_11       1.076331e-16    8.957217e-02    4.152295e-02</span></span>
<span id="cb43-272"><a href="#cb43-272" tabindex="-1"></a><span class="co">#&gt; patient_2        3.782588e-02    4.087513e-01    0.000000e+00</span></span>
<span id="cb43-273"><a href="#cb43-273" tabindex="-1"></a><span class="co">#&gt; patient_3        1.017413e-03    3.150856e-01    0.000000e+00</span></span>
<span id="cb43-274"><a href="#cb43-274" tabindex="-1"></a><span class="co">#&gt; patient_4        5.101980e-18    3.745663e-01    4.812461e-18</span></span>
<span id="cb43-275"><a href="#cb43-275" tabindex="-1"></a><span class="co">#&gt;            T_cell_regulatory_(Tregs) immune_score stroma_score</span></span>
<span id="cb43-276"><a href="#cb43-276" tabindex="-1"></a><span class="co">#&gt; patient_1               3.903347e-18   0.35499380 5.966090e-04</span></span>
<span id="cb43-277"><a href="#cb43-277" tabindex="-1"></a><span class="co">#&gt; patient_10              1.785058e-02   0.09337039 1.915915e-17</span></span>
<span id="cb43-278"><a href="#cb43-278" tabindex="-1"></a><span class="co">#&gt; patient_11              0.000000e+00   0.05980657 3.342654e-02</span></span>
<span id="cb43-279"><a href="#cb43-279" tabindex="-1"></a><span class="co">#&gt; patient_2               0.000000e+00   0.52851460 1.349824e-03</span></span>
<span id="cb43-280"><a href="#cb43-280" tabindex="-1"></a><span class="co">#&gt; patient_3               3.003288e-18   0.54084280 2.738785e-02</span></span>
<span id="cb43-281"><a href="#cb43-281" tabindex="-1"></a><span class="co">#&gt; patient_4               1.165112e-17   0.29118570 1.772281e-02</span></span>
<span id="cb43-282"><a href="#cb43-282" tabindex="-1"></a><span class="co">#&gt;            microenvironment_score  HPV_status</span></span>
<span id="cb43-283"><a href="#cb43-283" tabindex="-1"></a><span class="co">#&gt; patient_1              0.35559040  Associated</span></span>
<span id="cb43-284"><a href="#cb43-284" tabindex="-1"></a><span class="co">#&gt; patient_10             0.09337039 Independent</span></span>
<span id="cb43-285"><a href="#cb43-285" tabindex="-1"></a><span class="co">#&gt; patient_11             0.09323310 Independent</span></span>
<span id="cb43-286"><a href="#cb43-286" tabindex="-1"></a><span class="co">#&gt; patient_2              0.52986440  Associated</span></span>
<span id="cb43-287"><a href="#cb43-287" tabindex="-1"></a><span class="co">#&gt; patient_3              0.56823060  Associated</span></span>
<span id="cb43-288"><a href="#cb43-288" tabindex="-1"></a><span class="co">#&gt; patient_4              0.30890850  Associated</span></span></code></pre></div>
</div>
<div id="survival-analysis-htg_survival-for-rnaseq" class="section level4">
<h4>Survival analysis: HTG_survival() for RNAseq:</h4>
<p>Survival analysis can be performed using the results from DEA (res),
specific genes, or TME results. The function prioritizes inputs: first
genes_to_use, then DEA results, and finally TME results if none are
provided.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="co"># survival using res</span></span>
<span id="cb44-2"><a href="#cb44-2" tabindex="-1"></a><span class="fu">HTG_survival</span>(<span class="at">variable_01 =</span> <span class="st">&quot;Recurrence_01&quot;</span>,</span>
<span id="cb44-3"><a href="#cb44-3" tabindex="-1"></a>             <span class="at">time =</span> <span class="st">&quot;Time_to_death_surv&quot;</span>,</span>
<span id="cb44-4"><a href="#cb44-4" tabindex="-1"></a>             <span class="at">col_data =</span> AnnotData_tutorial,</span>
<span id="cb44-5"><a href="#cb44-5" tabindex="-1"></a>             <span class="at">counts_data =</span> counts_data_tutorial,</span>
<span id="cb44-6"><a href="#cb44-6" tabindex="-1"></a>             <span class="at">res =</span> res_tutorial,</span>
<span id="cb44-7"><a href="#cb44-7" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;median&quot;</span>,</span>
<span id="cb44-8"><a href="#cb44-8" tabindex="-1"></a>             <span class="at">genes_to_use =</span> <span class="cn">NULL</span>,</span>
<span id="cb44-9"><a href="#cb44-9" tabindex="-1"></a>             <span class="at">TME =</span> <span class="cn">NULL</span>,</span>
<span id="cb44-10"><a href="#cb44-10" tabindex="-1"></a>             <span class="at">outliers =</span> <span class="cn">NULL</span>,</span>
<span id="cb44-11"><a href="#cb44-11" tabindex="-1"></a>             <span class="at">pattern =</span> <span class="cn">NULL</span>,</span>
<span id="cb44-12"><a href="#cb44-12" tabindex="-1"></a>             <span class="at">remove_outliers =</span> <span class="cn">FALSE</span>)</span>
<span id="cb44-13"><a href="#cb44-13" tabindex="-1"></a><span class="co">#&gt; [32mNormalizing data...[0m</span></span>
<span id="cb44-14"><a href="#cb44-14" tabindex="-1"></a><span class="co">#&gt; [33mOBTAINING NORMALIEZ COUNTS...[0m</span></span>
<span id="cb44-15"><a href="#cb44-15" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb44-16"><a href="#cb44-16" tabindex="-1"></a><span class="co">#&gt; [32mChecking for duplicate columns[0m</span></span>
<span id="cb44-17"><a href="#cb44-17" tabindex="-1"></a><span class="co">#&gt; [1] &quot;No duplicate columns.&quot;</span></span>
<span id="cb44-18"><a href="#cb44-18" tabindex="-1"></a><span class="co">#&gt; [32mFiltering data.[0m</span></span>
<span id="cb44-19"><a href="#cb44-19" tabindex="-1"></a><span class="co">#&gt; [32mSelecting TOP 10 genes with the lowest padj[0m</span></span>
<span id="cb44-20"><a href="#cb44-20" tabindex="-1"></a><span class="co">#&gt; [32mStarting survival analysis.[0m</span></span>
<span id="cb44-21"><a href="#cb44-21" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-22"><a href="#cb44-22" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  LCP1 </span></span>
<span id="cb44-23"><a href="#cb44-23" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb44-24"><a href="#cb44-24" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb44-25"><a href="#cb44-25" tabindex="-1"></a><span class="co">#&gt; [1] 494.7313</span></span>
<span id="cb44-26"><a href="#cb44-26" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-27"><a href="#cb44-27" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-28"><a href="#cb44-28" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-29"><a href="#cb44-29" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-30"><a href="#cb44-30" tabindex="-1"></a><span class="co">#&gt;                 LCP1_mRNA_expression=High </span></span>
<span id="cb44-31"><a href="#cb44-31" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-32"><a href="#cb44-32" tabindex="-1"></a><span class="co">#&gt;  1164      3       1    0.667   0.272       0.2995            1</span></span>
<span id="cb44-33"><a href="#cb44-33" tabindex="-1"></a><span class="co">#&gt;  1287      2       1    0.333   0.272       0.0673            1</span></span>
<span id="cb44-34"><a href="#cb44-34" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-35"><a href="#cb44-35" tabindex="-1"></a><span class="co">#&gt;                 LCP1_mRNA_expression=Low </span></span>
<span id="cb44-36"><a href="#cb44-36" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-37"><a href="#cb44-37" tabindex="-1"></a><span class="co">#&gt;   234      6       1    0.833   0.152       0.5827            1</span></span>
<span id="cb44-38"><a href="#cb44-38" tabindex="-1"></a><span class="co">#&gt;  2458      2       1    0.417   0.304       0.0996            1</span></span>
<span id="cb44-39"><a href="#cb44-39" tabindex="-1"></a><span class="co">#&gt;  5844      1       1    0.000     NaN           NA           NA</span></span>
<span id="cb44-40"><a href="#cb44-40" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-41"><a href="#cb44-41" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-42"><a href="#cb44-42" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-43"><a href="#cb44-43" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-44"><a href="#cb44-44" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-45"><a href="#cb44-45" tabindex="-1"></a><span class="co">#&gt;                           N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-46"><a href="#cb44-46" tabindex="-1"></a><span class="co">#&gt; LCP1_mRNA_expression=High 5        2     1.35     0.308     0.565</span></span>
<span id="cb44-47"><a href="#cb44-47" tabindex="-1"></a><span class="co">#&gt; LCP1_mRNA_expression=Low  6        3     3.65     0.114     0.565</span></span>
<span id="cb44-48"><a href="#cb44-48" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-49"><a href="#cb44-49" tabindex="-1"></a><span class="co">#&gt;  Chisq= 0.6  on 1 degrees of freedom, p= 0.5 </span></span>
<span id="cb44-50"><a href="#cb44-50" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-51"><a href="#cb44-51" tabindex="-1"></a><span class="co">#&gt; [1] 0.4524268</span></span>
<span id="cb44-52"><a href="#cb44-52" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-53"><a href="#cb44-53" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb44-54"><a href="#cb44-54" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-55"><a href="#cb44-55" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  LYPD3 </span></span>
<span id="cb44-56"><a href="#cb44-56" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb44-57"><a href="#cb44-57" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb44-58"><a href="#cb44-58" tabindex="-1"></a><span class="co">#&gt; [1] 311.7017</span></span>
<span id="cb44-59"><a href="#cb44-59" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-60"><a href="#cb44-60" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-61"><a href="#cb44-61" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-62"><a href="#cb44-62" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-63"><a href="#cb44-63" tabindex="-1"></a><span class="co">#&gt;                 LYPD3_mRNA_expression=High </span></span>
<span id="cb44-64"><a href="#cb44-64" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-65"><a href="#cb44-65" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb44-66"><a href="#cb44-66" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-67"><a href="#cb44-67" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-68"><a href="#cb44-68" tabindex="-1"></a><span class="co">#&gt;                 LYPD3_mRNA_expression=Low </span></span>
<span id="cb44-69"><a href="#cb44-69" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-70"><a href="#cb44-70" tabindex="-1"></a><span class="co">#&gt;   234      6       1    0.833   0.152       0.5827            1</span></span>
<span id="cb44-71"><a href="#cb44-71" tabindex="-1"></a><span class="co">#&gt;  1164      3       1    0.556   0.248       0.2312            1</span></span>
<span id="cb44-72"><a href="#cb44-72" tabindex="-1"></a><span class="co">#&gt;  1287      2       1    0.278   0.232       0.0539            1</span></span>
<span id="cb44-73"><a href="#cb44-73" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-74"><a href="#cb44-74" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-75"><a href="#cb44-75" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-76"><a href="#cb44-76" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-77"><a href="#cb44-77" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-78"><a href="#cb44-78" tabindex="-1"></a><span class="co">#&gt;                            N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-79"><a href="#cb44-79" tabindex="-1"></a><span class="co">#&gt; LYPD3_mRNA_expression=High 5        2     3.55      0.68      3.27</span></span>
<span id="cb44-80"><a href="#cb44-80" tabindex="-1"></a><span class="co">#&gt; LYPD3_mRNA_expression=Low  6        3     1.45      1.67      3.27</span></span>
<span id="cb44-81"><a href="#cb44-81" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-82"><a href="#cb44-82" tabindex="-1"></a><span class="co">#&gt;  Chisq= 3.3  on 1 degrees of freedom, p= 0.07 </span></span>
<span id="cb44-83"><a href="#cb44-83" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-84"><a href="#cb44-84" tabindex="-1"></a><span class="co">#&gt; [1] 0.07035001</span></span>
<span id="cb44-85"><a href="#cb44-85" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-86"><a href="#cb44-86" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb44-87"><a href="#cb44-87" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-88"><a href="#cb44-88" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  SLAMF7 </span></span>
<span id="cb44-89"><a href="#cb44-89" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb44-90"><a href="#cb44-90" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb44-91"><a href="#cb44-91" tabindex="-1"></a><span class="co">#&gt; [1] 532.8931</span></span>
<span id="cb44-92"><a href="#cb44-92" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-93"><a href="#cb44-93" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-94"><a href="#cb44-94" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-95"><a href="#cb44-95" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-96"><a href="#cb44-96" tabindex="-1"></a><span class="co">#&gt;                 SLAMF7_mRNA_expression=High </span></span>
<span id="cb44-97"><a href="#cb44-97" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-98"><a href="#cb44-98" tabindex="-1"></a><span class="co">#&gt;   234      5       1      0.8   0.179       0.5161            1</span></span>
<span id="cb44-99"><a href="#cb44-99" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.4   0.297       0.0935            1</span></span>
<span id="cb44-100"><a href="#cb44-100" tabindex="-1"></a><span class="co">#&gt;  1287      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-101"><a href="#cb44-101" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-102"><a href="#cb44-102" tabindex="-1"></a><span class="co">#&gt;                 SLAMF7_mRNA_expression=Low </span></span>
<span id="cb44-103"><a href="#cb44-103" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-104"><a href="#cb44-104" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb44-105"><a href="#cb44-105" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-106"><a href="#cb44-106" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-107"><a href="#cb44-107" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-108"><a href="#cb44-108" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-109"><a href="#cb44-109" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-110"><a href="#cb44-110" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-111"><a href="#cb44-111" tabindex="-1"></a><span class="co">#&gt;                             N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-112"><a href="#cb44-112" tabindex="-1"></a><span class="co">#&gt; SLAMF7_mRNA_expression=High 5        3    0.988      4.10      6.42</span></span>
<span id="cb44-113"><a href="#cb44-113" tabindex="-1"></a><span class="co">#&gt; SLAMF7_mRNA_expression=Low  6        2    4.012      1.01      6.42</span></span>
<span id="cb44-114"><a href="#cb44-114" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-115"><a href="#cb44-115" tabindex="-1"></a><span class="co">#&gt;  Chisq= 6.4  on 1 degrees of freedom, p= 0.01 </span></span>
<span id="cb44-116"><a href="#cb44-116" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-117"><a href="#cb44-117" tabindex="-1"></a><span class="co">#&gt; [1] 0.01125371</span></span>
<span id="cb44-118"><a href="#cb44-118" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-119"><a href="#cb44-119" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb44-120"><a href="#cb44-120" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-121"><a href="#cb44-121" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  OMA1 </span></span>
<span id="cb44-122"><a href="#cb44-122" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb44-123"><a href="#cb44-123" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb44-124"><a href="#cb44-124" tabindex="-1"></a><span class="co">#&gt; [1] 149.9492</span></span>
<span id="cb44-125"><a href="#cb44-125" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-126"><a href="#cb44-126" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-127"><a href="#cb44-127" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-128"><a href="#cb44-128" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-129"><a href="#cb44-129" tabindex="-1"></a><span class="co">#&gt;                 OMA1_mRNA_expression=High </span></span>
<span id="cb44-130"><a href="#cb44-130" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-131"><a href="#cb44-131" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb44-132"><a href="#cb44-132" tabindex="-1"></a><span class="co">#&gt;  1287      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-133"><a href="#cb44-133" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-134"><a href="#cb44-134" tabindex="-1"></a><span class="co">#&gt;                 OMA1_mRNA_expression=Low </span></span>
<span id="cb44-135"><a href="#cb44-135" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-136"><a href="#cb44-136" tabindex="-1"></a><span class="co">#&gt;   234      6       1    0.833   0.152       0.5827            1</span></span>
<span id="cb44-137"><a href="#cb44-137" tabindex="-1"></a><span class="co">#&gt;  2458      2       1    0.417   0.304       0.0996            1</span></span>
<span id="cb44-138"><a href="#cb44-138" tabindex="-1"></a><span class="co">#&gt;  5844      1       1    0.000     NaN           NA           NA</span></span>
<span id="cb44-139"><a href="#cb44-139" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-140"><a href="#cb44-140" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-141"><a href="#cb44-141" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-142"><a href="#cb44-142" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-143"><a href="#cb44-143" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-144"><a href="#cb44-144" tabindex="-1"></a><span class="co">#&gt;                           N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-145"><a href="#cb44-145" tabindex="-1"></a><span class="co">#&gt; OMA1_mRNA_expression=High 5        2    0.988     1.037      1.63</span></span>
<span id="cb44-146"><a href="#cb44-146" tabindex="-1"></a><span class="co">#&gt; OMA1_mRNA_expression=Low  6        3    4.012     0.255      1.63</span></span>
<span id="cb44-147"><a href="#cb44-147" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-148"><a href="#cb44-148" tabindex="-1"></a><span class="co">#&gt;  Chisq= 1.6  on 1 degrees of freedom, p= 0.2 </span></span>
<span id="cb44-149"><a href="#cb44-149" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-150"><a href="#cb44-150" tabindex="-1"></a><span class="co">#&gt; [1] 0.202311</span></span>
<span id="cb44-151"><a href="#cb44-151" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-152"><a href="#cb44-152" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb44-153"><a href="#cb44-153" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-154"><a href="#cb44-154" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  VOPP1 </span></span>
<span id="cb44-155"><a href="#cb44-155" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb44-156"><a href="#cb44-156" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb44-157"><a href="#cb44-157" tabindex="-1"></a><span class="co">#&gt; [1] 441.7856</span></span>
<span id="cb44-158"><a href="#cb44-158" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-159"><a href="#cb44-159" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-160"><a href="#cb44-160" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-161"><a href="#cb44-161" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-162"><a href="#cb44-162" tabindex="-1"></a><span class="co">#&gt;                 VOPP1_mRNA_expression=High </span></span>
<span id="cb44-163"><a href="#cb44-163" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-164"><a href="#cb44-164" tabindex="-1"></a><span class="co">#&gt;   234      5       1      0.8   0.179       0.5161            1</span></span>
<span id="cb44-165"><a href="#cb44-165" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.4   0.297       0.0935            1</span></span>
<span id="cb44-166"><a href="#cb44-166" tabindex="-1"></a><span class="co">#&gt;  1287      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-167"><a href="#cb44-167" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-168"><a href="#cb44-168" tabindex="-1"></a><span class="co">#&gt;                 VOPP1_mRNA_expression=Low </span></span>
<span id="cb44-169"><a href="#cb44-169" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-170"><a href="#cb44-170" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb44-171"><a href="#cb44-171" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-172"><a href="#cb44-172" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-173"><a href="#cb44-173" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-174"><a href="#cb44-174" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-175"><a href="#cb44-175" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-176"><a href="#cb44-176" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-177"><a href="#cb44-177" tabindex="-1"></a><span class="co">#&gt;                            N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-178"><a href="#cb44-178" tabindex="-1"></a><span class="co">#&gt; VOPP1_mRNA_expression=High 5        3    0.988      4.10      6.42</span></span>
<span id="cb44-179"><a href="#cb44-179" tabindex="-1"></a><span class="co">#&gt; VOPP1_mRNA_expression=Low  6        2    4.012      1.01      6.42</span></span>
<span id="cb44-180"><a href="#cb44-180" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-181"><a href="#cb44-181" tabindex="-1"></a><span class="co">#&gt;  Chisq= 6.4  on 1 degrees of freedom, p= 0.01 </span></span>
<span id="cb44-182"><a href="#cb44-182" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-183"><a href="#cb44-183" tabindex="-1"></a><span class="co">#&gt; [1] 0.01125371</span></span>
<span id="cb44-184"><a href="#cb44-184" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-185"><a href="#cb44-185" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb44-186"><a href="#cb44-186" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-187"><a href="#cb44-187" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  COL5A3 </span></span>
<span id="cb44-188"><a href="#cb44-188" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb44-189"><a href="#cb44-189" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb44-190"><a href="#cb44-190" tabindex="-1"></a><span class="co">#&gt; [1] 87.16655</span></span>
<span id="cb44-191"><a href="#cb44-191" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-192"><a href="#cb44-192" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-193"><a href="#cb44-193" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-194"><a href="#cb44-194" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-195"><a href="#cb44-195" tabindex="-1"></a><span class="co">#&gt;                 COL5A3_mRNA_expression=High </span></span>
<span id="cb44-196"><a href="#cb44-196" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-197"><a href="#cb44-197" tabindex="-1"></a><span class="co">#&gt;  1287      5       1      0.8   0.179       0.5161            1</span></span>
<span id="cb44-198"><a href="#cb44-198" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.4   0.297       0.0935            1</span></span>
<span id="cb44-199"><a href="#cb44-199" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-200"><a href="#cb44-200" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-201"><a href="#cb44-201" tabindex="-1"></a><span class="co">#&gt;                 COL5A3_mRNA_expression=Low </span></span>
<span id="cb44-202"><a href="#cb44-202" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-203"><a href="#cb44-203" tabindex="-1"></a><span class="co">#&gt;   234      6       1    0.833   0.152        0.583            1</span></span>
<span id="cb44-204"><a href="#cb44-204" tabindex="-1"></a><span class="co">#&gt;  1164      1       1    0.000     NaN           NA           NA</span></span>
<span id="cb44-205"><a href="#cb44-205" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-206"><a href="#cb44-206" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-207"><a href="#cb44-207" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-208"><a href="#cb44-208" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-209"><a href="#cb44-209" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-210"><a href="#cb44-210" tabindex="-1"></a><span class="co">#&gt;                             N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-211"><a href="#cb44-211" tabindex="-1"></a><span class="co">#&gt; COL5A3_mRNA_expression=High 5        3    4.288     0.387      4.29</span></span>
<span id="cb44-212"><a href="#cb44-212" tabindex="-1"></a><span class="co">#&gt; COL5A3_mRNA_expression=Low  6        2    0.712     2.329      4.29</span></span>
<span id="cb44-213"><a href="#cb44-213" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-214"><a href="#cb44-214" tabindex="-1"></a><span class="co">#&gt;  Chisq= 4.3  on 1 degrees of freedom, p= 0.04 </span></span>
<span id="cb44-215"><a href="#cb44-215" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-216"><a href="#cb44-216" tabindex="-1"></a><span class="co">#&gt; [1] 0.03838604</span></span>
<span id="cb44-217"><a href="#cb44-217" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-218"><a href="#cb44-218" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb44-219"><a href="#cb44-219" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-220"><a href="#cb44-220" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  MMP10 </span></span>
<span id="cb44-221"><a href="#cb44-221" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb44-222"><a href="#cb44-222" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb44-223"><a href="#cb44-223" tabindex="-1"></a><span class="co">#&gt; [1] 49.05272</span></span>
<span id="cb44-224"><a href="#cb44-224" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-225"><a href="#cb44-225" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-226"><a href="#cb44-226" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-227"><a href="#cb44-227" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-228"><a href="#cb44-228" tabindex="-1"></a><span class="co">#&gt;                 MMP10_mRNA_expression=High </span></span>
<span id="cb44-229"><a href="#cb44-229" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-230"><a href="#cb44-230" tabindex="-1"></a><span class="co">#&gt;   234      5       1      0.8   0.179        0.516            1</span></span>
<span id="cb44-231"><a href="#cb44-231" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-232"><a href="#cb44-232" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-233"><a href="#cb44-233" tabindex="-1"></a><span class="co">#&gt;                 MMP10_mRNA_expression=Low </span></span>
<span id="cb44-234"><a href="#cb44-234" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-235"><a href="#cb44-235" tabindex="-1"></a><span class="co">#&gt;  1164      4       1     0.75   0.217        0.426            1</span></span>
<span id="cb44-236"><a href="#cb44-236" tabindex="-1"></a><span class="co">#&gt;  1287      3       1     0.50   0.250        0.188            1</span></span>
<span id="cb44-237"><a href="#cb44-237" tabindex="-1"></a><span class="co">#&gt;  2458      1       1     0.00     NaN           NA           NA</span></span>
<span id="cb44-238"><a href="#cb44-238" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-239"><a href="#cb44-239" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-240"><a href="#cb44-240" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-241"><a href="#cb44-241" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-242"><a href="#cb44-242" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-243"><a href="#cb44-243" tabindex="-1"></a><span class="co">#&gt;                            N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-244"><a href="#cb44-244" tabindex="-1"></a><span class="co">#&gt; MMP10_mRNA_expression=High 5        2     2.69     0.176     0.493</span></span>
<span id="cb44-245"><a href="#cb44-245" tabindex="-1"></a><span class="co">#&gt; MMP10_mRNA_expression=Low  6        3     2.31     0.205     0.493</span></span>
<span id="cb44-246"><a href="#cb44-246" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-247"><a href="#cb44-247" tabindex="-1"></a><span class="co">#&gt;  Chisq= 0.5  on 1 degrees of freedom, p= 0.5 </span></span>
<span id="cb44-248"><a href="#cb44-248" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-249"><a href="#cb44-249" tabindex="-1"></a><span class="co">#&gt; [1] 0.4826753</span></span>
<span id="cb44-250"><a href="#cb44-250" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-251"><a href="#cb44-251" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb44-252"><a href="#cb44-252" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-253"><a href="#cb44-253" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  CDCA4 </span></span>
<span id="cb44-254"><a href="#cb44-254" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb44-255"><a href="#cb44-255" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb44-256"><a href="#cb44-256" tabindex="-1"></a><span class="co">#&gt; [1] 234.6449</span></span>
<span id="cb44-257"><a href="#cb44-257" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-258"><a href="#cb44-258" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-259"><a href="#cb44-259" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-260"><a href="#cb44-260" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-261"><a href="#cb44-261" tabindex="-1"></a><span class="co">#&gt;                 CDCA4_mRNA_expression=High </span></span>
<span id="cb44-262"><a href="#cb44-262" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-263"><a href="#cb44-263" tabindex="-1"></a><span class="co">#&gt;   234      5       1      0.8   0.179       0.5161            1</span></span>
<span id="cb44-264"><a href="#cb44-264" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.4   0.297       0.0935            1</span></span>
<span id="cb44-265"><a href="#cb44-265" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-266"><a href="#cb44-266" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-267"><a href="#cb44-267" tabindex="-1"></a><span class="co">#&gt;                 CDCA4_mRNA_expression=Low </span></span>
<span id="cb44-268"><a href="#cb44-268" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-269"><a href="#cb44-269" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb44-270"><a href="#cb44-270" tabindex="-1"></a><span class="co">#&gt;  1287      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-271"><a href="#cb44-271" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-272"><a href="#cb44-272" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-273"><a href="#cb44-273" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-274"><a href="#cb44-274" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-275"><a href="#cb44-275" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-276"><a href="#cb44-276" tabindex="-1"></a><span class="co">#&gt;                            N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-277"><a href="#cb44-277" tabindex="-1"></a><span class="co">#&gt; CDCA4_mRNA_expression=High 5        3     3.92     0.216      1.35</span></span>
<span id="cb44-278"><a href="#cb44-278" tabindex="-1"></a><span class="co">#&gt; CDCA4_mRNA_expression=Low  6        2     1.08     0.787      1.35</span></span>
<span id="cb44-279"><a href="#cb44-279" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-280"><a href="#cb44-280" tabindex="-1"></a><span class="co">#&gt;  Chisq= 1.3  on 1 degrees of freedom, p= 0.2 </span></span>
<span id="cb44-281"><a href="#cb44-281" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-282"><a href="#cb44-282" tabindex="-1"></a><span class="co">#&gt; [1] 0.2458557</span></span>
<span id="cb44-283"><a href="#cb44-283" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-284"><a href="#cb44-284" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb44-285"><a href="#cb44-285" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-286"><a href="#cb44-286" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  TACSTD2 </span></span>
<span id="cb44-287"><a href="#cb44-287" tabindex="-1"></a><span class="co">#&gt; [32mUsing Median to determine cutoff...[0m</span></span>
<span id="cb44-288"><a href="#cb44-288" tabindex="-1"></a><span class="co">#&gt; [32mMEDIAN CUT OFF:[0m</span></span>
<span id="cb44-289"><a href="#cb44-289" tabindex="-1"></a><span class="co">#&gt; [1] 15.75121</span></span>
<span id="cb44-290"><a href="#cb44-290" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-291"><a href="#cb44-291" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-292"><a href="#cb44-292" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-293"><a href="#cb44-293" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-294"><a href="#cb44-294" tabindex="-1"></a><span class="co">#&gt;                 TACSTD2_mRNA_expression=High </span></span>
<span id="cb44-295"><a href="#cb44-295" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-296"><a href="#cb44-296" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb44-297"><a href="#cb44-297" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-298"><a href="#cb44-298" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-299"><a href="#cb44-299" tabindex="-1"></a><span class="co">#&gt;                 TACSTD2_mRNA_expression=Low </span></span>
<span id="cb44-300"><a href="#cb44-300" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-301"><a href="#cb44-301" tabindex="-1"></a><span class="co">#&gt;   234      6       1    0.833   0.152       0.5827            1</span></span>
<span id="cb44-302"><a href="#cb44-302" tabindex="-1"></a><span class="co">#&gt;  1164      3       1    0.556   0.248       0.2312            1</span></span>
<span id="cb44-303"><a href="#cb44-303" tabindex="-1"></a><span class="co">#&gt;  1287      2       1    0.278   0.232       0.0539            1</span></span>
<span id="cb44-304"><a href="#cb44-304" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-305"><a href="#cb44-305" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-306"><a href="#cb44-306" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-307"><a href="#cb44-307" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-308"><a href="#cb44-308" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-309"><a href="#cb44-309" tabindex="-1"></a><span class="co">#&gt;                              N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-310"><a href="#cb44-310" tabindex="-1"></a><span class="co">#&gt; TACSTD2_mRNA_expression=High 5        2     3.55      0.68      3.27</span></span>
<span id="cb44-311"><a href="#cb44-311" tabindex="-1"></a><span class="co">#&gt; TACSTD2_mRNA_expression=Low  6        3     1.45      1.67      3.27</span></span>
<span id="cb44-312"><a href="#cb44-312" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-313"><a href="#cb44-313" tabindex="-1"></a><span class="co">#&gt;  Chisq= 3.3  on 1 degrees of freedom, p= 0.07 </span></span>
<span id="cb44-314"><a href="#cb44-314" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-315"><a href="#cb44-315" tabindex="-1"></a><span class="co">#&gt; [1] 0.07035001</span></span>
<span id="cb44-316"><a href="#cb44-316" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-317"><a href="#cb44-317" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_top_10.pdf[0m</span></span>
<span id="cb44-318"><a href="#cb44-318" tabindex="-1"></a><span class="co">#&gt; png </span></span>
<span id="cb44-319"><a href="#cb44-319" tabindex="-1"></a><span class="co">#&gt;   2</span></span>
<span id="cb44-320"><a href="#cb44-320" tabindex="-1"></a></span>
<span id="cb44-321"><a href="#cb44-321" tabindex="-1"></a><span class="co"># USING SPECIFIC GENES</span></span>
<span id="cb44-322"><a href="#cb44-322" tabindex="-1"></a><span class="fu">HTG_survival</span>(<span class="at">variable_01 =</span> <span class="st">&quot;Recurrence_01&quot;</span>,</span>
<span id="cb44-323"><a href="#cb44-323" tabindex="-1"></a>             <span class="at">time =</span> <span class="st">&quot;Time_to_death_surv&quot;</span>,</span>
<span id="cb44-324"><a href="#cb44-324" tabindex="-1"></a>             <span class="at">col_data =</span> AnnotData_tutorial,</span>
<span id="cb44-325"><a href="#cb44-325" tabindex="-1"></a>             <span class="at">counts_data =</span> counts_data_tutorial,</span>
<span id="cb44-326"><a href="#cb44-326" tabindex="-1"></a>             <span class="at">res =</span> <span class="cn">NULL</span>,</span>
<span id="cb44-327"><a href="#cb44-327" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;quartiles&quot;</span>,</span>
<span id="cb44-328"><a href="#cb44-328" tabindex="-1"></a>             <span class="at">genes_to_use =</span> <span class="fu">c</span>(<span class="st">&quot;LCP1&quot;</span>, <span class="st">&quot;OMA1&quot;</span>),</span>
<span id="cb44-329"><a href="#cb44-329" tabindex="-1"></a>             <span class="at">TME =</span> <span class="cn">NULL</span>,</span>
<span id="cb44-330"><a href="#cb44-330" tabindex="-1"></a>             <span class="at">outliers =</span> <span class="cn">NULL</span>,</span>
<span id="cb44-331"><a href="#cb44-331" tabindex="-1"></a>             <span class="at">pattern =</span> <span class="cn">NULL</span>,</span>
<span id="cb44-332"><a href="#cb44-332" tabindex="-1"></a>             <span class="at">remove_outliers =</span> <span class="cn">FALSE</span>)</span>
<span id="cb44-333"><a href="#cb44-333" tabindex="-1"></a><span class="co">#&gt; [32mNormalizing data...[0m</span></span>
<span id="cb44-334"><a href="#cb44-334" tabindex="-1"></a><span class="co">#&gt; [33mOBTAINING NORMALIEZ COUNTS...[0m</span></span>
<span id="cb44-335"><a href="#cb44-335" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb44-336"><a href="#cb44-336" tabindex="-1"></a><span class="co">#&gt; [32mChecking for duplicate columns[0m</span></span>
<span id="cb44-337"><a href="#cb44-337" tabindex="-1"></a><span class="co">#&gt; [1] &quot;No duplicate columns.&quot;</span></span>
<span id="cb44-338"><a href="#cb44-338" tabindex="-1"></a><span class="co">#&gt; [32mFiltering data.[0m</span></span>
<span id="cb44-339"><a href="#cb44-339" tabindex="-1"></a><span class="co">#&gt; [32mUsing provided genes[0m</span></span>
<span id="cb44-340"><a href="#cb44-340" tabindex="-1"></a><span class="co">#&gt; [32mStarting survival analysis.[0m</span></span>
<span id="cb44-341"><a href="#cb44-341" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-342"><a href="#cb44-342" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  LCP1 </span></span>
<span id="cb44-343"><a href="#cb44-343" tabindex="-1"></a><span class="co">#&gt; [32mUsing Quartiles to determine cutoff...[0m</span></span>
<span id="cb44-344"><a href="#cb44-344" tabindex="-1"></a><span class="co">#&gt; [32mQuartile 1 (Q1):[0m  396.9476 </span></span>
<span id="cb44-345"><a href="#cb44-345" tabindex="-1"></a><span class="co">#&gt; [32mQuartile 3 (Q3):[0m  3921.42 </span></span>
<span id="cb44-346"><a href="#cb44-346" tabindex="-1"></a><span class="co">#&gt; [32mUsing Q3 Quartile to determine cutoff...[0m</span></span>
<span id="cb44-347"><a href="#cb44-347" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-348"><a href="#cb44-348" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-349"><a href="#cb44-349" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-350"><a href="#cb44-350" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-351"><a href="#cb44-351" tabindex="-1"></a><span class="co">#&gt;                 LCP1_mRNA_expression=High </span></span>
<span id="cb44-352"><a href="#cb44-352" tabindex="-1"></a><span class="co">#&gt;         time       n.risk      n.event     survival      std.err lower 95% CI </span></span>
<span id="cb44-353"><a href="#cb44-353" tabindex="-1"></a><span class="co">#&gt;         1164            1            1            0          NaN           NA </span></span>
<span id="cb44-354"><a href="#cb44-354" tabindex="-1"></a><span class="co">#&gt; upper 95% CI </span></span>
<span id="cb44-355"><a href="#cb44-355" tabindex="-1"></a><span class="co">#&gt;           NA </span></span>
<span id="cb44-356"><a href="#cb44-356" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-357"><a href="#cb44-357" tabindex="-1"></a><span class="co">#&gt;                 LCP1_mRNA_expression=Low </span></span>
<span id="cb44-358"><a href="#cb44-358" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-359"><a href="#cb44-359" tabindex="-1"></a><span class="co">#&gt;   234      8       1    0.875   0.117       0.6734            1</span></span>
<span id="cb44-360"><a href="#cb44-360" tabindex="-1"></a><span class="co">#&gt;  1287      5       1    0.700   0.182       0.4201            1</span></span>
<span id="cb44-361"><a href="#cb44-361" tabindex="-1"></a><span class="co">#&gt;  2458      2       1    0.350   0.264       0.0799            1</span></span>
<span id="cb44-362"><a href="#cb44-362" tabindex="-1"></a><span class="co">#&gt;  5844      1       1    0.000     NaN           NA           NA</span></span>
<span id="cb44-363"><a href="#cb44-363" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-364"><a href="#cb44-364" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-365"><a href="#cb44-365" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-366"><a href="#cb44-366" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-367"><a href="#cb44-367" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-368"><a href="#cb44-368" tabindex="-1"></a><span class="co">#&gt;                           N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-369"><a href="#cb44-369" tabindex="-1"></a><span class="co">#&gt; LCP1_mRNA_expression=High 3        1    0.439    0.7153     0.932</span></span>
<span id="cb44-370"><a href="#cb44-370" tabindex="-1"></a><span class="co">#&gt; LCP1_mRNA_expression=Low  8        4    4.561    0.0689     0.932</span></span>
<span id="cb44-371"><a href="#cb44-371" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-372"><a href="#cb44-372" tabindex="-1"></a><span class="co">#&gt;  Chisq= 0.9  on 1 degrees of freedom, p= 0.3 </span></span>
<span id="cb44-373"><a href="#cb44-373" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-374"><a href="#cb44-374" tabindex="-1"></a><span class="co">#&gt; [1] 0.3343628</span></span>
<span id="cb44-375"><a href="#cb44-375" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-376"><a href="#cb44-376" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_SELECTED_GENES.pdf[0m</span></span>
<span id="cb44-377"><a href="#cb44-377" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-378"><a href="#cb44-378" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  OMA1 </span></span>
<span id="cb44-379"><a href="#cb44-379" tabindex="-1"></a><span class="co">#&gt; [32mUsing Quartiles to determine cutoff...[0m</span></span>
<span id="cb44-380"><a href="#cb44-380" tabindex="-1"></a><span class="co">#&gt; [32mQuartile 1 (Q1):[0m  109.8032 </span></span>
<span id="cb44-381"><a href="#cb44-381" tabindex="-1"></a><span class="co">#&gt; [32mQuartile 3 (Q3):[0m  623.4896 </span></span>
<span id="cb44-382"><a href="#cb44-382" tabindex="-1"></a><span class="co">#&gt; [32mUsing Q3 Quartile to determine cutoff...[0m</span></span>
<span id="cb44-383"><a href="#cb44-383" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-384"><a href="#cb44-384" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-385"><a href="#cb44-385" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-386"><a href="#cb44-386" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-387"><a href="#cb44-387" tabindex="-1"></a><span class="co">#&gt;                 OMA1_mRNA_expression=High </span></span>
<span id="cb44-388"><a href="#cb44-388" tabindex="-1"></a><span class="co">#&gt;         time       n.risk      n.event     survival      std.err lower 95% CI </span></span>
<span id="cb44-389"><a href="#cb44-389" tabindex="-1"></a><span class="co">#&gt;         1164            1            1            0          NaN           NA </span></span>
<span id="cb44-390"><a href="#cb44-390" tabindex="-1"></a><span class="co">#&gt; upper 95% CI </span></span>
<span id="cb44-391"><a href="#cb44-391" tabindex="-1"></a><span class="co">#&gt;           NA </span></span>
<span id="cb44-392"><a href="#cb44-392" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-393"><a href="#cb44-393" tabindex="-1"></a><span class="co">#&gt;                 OMA1_mRNA_expression=Low </span></span>
<span id="cb44-394"><a href="#cb44-394" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-395"><a href="#cb44-395" tabindex="-1"></a><span class="co">#&gt;   234      8       1    0.875   0.117       0.6734            1</span></span>
<span id="cb44-396"><a href="#cb44-396" tabindex="-1"></a><span class="co">#&gt;  1287      5       1    0.700   0.182       0.4201            1</span></span>
<span id="cb44-397"><a href="#cb44-397" tabindex="-1"></a><span class="co">#&gt;  2458      2       1    0.350   0.264       0.0799            1</span></span>
<span id="cb44-398"><a href="#cb44-398" tabindex="-1"></a><span class="co">#&gt;  5844      1       1    0.000     NaN           NA           NA</span></span>
<span id="cb44-399"><a href="#cb44-399" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-400"><a href="#cb44-400" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-401"><a href="#cb44-401" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-402"><a href="#cb44-402" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-403"><a href="#cb44-403" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-404"><a href="#cb44-404" tabindex="-1"></a><span class="co">#&gt;                           N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-405"><a href="#cb44-405" tabindex="-1"></a><span class="co">#&gt; OMA1_mRNA_expression=High 3        1    0.439    0.7153     0.932</span></span>
<span id="cb44-406"><a href="#cb44-406" tabindex="-1"></a><span class="co">#&gt; OMA1_mRNA_expression=Low  8        4    4.561    0.0689     0.932</span></span>
<span id="cb44-407"><a href="#cb44-407" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-408"><a href="#cb44-408" tabindex="-1"></a><span class="co">#&gt;  Chisq= 0.9  on 1 degrees of freedom, p= 0.3 </span></span>
<span id="cb44-409"><a href="#cb44-409" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-410"><a href="#cb44-410" tabindex="-1"></a><span class="co">#&gt; [1] 0.3343628</span></span>
<span id="cb44-411"><a href="#cb44-411" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-412"><a href="#cb44-412" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_SELECTED_GENES.pdf[0m</span></span>
<span id="cb44-413"><a href="#cb44-413" tabindex="-1"></a><span class="co">#&gt; png </span></span>
<span id="cb44-414"><a href="#cb44-414" tabindex="-1"></a><span class="co">#&gt;   2</span></span>
<span id="cb44-415"><a href="#cb44-415" tabindex="-1"></a></span>
<span id="cb44-416"><a href="#cb44-416" tabindex="-1"></a><span class="co"># USING TME DATA</span></span>
<span id="cb44-417"><a href="#cb44-417" tabindex="-1"></a><span class="fu">HTG_survival</span>(<span class="at">variable_01 =</span> <span class="st">&quot;Recurrence_01&quot;</span>,</span>
<span id="cb44-418"><a href="#cb44-418" tabindex="-1"></a>             <span class="at">time =</span> <span class="st">&quot;Time_to_death_surv&quot;</span>,</span>
<span id="cb44-419"><a href="#cb44-419" tabindex="-1"></a>             <span class="at">col_data =</span> AnnotData_tutorial,</span>
<span id="cb44-420"><a href="#cb44-420" tabindex="-1"></a>             <span class="at">counts_data =</span> counts_data_tutorial,</span>
<span id="cb44-421"><a href="#cb44-421" tabindex="-1"></a>             <span class="at">res =</span> <span class="cn">NULL</span>,</span>
<span id="cb44-422"><a href="#cb44-422" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;maxstat&quot;</span>,</span>
<span id="cb44-423"><a href="#cb44-423" tabindex="-1"></a>             <span class="at">genes_to_use =</span> <span class="cn">NULL</span>,</span>
<span id="cb44-424"><a href="#cb44-424" tabindex="-1"></a>             <span class="at">TME =</span> TME_data_tutorial<span class="sc">$</span>EPIC,</span>
<span id="cb44-425"><a href="#cb44-425" tabindex="-1"></a>             <span class="at">outliers =</span> <span class="cn">NULL</span>,</span>
<span id="cb44-426"><a href="#cb44-426" tabindex="-1"></a>             <span class="at">pattern =</span> <span class="cn">NULL</span>,</span>
<span id="cb44-427"><a href="#cb44-427" tabindex="-1"></a>             <span class="at">remove_outliers =</span> <span class="cn">FALSE</span>)</span>
<span id="cb44-428"><a href="#cb44-428" tabindex="-1"></a><span class="co">#&gt; [32mNormalizing data...[0m</span></span>
<span id="cb44-429"><a href="#cb44-429" tabindex="-1"></a><span class="co">#&gt; [33mOBTAINING NORMALIEZ COUNTS...[0m</span></span>
<span id="cb44-430"><a href="#cb44-430" tabindex="-1"></a><span class="co">#&gt; converting counts to integer mode</span></span>
<span id="cb44-431"><a href="#cb44-431" tabindex="-1"></a><span class="co">#&gt; [32mChecking for duplicate columns[0m</span></span>
<span id="cb44-432"><a href="#cb44-432" tabindex="-1"></a><span class="co">#&gt; [1] &quot;No duplicate columns.&quot;</span></span>
<span id="cb44-433"><a href="#cb44-433" tabindex="-1"></a><span class="co">#&gt; [32mFiltering data.[0m</span></span>
<span id="cb44-434"><a href="#cb44-434" tabindex="-1"></a><span class="co">#&gt; [32mUsing rownames from TME[0m</span></span>
<span id="cb44-435"><a href="#cb44-435" tabindex="-1"></a><span class="co">#&gt; [32mStarting survival analysis.[0m</span></span>
<span id="cb44-436"><a href="#cb44-436" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-437"><a href="#cb44-437" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  B_cell </span></span>
<span id="cb44-438"><a href="#cb44-438" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb44-439"><a href="#cb44-439" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb44-440"><a href="#cb44-440" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb44-441"><a href="#cb44-441" tabindex="-1"></a><span class="co">#&gt;         0.01396394 </span></span>
<span id="cb44-442"><a href="#cb44-442" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-443"><a href="#cb44-443" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-444"><a href="#cb44-444" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-445"><a href="#cb44-445" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-446"><a href="#cb44-446" tabindex="-1"></a><span class="co">#&gt;                 B_cell=High </span></span>
<span id="cb44-447"><a href="#cb44-447" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-448"><a href="#cb44-448" tabindex="-1"></a><span class="co">#&gt;  1164      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb44-449"><a href="#cb44-449" tabindex="-1"></a><span class="co">#&gt;  2458      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-450"><a href="#cb44-450" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-451"><a href="#cb44-451" tabindex="-1"></a><span class="co">#&gt;                 B_cell=Low </span></span>
<span id="cb44-452"><a href="#cb44-452" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-453"><a href="#cb44-453" tabindex="-1"></a><span class="co">#&gt;  1287      4       1     0.75   0.217        0.426            1</span></span>
<span id="cb44-454"><a href="#cb44-454" tabindex="-1"></a><span class="co">#&gt;  5844      1       1     0.00     NaN           NA           NA</span></span>
<span id="cb44-455"><a href="#cb44-455" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-456"><a href="#cb44-456" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-457"><a href="#cb44-457" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-458"><a href="#cb44-458" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-459"><a href="#cb44-459" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-460"><a href="#cb44-460" tabindex="-1"></a><span class="co">#&gt;             N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-461"><a href="#cb44-461" tabindex="-1"></a><span class="co">#&gt; B_cell=High 4        2     1.03     0.904      1.48</span></span>
<span id="cb44-462"><a href="#cb44-462" tabindex="-1"></a><span class="co">#&gt; B_cell=Low  4        2     2.97     0.315      1.48</span></span>
<span id="cb44-463"><a href="#cb44-463" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-464"><a href="#cb44-464" tabindex="-1"></a><span class="co">#&gt;  Chisq= 1.5  on 1 degrees of freedom, p= 0.2 </span></span>
<span id="cb44-465"><a href="#cb44-465" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-466"><a href="#cb44-466" tabindex="-1"></a><span class="co">#&gt; [1] 0.2240827</span></span>
<span id="cb44-467"><a href="#cb44-467" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-468"><a href="#cb44-468" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb44-469"><a href="#cb44-469" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-470"><a href="#cb44-470" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  Cancer_associated_fibroblast </span></span>
<span id="cb44-471"><a href="#cb44-471" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb44-472"><a href="#cb44-472" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb44-473"><a href="#cb44-473" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb44-474"><a href="#cb44-474" tabindex="-1"></a><span class="co">#&gt;        0.008256491 </span></span>
<span id="cb44-475"><a href="#cb44-475" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-476"><a href="#cb44-476" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-477"><a href="#cb44-477" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-478"><a href="#cb44-478" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-479"><a href="#cb44-479" tabindex="-1"></a><span class="co">#&gt;                 Cancer_associated_fibroblast=High </span></span>
<span id="cb44-480"><a href="#cb44-480" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-481"><a href="#cb44-481" tabindex="-1"></a><span class="co">#&gt;  1164      4       1    0.750   0.217       0.4259            1</span></span>
<span id="cb44-482"><a href="#cb44-482" tabindex="-1"></a><span class="co">#&gt;  2458      2       1    0.375   0.286       0.0839            1</span></span>
<span id="cb44-483"><a href="#cb44-483" tabindex="-1"></a><span class="co">#&gt;  5844      1       1    0.000     NaN           NA           NA</span></span>
<span id="cb44-484"><a href="#cb44-484" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-485"><a href="#cb44-485" tabindex="-1"></a><span class="co">#&gt;                 Cancer_associated_fibroblast=Low </span></span>
<span id="cb44-486"><a href="#cb44-486" tabindex="-1"></a><span class="co">#&gt;         time       n.risk      n.event     survival      std.err lower 95% CI </span></span>
<span id="cb44-487"><a href="#cb44-487" tabindex="-1"></a><span class="co">#&gt;     1287.000        2.000        1.000        0.500        0.354        0.125 </span></span>
<span id="cb44-488"><a href="#cb44-488" tabindex="-1"></a><span class="co">#&gt; upper 95% CI </span></span>
<span id="cb44-489"><a href="#cb44-489" tabindex="-1"></a><span class="co">#&gt;        1.000 </span></span>
<span id="cb44-490"><a href="#cb44-490" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-491"><a href="#cb44-491" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-492"><a href="#cb44-492" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-493"><a href="#cb44-493" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-494"><a href="#cb44-494" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-495"><a href="#cb44-495" tabindex="-1"></a><span class="co">#&gt;                                   N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-496"><a href="#cb44-496" tabindex="-1"></a><span class="co">#&gt; Cancer_associated_fibroblast=High 5        3    3.267    0.0218     0.154</span></span>
<span id="cb44-497"><a href="#cb44-497" tabindex="-1"></a><span class="co">#&gt; Cancer_associated_fibroblast=Low  3        1    0.733    0.0970     0.154</span></span>
<span id="cb44-498"><a href="#cb44-498" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-499"><a href="#cb44-499" tabindex="-1"></a><span class="co">#&gt;  Chisq= 0.2  on 1 degrees of freedom, p= 0.7 </span></span>
<span id="cb44-500"><a href="#cb44-500" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-501"><a href="#cb44-501" tabindex="-1"></a><span class="co">#&gt; [1] 0.6948866</span></span>
<span id="cb44-502"><a href="#cb44-502" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-503"><a href="#cb44-503" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb44-504"><a href="#cb44-504" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-505"><a href="#cb44-505" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  T_cell_CD4_ </span></span>
<span id="cb44-506"><a href="#cb44-506" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb44-507"><a href="#cb44-507" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb44-508"><a href="#cb44-508" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb44-509"><a href="#cb44-509" tabindex="-1"></a><span class="co">#&gt;       8.969556e-07 </span></span>
<span id="cb44-510"><a href="#cb44-510" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-511"><a href="#cb44-511" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-512"><a href="#cb44-512" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-513"><a href="#cb44-513" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-514"><a href="#cb44-514" tabindex="-1"></a><span class="co">#&gt;                 T_cell_CD4_=High </span></span>
<span id="cb44-515"><a href="#cb44-515" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-516"><a href="#cb44-516" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb44-517"><a href="#cb44-517" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-518"><a href="#cb44-518" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-519"><a href="#cb44-519" tabindex="-1"></a><span class="co">#&gt;                 T_cell_CD4_=Low </span></span>
<span id="cb44-520"><a href="#cb44-520" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-521"><a href="#cb44-521" tabindex="-1"></a><span class="co">#&gt;  1164      3       1    0.667   0.272       0.2995            1</span></span>
<span id="cb44-522"><a href="#cb44-522" tabindex="-1"></a><span class="co">#&gt;  1287      2       1    0.333   0.272       0.0673            1</span></span>
<span id="cb44-523"><a href="#cb44-523" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-524"><a href="#cb44-524" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-525"><a href="#cb44-525" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-526"><a href="#cb44-526" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-527"><a href="#cb44-527" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-528"><a href="#cb44-528" tabindex="-1"></a><span class="co">#&gt;                  N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-529"><a href="#cb44-529" tabindex="-1"></a><span class="co">#&gt; T_cell_CD4_=High 5        2      3.1      0.39      2.47</span></span>
<span id="cb44-530"><a href="#cb44-530" tabindex="-1"></a><span class="co">#&gt; T_cell_CD4_=Low  3        2      0.9      1.34      2.47</span></span>
<span id="cb44-531"><a href="#cb44-531" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-532"><a href="#cb44-532" tabindex="-1"></a><span class="co">#&gt;  Chisq= 2.5  on 1 degrees of freedom, p= 0.1 </span></span>
<span id="cb44-533"><a href="#cb44-533" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-534"><a href="#cb44-534" tabindex="-1"></a><span class="co">#&gt; [1] 0.1160831</span></span>
<span id="cb44-535"><a href="#cb44-535" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-536"><a href="#cb44-536" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb44-537"><a href="#cb44-537" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-538"><a href="#cb44-538" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  T_cell_CD8_ </span></span>
<span id="cb44-539"><a href="#cb44-539" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb44-540"><a href="#cb44-540" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb44-541"><a href="#cb44-541" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb44-542"><a href="#cb44-542" tabindex="-1"></a><span class="co">#&gt;       5.286423e-07 </span></span>
<span id="cb44-543"><a href="#cb44-543" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-544"><a href="#cb44-544" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-545"><a href="#cb44-545" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-546"><a href="#cb44-546" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-547"><a href="#cb44-547" tabindex="-1"></a><span class="co">#&gt;                 T_cell_CD8_=High </span></span>
<span id="cb44-548"><a href="#cb44-548" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-549"><a href="#cb44-549" tabindex="-1"></a><span class="co">#&gt;  1164      3       1    0.667   0.272       0.2995            1</span></span>
<span id="cb44-550"><a href="#cb44-550" tabindex="-1"></a><span class="co">#&gt;  1287      2       1    0.333   0.272       0.0673            1</span></span>
<span id="cb44-551"><a href="#cb44-551" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-552"><a href="#cb44-552" tabindex="-1"></a><span class="co">#&gt;                 T_cell_CD8_=Low </span></span>
<span id="cb44-553"><a href="#cb44-553" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-554"><a href="#cb44-554" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb44-555"><a href="#cb44-555" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-556"><a href="#cb44-556" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-557"><a href="#cb44-557" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-558"><a href="#cb44-558" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-559"><a href="#cb44-559" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-560"><a href="#cb44-560" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-561"><a href="#cb44-561" tabindex="-1"></a><span class="co">#&gt;                  N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-562"><a href="#cb44-562" tabindex="-1"></a><span class="co">#&gt; T_cell_CD8_=High 5        2      0.9      1.34      2.47</span></span>
<span id="cb44-563"><a href="#cb44-563" tabindex="-1"></a><span class="co">#&gt; T_cell_CD8_=Low  3        2      3.1      0.39      2.47</span></span>
<span id="cb44-564"><a href="#cb44-564" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-565"><a href="#cb44-565" tabindex="-1"></a><span class="co">#&gt;  Chisq= 2.5  on 1 degrees of freedom, p= 0.1 </span></span>
<span id="cb44-566"><a href="#cb44-566" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-567"><a href="#cb44-567" tabindex="-1"></a><span class="co">#&gt; [1] 0.1160831</span></span>
<span id="cb44-568"><a href="#cb44-568" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-569"><a href="#cb44-569" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb44-570"><a href="#cb44-570" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-571"><a href="#cb44-571" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  Endothelial_cell </span></span>
<span id="cb44-572"><a href="#cb44-572" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb44-573"><a href="#cb44-573" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb44-574"><a href="#cb44-574" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb44-575"><a href="#cb44-575" tabindex="-1"></a><span class="co">#&gt;        0.004808822 </span></span>
<span id="cb44-576"><a href="#cb44-576" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-577"><a href="#cb44-577" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-578"><a href="#cb44-578" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-579"><a href="#cb44-579" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-580"><a href="#cb44-580" tabindex="-1"></a><span class="co">#&gt;                 Endothelial_cell=High </span></span>
<span id="cb44-581"><a href="#cb44-581" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-582"><a href="#cb44-582" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb44-583"><a href="#cb44-583" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-584"><a href="#cb44-584" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-585"><a href="#cb44-585" tabindex="-1"></a><span class="co">#&gt;                 Endothelial_cell=Low </span></span>
<span id="cb44-586"><a href="#cb44-586" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-587"><a href="#cb44-587" tabindex="-1"></a><span class="co">#&gt;  1164      3       1    0.667   0.272       0.2995            1</span></span>
<span id="cb44-588"><a href="#cb44-588" tabindex="-1"></a><span class="co">#&gt;  1287      2       1    0.333   0.272       0.0673            1</span></span>
<span id="cb44-589"><a href="#cb44-589" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-590"><a href="#cb44-590" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-591"><a href="#cb44-591" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-592"><a href="#cb44-592" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-593"><a href="#cb44-593" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-594"><a href="#cb44-594" tabindex="-1"></a><span class="co">#&gt;                       N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-595"><a href="#cb44-595" tabindex="-1"></a><span class="co">#&gt; Endothelial_cell=High 5        2      3.1      0.39      2.47</span></span>
<span id="cb44-596"><a href="#cb44-596" tabindex="-1"></a><span class="co">#&gt; Endothelial_cell=Low  3        2      0.9      1.34      2.47</span></span>
<span id="cb44-597"><a href="#cb44-597" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-598"><a href="#cb44-598" tabindex="-1"></a><span class="co">#&gt;  Chisq= 2.5  on 1 degrees of freedom, p= 0.1 </span></span>
<span id="cb44-599"><a href="#cb44-599" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-600"><a href="#cb44-600" tabindex="-1"></a><span class="co">#&gt; [1] 0.1160831</span></span>
<span id="cb44-601"><a href="#cb44-601" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-602"><a href="#cb44-602" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb44-603"><a href="#cb44-603" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-604"><a href="#cb44-604" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  Macrophage </span></span>
<span id="cb44-605"><a href="#cb44-605" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb44-606"><a href="#cb44-606" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb44-607"><a href="#cb44-607" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb44-608"><a href="#cb44-608" tabindex="-1"></a><span class="co">#&gt;         0.03831832 </span></span>
<span id="cb44-609"><a href="#cb44-609" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-610"><a href="#cb44-610" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-611"><a href="#cb44-611" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-612"><a href="#cb44-612" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-613"><a href="#cb44-613" tabindex="-1"></a><span class="co">#&gt;                 Macrophage=High </span></span>
<span id="cb44-614"><a href="#cb44-614" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-615"><a href="#cb44-615" tabindex="-1"></a><span class="co">#&gt;  1164      4       1     0.75   0.217        0.426            1</span></span>
<span id="cb44-616"><a href="#cb44-616" tabindex="-1"></a><span class="co">#&gt;  1287      3       1     0.50   0.250        0.188            1</span></span>
<span id="cb44-617"><a href="#cb44-617" tabindex="-1"></a><span class="co">#&gt;  2458      1       1     0.00     NaN           NA           NA</span></span>
<span id="cb44-618"><a href="#cb44-618" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-619"><a href="#cb44-619" tabindex="-1"></a><span class="co">#&gt;                 Macrophage=Low </span></span>
<span id="cb44-620"><a href="#cb44-620" tabindex="-1"></a><span class="co">#&gt;         time       n.risk      n.event     survival      std.err lower 95% CI </span></span>
<span id="cb44-621"><a href="#cb44-621" tabindex="-1"></a><span class="co">#&gt;         5844            1            1            0          NaN           NA </span></span>
<span id="cb44-622"><a href="#cb44-622" tabindex="-1"></a><span class="co">#&gt; upper 95% CI </span></span>
<span id="cb44-623"><a href="#cb44-623" tabindex="-1"></a><span class="co">#&gt;           NA </span></span>
<span id="cb44-624"><a href="#cb44-624" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-625"><a href="#cb44-625" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-626"><a href="#cb44-626" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-627"><a href="#cb44-627" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-628"><a href="#cb44-628" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-629"><a href="#cb44-629" tabindex="-1"></a><span class="co">#&gt;                 N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-630"><a href="#cb44-630" tabindex="-1"></a><span class="co">#&gt; Macrophage=High 5        3     1.77     0.861      2.14</span></span>
<span id="cb44-631"><a href="#cb44-631" tabindex="-1"></a><span class="co">#&gt; Macrophage=Low  3        1     2.23     0.681      2.14</span></span>
<span id="cb44-632"><a href="#cb44-632" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-633"><a href="#cb44-633" tabindex="-1"></a><span class="co">#&gt;  Chisq= 2.1  on 1 degrees of freedom, p= 0.1 </span></span>
<span id="cb44-634"><a href="#cb44-634" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-635"><a href="#cb44-635" tabindex="-1"></a><span class="co">#&gt; [1] 0.1439024</span></span>
<span id="cb44-636"><a href="#cb44-636" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-637"><a href="#cb44-637" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb44-638"><a href="#cb44-638" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-639"><a href="#cb44-639" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  NK_cell </span></span>
<span id="cb44-640"><a href="#cb44-640" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb44-641"><a href="#cb44-641" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb44-642"><a href="#cb44-642" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb44-643"><a href="#cb44-643" tabindex="-1"></a><span class="co">#&gt;       1.486227e-07 </span></span>
<span id="cb44-644"><a href="#cb44-644" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-645"><a href="#cb44-645" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-646"><a href="#cb44-646" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-647"><a href="#cb44-647" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-648"><a href="#cb44-648" tabindex="-1"></a><span class="co">#&gt;                 NK_cell=High </span></span>
<span id="cb44-649"><a href="#cb44-649" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-650"><a href="#cb44-650" tabindex="-1"></a><span class="co">#&gt;  1164      3       1    0.667   0.272       0.2995            1</span></span>
<span id="cb44-651"><a href="#cb44-651" tabindex="-1"></a><span class="co">#&gt;  1287      2       1    0.333   0.272       0.0673            1</span></span>
<span id="cb44-652"><a href="#cb44-652" tabindex="-1"></a><span class="co">#&gt;  2458      1       1    0.000     NaN           NA           NA</span></span>
<span id="cb44-653"><a href="#cb44-653" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-654"><a href="#cb44-654" tabindex="-1"></a><span class="co">#&gt;                 NK_cell=Low </span></span>
<span id="cb44-655"><a href="#cb44-655" tabindex="-1"></a><span class="co">#&gt;         time       n.risk      n.event     survival      std.err lower 95% CI </span></span>
<span id="cb44-656"><a href="#cb44-656" tabindex="-1"></a><span class="co">#&gt;         5844            1            1            0          NaN           NA </span></span>
<span id="cb44-657"><a href="#cb44-657" tabindex="-1"></a><span class="co">#&gt; upper 95% CI </span></span>
<span id="cb44-658"><a href="#cb44-658" tabindex="-1"></a><span class="co">#&gt;           NA </span></span>
<span id="cb44-659"><a href="#cb44-659" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-660"><a href="#cb44-660" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-661"><a href="#cb44-661" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-662"><a href="#cb44-662" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-663"><a href="#cb44-663" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-664"><a href="#cb44-664" tabindex="-1"></a><span class="co">#&gt;              N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-665"><a href="#cb44-665" tabindex="-1"></a><span class="co">#&gt; NK_cell=High 5        3      1.4     1.829      3.46</span></span>
<span id="cb44-666"><a href="#cb44-666" tabindex="-1"></a><span class="co">#&gt; NK_cell=Low  3        1      2.6     0.985      3.46</span></span>
<span id="cb44-667"><a href="#cb44-667" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-668"><a href="#cb44-668" tabindex="-1"></a><span class="co">#&gt;  Chisq= 3.5  on 1 degrees of freedom, p= 0.06 </span></span>
<span id="cb44-669"><a href="#cb44-669" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-670"><a href="#cb44-670" tabindex="-1"></a><span class="co">#&gt; [1] 0.06289087</span></span>
<span id="cb44-671"><a href="#cb44-671" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-672"><a href="#cb44-672" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb44-673"><a href="#cb44-673" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-674"><a href="#cb44-674" tabindex="-1"></a><span class="co">#&gt; [32mPerforming analysis for column:[0m  uncharacterized_cell </span></span>
<span id="cb44-675"><a href="#cb44-675" tabindex="-1"></a><span class="co">#&gt; [32mUsing MAXSTAT to determine cutoff...[0m</span></span>
<span id="cb44-676"><a href="#cb44-676" tabindex="-1"></a><span class="co">#&gt; [32mMAXSTAT CUT OFF:[0m</span></span>
<span id="cb44-677"><a href="#cb44-677" tabindex="-1"></a><span class="co">#&gt; estimated cutpoint </span></span>
<span id="cb44-678"><a href="#cb44-678" tabindex="-1"></a><span class="co">#&gt;          0.4015231 </span></span>
<span id="cb44-679"><a href="#cb44-679" tabindex="-1"></a><span class="co">#&gt; [32mFitting survival model[0m</span></span>
<span id="cb44-680"><a href="#cb44-680" tabindex="-1"></a><span class="co">#&gt; [32mSummary of the fit[0m</span></span>
<span id="cb44-681"><a href="#cb44-681" tabindex="-1"></a><span class="co">#&gt; Call: survfit(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-682"><a href="#cb44-682" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-683"><a href="#cb44-683" tabindex="-1"></a><span class="co">#&gt;                 uncharacterized_cell=High </span></span>
<span id="cb44-684"><a href="#cb44-684" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-685"><a href="#cb44-685" tabindex="-1"></a><span class="co">#&gt;  2458      2       1      0.5   0.354        0.125            1</span></span>
<span id="cb44-686"><a href="#cb44-686" tabindex="-1"></a><span class="co">#&gt;  5844      1       1      0.0     NaN           NA           NA</span></span>
<span id="cb44-687"><a href="#cb44-687" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-688"><a href="#cb44-688" tabindex="-1"></a><span class="co">#&gt;                 uncharacterized_cell=Low </span></span>
<span id="cb44-689"><a href="#cb44-689" tabindex="-1"></a><span class="co">#&gt;  time n.risk n.event survival std.err lower 95% CI upper 95% CI</span></span>
<span id="cb44-690"><a href="#cb44-690" tabindex="-1"></a><span class="co">#&gt;  1164      3       1    0.667   0.272       0.2995            1</span></span>
<span id="cb44-691"><a href="#cb44-691" tabindex="-1"></a><span class="co">#&gt;  1287      2       1    0.333   0.272       0.0673            1</span></span>
<span id="cb44-692"><a href="#cb44-692" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-693"><a href="#cb44-693" tabindex="-1"></a><span class="co">#&gt; [32mPerforming log-rank test and obtaining p-value[0m</span></span>
<span id="cb44-694"><a href="#cb44-694" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb44-695"><a href="#cb44-695" tabindex="-1"></a><span class="co">#&gt; survival::survdiff(formula = surv_formula, data = merged_data)</span></span>
<span id="cb44-696"><a href="#cb44-696" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-697"><a href="#cb44-697" tabindex="-1"></a><span class="co">#&gt;                           N Observed Expected (O-E)^2/E (O-E)^2/V</span></span>
<span id="cb44-698"><a href="#cb44-698" tabindex="-1"></a><span class="co">#&gt; uncharacterized_cell=High 5        2      3.1      0.39      2.47</span></span>
<span id="cb44-699"><a href="#cb44-699" tabindex="-1"></a><span class="co">#&gt; uncharacterized_cell=Low  3        2      0.9      1.34      2.47</span></span>
<span id="cb44-700"><a href="#cb44-700" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb44-701"><a href="#cb44-701" tabindex="-1"></a><span class="co">#&gt;  Chisq= 2.5  on 1 degrees of freedom, p= 0.1 </span></span>
<span id="cb44-702"><a href="#cb44-702" tabindex="-1"></a><span class="co">#&gt; [32mP-value[0m</span></span>
<span id="cb44-703"><a href="#cb44-703" tabindex="-1"></a><span class="co">#&gt; [1] 0.1160831</span></span>
<span id="cb44-704"><a href="#cb44-704" tabindex="-1"></a><span class="co">#&gt; [32mGenerating Kaplan-Meier plot[0m</span></span>
<span id="cb44-705"><a href="#cb44-705" tabindex="-1"></a><span class="co">#&gt; [32mPlots saved in survival_analysis_plots_TME.pdf[0m</span></span>
<span id="cb44-706"><a href="#cb44-706" tabindex="-1"></a><span class="co">#&gt; png </span></span>
<span id="cb44-707"><a href="#cb44-707" tabindex="-1"></a><span class="co">#&gt;   2</span></span></code></pre></div>
</div>
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
